# Scaffolding the *Pipra filicauda* Genome
This pipeline outlines how we scaffolded the *Pipra filicauda* genome assembly to pseudochromosome scale based on synteny with *Chiroxiphia lanceolata*.  

**Input:** reference genome from NCBI  
**Output** reference genome scaffolded into pseudochromosomes and ready to be used for read mapping.  

Currently, there are several Pipridae reference genomes available. We will use *Pipra filicauda*, which is closely related to our study system. We want our reference genome to be as close as possible, while preferably being an outgroup so that all samples are equally distant to the reference (reducing bias).  
* ASM394559v2, GCF_003945595.2, Pipra filicauda  

There is one Pipridae genome that is chromosome scale:  
* bChiLan1.pri, GCF_009829145.1, Chiroxiphia lanceolata  

We can make the closer reference genome of *Pipra* pseudo-chromosome scale based on synteny with the *Chiroxiphia* genome.  
I am downloading the dataset from NCBI using the NCBI datasets tool.  
```sh
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome

/home/0_PROGRAMS/datasets download genome accession GCF_003945595.2 --filename GCF_003945595.2.zip 
/home/0_PROGRAMS/datasets download genome accession GCF_009829145.1 --filename GCF_009829145.1.zip 

unzip GCF_003945595.2.zip
mv ncbi_dataset GCF_003945595.2
unzip GCF_009829145.1.zip
mv ncbi_dataset GCF_009829145.1
rm GCF_003945595.2.zip GCF_009829145.1.zip #save space

```

# Reference-based scaffolding with RagTag  
Now we have a reference assembly, but it is not chromosome scale. To get there, we will rely on other published genomes. Luckily, there is one chromosome scale reference of Pipridae as of now (May 5 2023). I will use the program RagTag to improve contiguity by aligning the genome to the chromosome-scale reference (with minimap2) and joining scaffolds that map unambiguously to the adjacent regions of same chromosome, separated by gaps of N's of estimated sizes. Note: RagTag will not break or alter your existing scaffolds at all. It will only join them together.  

Note on picking the aligner: RagTag can use either minimap2 or nucmer. You can adjust the settings, but when I compared the defaults of minimap2 (took 1m1 real 3m36 user) vs nucmer (took 15m54 real 205m17 user) with my songbird dataset, minimap2 actually gave a more contiguous assembly with 8684 scaffolds, N50=74Mb, while nucmer only joined the biggest scaffolds resulting in 10612 scaffolds, N50=73 Mb. So, the N50 statistics are very similar but the difference is that minimap2 incorporated more of the small scaffolds. In terms of correctness, the minimap2 assembly appears to not show any interchromosomal rearrangements that could be detected by inspecting with jupiter plots, while three supposed tiny translocations from the autosomes to the Z chromosome appeared with the nucmer assembly - these may be errors since they are not supported by any other data. Based on this, I decided to go with minimap2 since it is 1) 15 times faster 2) the default of RagTag 3) gave a more contiguous genome in the samples I tested 4) did not give any suspicious-looking interchromosomal rearrangements in the samples I tested.  

```bash
#set up a directory
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/ragtag
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/ragtag

#get a copy of your genome assembly using soft links to save space
ln -s ../GCF_003945595.2/data/GCF_003945595.2/GCF_003945595.2_ASM394559v2_genomic.fna .

#get a copy of your reference genome assembly using soft links to save space
ln -s ../GCF_009829145.1/data/GCF_009829145.1/GCF_009829145.1_bChiLan1.pri_genomic.fna .

#now scaffold further with RagTag, with each available assembly
conda activate ragtag

#Here I am saving all my assembly names as variables to more easily change the pipeline
DRAFT1=GCF_003945595.2_ASM394559v2_genomic.fna #this is the target assembly to scaffold
ref2=GCF_009829145.1_bChiLan1.pri_genomic.fna #this is the best, chromosome-scale reference available

#run RagTag
time python /home/0_PROGRAMS/RagTag/ragtag.py scaffold "$ref2" "$DRAFT1" -f 1000 -q 10 -d 100000 -r -g 100 -m 100000 -o "$DRAFT1"_ragtag1_"$ref2" -t 50 --aligner /home/0_PROGRAMS/minimap2-2.17_x64-linux/minimap2 -u 
#real	0m51.279s
#user	2m56.683s
#sys	0m34.100s

#I'm using samtools faidx just to be able to view the lengths of scaffolds 
/home/0_PROGRAMS/samtools-1.17/samtools faidx "$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta

#check the resulting continguity of the assemblies at each step
#Ideally, the contiguity should be similar to a chromosome-scale assembly (ie, on the order of N50=74 Mb). If it is much lower, that is too bad - maybe you need better references, or better quality (eg sequencing depth) of your draft assembly. If it is much higher, then maybe something went wrong (unless you had a real chromosomal fusion in your taxon).
conda activate abyss #conda install -y -c bioconda abyss #conda activate abyss on new server, was /home/0_PROGRAMS/else_conda/conda/envs/abyss on old server
abyss-fac "$DRAFT1"
abyss-fac "$ref2"
abyss-fac "$DRAFT1"_ragtag1_"$ref2"/ragtag.scaffold.fasta
```
Ragtag settings:   
* the first two arguments are the (not gzipped) reference and the target assembly to scaffold.  
* `-f `: minimum length of unique alignment to consider for scaffolding  
* `-q`: min mapping quality  
* `-d`: syntenic alignments within this distance of each other on the query and merged into one alignment.  
* `-r`: infer gap sizes  
* `-g`: min gap size  
* `-m`: max gap size  
* `-o`: output directory  
* `-t`: number of threads for the mapping step  
* `--aligner`: path to minimap2 (or nucmer)  
* `-j`: txt file listing query scaffolds to skip (leave unplaced)  

**Timing**: Running ragtag with minimap2 took around 1m to 1m30 for each iteration.   

Results:  
| n    | n:500 | L50 | min  | N75      | N50      | N25      | E-size   | max      | sum      | name                                                                                                           |
| ---- | ----- | --- | ---- | -------- | -------- | -------- | -------- | -------- | -------- | -------------------------------------------------------------------------------------------------------------- |
| 1768 | 1768  | 122 | 2381 | 1184136  | 2585169  | 4446957  | 3150527  | 1.10E+07 | 1.08E+09 | GCF_003945595.2_ASM394559v2_genomic.fna                                                                        |
| 93   | 92    | 5   | 6217 | 2.65E+07 | 7.54E+07 | 1.20E+08 | 7.60E+07 | 1.56E+08 | 1.09E+09 | GCF_009829145.1_bChiLan1.pri_genomic.fna                                                                       |
| 666  | 666   | 5   | 2381 | 2.51E+07 | 7.66E+07 | 1.30E+08 | 7.71E+07 | 1.53E+08 | 1.08E+09 | GCF_003945595.2_ASM394559v2_genomic.fna_ragtag1_GCF_009829145.1_bChiLan1.pri_genomic.fna/ragtag.scaffold.fasta |

The contiguity of the *Pipra* genome is now similar to the chromosome-scale reference.  

# Rename  
Now that we have the reference genome, I will just clean up the scaffold names a bit for aesthetics.  

Here is a renaming table to rename scaffolds in the assembly. This has the first column: chromosome name; second column: Taeniopygia accession # for orthologous chromosome (for my reference); third column: Chiroxiphia reference-based scaffold names to be renamed.  
```bash
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference

cat > naming_table #contents are below
```
```
chr_1 NC_044998.1 NC_045638.1_RagTag
chr_1A NC_044999.1 NC_045641.1_RagTag
chr_2 NC_045000.1 NC_045637.1_RagTag
chr_3 NC_045001.1 NC_045639.1_RagTag
chr_4 NC_045002.1 NC_045640.1_RagTag
chr_4A NC_045003.1 NC_045650.1_RagTag
chr_5 NC_045004.1 NC_045642.1_RagTag
chr_6 NC_045005.1 NC_045644.1_RagTag
chr_7 NC_045006.1 NC_045643.1_RagTag
chr_8 NC_045007.1 NC_045645.1_RagTag
chr_9 NC_045008.1 NC_045646.1_RagTag
chr_10 NC_045009.1 NC_045648.1_RagTag
chr_11 NC_045010.1 NC_045649.1_RagTag
chr_12 NC_045011.1 NC_045647.1_RagTag
chr_13 NC_045012.1 NC_045651.1_RagTag
chr_14 NC_045013.1 NC_045652.1_RagTag
chr_15 NC_045014.1 NC_045654.1_RagTag
chr_17 NC_045015.1 NC_045657.1_RagTag
chr_18 NC_045016.1 NC_045655.1_RagTag
chr_19 NC_045017.1 NC_045656.1_RagTag
chr_20 NC_045018.1 NC_045653.1_RagTag
chr_21 NC_045019.1 NC_045658.1_RagTag
chr_22 NC_045020.1 NC_045664.1_RagTag
chr_23 NC_045021.1 NC_045660.1_RagTag
chr_24 NC_045022.1 NC_045659.1_RagTag
chr_25 NC_045023.1 NC_045665.1_RagTag
chr_26 NC_045024.1 NC_045661.1_RagTag
chr_27 NC_045025.1 NC_045662.1_RagTag
chr_28 NC_045026.1 NC_045663.1_RagTag
chr_Z NC_045027.1 NC_045671.1_RagTag
chr_W NC_045028.1 NC_045670.1_RagTag
chr_29 NC_045955.1 NC_045666.1_RagTag
```
Use this naming table to rename the scaffolds:
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference

#get raw data here
cp ../ragtag/GCF_003945595.2_ASM394559v2_genomic.fna_ragtag1_GCF_009829145.1_bChiLan1.pri_genomic.fna/ragtag.scaffold.fasta ./Pipra_filicauda.temp
#rename chromosomes
time cat naming_table | while read chromosome Taeniopygia scaffold ; do sed -i "s/$scaffold/$chromosome/g" ./Pipra_filicauda.temp ; done #took 6m30-7m30 per genome

#separate the chromosomes from the unplaced scaffolds, then rename the unplaced scaffolds
/home/0_PROGRAMS/seqkit grep -r -p "chr_" ./Pipra_filicauda.temp > Pipra_filicauda.chromosomes.fasta
/home/0_PROGRAMS/seqkit grep -v -r -p "chr_" ./Pipra_filicauda.temp > Pipra_filicauda.unplaced.fasta

#I am renaming scaffolds to something more manageable: a prefix and a number that is consistent
/home/0_PROGRAMS/bbmap/rename.sh in=Pipra_filicauda.unplaced.fasta prefix=Pipfil -Xmx20g out=Pipra_filicauda.unplaced_named.fa

#put the parts together to get the final reference genome
cat Pipra_filicauda.chromosomes.fasta Pipra_filicauda.unplaced_named.fa > Pipra_filicauda.ref.fa

#delete unnecessary intermediate files to save space. Careful not to run this too early.
#rm ./*.chromosomes.fasta
#rm ./*.unplaced.fasta
#rm ./*.unplaced_named.fa
#rm ./*.temp
```

# Index  
Now I will index the reference genome to be used for downstream mapping steps.  
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference

#index the reference
time /home/0_PROGRAMS/samtools-1.17/samtools faidx Pipra_filicauda.ref.fa #0m4.733s

#Before we can map our sequences, we must prepare the reference genome for Bowtie2:
time /home/0_PROGRAMS/bowtie2-2.4.4-linux-x86_64/bowtie2-build Pipra_filicauda.ref.fa Pipra_filicauda.ref #39m56.672s

#or bwa
time /home/0_PROGRAMS/bwa/bwa index Pipra_filicauda.ref.fa #25m29.456s

```

# Lift over genome annotation
This pipeline is how I obtained a gene annotation for the reference genome.  

First, we need to gather the reference annotation. The *Pipra filicauda* genome is already annotated, and the gff file came with the genome that we downloaded. It is in `/home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/GCF_003945595.2/data/GCF_003945595.2/genomic.gff`. We just need to lift it onto our scaffolded genome so that the coordinates match the pseudochromosome assembly.  


# Liftover genome annotation  

Now, we are going to transfer the annotation using liftoff, a program that takes a gff file and maps its features onto another genome.  
```bash
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0.3_annotation
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0.3_annotation
conda activate liftoff 

mkdir -p liftoff
#lift annotation to scaffolded reference
time liftoff -g /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/GCF_003945595.2/data/GCF_003945595.2/genomic.gff -dir liftoff/Pipra_filicauda_liftoff -o liftoff/Pipra_filicauda_liftoff.gff -u liftoff/Pipra_filicauda_liftoff.unmapped_features.txt /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/GCF_003945595.2/data/GCF_003945595.2/GCF_003945595.2_ASM394559v2_genomic.fna

#make a copy that just lists exons
PATH=$PATH:/home/0_PROGRAMS/bedops/bin
grep -P "Liftoff\texon" liftoff/Pipra_filicauda_liftoff.gff | /home/0_PROGRAMS/bedops/bin/gff2bed > liftoff/Pipra_filicauda_liftoff.exons.bed
grep -P "Liftoff\tgene" liftoff/Pipra_filicauda_liftoff.gff | /home/0_PROGRAMS/bedops/bin/gff2bed > liftoff/Pipra_filicauda_liftoff.genes.bed

```

# add repeat annotation
```bash
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0.4_repeats
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0.4_repeats
#set up environment
export PATH=$PATH:/home/0_PROGRAMS/RepeatMasker 
export PATH=$PATH:/home/0_PROGRAMS/cdhit 
export PATH=$PATH:/home/0_PROGRAMS/rmblast-2.11.0/bin
export PATH=$PATH:/home/0_PROGRAMS/RepeatScout-1.0.6 
export PATH=$PATH:/home/0_PROGRAMS/ucscTwoBitTools 
export PATH=$PATH:/home/0_PROGRAMS/RepeatMasker/util/ 
export PATH=$PATH:/home/0_PROGRAMS/ucscTwoBitTools 
export PATH=$PATH:/home/0_PROGRAMS/RECON-1.08/bin
export PATH=$PATH:/home/0_PROGRAMS/RepeatModeler-2.0.2a/
export PATH=$PATH:/home/0_PROGRAMS/EarlGrey

conda activate earlGrey

#run earlgrey
time /home/0_PROGRAMS/EarlGrey/earlGrey -g /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa -s Pipra_filicauda -o Pipra_filicauda_EarlGrey -r aves -t 60


#now run Tandem Repeats Finder
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0.4_repeats/trf
cd /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0.4_repeats/trf

#run trf
time /home/0_PROGRAMS/trf /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa 2 7 7 80 10 50 2000 -d #28m47.969s Lepidothrix; 39m Ceratopipra

#convert TRF output to bed format
conda activate py2
time python /home/0_PROGRAMS/TandemRepeatFinder_scripts/TRFdat_to_bed.py --dat Pipra_filicauda.ref.fa.2.7.7.80.10.50.2000.dat --bed Pipra_filicauda.ref.fa.2.7.7.80.10.50.2000.trf.bed #1s

#convert to different bed format
#tandem repeat
sort -f -k1,1 -k2,2n Pipra_filicauda.ref.fa.2.7.7.80.10.50.2000.trf.bed > Pipra_filicauda.trf.bed

#Telomeric repeat
grep -e "TTAGGG" -e "CCCTAA" Pipra_filicauda.ref.fa.2.7.7.80.10.50.2000.trf.bed | sort -f -k1,1 -k2,2n > Pipra_filicauda.telomeric.bed

#windowed format - easier to plot in my ggplots
#get windows, re-using the same windows from pixy
for chr in chr_1 chr_1A chr_2 chr_3 chr_4A chr_4 chr_5 chr_6 chr_7 chr_8 chr_9 chr_10 chr_11 chr_12 chr_13 chr_14 chr_15 chr_17 chr_18 chr_19 chr_20 chr_21 chr_22 chr_23 chr_24 chr_25 chr_26 chr_27 chr_28 chr_29 chr_Z ; do cut -f 2-4 /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy/Ceratopipra_rubrocapilla/window_500k/"$chr"_pi.txt | tail -n +2 | uniq | awk -v OFS='' '{print $1, "\t", $2-1, "\t", $3-1}' >> windows.bed ; done
sort -f -k1,1 -k2,2n windows.bed > temp && mv temp windows.bed

/home/0_PROGRAMS/bedops/bin/bedmap --echo --bases --bases-uniq --bases-uniq-f windows.bed Pipra_filicauda.trf.bed | tr "|" "\t" > trfcontent.txt

```

Now we have a chromosome-scale, annotated reference genome ready for use.  
