# Dataset
This page describes the raw data used in the analyses for the *Ceratopipra rubrocapilla* project. This dataset consists of 90-bp reads for 70 samples of *Ceratopipra rubrocapilla* and an outgroup of 5 samples of *Ceratopipra erythrocephala*. The samples were sequenced using a reduced representation RAD-seq approach with Pst.    

# Step 1: Obtain data  
First, we need to obtain the raw data. If this project is being replicated, the raw reads will be available to download from the NCBI SRA (accession numbers to be received upon publication).  

I made a new directory on the new server and copied the data over from the server where it was stored. The whole project will be stored in the folder `/home/1_Else/Ceratopipra`. The contents of this folder will have the same folder structure as the Github repository.
```bash
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.1_GBSDatasets
cd /home/1_Else/Ceratopipra/1_Datasets/1.1_GBSDatasets

#obtain raw data. I am placing them into 6 subfolders, each for a different lane of sequencing.  
mkdir -p rawdata/C6G73ANXX_6
scp else@$IP_ADDRESS:$PATH_TO_DATA/C6G73ANXX_6/Cerato*fq.gz ./rawdata/C6G73ANXX_6
mkdir -p rawdata/C6RM5ANXX_5
scp else@$IP_ADDRESS:$PATH_TO_DATA/C6RM5ANXX_5/Cerato*fq.gz ./rawdata/C6RM5ANXX_5
mkdir -p rawdata/C50W3ACXX_7
scp else@$IP_ADDRESS:$PATH_TO_DATA/C50W3ACXX_7/Cerato*fq.gz ./rawdata/C50W3ACXX_7
mkdir -p rawdata/C4BWGACXX_5
scp else@$IP_ADDRESS:$PATH_TO_DATA/C4BWGACXX_5/Cerato*fq.gz ./rawdata/C4BWGACXX_5
mkdir -p rawdata/C6JATANXX_8
scp else@$IP_ADDRESS:$PATH_TO_DATA/C6JATANXX_8/Cerato*fq.gz ./rawdata/C6JATANXX_8
mkdir -p rawdata/C50W3ACXX_8
scp else@$IP_ADDRESS:$PATH_TO_DATA/C50W3ACXX_8/Cerato*fq.gz ./rawdata/C50W3ACXX_8

```

# Assess quality

Now that we have the raw data, assess the reads with FastQC. This will give a glimpse about the quality of the sequencing reads, the number of reads per sample, and the presence of adapter contamination or other problems.    

```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.1_GBSDatasets

conda activate fastqc
fastqc rawdata/*/*fq.gz --threads 100

cd rawdata
cd C6G73ANXX_6
unzip '*fastqc.zip' #if you want to unzip all of them at once and there are no other files you don't want unzipped
cd ../C6RM5ANXX_5
unzip '*fastqc.zip' #if you want to unzip all of them at once and there are no other files you don't want unzipped
cd ../C50W3ACXX_7
unzip '*fastqc.zip' #if you want to unzip all of them at once and there are no other files you don't want unzipped
cd ../C4BWGACXX_5
unzip '*fastqc.zip' #if you want to unzip all of them at once and there are no other files you don't want unzipped
cd ../C6JATANXX_8
unzip '*fastqc.zip' #if you want to unzip all of them at once and there are no other files you don't want unzipped
cd ../C50W3ACXX_8
unzip '*fastqc.zip' #if you want to unzip all of them at once and there are no other files you don't want unzipped
cd ..

#get names of sequences
ls */*html | sed 's/.html//g' 
#verify that these are conventional file types
ls */*html | sed 's/.html//g' | while read file ; do grep "File type" "$file"/fastqc_data.txt | sed 's/File type\t//g' | wc -l ; done
#verify the encoding
ls */*html | sed 's/.html//g' | while read file ; do grep "Encoding" "$file"/fastqc_data.txt | sed 's/Encoding\t//g' ; done

#count number of sequences
ls */*html | sed 's/.html//g' | while read file ; do grep "Total Sequences" "$file"/fastqc_data.txt | sed 's/Total Sequences\t//g' ; done
#check if there are any flagged as poor quality
ls */*html | sed 's/.html//g' | while read file ; do grep "Sequences flagged as poor quality" "$file"/fastqc_data.txt | sed 's/Sequences flagged as poor quality\t//g' ; done
#check sequence lengths
ls */*html | sed 's/.html//g' | while read file ; do grep "Sequence length" "$file"/fastqc_data.txt | sed 's/Sequence length\t//g' ; done
#check GC content. Should be 41-47
ls */*html | sed 's/.html//g' | while read file ; do grep "%GC" "$file"/fastqc_data.txt | sed 's/%GC\t//g' ; done

ls */*html | sed 's/.html//g' | while read file ; do grep "Per base sequence quality" "$file"/fastqc_data.txt | sed 's/>>Per base sequence quality\t//g' ; done
#check Per tile sequence quality (some samples don't have this module)
#check Per sequence quality scores
ls */*html | sed 's/.html//g' | while read file ; do grep "Per sequence quality scores" "$file"/fastqc_data.txt | sed 's/>>Per sequence quality scores\t//g' ; done
#check Per base sequence content
ls */*html | sed 's/.html//g' | while read file ; do grep "Per base sequence content" "$file"/fastqc_data.txt | sed 's/>>Per base sequence content\t//g' ; done
#check Per sequence GC content
ls */*html | sed 's/.html//g' | while read file ; do grep "Per sequence GC content" "$file"/fastqc_data.txt | sed 's/>>Per sequence GC content\t//g' ; done
#check Per base N content
ls */*html | sed 's/.html//g' | while read file ; do grep "Per base N content" "$file"/fastqc_data.txt | sed 's/>>Per base N content\t//g' ; done
#Sequence Length Distribution
ls */*html | sed 's/.html//g' | while read file ; do grep "Sequence Length Distribution" "$file"/fastqc_data.txt | sed 's/>>Sequence Length Distribution\t//g' ; done
#Sequence Duplication Levels
ls */*html | sed 's/.html//g' | while read file ; do grep "Sequence Duplication Levels" "$file"/fastqc_data.txt | sed 's/>>Sequence Duplication Levels\t//g' ; done
#Overrepresented sequences
ls */*html | sed 's/.html//g' | while read file ; do grep "Overrepresented sequences" "$file"/fastqc_data.txt | sed 's/>>Overrepresented sequences\t//g' ; done
#Adapter Content
ls */*html | sed 's/.html//g' | while read file ; do grep "Adapter Content" "$file"/fastqc_data.txt | sed 's/>>Adapter Content\t//g' ; done
#look at amount of adapters at very ends of reads
ls */*html | sed 's/.html//g' | while read file ; do tail -n 2 "$file"/fastqc_data.txt | head -n 1 | cut -f 2-6; done
#and very start of reads
ls */*html | sed 's/.html//g' | while read file ; do grep -A 1 "Illumina Universal Adapter" "$file"/fastqc_data.txt | tail -n 1 | cut -f 2-6; done

```

The raw reads are 90 bp in length, starting with the PstI restriction cut site (TGCAG). Our data shows a large number of sequences with adapter contamination that must be removed before proceeding to the mapping stage.  

# Trim reads

We will trim reads using Trimmomatic. In order to loop through the files in parallel, I will make a text file listing the reads with two columns:  
1) Name of the sequencing run (specifying what folder the reads are in, and how to name the output)  
2) Name of sample (to specify name of input and output reads)  

```bash
cat > read_prefixes
```
Please ignore the fact that some Ceratopipra are spelled `Ceratopipraipra`, I don't want to change the names of the raw files.  
```
C50W3ACXX_7 Ceratopipraipra_eythrocephala_SGC288
C6G73ANXX_6 Ceratopipra_eythrocephala_B7534
C6G73ANXX_6 Ceratopipra_eythrocephala_B7539
C6G73ANXX_6 Ceratopipra_eythrocephala_B7536
C6G73ANXX_6 Ceratopipra_eythrocephala_B7537
C4BWGACXX_5 Ceratopipraipra_rubrocapila_PPS174
C4BWGACXX_5 Ceratopipraipra_rubrocapila_PPS221
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_AMA340
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_AMA601
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_AMA656
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_AMANA034
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_AMANA137
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_AMZ551
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_AMZ610
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_BR033
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_BR036
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_CAM117
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_CPE012
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_CPE025
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_CPE049
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_CPE069
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_CUJ027
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_CUJ072
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_FPR006
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_FPR015
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_FPR021
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_GAPTO259
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_MAM010
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_MAR092
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_MF005
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_MPDS628
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_MT087
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_MT123
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_OM022
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_OM112
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_OP100
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME024
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME062
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME079
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME134
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME169
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME460
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME482
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME521
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PIME522
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PPBIO221
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PPBIO305
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PPS041
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PPS352
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PUC007
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_PUC008
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_RDO007
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_TLP025
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UFAC1007
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UFAC305
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UFAC476
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UHE442
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UNA002
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UNA018
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UNA088
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UNA110
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_UNA124
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_WM281
C50W3ACXX_7 Ceratopipraipra_rubrocapilla_WM293
C6G73ANXX_6 Ceratopipra_rubrocapilla_B4615
C6G73ANXX_6 Ceratopipra_rubrocapilla_B5082
C6G73ANXX_6 Ceratopipra_rubrocapilla_CPEII008
C6G73ANXX_6 Ceratopipra_rubrocapilla_GAPTO268
C6G73ANXX_6 Ceratopipra_rubrocapilla_GAPTO277
C6G73ANXX_6 Ceratopipra_rubrocapilla_GAPTO327
C6G73ANXX_6 Ceratopipra_rubrocapilla_MSF107
C6G73ANXX_6 Ceratopipra_rubrocapilla_MT087
C6G73ANXX_6 Ceratopipra_rubrocapilla_MT123
C6JATANXX_8 Ceratopipra_rubrocapilla_AMZ551
C6JATANXX_8 Ceratopipra_rubrocapilla_AMZ610
C6JATANXX_8 Ceratopipra_rubrocapilla_BR36440
C6JATANXX_8 Ceratopipra_rubrocapilla_OM172
C6JATANXX_8 Ceratopipra_rubrocapilla_PIME135
C6JATANXX_8 Ceratopipra_rubrocapilla_TLP123
C6JATANXX_8 Ceratopipra_rubrocapilla_WM293
```
Note on choice of trimmer: Our data is too heavily contaminated by adapter sequences to use fastp: fastp currently does not allow any indels within the adapter sequence, but we have many sequences contaminated by an adapter with an indel in the sequence. I am also avoiding `process_radtags` due to the level of adapter contamination: `process_radtags` removed any sequence that contained an adapter, which excluded most of the data (about 70% of the reads for most samples). Some adapters still remained (eg, adapters with an indel and a SNP). I am using Trimmomatic for a bit more flexibility removing adapters that contain sequencing errors, without throwing away the whole read.  

The data are contaminated by this adapter: `AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG`  

Here are some examples of adapter sequences that I found by looking at the raw data. They are cherry-picked to illustrate how they can be variable in length, and contain errors (deletions or incorrect bases). Some time the data contains the full adapter sequence, and then a run of A's.
```
AGATCGGAAGAGGGTTCAGCAGGAATGCCGAGACCGTTCGTATGCCGTCT
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCT
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGC
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCT
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTC
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGC
AGATCGGAAGAGCGGTTCAGCAG_AATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAA
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG AAAAAA
AGATCGGAAGAGCGGT CAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG AAA
AGATCGGAAGAGCGGT CAGCAGGAATGCCGAGA
AGATCGGAAGAGCGGTTCAGCAGGAAT CCGAGACCGATCTCGTATGCCGTCTTCTGCTTGAAAAA
AGATCGGAAGAGCACACGTCTGAACTCCAGTCA
```

I will specify the adapter with an adapter file (fasta format) so that Trimmomatic knows what to trim out:  
```bash
cat > adapters.fa
```
```
>TruSeq3_IndexedAdapter
AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC
>TruSeq3_UniversalAdapter
AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGTA
>primer
AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG
```
Now, we can run Trimmomatic.  
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.1_GBSDatasets

mkdir -p trimlogs
mkdir -p fltrd_adptrlss

#trim with trimmomatic

time cat read_prefixes | parallel --jobs 40 --colsep " " time java -jar /home/0_PROGRAMS/Trimmomatic-0.39/trimmomatic-0.39.jar SE -phred33 rawdata/{1}/{2}.fq.gz fltrd_adptrlss/fltrd_adptrlss.{1}_{2}.fastq.gz ILLUMINACLIP:adapters.fa:2:7:7 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:40 "2>" trimlogs/{1}_{2}_qualtrim_out.log
#real	4m45.496s
#user	263m49.389s
#sys	13m11.014s

#on old server (48 threads, 40 jobs at a time)
#real	9m12.614s
#user	285m41.488s
#sys	20m42.620s


#assess results
conda activate fastqc
fastqc fltrd_adptrlss/fltrd_adptrlss.*.fastq.gz --threads 100
cd /home/1_Else/Ceratopipra/1_Datasets/1.1_GBSDatasets/fltrd_adptrlss
unzip '*fastqc.zip' #if you want to unzip all of them at once and there are no other files you don't want unzipped


#extract info from fastQC runs
#get names of sequences
ls *html | sed 's/.html//g' 
#verify that these are conventional file types
ls *html | sed 's/.html//g' | while read file ; do grep "File type" "$file"/fastqc_data.txt | sed 's/File type\t//g' | wc -l ; done
#verify the encoding
ls *html | sed 's/.html//g' | while read file ; do grep "Encoding" "$file"/fastqc_data.txt | sed 's/Encoding\t//g' ; done

#count number of sequences
ls *html | sed 's/.html//g' | while read file ; do grep "Total Sequences" "$file"/fastqc_data.txt | sed 's/Total Sequences\t//g' ; done
#check if there are any flagged as poor quality
ls *html | sed 's/.html//g' | while read file ; do grep "Sequences flagged as poor quality" "$file"/fastqc_data.txt | sed 's/Sequences flagged as poor quality\t//g' ; done
#check sequence lengths
ls *html | sed 's/.html//g' | while read file ; do grep "Sequence length" "$file"/fastqc_data.txt | sed 's/Sequence length\t//g' ; done
#check GC content. Should be 41-47
ls *html | sed 's/.html//g' | while read file ; do grep "%GC" "$file"/fastqc_data.txt | sed 's/%GC\t//g' ; done

ls *html | sed 's/.html//g' | while read file ; do grep "Per base sequence quality" "$file"/fastqc_data.txt | sed 's/>>Per base sequence quality\t//g' ; done
#check Per tile sequence quality (some samples don't have this module)
#check Per sequence quality scores
ls *html | sed 's/.html//g' | while read file ; do grep "Per sequence quality scores" "$file"/fastqc_data.txt | sed 's/>>Per sequence quality scores\t//g' ; done
#check Per base sequence content
ls *html | sed 's/.html//g' | while read file ; do grep "Per base sequence content" "$file"/fastqc_data.txt | sed 's/>>Per base sequence content\t//g' ; done
#check Per sequence GC content
ls *html | sed 's/.html//g' | while read file ; do grep "Per sequence GC content" "$file"/fastqc_data.txt | sed 's/>>Per sequence GC content\t//g' ; done
#check Per base N content
ls *html | sed 's/.html//g' | while read file ; do grep "Per base N content" "$file"/fastqc_data.txt | sed 's/>>Per base N content\t//g' ; done
#Sequence Length Distribution
ls *html | sed 's/.html//g' | while read file ; do grep "Sequence Length Distribution" "$file"/fastqc_data.txt | sed 's/>>Sequence Length Distribution\t//g' ; done
#Sequence Duplication Levels
ls *html | sed 's/.html//g' | while read file ; do grep "Sequence Duplication Levels" "$file"/fastqc_data.txt | sed 's/>>Sequence Duplication Levels\t//g' ; done
#Overrepresented sequences
ls *html | sed 's/.html//g' | while read file ; do grep "Overrepresented sequences" "$file"/fastqc_data.txt | sed 's/>>Overrepresented sequences\t//g' ; done
#Adapter Content
ls *html | sed 's/.html//g' | while read file ; do grep "Adapter Content" "$file"/fastqc_data.txt | sed 's/>>Adapter Content\t//g' ; done
#look at amount of adapters at very ends of reads
ls *html | sed 's/.html//g' | while read file ; do tail -n 2 "$file"/fastqc_data.txt | head -n 1 | cut -f 2-6; done
#and very start of reads
ls *html | sed 's/.html//g' | while read file ; do grep -A 1 "Illumina Universal Adapter" "$file"/fastqc_data.txt | tail -n 1 | cut -f 2-6; done





#for my interest to compare alternative trimmers (not used for downstream analyses) :
#trim adapters and low quality bases with fastp
#time cat read_prefixes | parallel --colsep " " 'time /home/0_PROGRAMS/fastp -i rawdata/{1}/{2}.fq.gz -o fltrd_adptrlss/fltrd_adptrlss.{1}.{2}.fastq.gz  --adapter_sequence=AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG --dont_overwrite --cut_front --cut_front_window_size 1 --cut_front_mean_quality 20 --cut_right --cut_right_window_size 4 --cut_right_mean_quality 20 --qualified_quality_phred 15 --unqualified_percent_limit 40 --length_required 40 --overrepresentation_analysis -h trimlogs/{1}.{2}.html -j trimlogs/{1}.{2}.json &> trimlogs/{1}.{2}.log'

#trim with process_radtags
#parallel mkdir -p out_process_radtags/{1} :::  C6G73ANXX_6 C6RM5ANXX_5 C50W3ACXX_7 C4BWGACXX_5 C6JATANXX_8 C50W3ACXX_8
#time cat read_prefixes | parallel --colsep " " time /home/0_PROGRAMS/stacks-2.64/process_radtags -f rawdata/{1}/{2}.fq.gz --out-path out_process_radtags/{1} --threads 100 --quality --clean --rescue -e pstI --adapter-1 AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG --window-size 0.15 --score-limit 10
#real	22m22.443s
#user	2387m48.127s
#sys	1m13.136s

```


Next, I want to merge files of the same sample such that there is only one fastq file per sample. I make a file listing the full filename (that includes the name of the sequencing run) in column 1 and the simpler sample name in column 2. I am just concatenating all files with the same sample name. This is not a very efficient strategy for the samples that have only a single file (it essentially copies the file), but I am not low on space and would rather all files have the same name structure for the raw input and be together in the same folder so that there is no ambiguity in five years when I forget which files were used for which samples.  

```bash
cat > samples
```
```
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_eythrocephala_SGC288 Ceratopipra_eythrocephala_SGC288
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_eythrocephala_B7534 Ceratopipra_eythrocephala_B7534
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_eythrocephala_B7539 Ceratopipra_eythrocephala_B7539
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_eythrocephala_B7536 Ceratopipra_eythrocephala_B7536
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_eythrocephala_B7537 Ceratopipra_eythrocephala_B7537
fltrd_adptrlss.C4BWGACXX_5_Ceratopipraipra_rubrocapila_PPS174 Ceratopipra_rubrocapilla_PPS174
fltrd_adptrlss.C4BWGACXX_5_Ceratopipraipra_rubrocapila_PPS221 Ceratopipra_rubrocapilla_PPS221
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_AMA340 Ceratopipra_rubrocapilla_AMA340
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_AMA601 Ceratopipra_rubrocapilla_AMA601
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_AMA656 Ceratopipra_rubrocapilla_AMA656
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_AMANA034 Ceratopipra_rubrocapilla_AMANA034
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_AMANA137 Ceratopipra_rubrocapilla_AMANA137
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_AMZ551
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_AMZ610
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_BR033 Ceratopipra_rubrocapilla_BR033
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_BR036 Ceratopipra_rubrocapilla_BR036
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_CAM117 Ceratopipra_rubrocapilla_CAM117
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_CPE012 Ceratopipra_rubrocapilla_CPE012
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_CPE025 Ceratopipra_rubrocapilla_CPE025
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_CPE049 Ceratopipra_rubrocapilla_CPE049
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_CPE069 Ceratopipra_rubrocapilla_CPE069
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_CUJ027 Ceratopipra_rubrocapilla_CUJ027
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_CUJ072 Ceratopipra_rubrocapilla_CUJ072
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_FPR006 Ceratopipra_rubrocapilla_FPR006
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_FPR015 Ceratopipra_rubrocapilla_FPR015
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_FPR021 Ceratopipra_rubrocapilla_FPR021
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_GAPTO259 Ceratopipra_rubrocapilla_GAPTO259
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_MAM010 Ceratopipra_rubrocapilla_MAM010
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_MAR092 Ceratopipra_rubrocapilla_MAR092
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_MF005 Ceratopipra_rubrocapilla_MF005
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_MPDS628 Ceratopipra_rubrocapilla_MPDS628
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_MT087 Ceratopipra_rubrocapilla_MT087
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_MT123 Ceratopipra_rubrocapilla_MT123
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_OM022 Ceratopipra_rubrocapilla_OM022
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_OM112 Ceratopipra_rubrocapilla_OM112
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_OP100 Ceratopipra_rubrocapilla_OP100
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME024 Ceratopipra_rubrocapilla_PIME024
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME062 Ceratopipra_rubrocapilla_PIME062
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME079 Ceratopipra_rubrocapilla_PIME079
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME134 Ceratopipra_rubrocapilla_PIME134
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME169 Ceratopipra_rubrocapilla_PIME169
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME460 Ceratopipra_rubrocapilla_PIME460
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME482 Ceratopipra_rubrocapilla_PIME482
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME521 Ceratopipra_rubrocapilla_PIME521
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PIME522 Ceratopipra_rubrocapilla_PIME522
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PPBIO221 Ceratopipra_rubrocapilla_PPBIO221
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PPBIO305 Ceratopipra_rubrocapilla_PPBIO305
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PPS041 Ceratopipra_rubrocapilla_PPS041
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PPS352 Ceratopipra_rubrocapilla_PPS352
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PUC007 Ceratopipra_rubrocapilla_PUC007
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_PUC008 Ceratopipra_rubrocapilla_PUC008
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_RDO007 Ceratopipra_rubrocapilla_RDO007
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_TLP025 Ceratopipra_rubrocapilla_TLP025
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UFAC1007 Ceratopipra_rubrocapilla_UFAC1007
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UFAC305 Ceratopipra_rubrocapilla_UFAC305
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UFAC476 Ceratopipra_rubrocapilla_UFAC476
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_UHE442
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UNA002 Ceratopipra_rubrocapilla_UNA002
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UNA018 Ceratopipra_rubrocapilla_UNA018
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UNA088 Ceratopipra_rubrocapilla_UNA088
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UNA110 Ceratopipra_rubrocapilla_UNA110
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_UNA124 Ceratopipra_rubrocapilla_UNA124
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_WM281 Ceratopipra_rubrocapilla_WM281
fltrd_adptrlss.C50W3ACXX_7_Ceratopipraipra_rubrocapilla_WM293 Ceratopipra_rubrocapilla_WM293
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_B4615 Ceratopipra_rubrocapilla_B4615
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_B5082 Ceratopipra_rubrocapilla_B5082
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_CPEII008 Ceratopipra_rubrocapilla_CPEII008
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_GAPTO268 Ceratopipra_rubrocapilla_GAPTO268
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_GAPTO277 Ceratopipra_rubrocapilla_GAPTO277
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_GAPTO327 Ceratopipra_rubrocapilla_GAPTO327
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_MSF107 Ceratopipra_rubrocapilla_MSF107
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_MT087 Ceratopipra_rubrocapilla_MT087
fltrd_adptrlss.C6G73ANXX_6_Ceratopipra_rubrocapilla_MT123 Ceratopipra_rubrocapilla_MT123
fltrd_adptrlss.C6JATANXX_8_Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_AMZ551
fltrd_adptrlss.C6JATANXX_8_Ceratopipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_AMZ610
fltrd_adptrlss.C6JATANXX_8_Ceratopipra_rubrocapilla_BR36440 Ceratopipra_rubrocapilla_BR36440
fltrd_adptrlss.C6JATANXX_8_Ceratopipra_rubrocapilla_OM172 Ceratopipra_rubrocapilla_OM172
fltrd_adptrlss.C6JATANXX_8_Ceratopipra_rubrocapilla_PIME135 Ceratopipra_rubrocapilla_PIME135
fltrd_adptrlss.C6JATANXX_8_Ceratopipra_rubrocapilla_TLP123 Ceratopipra_rubrocapilla_TLP123
fltrd_adptrlss.C6JATANXX_8_Ceratopipra_rubrocapilla_WM293 Ceratopipra_rubrocapilla_WM293
```

Repeat for the Radtag test dataset (not used downstream)  
```bash
cat > radtagsamples
```

```
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_eythrocephala_SGC288.fq.gz Ceratopipra_eythrocephala_SGC288
out_process_radtags/C6G73ANXX_6/Ceratopipra_eythrocephala_B7534.fq.gz Ceratopipra_eythrocephala_B7534
out_process_radtags/C6G73ANXX_6/Ceratopipra_eythrocephala_B7539.fq.gz Ceratopipra_eythrocephala_B7539
out_process_radtags/C6G73ANXX_6/Ceratopipra_eythrocephala_B7536.fq.gz Ceratopipra_eythrocephala_B7536
out_process_radtags/C6G73ANXX_6/Ceratopipra_eythrocephala_B7537.fq.gz Ceratopipra_eythrocephala_B7537
out_process_radtags/C4BWGACXX_5/Ceratopipraipra_rubrocapila_PPS174.fq.gz Ceratopipra_rubrocapilla_PPS174
out_process_radtags/C4BWGACXX_5/Ceratopipraipra_rubrocapila_PPS221.fq.gz Ceratopipra_rubrocapilla_PPS221
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_AMA340.fq.gz Ceratopipra_rubrocapilla_AMA340
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_AMA601.fq.gz Ceratopipra_rubrocapilla_AMA601
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_AMA656.fq.gz Ceratopipra_rubrocapilla_AMA656
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_AMANA034.fq.gz Ceratopipra_rubrocapilla_AMANA034
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_AMANA137.fq.gz Ceratopipra_rubrocapilla_AMANA137
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_AMZ551.fq.gz Ceratopipra_rubrocapilla_AMZ551
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_AMZ610.fq.gz Ceratopipra_rubrocapilla_AMZ610
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_BR033.fq.gz Ceratopipra_rubrocapilla_BR033
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_BR036.fq.gz Ceratopipra_rubrocapilla_BR036
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_CAM117.fq.gz Ceratopipra_rubrocapilla_CAM117
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_CPE012.fq.gz Ceratopipra_rubrocapilla_CPE012
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_CPE025.fq.gz Ceratopipra_rubrocapilla_CPE025
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_CPE049.fq.gz Ceratopipra_rubrocapilla_CPE049
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_CPE069.fq.gz Ceratopipra_rubrocapilla_CPE069
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_CUJ027.fq.gz Ceratopipra_rubrocapilla_CUJ027
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_CUJ072.fq.gz Ceratopipra_rubrocapilla_CUJ072
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_FPR006.fq.gz Ceratopipra_rubrocapilla_FPR006
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_FPR015.fq.gz Ceratopipra_rubrocapilla_FPR015
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_FPR021.fq.gz Ceratopipra_rubrocapilla_FPR021
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_GAPTO259.fq.gz Ceratopipra_rubrocapilla_GAPTO259
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_MAM010.fq.gz Ceratopipra_rubrocapilla_MAM010
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_MAR092.fq.gz Ceratopipra_rubrocapilla_MAR092
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_MF005.fq.gz Ceratopipra_rubrocapilla_MF005
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_MPDS628.fq.gz Ceratopipra_rubrocapilla_MPDS628
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_MT087.fq.gz Ceratopipra_rubrocapilla_MT087
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_MT123.fq.gz Ceratopipra_rubrocapilla_MT123
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_OM022.fq.gz Ceratopipra_rubrocapilla_OM022
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_OM112.fq.gz Ceratopipra_rubrocapilla_OM112
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_OP100.fq.gz Ceratopipra_rubrocapilla_OP100
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME024.fq.gz Ceratopipra_rubrocapilla_PIME024
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME062.fq.gz Ceratopipra_rubrocapilla_PIME062
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME079.fq.gz Ceratopipra_rubrocapilla_PIME079
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME134.fq.gz Ceratopipra_rubrocapilla_PIME134
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME169.fq.gz Ceratopipra_rubrocapilla_PIME169
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME460.fq.gz Ceratopipra_rubrocapilla_PIME460
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME482.fq.gz Ceratopipra_rubrocapilla_PIME482
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME521.fq.gz Ceratopipra_rubrocapilla_PIME521
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PIME522.fq.gz Ceratopipra_rubrocapilla_PIME522
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PPBIO221.fq.gz Ceratopipra_rubrocapilla_PPBIO221
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PPBIO305.fq.gz Ceratopipra_rubrocapilla_PPBIO305
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PPS041.fq.gz Ceratopipra_rubrocapilla_PPS041
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PPS352.fq.gz Ceratopipra_rubrocapilla_PPS352
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PUC007.fq.gz Ceratopipra_rubrocapilla_PUC007
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_PUC008.fq.gz Ceratopipra_rubrocapilla_PUC008
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_RDO007.fq.gz Ceratopipra_rubrocapilla_RDO007
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_TLP025.fq.gz Ceratopipra_rubrocapilla_TLP025
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UFAC1007.fq.gz Ceratopipra_rubrocapilla_UFAC1007
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UFAC305.fq.gz Ceratopipra_rubrocapilla_UFAC305
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UFAC476.fq.gz Ceratopipra_rubrocapilla_UFAC476
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UHE442.fq.gz Ceratopipra_rubrocapilla_UHE442
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UNA002.fq.gz Ceratopipra_rubrocapilla_UNA002
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UNA018.fq.gz Ceratopipra_rubrocapilla_UNA018
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UNA088.fq.gz Ceratopipra_rubrocapilla_UNA088
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UNA110.fq.gz Ceratopipra_rubrocapilla_UNA110
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_UNA124.fq.gz Ceratopipra_rubrocapilla_UNA124
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_WM281.fq.gz Ceratopipra_rubrocapilla_WM281
out_process_radtags/C50W3ACXX_7/Ceratopipraipra_rubrocapilla_WM293.fq.gz Ceratopipra_rubrocapilla_WM293
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_B4615.fq.gz Ceratopipra_rubrocapilla_B4615
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_B5082.fq.gz Ceratopipra_rubrocapilla_B5082
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_CPEII008.fq.gz Ceratopipra_rubrocapilla_CPEII008
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_GAPTO268.fq.gz Ceratopipra_rubrocapilla_GAPTO268
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_GAPTO277.fq.gz Ceratopipra_rubrocapilla_GAPTO277
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_GAPTO327.fq.gz Ceratopipra_rubrocapilla_GAPTO327
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_MSF107.fq.gz Ceratopipra_rubrocapilla_MSF107
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_MT087.fq.gz Ceratopipra_rubrocapilla_MT087
out_process_radtags/C6G73ANXX_6/Ceratopipra_rubrocapilla_MT123.fq.gz Ceratopipra_rubrocapilla_MT123
out_process_radtags/C6JATANXX_8/Ceratopipra_rubrocapilla_AMZ551.fq.gz Ceratopipra_rubrocapilla_AMZ551
out_process_radtags/C6JATANXX_8/Ceratopipra_rubrocapilla_AMZ610.fq.gz Ceratopipra_rubrocapilla_AMZ610
out_process_radtags/C6JATANXX_8/Ceratopipra_rubrocapilla_BR36440.fq.gz Ceratopipra_rubrocapilla_BR36440
out_process_radtags/C6JATANXX_8/Ceratopipra_rubrocapilla_OM172.fq.gz Ceratopipra_rubrocapilla_OM172
out_process_radtags/C6JATANXX_8/Ceratopipra_rubrocapilla_PIME135.fq.gz Ceratopipra_rubrocapilla_PIME135
out_process_radtags/C6JATANXX_8/Ceratopipra_rubrocapilla_TLP123.fq.gz Ceratopipra_rubrocapilla_TLP123
out_process_radtags/C6JATANXX_8/Ceratopipra_rubrocapilla_WM293.fq.gz Ceratopipra_rubrocapilla_WM293
```

```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.1_GBSDatasets

#merge samples that were trimmed by Trimmomatic
mkdir -p merged_fltrd_adptrlss
cat samples | while read run sample ; do cat fltrd_adptrlss/"$run".fastq.gz >> merged_fltrd_adptrlss/fltrd_adptrlss."$sample".fastq.gz ; done


#to compare Trimmomatic with fastp and process_radtags:
#merge samples that were trimmed by Stacks process_radtags
#mkdir -p merged_fltrd_adptrlss_process_radtags
#cat radtagsamples | while read file sample ; do cat "$file" >> merged_fltrd_adptrlss_process_radtags/fltrd_adptrlss."$sample".fastq.gz ; done

#merge samples trimmed by fastp
#mkdir -p merged_fltrd_adptrlss_fastp
#cat samples | sed 's/_Cerato/.Cerato/g' | while read run sample ; do cat failed_to_remove_adapters/fastp_indel_adapter_issue_fltrd_adptrlss/"$run".fastq.gz >> merged_fltrd_adptrlss_fastp/fltrd_adptrlss."$sample".fastq.gz ; done

```
Now we have our files that can be sent to the downstream mapping steps.  

# Appendix: counting reads
I am going to report the number of raw sequences input into the project. I summed the number of reads for each of the 70 samples of *Ceratopipra rubrocapilla* to be included in the publication: `193,788,016`. Here are the counts for each of the 70 samples that went into this sum (note five samples are included twice as they were sequennced across two lanes)    
```
Ceratopipra_rubrocapilla_AMZ551	256825
Ceratopipra_rubrocapilla_AMZ610	160163
Ceratopipra_rubrocapilla_B4615	3067067
Ceratopipra_rubrocapilla_B5082	2286319
Ceratopipra_rubrocapilla_BR36440	209639
Ceratopipra_rubrocapilla_CPEII008	4950262
Ceratopipra_rubrocapilla_GAPTO268	3455456
Ceratopipra_rubrocapilla_GAPTO277	745397
Ceratopipra_rubrocapilla_GAPTO327	2481619
Ceratopipra_rubrocapilla_MSF107	263543
Ceratopipra_rubrocapilla_MT087	3870193
Ceratopipra_rubrocapilla_MT123	4808262
Ceratopipra_rubrocapilla_OM172	3011928
Ceratopipra_rubrocapilla_PIME135	2994778
Ceratopipra_rubrocapilla_TLP123	1723537
Ceratopipra_rubrocapilla_WM293	636480
Ceratopipraipra_rubrocapila_PPS174	1845262
Ceratopipraipra_rubrocapila_PPS221	1676869
Ceratopipraipra_rubrocapilla_AMA340	1926046
Ceratopipraipra_rubrocapilla_AMA601	1453510
Ceratopipraipra_rubrocapilla_AMA656	686843
Ceratopipraipra_rubrocapilla_AMANA034	282517
Ceratopipraipra_rubrocapilla_AMANA137	464707
Ceratopipraipra_rubrocapilla_AMZ551	426105
Ceratopipraipra_rubrocapilla_AMZ610	355212
Ceratopipraipra_rubrocapilla_BR033	4220626
Ceratopipraipra_rubrocapilla_BR036	1830424
Ceratopipraipra_rubrocapilla_CAM117	7963151
Ceratopipraipra_rubrocapilla_CPE012	3530268
Ceratopipraipra_rubrocapilla_CPE025	2694211
Ceratopipraipra_rubrocapilla_CPE049	1227669
Ceratopipraipra_rubrocapilla_CPE069	6634326
Ceratopipraipra_rubrocapilla_CUJ027	1390178
Ceratopipraipra_rubrocapilla_CUJ072	1107820
Ceratopipraipra_rubrocapilla_FPR006	3914783
Ceratopipraipra_rubrocapilla_FPR015	921311
Ceratopipraipra_rubrocapilla_FPR021	7215711
Ceratopipraipra_rubrocapilla_GAPTO259	3268682
Ceratopipraipra_rubrocapilla_MAM010	1201579
Ceratopipraipra_rubrocapilla_MAR092	910916
Ceratopipraipra_rubrocapilla_MF005	8109064
Ceratopipraipra_rubrocapilla_MPDS628	651963
Ceratopipraipra_rubrocapilla_MT087	5196789
Ceratopipraipra_rubrocapilla_MT123	6526282
Ceratopipraipra_rubrocapilla_OM022	7585391
Ceratopipraipra_rubrocapilla_OM112	4774464
Ceratopipraipra_rubrocapilla_OP100	3494609
Ceratopipraipra_rubrocapilla_PIME024	6634244
Ceratopipraipra_rubrocapilla_PIME062	6988442
Ceratopipraipra_rubrocapilla_PIME079	738246
Ceratopipraipra_rubrocapilla_PIME134	1359601
Ceratopipraipra_rubrocapilla_PIME169	1262454
Ceratopipraipra_rubrocapilla_PIME460	2194098
Ceratopipraipra_rubrocapilla_PIME482	1040018
Ceratopipraipra_rubrocapilla_PIME521	382904
Ceratopipraipra_rubrocapilla_PIME522	1453857
Ceratopipraipra_rubrocapilla_PPBIO221	626894
Ceratopipraipra_rubrocapilla_PPBIO305	875963
Ceratopipraipra_rubrocapilla_PPS041	765780
Ceratopipraipra_rubrocapilla_PPS352	229939
Ceratopipraipra_rubrocapilla_PUC007	4993349
Ceratopipraipra_rubrocapilla_PUC008	6371310
Ceratopipraipra_rubrocapilla_RDO007	1421193
Ceratopipraipra_rubrocapilla_TLP025	1095049
Ceratopipraipra_rubrocapilla_UFAC1007	613532
Ceratopipraipra_rubrocapilla_UFAC305	3273321
Ceratopipraipra_rubrocapilla_UFAC476	762363
Ceratopipraipra_rubrocapilla_UHE442	1589369
Ceratopipraipra_rubrocapilla_UNA002	3054850
Ceratopipraipra_rubrocapilla_UNA018	6973565
Ceratopipraipra_rubrocapilla_UNA088	1322785
Ceratopipraipra_rubrocapilla_UNA110	6539657
Ceratopipraipra_rubrocapilla_UNA124	1560989
Ceratopipraipra_rubrocapilla_WM281	388256
Ceratopipraipra_rubrocapilla_WM293	867232
```

I did not include the outgroup in this count, but here they are for the record:  
```
Ceratopipra_erythrocephala_SGC288	7419354
Ceratopipra_erythrocephala_B7534	3360818
Ceratopipra_erythrocephala_B7536	2896915
Ceratopipra_erythrocephala_B7537	3286821
Ceratopipra_erythrocephala_B7539	3421640
```

## Sample metadata notes  
Note that PIME079 and PIME062 do not have coordinates that I could find. Their metadata in the museum archive states "Itaituba, Parna da Amazonia km 95". I interpret this to mean Parque National da Amazonia, 95 km down the road from Itaituba, since that is how the road markers are named. This would be approximately -4.7 N, -56.5 E, on the left (west) bank of the Tapajos, in the Rondonia area of endemism. It was collected in a series with other species that are associated with Rondonia, so I am confident that it is from the left bank of the river in Rondonia, not Tapajos.  


