# Mapping to reference genome
This page describes the pipeline used to map the trimmed sequencing reads to the reference genome.

First, I will make a list of samples to loop through:  
```bash
cat > samples.txt
```
```
Ceratopipra_chloromeros_B106768
Ceratopipra_chloromeros_B9014
Ceratopipra_chloromeros_DOT18183
Ceratopipra_chloromeros_FM18593
Ceratopipra_chloromeros_FM433680
Ceratopipra_chloromeros_Fm433681
Ceratopipra_cornuta_DOT11865
Ceratopipra_cornuta_DOT4822
Ceratopipra_erythrocephala_A12V3
Ceratopipra_erythrocephala_AMA197
Ceratopipra_erythrocephala_AMA216
Ceratopipra_erythrocephala_AMZ073
Ceratopipra_erythrocephala_AMZ098
Ceratopipra_erythrocephala_AMZ246
Ceratopipra_erythrocephala_AMZ247
Ceratopipra_erythrocephala_AMZ268
Ceratopipra_erythrocephala_AMZ291
Ceratopipra_erythrocephala_AMZ312
Ceratopipra_erythrocephala_AMZ313
Ceratopipra_erythrocephala_AMZ326
Ceratopipra_erythrocephala_AMZ477
Ceratopipra_erythrocephala_AMZ478
Ceratopipra_erythrocephala_B20441
Ceratopipra_erythrocephala_B2615
Ceratopipra_erythrocephala_B27528
Ceratopipra_erythrocephala_B27638
Ceratopipra_erythrocephala_B27671
Ceratopipra_erythrocephala_B27704
Ceratopipra_erythrocephala_B27884
Ceratopipra_erythrocephala_B35956
Ceratopipra_erythrocephala_B35977
Ceratopipra_erythrocephala_B4161
Ceratopipra_erythrocephala_B42717
Ceratopipra_erythrocephala_B42741
Ceratopipra_erythrocephala_B42907
Ceratopipra_erythrocephala_B46548
Ceratopipra_erythrocephala_B46584
Ceratopipra_erythrocephala_B48438
Ceratopipra_erythrocephala_B52929
Ceratopipra_erythrocephala_B52935
Ceratopipra_erythrocephala_B52936
Ceratopipra_erythrocephala_B5443
Ceratopipra_erythrocephala_B55344
Ceratopipra_erythrocephala_B65854
Ceratopipra_erythrocephala_B69282
Ceratopipra_erythrocephala_B69362
Ceratopipra_erythrocephala_B69400
Ceratopipra_erythrocephala_B69466
Ceratopipra_erythrocephala_B69478
Ceratopipra_erythrocephala_B69485
Ceratopipra_erythrocephala_B69488
Ceratopipra_erythrocephala_B69684
Ceratopipra_erythrocephala_B7014
Ceratopipra_erythrocephala_B7312
Ceratopipra_erythrocephala_B7455
Ceratopipra_erythrocephala_B7502
Ceratopipra_erythrocephala_B7533
Ceratopipra_erythrocephala_B7534
Ceratopipra_erythrocephala_B7535
Ceratopipra_erythrocephala_B7536
Ceratopipra_erythrocephala_B7537
Ceratopipra_erythrocephala_B7538
Ceratopipra_erythrocephala_B7539
Ceratopipra_erythrocephala_B7560
Ceratopipra_erythrocephala_CN225
Ceratopipra_erythrocephala_CN471
Ceratopipra_erythrocephala_CN495
Ceratopipra_erythrocephala_CN590
Ceratopipra_erythrocephala_CN626
Ceratopipra_erythrocephala_CN931
Ceratopipra_erythrocephala_CN932
Ceratopipra_erythrocephala_JAP008
Ceratopipra_erythrocephala_JAP015
Ceratopipra_erythrocephala_JAP025
Ceratopipra_erythrocephala_JAP472
Ceratopipra_erythrocephala_MPDS0342
Ceratopipra_erythrocephala_MPDS035
Ceratopipra_erythrocephala_MPDS0364
Ceratopipra_erythrocephala_MPDS342
Ceratopipra_erythrocephala_MPDS398
Ceratopipra_erythrocephala_ORX044
Ceratopipra_erythrocephala_ORX264
Ceratopipra_erythrocephala_ORX345
Ceratopipra_erythrocephala_ORX370
Ceratopipra_erythrocephala_ORX466
Ceratopipra_erythrocephala_SGC053
Ceratopipra_erythrocephala_SGC057
Ceratopipra_erythrocephala_SGC106
Ceratopipra_erythrocephala_SGC288
Ceratopipra_erythrocephala_SGC436
Ceratopipra_erythrocephala_SGC477
Ceratopipra_erythrocephala_SGC519
Ceratopipra_erythrocephala_SGC591
Ceratopipra_erythrocephala_SGC731
Ceratopipra_erythrocephala_SIL076
Ceratopipra_erythrocephala_SIL109
Ceratopipra_erythrocephala_V836
Ceratopipra_mentalis_B16114
Ceratopipra_mentalis_B26931
Ceratopipra_mentalis_B28451
Ceratopipra_mentalis_B28462
Ceratopipra_mentalis_B28463
Ceratopipra_mentalis_B28635
Ceratopipra_mentalis_B60669
Ceratopipra_mentalis_B60707
Ceratopipra_mentalis_B60938
Ceratopipra_mentalis_ORN335642
Ceratopipra_mentalis_ORN337342
Ceratopipra_mentalis_ORN337343
Ceratopipra_mentalis_ORN348320
Ceratopipra_rubrocapilla_AMA340
Ceratopipra_rubrocapilla_AMA601
Ceratopipra_rubrocapilla_AMA656
Ceratopipra_rubrocapilla_AMANA034
Ceratopipra_rubrocapilla_AMANA137
Ceratopipra_rubrocapilla_AMZ551
Ceratopipra_rubrocapilla_AMZ610
Ceratopipra_rubrocapilla_B4615
Ceratopipra_rubrocapilla_B5082
Ceratopipra_rubrocapilla_BR033
Ceratopipra_rubrocapilla_BR036
Ceratopipra_rubrocapilla_BR36440
Ceratopipra_rubrocapilla_CAM117
Ceratopipra_rubrocapilla_CPE012
Ceratopipra_rubrocapilla_CPE025
Ceratopipra_rubrocapilla_CPE027
Ceratopipra_rubrocapilla_CPE049
Ceratopipra_rubrocapilla_CPE066
Ceratopipra_rubrocapilla_CPE067
Ceratopipra_rubrocapilla_CPE069
Ceratopipra_rubrocapilla_CPE071
Ceratopipra_rubrocapilla_CPEII008
Ceratopipra_rubrocapilla_CUJ027
Ceratopipra_rubrocapilla_CUJ072
Ceratopipra_rubrocapilla_FPR006
Ceratopipra_rubrocapilla_FPR015
Ceratopipra_rubrocapilla_FPR021
Ceratopipra_rubrocapilla_GAPTO140
Ceratopipra_rubrocapilla_GAPTO259
Ceratopipra_rubrocapilla_GAPTO268
Ceratopipra_rubrocapilla_GAPTO277
Ceratopipra_rubrocapilla_GAPTO289
Ceratopipra_rubrocapilla_GAPTO327
Ceratopipra_rubrocapilla_JRT021
Ceratopipra_rubrocapilla_MAM010
Ceratopipra_rubrocapilla_MAR092
Ceratopipra_rubrocapilla_MELO34
Ceratopipra_rubrocapilla_MF005
Ceratopipra_rubrocapilla_MPDS628
Ceratopipra_rubrocapilla_MPDS776
Ceratopipra_rubrocapilla_MSF107
Ceratopipra_rubrocapilla_MT087
Ceratopipra_rubrocapilla_MT099
Ceratopipra_rubrocapilla_MT123
Ceratopipra_rubrocapilla_OM022
Ceratopipra_rubrocapilla_OM112
Ceratopipra_rubrocapilla_OM172
Ceratopipra_rubrocapilla_OM215
Ceratopipra_rubrocapilla_OP100
Ceratopipra_rubrocapilla_PIME024
Ceratopipra_rubrocapilla_PIME062
Ceratopipra_rubrocapilla_PIME079
Ceratopipra_rubrocapilla_PIME134
Ceratopipra_rubrocapilla_PIME135
Ceratopipra_rubrocapilla_PIME169
Ceratopipra_rubrocapilla_PIME460
Ceratopipra_rubrocapilla_PIME482
Ceratopipra_rubrocapilla_PIME521
Ceratopipra_rubrocapilla_PIME522
Ceratopipra_rubrocapilla_PPBIO221
Ceratopipra_rubrocapilla_PPBIO305
Ceratopipra_rubrocapilla_PPBIO312
Ceratopipra_rubrocapilla_PPS041
Ceratopipra_rubrocapilla_PPS174
Ceratopipra_rubrocapilla_PPS221
Ceratopipra_rubrocapilla_PPS352
Ceratopipra_rubrocapilla_PUC007
Ceratopipra_rubrocapilla_PUC008
Ceratopipra_rubrocapilla_RDO007
Ceratopipra_rubrocapilla_TLP025
Ceratopipra_rubrocapilla_TLP123
Ceratopipra_rubrocapilla_UFAC1007
Ceratopipra_rubrocapilla_UFAC1687
Ceratopipra_rubrocapilla_UFAC305
Ceratopipra_rubrocapilla_UFAC476
Ceratopipra_rubrocapilla_UHE442
Ceratopipra_rubrocapilla_UNA002
Ceratopipra_rubrocapilla_UNA014
Ceratopipra_rubrocapilla_UNA018
Ceratopipra_rubrocapilla_UNA088
Ceratopipra_rubrocapilla_UNA110
Ceratopipra_rubrocapilla_UNA124
Ceratopipra_rubrocapilla_WM281
Ceratopipra_rubrocapilla_WM293
```

Then, I will loop through this list and map each sample to the reference genome using bwa mem. I am immediately sorting the bam files (without storing the intermediate unsorted files).  
```bash
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.2a_mapping
cd /home/1_Else/Ceratopipra/1_Datasets/1.2a_mapping

mkdir -p mapping_logs
mkdir -p trimmomaticbwamapped

#index the reference genome
#time /home/0_PROGRAMS/bwa/bwa index /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa #25m

#get readgroups file (see above)

#run bwa and pipe output to fixmate
#Fixmate adds information on insert size and mate pair coordinates. note fixmate normally requires bam file to be sorted by read name, but bwa outputs the sam file essentially sorted by name already with mates one after the other. 
#before we can merge our sequencing runs, we must also sort each run by genomic position. 
cat samples.txt | time parallel --jobs 40 --colsep '\t' 'time /home/0_PROGRAMS/bwa/bwa mem -R "@RG\tID:Missing.1.{1}\tSM:{1}\tLB:Lib1\tPU:Missing.1.NA\tPL:ILLUMINA" /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa /home/1_Else/Ceratopipra/1_Datasets/1.1_GBSDatasets/merged_fltrd_adptrlss/fltrd_adptrlss.{1}.fastq.gz 2>> mapping_logs/{1}_GBSPipfilbwa.log | /home/0_PROGRAMS/samtools-1.17/samtools sort -O BAM > trimmomaticbwamapped/{1}.GBSPipfilbwa.bam'
#60898.96user 4736.72system 38:50.81elapsed 2815%CPU (0avgtext+0avgdata 2286548maxresident)k without -R
#61007.75user 4811.17system 38:34.84elapsed 2843%CPU (0avgtext+0avgdata 2286744maxresident)k with -R

#index the bam with samtools for downstream programs that require an index
cat samples.txt | parallel --jobs 40 --colsep '\t' 'time /home/0_PROGRAMS/samtools-1.17/samtools index trimmomaticbwamapped/{1}.GBSPipfilbwa.bam'

#Now we have a bam file ready to use! BE AWARE: there has been no filtering. You will have to implement some filtering before calling genotypes etc.
```

### Quality control: mapping stats
BWA does not output a log of mapping success, but this can be easily calculated. I will run bamstats to assess the mapping.    
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.2a_mapping

mkdir -p QC_reports
#bamstats
cat samples.txt | parallel --jobs 60 --colsep " " 'time /home/0_PROGRAMS/bamtools/bin/bamtools stats -in trimmomaticbwamapped/{1}.GBSPipfilbwa.bam > QC_reports/{1}.bamstats.txt'

#in order to read the bamstats results, MultiQC sometimes needs some formatting fixed, excuse the messiness. This just removes the first line if it is blank.
cat samples.txt | while read sample ; do sed -i '1{/^$/d}' QC_reports/"$sample".bamstats.txt ; done #removes first line if it is blank

#read results
grep "Mapped reads" QC_reports/*.bamstats.txt
```

### Quality control: general data
There are several tools that calculate various stats about your data. Here are a few faves. First, I will get some general stats from samtools (manual for these commands [here](http://www.htslib.org/doc/samtools-1.6.html)).  
Then I will look at some more details with [Qualimap](http://qualimap.bioinfo.cipf.es/doc_html/analysis.html). It takes longer to run but I love looking at Qualimap reports.    

(Note I am not running Qualimap in parallel because that would take too much memory and crash on this server. One sample at a time.)  
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.2a_mapping

cat samples.txt | parallel --jobs 40 --colsep " " 'time /home/0_PROGRAMS/samtools-1.17/samtools flagstat trimmomaticbwamapped/{1}.GBSPipfilbwa.bam > QC_reports/flagstats_{1}.txt' &
cat samples.txt | parallel --jobs 40 --colsep " " 'time /home/0_PROGRAMS/samtools-1.17/samtools idxstats trimmomaticbwamapped/{1}.GBSPipfilbwa.bam > QC_reports/idxstats_{1}.txt' &
cat samples.txt | parallel --jobs 40 --colsep " " 'time /home/0_PROGRAMS/samtools-1.17/samtools stats trimmomaticbwamapped/{1}.GBSPipfilbwa.bam > QC_reports/samstats_{1}.txt'

cat samples.txt | while read sample data; do time /home/0_PROGRAMS/qualimap_v2.2.1/qualimap bamqc -bam trimmomaticbwamapped/"$sample".GBSPipfilbwa.bam -c -ip -nt 80 --java-mem-size=8G -outdir QC_reports/"$sample"_qualimap ; done

#now read the results with multiqc!
conda activate fastqc #conda install -c bioconda fastqc #conda install -c bioconda multiqc
multiqc QC_reports -n bwa_mapping_stats -o QC_reports -f

#get average depths of each sample
cat samples.txt | while read sample ; do echo "$sample" ; time /home/0_PROGRAMS/samtools-1.13/samtools depth trimmomaticbwamapped/"$sample".GBSPipfilbwa.bam | awk '{sum+=$3; sumsq+=$3*$3} END { print "Average = ",sum/NR; print "Stdev = ",sqrt(sumsq/NR - (sum/NR)**2)}' ; done

#check the sex of each sample

#get the idx_Utilities.R file into your working directory
cd /home/1_Else/Ceratopipra/1_Datasets/1.2a_mapping/QC_reports
cp /home/0_PROGRAMS/elsemikk_scripts/idx_Utilities.R .

#run Process_idxstats.R, a script that I made to use the relative depth of the Z chromosome to identify the sex of a sample
cat ../samples.txt | parallel --jobs 40 'Rscript /home/0_PROGRAMS/elsemikk_scripts/Process_idxstats.R idxstats_{1}.txt 150 >> {1}.processidx'

#view results
cat *.processidx | grep -E "file being|Cong"

#test putative males now that we know which scaffold is the Z (note this looks a little backwards because after I figured out which chromosome was Z, I named it chrZ, making it look like it was unnecessary to try figure out which chromosome was Z)
cat ../samples.txt | parallel --jobs 40 'Rscript /home/0_PROGRAMS/elsemikk_scripts/testMale.R idxstats_{1}.txt chr_Z >> {1}.testmale'
#make an easy-to-parse file
cat ../samples.txt | while read sample ; do sex=$(grep "male" "$sample".testmale | sed 's/^.* //g; s/female/F/g; s/male/M/g') ; reldepth=$(grep " the depth" "$sample".testmale | sed 's/^.*Z is //g; s/ .*$//g') ; printf "$sample\t$sex\t$reldepth\n" >> Ceratopipra.sexes ; done
```
* `-c`:Paint chromosome limits inside charts
* `-ip`: collect information on reads overlapping between paired end mates.
* `-nt`: number of threads to use
* `--java-mem-size=4G`: increases memory allowance, it was crashing after around half way done otherwise.

Here are the predicted sexes from my script based on relative depth of the Z. They match my mannually assigned sexes based on looking at the plots of read counts vs chromosome size, except for samples that failed to sequence properly and had extremely low depth, so there was not much to go on. The failed samples had relative chrZ depth between 0.8 and 0.7, so got assigned male automatically, but some of them look like female to me, though it is hard to tell based on so few reads (looking at heterozygosity would be more informative, or better, looking for chrW reads). These are excluded from the project due to their extremely low depth and so it does not matter for out project.  
```
sample	sex	relative_depth_of_Z
Ceratopipra_erythrocephala_B7534	M	0.966288
Ceratopipra_erythrocephala_B7536	M	0.9643778
Ceratopipra_erythrocephala_B7537	M	0.9687279
Ceratopipra_erythrocephala_B7539	M	0.9622178
Ceratopipra_erythrocephala_SGC288	M	0.9696757
Ceratopipra_rubrocapilla_AMA340	M	0.9626801
Ceratopipra_rubrocapilla_AMA601	M	0.9555621
Ceratopipra_rubrocapilla_AMA656	M	0.954985
Ceratopipra_rubrocapilla_AMANA034	M	0.9586599
Ceratopipra_rubrocapilla_AMANA137	M	0.9570945
Ceratopipra_rubrocapilla_AMZ551	F	0.6398828
Ceratopipra_rubrocapilla_AMZ610	F	0.6007255
Ceratopipra_rubrocapilla_B4615	M	0.9678645
Ceratopipra_rubrocapilla_B5082	M	0.946409
Ceratopipra_rubrocapilla_BR033	F	0.6462398
Ceratopipra_rubrocapilla_BR036	M	0.9779082
Ceratopipra_rubrocapilla_BR36440	M	0.9922351
Ceratopipra_rubrocapilla_CAM117	M	0.9624148
Ceratopipra_rubrocapilla_CPE012	M	0.9823789
Ceratopipra_rubrocapilla_CPE025	M	0.9826438
Ceratopipra_rubrocapilla_CPE027	M	0.9990412
Ceratopipra_rubrocapilla_CPE049	F	0.6387009
Ceratopipra_rubrocapilla_CPE066	M	0.7867647
Ceratopipra_rubrocapilla_CPE067	M	0.748227
Ceratopipra_rubrocapilla_CPE069	M	0.9809381
Ceratopipra_rubrocapilla_CPE071	F	0.65063
Ceratopipra_rubrocapilla_CPEII008	M	0.9888562
Ceratopipra_rubrocapilla_CUJ027	F	0.6092802
Ceratopipra_rubrocapilla_CUJ072	F	0.6057681
Ceratopipra_rubrocapilla_FPR006	F	0.6088517
Ceratopipra_rubrocapilla_FPR015	M	0.9441761
Ceratopipra_rubrocapilla_FPR021	M	0.9676323
Ceratopipra_rubrocapilla_GAPTO140	F	
Ceratopipra_rubrocapilla_GAPTO259	F	0.6980761
Ceratopipra_rubrocapilla_GAPTO268	F	0.6632395
Ceratopipra_rubrocapilla_GAPTO277	M	0.9925182
Ceratopipra_rubrocapilla_GAPTO289	F	0.6400552
Ceratopipra_rubrocapilla_GAPTO327	F	0.6286752
Ceratopipra_rubrocapilla_JRT021	M	0.9254658
Ceratopipra_rubrocapilla_MAM010	M	0.971104
Ceratopipra_rubrocapilla_MAR092	M	0.9526952
Ceratopipra_rubrocapilla_MELO34	M	0.8043478
Ceratopipra_rubrocapilla_MF005	M	0.9664961
Ceratopipra_rubrocapilla_MPDS628	F	0.5983577
Ceratopipra_rubrocapilla_MPDS776	F	0.5805085
Ceratopipra_rubrocapilla_MSF107	F	0.6221935
Ceratopipra_rubrocapilla_MT087	F	0.6166953
Ceratopipra_rubrocapilla_MT099	F	
Ceratopipra_rubrocapilla_MT123	M	0.9625963
Ceratopipra_rubrocapilla_OM022	F	0.6220103
Ceratopipra_rubrocapilla_OM112	M	0.9653893
Ceratopipra_rubrocapilla_OM172	F	0.6042036
Ceratopipra_rubrocapilla_OM215	F	0.6309717
Ceratopipra_rubrocapilla_OP100	F	0.62616
Ceratopipra_rubrocapilla_PIME024	M	0.9636559
Ceratopipra_rubrocapilla_PIME062	M	0.9690425
Ceratopipra_rubrocapilla_PIME079	F	0.6217783
Ceratopipra_rubrocapilla_PIME134	M	0.9691162
Ceratopipra_rubrocapilla_PIME135	F	0.6095504
Ceratopipra_rubrocapilla_PIME169	F	0.6052931
Ceratopipra_rubrocapilla_PIME460	M	0.9676999
Ceratopipra_rubrocapilla_PIME482	F	0.6119122
Ceratopipra_rubrocapilla_PIME521	M	0.9925229
Ceratopipra_rubrocapilla_PIME522	F	0.6224734
Ceratopipra_rubrocapilla_PPBIO221	F	0.6011726
Ceratopipra_rubrocapilla_PPBIO305	F	0.6160478
Ceratopipra_rubrocapilla_PPBIO312	F	0.619019
Ceratopipra_rubrocapilla_PPS041	M	0.9502862
Ceratopipra_rubrocapilla_PPS174	F	0.5943382
Ceratopipra_rubrocapilla_PPS221	M	0.9821685
Ceratopipra_rubrocapilla_PPS352	M	0.9819714
Ceratopipra_rubrocapilla_PUC007	M	0.9699252
Ceratopipra_rubrocapilla_PUC008	M	0.9678907
Ceratopipra_rubrocapilla_RDO007	F	0.6271393
Ceratopipra_rubrocapilla_TLP025	F	0.6253759
Ceratopipra_rubrocapilla_TLP123	M	0.9603932
Ceratopipra_rubrocapilla_UFAC1007		
Ceratopipra_rubrocapilla_UFAC1687	M	0.7704918
Ceratopipra_rubrocapilla_UFAC305	F	0.6080613
Ceratopipra_rubrocapilla_UFAC476	F	0.6287646
Ceratopipra_rubrocapilla_UHE442	M	0.9719096
Ceratopipra_rubrocapilla_UNA002	F	0.6378416
Ceratopipra_rubrocapilla_UNA014	F	0.6070234
Ceratopipra_rubrocapilla_UNA018	M	0.9911094
Ceratopipra_rubrocapilla_UNA088	M	0.9645088
Ceratopipra_rubrocapilla_UNA110	M	0.989594
Ceratopipra_rubrocapilla_UNA124	M	0.9706357
Ceratopipra_rubrocapilla_WM281	F	0.6178078
Ceratopipra_rubrocapilla_WM293	F	0.6166307
```