# Genotype calling

This page contains the pipeline used to call and filter genotypes, using the mapped bam files.  

I am going to call genotypes for *Ceratopipra rubrocapilla* alone, and then again with the outgroup species, in order to have two datasets: one for population genetics analyses that do not need an outgroup, and a phylogenetic dataset that contains the outgroup. Note that in this repository I refer only to the outgroup *C. erythrocephala* samples. I called additional species at the same time, but have excluded them here as they are for a different unpublished project; they were not used in this project and so are not listed in this repository, though they were present in the original pipeline. I provided the genotype caller with population divisions so that the different species would not influence the genotype calls of each other (otherwise, the genotype caller may alter its genotye calls to attempt to make calls that are more probable under Hardy-Weinberg equilibrium, i.e., making a low-depth homozygote into a heterozygote if its allele is very rare, which it may be in the dataset if it is from a different species).  


First, I need to list all the samples that we will analyze, and their sex, so that bcftools can treat the hemizygous Z chromosome of females correctly as haploid. I will make a list for each species.  


```bash
cat > Cerythrocephala.list
```
```
Ceratopipra_erythrocephala_B7534 M
Ceratopipra_erythrocephala_B7536 M
Ceratopipra_erythrocephala_B7537 M
Ceratopipra_erythrocephala_B7539 M
Ceratopipra_erythrocephala_SGC288 M
```
```bash
cat > Crubrocapilla.list
```
```
Ceratopipra_chloromeros_Fm433681 F
Ceratopipra_rubrocapilla_AMA340 M
Ceratopipra_rubrocapilla_AMA601 M
Ceratopipra_rubrocapilla_AMA656 M
Ceratopipra_rubrocapilla_AMANA034 M
Ceratopipra_rubrocapilla_AMANA137 M
Ceratopipra_rubrocapilla_AMZ551 F
Ceratopipra_rubrocapilla_AMZ610 F
Ceratopipra_rubrocapilla_B4615 M
Ceratopipra_rubrocapilla_B5082 M
Ceratopipra_rubrocapilla_BR033 F
Ceratopipra_rubrocapilla_BR036 M
Ceratopipra_rubrocapilla_BR36440 M
Ceratopipra_rubrocapilla_CAM117 M
Ceratopipra_rubrocapilla_CPE012 M
Ceratopipra_rubrocapilla_CPE025 M
Ceratopipra_rubrocapilla_CPE027 M
Ceratopipra_rubrocapilla_CPE049 F
Ceratopipra_rubrocapilla_CPE066 F
Ceratopipra_rubrocapilla_CPE067 F
Ceratopipra_rubrocapilla_CPE069 M
Ceratopipra_rubrocapilla_CPE071 F
Ceratopipra_rubrocapilla_CPEII008 M
Ceratopipra_rubrocapilla_CUJ027 F
Ceratopipra_rubrocapilla_CUJ072 F
Ceratopipra_rubrocapilla_FPR006 F
Ceratopipra_rubrocapilla_FPR015 M
Ceratopipra_rubrocapilla_FPR021 M
Ceratopipra_rubrocapilla_GAPTO140 F
Ceratopipra_rubrocapilla_GAPTO259 F
Ceratopipra_rubrocapilla_GAPTO268 F
Ceratopipra_rubrocapilla_GAPTO277 M
Ceratopipra_rubrocapilla_GAPTO289 F
Ceratopipra_rubrocapilla_GAPTO327 F
Ceratopipra_rubrocapilla_JRT021 M
Ceratopipra_rubrocapilla_MAM010 M
Ceratopipra_rubrocapilla_MAR092 M
Ceratopipra_rubrocapilla_MELO34 F
Ceratopipra_rubrocapilla_MF005 M
Ceratopipra_rubrocapilla_MPDS628 F
Ceratopipra_rubrocapilla_MPDS776 F
Ceratopipra_rubrocapilla_MSF107 F
Ceratopipra_rubrocapilla_MT087 F
Ceratopipra_rubrocapilla_MT099 F
Ceratopipra_rubrocapilla_MT123 M
Ceratopipra_rubrocapilla_OM022 F
Ceratopipra_rubrocapilla_OM112 M
Ceratopipra_rubrocapilla_OM172 F
Ceratopipra_rubrocapilla_OM215 F
Ceratopipra_rubrocapilla_OP100 F
Ceratopipra_rubrocapilla_PIME024 M
Ceratopipra_rubrocapilla_PIME062 M
Ceratopipra_rubrocapilla_PIME079 F
Ceratopipra_rubrocapilla_PIME134 M
Ceratopipra_rubrocapilla_PIME135 F
Ceratopipra_rubrocapilla_PIME169 F
Ceratopipra_rubrocapilla_PIME460 M
Ceratopipra_rubrocapilla_PIME482 F
Ceratopipra_rubrocapilla_PIME521 M
Ceratopipra_rubrocapilla_PIME522 F
Ceratopipra_rubrocapilla_PPBIO221 F
Ceratopipra_rubrocapilla_PPBIO305 F
Ceratopipra_rubrocapilla_PPBIO312 F
Ceratopipra_rubrocapilla_PPS041 M
Ceratopipra_rubrocapilla_PPS174 F
Ceratopipra_rubrocapilla_PPS221 M
Ceratopipra_rubrocapilla_PPS352 M
Ceratopipra_rubrocapilla_PUC007 M
Ceratopipra_rubrocapilla_PUC008 M
Ceratopipra_rubrocapilla_RDO007 F
Ceratopipra_rubrocapilla_TLP025 F
Ceratopipra_rubrocapilla_TLP123 M
Ceratopipra_rubrocapilla_UFAC1007 M
Ceratopipra_rubrocapilla_UFAC1687 M
Ceratopipra_rubrocapilla_UFAC305 F
Ceratopipra_rubrocapilla_UFAC476 F
Ceratopipra_rubrocapilla_UHE442 M
Ceratopipra_rubrocapilla_UNA002 F
Ceratopipra_rubrocapilla_UNA014 F
Ceratopipra_rubrocapilla_UNA018 M
Ceratopipra_rubrocapilla_UNA088 M
Ceratopipra_rubrocapilla_UNA110 M
Ceratopipra_rubrocapilla_UNA124 M
Ceratopipra_rubrocapilla_WM281 F
Ceratopipra_rubrocapilla_WM293 F
```
(Note: a few of the above samples were excluded downstream and not used in the project due to their extremely low depth - their sequencing failed unfortunately. I have kept them in the list here as a record. These are the failed samples: Ceratopipra_rubrocapilla_UNA014,Ceratopipra_rubrocapilla_CPE027,Ceratopipra_rubrocapilla_CPE066,Ceratopipra_rubrocapilla_CPE067,Ceratopipra_rubrocapilla_CPE071,Ceratopipra_rubrocapilla_GAPTO140,Ceratopipra_rubrocapilla_GAPTO289,Ceratopipra_rubrocapilla_JRT021,Ceratopipra_rubrocapilla_MELO34,Ceratopipra_rubrocapilla_MPDS776,Ceratopipra_rubrocapilla_MT099,Ceratopipra_rubrocapilla_OM215,Ceratopipra_rubrocapilla_PPBIO312,Ceratopipra_rubrocapilla_UFAC1687,Ceratopipra_chloromeros_Fm433681)

Once we have a list of the samples, we need a list of the paths to their bam files.  
```bash
#make a file listing the locations of the bam files for C. rubrocapilla
sed 's/ .*$/.GBSPipfilbwa.bam/g' ./Crubrocapilla.list | sed 's|^|../1.2a_mapping/trimmomaticbwamapped/|g' > Crubrocapilla.bam.list

#combine all samples, including outgroups (this included some other species that were not used in the C. rubrocapilla project, but since they were grouped into separate populations during genotype calling, they won't affect the C. rubrocapilla genotype calls)
cat  Crubrocapilla.list Cmentalis.list Cerythrocephala.list Ccornuta.list Cchloromeros.list > Ceratopipra.list
sed 's/ .*$/.GBSPipfilbwa.bam/g' ./Ceratopipra.list | sed 's|^|../1.2a_mapping/trimmomaticbwamapped/|g' > Ceratopipra.bam.list

cat > ploidy_file
```
We also need a ploidy file that tells bcftools to call the Z chromosome as haploid for females. Note that there is no W chromosome in our reference genome as it was based on a male sample. 
```
chr_Z 1 67472377 F 1
* * * M 2
* * * F 2
```

Next, we can use give files to bcftools are call genotypes. I am generating three VCF files:  
1) *C. rubrocapilla* only (population genetics dataset)  
2) *C. rubrocapilla* and outgroup samples (phylogenetic dataset)  
3) Dataset with female samples treated as diploid on chrZ. This is not correct but will allow us to identify sites that should be excluded: females should never be heterozygous on chrZ outside of the pseudoautosomal region, but may be called as diploid due to genotyping errors, collapsed duplicates, or chrW reads that retain enough homology to chrZ to map to the chrZ. The chrW sequences can severely bias downstream analyses, so I will identify those sites with this dataset and then I can exclude those sites from the properly-called dataset where females are haploid. I put this pipeline lower down the page, in the chrZ dataset section.  

```bash
#set up environment
mkdir -p /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

#call all sites including invariants
time /home/0_PROGRAMS/bcftools-1.17/bcftools mpileup -B -a AD,DP,SP -Ou -f /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa --threads 10 -b Crubrocapilla.bam.list | time /home/0_PROGRAMS/bcftools-1.17/bcftools call --samples-file ./Crubrocapilla.list --ploidy-file ploidy_file --threads 10 -f GQ,GP -m -O z - > Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.vcf.gz #52m58.460s (72m49.524s)
time /home/0_PROGRAMS/bcftools-1.17/bcftools index Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.vcf.gz #index the vcf file

#ignoring 'Note: could not parse as PED:' and verified that it correctly called females as hemizygous on chrZ

#repeat to generate a dataset with the outgroup species
time /home/0_PROGRAMS/bcftools-1.17/bcftools mpileup -B -a AD,DP,SP -Ou -f /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa --threads 10 -b Ceratopipra.bam.list | time /home/0_PROGRAMS/bcftools-1.17/bcftools call --samples-file ./Ceratopipra.list --ploidy-file ploidy_file --group-samples Ceratopipra.groups --threads 10 -f GQ,GP -m -O z - > Ceratopipra.GBSPipfilbwa.allsites.vcf.gz #154m55.565s (234m23.598s)
time /home/0_PROGRAMS/bcftools-1.17/bcftools index Ceratopipra.GBSPipfilbwa.allsites.vcf.gz

```
`bcftool mpileup`: creates genotype likelihoods at each position in the genome, with their depth of coverage
* `-B`: avoid false SNPs by disabling probabilistic realignment
* `-b`: "list of input BAM filenames, one per line"
* `-Ou`: save time by not compressing the VCF file before piping it to the next command
* `-f`: FASTA file of indexed reference sequence
* `-a AD,DP,SP`: annotate the VCF with AD, DP, and SP fields.

`bcftools call`: outputs the VCF file with only variant sites
* `--threads`: specifies number of threads that can be used
* `-m`: use the newer multiallelic caller (opposite of `-c`)
* `-O z`: specifices that the output type is a compressed VCF file


# Evaluate dataset

Before proceeding, we can assess whether any samples failed, and assess a few quality statistics to help with deciding appropriate filters.  

```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

mkdir -p VCF_QC

#Evaluate
#First evaluate how much missing data you have
#missing data per site
taxon=Ceratopipra_rubrocapilla
time /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf "$taxon".GBSPipfilbwa.allsites.vcf.gz --missing-site --out VCF_QC/"$taxon".GBSPipfilbwa.allsites
#Most sites have over 90% missing data, as expected for GBS data. Most sites that worked had less than 10% missing data.

#make a dataset with less missing data in order to evaluate the quality of those sites and decide how best to filter it
mkdir -p maxmissing50
parallel time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "'F_MISSING > 0.50'" -O z {1}.GBSPipfilbwa.allsites.vcf.gz '>' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz ::: Ceratopipra_rubrocapilla Ceratopipra

#make a dataset with the failed samples removed to quantify stats to report that are specific to the samples used in the final rubrocapilla project
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples ^Ceratopipra_rubrocapilla_UNA014,Ceratopipra_rubrocapilla_CPE027,Ceratopipra_rubrocapilla_CPE066,Ceratopipra_rubrocapilla_CPE067,Ceratopipra_rubrocapilla_CPE071,Ceratopipra_rubrocapilla_GAPTO140,Ceratopipra_rubrocapilla_GAPTO289,Ceratopipra_rubrocapilla_JRT021,Ceratopipra_rubrocapilla_MELO34,Ceratopipra_rubrocapilla_MPDS776,Ceratopipra_rubrocapilla_MT099,Ceratopipra_rubrocapilla_OM215,Ceratopipra_rubrocapilla_PPBIO312,Ceratopipra_rubrocapilla_UFAC1687,Ceratopipra_chloromeros_Fm433681 Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > 0.50" -O z > maxmissing50/Ceratopipra_rubrocapilla_70samples.GBSPipfilbwa.allsites.maxmiss50.vcf.gz 

#allele frequencies for bialleleic SNPs
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --freq2 --out VCF_QC/{1}.maxmiss50 --max-alleles 2 ::: Ceratopipra_rubrocapilla Ceratopipra
#mean depth of coverage per sample
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --depth --out VCF_QC/{1}.maxmiss50 ::: Ceratopipra_rubrocapilla_70samples Ceratopipra_rubrocapilla Ceratopipra

#mean depth of coverage per site
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --site-mean-depth --out VCF_QC/{1}.maxmiss50 ::: Ceratopipra_rubrocapilla_70samples Ceratopipra_rubrocapilla Ceratopipra
#site quality scores
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --site-quality --out VCF_QC/{1}.maxmiss50 ::: Ceratopipra_rubrocapilla Ceratopipra
#missing data
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --missing-indv --out VCF_QC/{1}.maxmiss50 ::: Ceratopipra_rubrocapilla Ceratopipra
#missing data per site
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --missing-site --out VCF_QC/{1}.maxmiss50 ::: Ceratopipra_rubrocapilla Ceratopipra
#heterozygosity
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --het --out VCF_QC/{1}.maxmiss50 ::: Ceratopipra_rubrocapilla Ceratopipra

#Then, I am checking the relatedness of my samples. I hope I don't have close relatives!
#Let's check if any of our samples are likely to be related. Hopefully not!
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz --relatedness2 --out VCF_QC/{1}.maxmiss50.relatedness2 ::: Ceratopipra_rubrocapilla_70samples Ceratopipra_rubrocapilla Ceratopipra #this will give us a good indication of relatedness. It uses KING inference of relatedness (Manichaikul et al., BIOINFORMATICS 2010 (doi:10.1093/bioinformatics/btq559))
```

**Statistics to report in manuscript**:  
I will report the number of bases sequenced and their average coverage, from the maxmiss50 dataset that includes sites where at least half of the 70 samples were genotyped. `wc -l Ceratopipra_rubrocapilla_70samples.maxmiss50.ldepth.mean` = 12447782 bp = 12.4 Mb. `zgrep -c -v "^#" Ceratopipra_rubrocapilla_70samples.GBSPipfilbwa.allsites.maxmiss50.vcf.gz` = 12447781 bp = 12.4 Mb (those two counts are off by 1 due to the header line of the ldepth.mean file).  
I will also report the average depth per sample across the covered bases: `tail -n +2 Ceratopipra_rubrocapilla_70samples.maxmiss50.ldepth.mean | awk '{ sum += $3; n++ } END { if (n > 0) print sum / n; }'` = 9.42767 = average of 9.4x coverage per sample per site.  

**Evaluating relatedness**:  
A few relatedness values are higher than 0.01:  
```
#When including only the 70 samples used in this project:  
Ceratopipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_UHE442 8007    3529    32867   56682   0.0105975
Ceratopipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_AMZ610 8007    3529    56682   32867   0.0105975
Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_WM293  10230   4296    46164   64339   0.0148231
Ceratopipra_rubrocapilla_WM293  Ceratopipra_rubrocapilla_AMZ551 10230   4296    64339   46164   0.0148231
Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_UHE442 9639    3875    46164   56682   0.0183673
Ceratopipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_AMZ551 9639    3875    56682   46164   0.0183673
Ceratopipra_rubrocapilla_PPBIO221       Ceratopipra_rubrocapilla_WM293  10684   4419    34476   64339   0.0186814
Ceratopipra_rubrocapilla_WM293  Ceratopipra_rubrocapilla_PPBIO221       10684   4419    64339   34476   0.0186814
Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_AMZ610 7037    2750    46164   32867   0.0194481
Ceratopipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_AMZ551 7037    2750    32867   46164   0.0194481
Ceratopipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_WM293  14623   5812    56682   64339   0.0247808
Ceratopipra_rubrocapilla_WM293  Ceratopipra_rubrocapilla_UHE442 14623   5812    64339   56682   0.0247808
Ceratopipra_rubrocapilla_UNA088 Ceratopipra_rubrocapilla_UNA124 13382   5414    47962   47582   0.0267311
Ceratopipra_rubrocapilla_UNA124 Ceratopipra_rubrocapilla_UNA088 13382   5414    47582   47962   0.0267311
Ceratopipra_rubrocapilla_PUC007 Ceratopipra_rubrocapilla_PUC008 20481   8520    61961   64867   0.0271312
Ceratopipra_rubrocapilla_PUC008 Ceratopipra_rubrocapilla_PUC007 20481   8520    64867   61961   0.0271312


#When including some failed samples that were not used in the project, just to see:  
Ceratopipra_rubrocapilla_PPBIO312       Ceratopipra_rubrocapilla_UHE442 2724    1044    9252    53617   0.0101163
Ceratopipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_PPBIO312       2724    1044    53617   9252    0.0101163
Ceratopipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_UHE442 7888    3435    31545   53617   0.0119537
Ceratopipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_AMZ610 7888    3435    53617   31545   0.0119537
Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_WM293  9960    4109    44184   58910   0.0168972
Ceratopipra_rubrocapilla_WM293  Ceratopipra_rubrocapilla_AMZ551 9960    4109    58910   44184   0.0168972
Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_UHE442 9460    3745    44184   53617   0.0201429
Ceratopipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_AMZ551 9460    3745    53617   44184   0.0201429
Ceratopipra_rubrocapilla_PPBIO221       Ceratopipra_rubrocapilla_WM293  10351   4250    32850   58910   0.0201722
Ceratopipra_rubrocapilla_WM293  Ceratopipra_rubrocapilla_PPBIO221       10351   4250    58910   32850   0.0201722
Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_AMZ610 6935    2701    44184   31545   0.0202432
Ceratopipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_AMZ551 6935    2701    31545   44184   0.0202432
Ceratopipra_chloromeros_Fm433681        Ceratopipra_rubrocapilla_MT099  1987    192     56680   6597    0.0253331
Ceratopipra_rubrocapilla_MT099  Ceratopipra_chloromeros_Fm433681        1987    192     6597    56680   0.0253331
Ceratopipra_rubrocapilla_UNA088 Ceratopipra_rubrocapilla_UNA124 13046   5253    45337   44460   0.028286
Ceratopipra_rubrocapilla_UNA124 Ceratopipra_rubrocapilla_UNA088 13046   5253    44460   45337   0.028286
Ceratopipra_rubrocapilla_GAPTO140       Ceratopipra_rubrocapilla_MELO34 11      0       139     237     0.0292553
Ceratopipra_rubrocapilla_MELO34 Ceratopipra_rubrocapilla_GAPTO140       11      0       237     139     0.0292553
Ceratopipra_rubrocapilla_UHE442 Ceratopipra_rubrocapilla_WM293  14142   5407    53617   58910   0.0295751
Ceratopipra_rubrocapilla_WM293  Ceratopipra_rubrocapilla_UHE442 14142   5407    58910   53617   0.0295751
Ceratopipra_rubrocapilla_PUC007 Ceratopipra_rubrocapilla_PUC008 18830   7685    55985   58170   0.0303097
Ceratopipra_rubrocapilla_PUC008 Ceratopipra_rubrocapilla_PUC007 18830   7685    58170   55985   0.0303097
Ceratopipra_chloromeros_Fm433681        Ceratopipra_rubrocapilla_FPR006 36691   586     56680   47396   0.341279
Ceratopipra_rubrocapilla_FPR006 Ceratopipra_chloromeros_Fm433681        36691   586     47396   56680   0.341279
```
It looks like Ceratopipra_rubrocapilla_PUC008 and Ceratopipra_rubrocapilla_PUC007 are somewhat related, with a slightly elevated relatedness phi. These samples are from the same site, so it is plausible that they could be relatives (eg., second cousins, etc). To be conservative, I will remove one for the PCoA, STRUCTURE, and estimates of genetic diversity so that their relatedness does not drive a PC axis or bias diversity downwards. A few other sample pairs have slightly elevated kinship coefficients, but were not collected back-to-back; I suspect that these may be more likely due to population structure or shared biases than close kinship. (when there is population structure within the dataset, samples from the same population will have higher than expected chance of sharing alleles based on an expectation of a single panmictic population, and therefore they will have elevated kinship coefficient)

# Filtering 
Below is the pipeline for filtering the dataset.

## Datasets without outgroups
The first set of datasets are for analyses that do not require an outgroup.  

### Dataset for pi and dxy

First, I will filter the Population genetics dataset that has only *C. rubrocapilla* without the outgroup samples. To make my pipeline for flexible, I set it up to read an input text file that lists various filters to implement. This way I can reuse the pipeline for different datasets that need different filters. This file has 6 space-delimited columns:
1) Tagname to identify which input VCF to use (Ceratopipra_rubrocapilla or Ceratopipra).  
2) max average depth per site  
3) comma-separated list of samples to exclude (failed samples, and Ceratopipra_rubrocapilla_PUC007 for analyses sensitive to the inclusion of sample pairs that are close kin)  
4) max missingness per site  
5) min sequencing depth per individual genotype  
6) min genotype quality per individual genotype  

```bash
cat > filters
```
```
Ceratopipra_rubrocapilla 37 Ceratopipra_rubrocapilla_UNA014,Ceratopipra_rubrocapilla_CPE027,Ceratopipra_rubrocapilla_CPE066,Ceratopipra_rubrocapilla_CPE067,Ceratopipra_rubrocapilla_CPE071,Ceratopipra_rubrocapilla_GAPTO140,Ceratopipra_rubrocapilla_GAPTO289,Ceratopipra_rubrocapilla_JRT021,Ceratopipra_rubrocapilla_MELO34,Ceratopipra_rubrocapilla_MPDS776,Ceratopipra_rubrocapilla_MT099,Ceratopipra_rubrocapilla_OM215,Ceratopipra_rubrocapilla_PPBIO312,Ceratopipra_rubrocapilla_UFAC1687,Ceratopipra_chloromeros_Fm433681,Ceratopipra_rubrocapilla_PUC007 0.2 5 20.0
```

First, I am evaluating the distribution of sequencing depth in order to decide an appropriate max depth cutoff. The purpose of this filter is to remove sites that are collapsed duplicates (such as repetitive elements) where paralogs both map to the same position and could therefore lead to a SNP that is actually a difference between paralogs, which could have unpredictable impacts on various analyses. Since we do not know how many of these cases exist in our dataset, and sequencing data varies across the genome due to other biases, it is difficult to objectively choose a cutoff, but something that is higher than the mean depth and lower than twice the mean and removes a few percent of the dataset is probably appropriate. I like incorporating the square root of the mean with the formula `max_depth  = mean_depth + 2*sqrt(mean_depth)`. This will be more lenient on low depth datasets where we expect a broader distribution of depths approaching double the mean, whereas higher depth datasets should not approach double the mean unless caused by collapsed duplicates (or the existence of strong biases in sequencing). (For example, with a mean depth of 100, a max depth of double (200) would probably be too lenient, but with a mean depth of 3, a max depth of double (6) might be too strict). 
```bash
#first, filter dataset without filtering for max depth cutoff
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m1 -M1 -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<5'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > 0.25'\'' -Oz -o {1}.GBSPipfilbwa.allsites.invariant.for_pixy.nomax.auto.vcf.gz' #7m58.584s

#decide on max depth cutoff based on the distribution of mean depth after removing the failed samples. Place this value in column to 2 of the filters file.
parallel /home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf {1}.GBSPipfilbwa.allsites.invariant.for_pixy.nomax.auto.vcf.gz --site-mean-depth --out VCF_QC/{1}.invariant.for_pixy.nomax.auto ::: Ceratopipra_rubrocapilla
awk 'NR>1 { sum += $3 } END { print sum / (NR-1) }' VCF_QC/Ceratopipra_rubrocapilla.invariant.for_pixy.nomax.auto.ldepth.mean
#mean depth: 26.7067
#26.7067 + 2*sqrt(26.7067) = 37.04241

#I will use a max depth filter of 37x for the population genetics dataset.
```

After choosing filters, I will pass my filters file to bcftools. My first set of filters use equivalent filters for both variant and invariant sites to avoid biasing the estimates of diversity by filtering the SNPs more stringently than invariant sites.
These datasets have:  
* biallelic SNPs and/or invariant sites  
* autosomes only  
* remove excluded samples  
* min individual depth 5X (to reduce missed heterozygotes or false heterozygotes)  
* max average depth less than `mean_depth + 3*sqrt(mean_depth)` (to reduce collapsed duplicates being mistaken for SNPs)  
* min mapping quality 25 (to reduce false SNPs due to mismapping)  
* max missingness 20%  


```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

mkdir -p fairfilters
#first, filter the autosomal dataset

#filter invariant sites
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m1 -M1 -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #3m for rubricapilla #9m20.095s (20m18.336s) for all combined

#repeat to filter biallelic SNPs. Do not yet apply GQ filters, as these cannot be applied to invariants the same way and would bias pi/dxy downwards depending on sequencing depth
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m2 -M2 -v snps -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #6m for rubricapilla #9m20.095s (20m18.336s) for all combined

#merge invariants with SNPs
conda activate basic
#*****This dataset is used for pi and dxy*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools concat --threads 30 --allow-overlaps fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz -O z -o ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #204m (360m)

```

### Dataset for PCoA, STRUCTURE
My next dataset is filtered more heavily, thinned for linkage, and contains only biallelic SNPs.  

For the PCoA we want the following filters:  

These are already applied for the pixy dataset:  
* biallelic SNPs only  
* autosomes only, no unplaced scaffolds  
* min mapping quality 25 (to reduce false SNPs due to mismapping)  
* max average depth less than `mean_depth + 3*sqrt(mean_depth)` (to reduce collapsed duplicates being mistaken for SNPs)  
* min individual depth 5X (to reduce missed heterozygotes or false heterozygotes)  
* max missing data 20%  

These now need to be applied:  
* min site variant quality 25 and individual genotype quality 20 (to reduce genotype errors)   
* no singletons or rare alleles (with allele count < 4, i.e. 4 heterozygotes or 2 homozygotes) (they are not informative)  
* remove sites with excess heterozygosity  
* each SNP spaced at least 10 kb apart (reduce linkage)    
* max missing data 20% (recalculated after adding the above filters)  

Next, we will filter the SNPs more heavily
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping
mkdir -p filteredSNPs

#This dataset is for PCoA, and STRUCTURE: biallelic autosomal SNPs, min qual 25, min depth 5 (per sample), 
mkdir -p filteredSNPs/thinned_datasets

#Raising AC limit to 4
#*****This dataset is used for structure and pcoa*****
parallel --colsep " " 'seed=$RANDOM ; echo "Random seed: "$seed ; time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC<=4 || AC>=AN-4 || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" | /home/0_PROGRAMS/bcftools-1.17/bcftools +prune --nsites-per-win 1 -w 10000bp --nsites-per-win-mode "rand" --random-seed "$seed" -O z - 2> filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.{2}_{4}_{5}_{6}.vcf.gz' :::: filters
parallel --colsep " " 'echo {1}_{2}_{4}_{5}_{6} ; zgrep -v "^#" filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.{2}_{4}_{5}_{6}.vcf.gz | wc -l'  :::: filters
#Ceratopipra_rubrocapilla_37_0.2_5_20.0: 7865

```

### Dataset for Fst, Tajima's D
These datasets are similar to the above, but are not thinned for linkage, and need careful attention to the MAF filter (no MAF filter should go into Tajima's D or it will severely bias the results).  
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

#the dataset for Fst will not be thinned for linkage, but we will filter out sites with low minor allele frequency (make sure not to compare values to Fst estimates from datasets where this MAF filter was not implemented)
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || AC==1 || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.gz' #0m19.856s
#add MAF filter
#*****This dataset is used for fst*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''MAF<0.05'\'' -Oz -o filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.maf5.vcf.gz filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.gz'
cat filters | parallel --colsep " " time tabix filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.maf5.vcf.gz
zgrep -v "^#" filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.4_5_20.0.maf5.vcf.gz | wc -l #40514

# now repeat but with no MAF filter (for Tajima's D)
#*****This dataset is used for tajima's D*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.{2}_{4}_{5}_{6}.vcf.gz' #0m19.856s
cat filters | parallel --colsep " " time tabix filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.{2}_{4}_{5}_{6}.vcf.gz
```

# chromosome Z filters

We need to use slightly different strategies for chromosome Z to accommodate hemizygous female genotypes.

First, we will re-call females as diploid on the Z chromosome, using individual (vs group) genotyping (setting: `--group-samples - `) to avoid assumptions of Hardy Weinberg Equilibrium from influences genotype calls. Then we can identify sites to exclude as sites where any 2 or more females are called as heterozygous with minimum genotype quality of 10.    
```bash
#Any two females
cd /home/1_Else/Ceratopipra/1_Datasets/1.3c_genotyping #keeping in a different folder since it is using different settings that I don't want to mix up

#repeat, treating chr_Z as diploid (incorrect for females)
time /home/0_PROGRAMS/bcftools-1.17/bcftools mpileup --regions chr_Z -B -a AD,DP,SP,QS -Ou -f /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa --threads 10 -b Ceratopipra.bam.list | time /home/0_PROGRAMS/bcftools-1.17/bcftools call --group-samples - --threads 10 -f GQ,GP -m -O z - > Ceratopipra.GBSPipfilbwa.chr_Z.diploid.G0.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index Ceratopipra.GBSPipfilbwa.chr_Z.diploid.G0.vcf.gz #6m25.011s
#filter for max 50% missing data
parallel time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "'F_MISSING > 0.50'" -O z {2}.GBSPipfilbwa.chr_Z.diploid.{1}.vcf.gz '>' maxmissing50/{2}.GBSPipfilbwa.chr_Z.diploid.maxmiss50.{1}.vcf.gz ::: G0 ::: Ceratopipra #2m22.790s

#filter to keep only females, then identify sites where any 2+ samples are heterozygous
#require GQ>10 for genotypes to exclude spurious heterozygous calls
time /home/0_PROGRAMS/bcftools-1.17/bcftools view -v snps --samples Ceratopipra_rubrocapilla_CPE049,Ceratopipra_rubrocapilla_UNA002,Ceratopipra_rubrocapilla_MT087,Ceratopipra_rubrocapilla_OM022,Ceratopipra_rubrocapilla_OM172,Ceratopipra_rubrocapilla_UFAC305,Ceratopipra_rubrocapilla_CUJ027,Ceratopipra_rubrocapilla_CUJ072,Ceratopipra_rubrocapilla_UFAC476,Ceratopipra_rubrocapilla_FPR006,Ceratopipra_rubrocapilla_MPDS628,Ceratopipra_rubrocapilla_PIME079,Ceratopipra_rubrocapilla_PIME169,Ceratopipra_rubrocapilla_AMZ551,Ceratopipra_rubrocapilla_AMZ610,Ceratopipra_rubrocapilla_OP100,Ceratopipra_rubrocapilla_BR033,Ceratopipra_rubrocapilla_PIME135,Ceratopipra_rubrocapilla_PIME482,Ceratopipra_rubrocapilla_PIME522,Ceratopipra_rubrocapilla_PPS174,Ceratopipra_rubrocapilla_TLP025,Ceratopipra_rubrocapilla_WM293,Ceratopipra_rubrocapilla_GAPTO259,Ceratopipra_rubrocapilla_GAPTO268,Ceratopipra_rubrocapilla_GAPTO327,Ceratopipra_rubrocapilla_RDO007,Ceratopipra_rubrocapilla_PPBIO221,Ceratopipra_rubrocapilla_PPBIO305 maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.maxmiss50.G0.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<10" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC_Het | /home/0_PROGRAMS/bcftools-1.17/bcftools view -i 'AC_Het > 1' -Oz -o maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.maxmiss50.G0.females.AC_Het2.gq10.vcf.gz 
zgrep -v "^#" maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.maxmiss50.G0.females.AC_Het2.gq10.vcf.gz | wc -l #1490 sites (for comparison: would be 1982 sites if using group vs individual genotype calling)
#get a list of sites to exclude
mkdir -p sites_to_exclude
zgrep -v "^#" maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.maxmiss50.G0.females.AC_Het2.gq10.vcf.gz | cut -f 1-2 > sites_to_exclude/chrZ_2heterozygousP0Females.tsv

#evaluate depth on chrZ to help to identify pseudoautosomal region
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples Ceratopipra_rubrocapilla_CPE049,Ceratopipra_rubrocapilla_UNA002,Ceratopipra_rubrocapilla_MT087,Ceratopipra_rubrocapilla_OM022,Ceratopipra_rubrocapilla_OM172,Ceratopipra_rubrocapilla_UFAC305,Ceratopipra_rubrocapilla_CUJ027,Ceratopipra_rubrocapilla_CUJ072,Ceratopipra_rubrocapilla_UFAC476,Ceratopipra_rubrocapilla_FPR006,Ceratopipra_rubrocapilla_MPDS628,Ceratopipra_rubrocapilla_PIME079,Ceratopipra_rubrocapilla_PIME169,Ceratopipra_rubrocapilla_AMZ551,Ceratopipra_rubrocapilla_AMZ610,Ceratopipra_rubrocapilla_OP100,Ceratopipra_rubrocapilla_BR033,Ceratopipra_rubrocapilla_PIME135,Ceratopipra_rubrocapilla_PIME482,Ceratopipra_rubrocapilla_PIME522,Ceratopipra_rubrocapilla_PPS174,Ceratopipra_rubrocapilla_TLP025,Ceratopipra_rubrocapilla_WM293,Ceratopipra_rubrocapilla_GAPTO259,Ceratopipra_rubrocapilla_GAPTO268,Ceratopipra_rubrocapilla_GAPTO327,Ceratopipra_rubrocapilla_RDO007,Ceratopipra_rubrocapilla_PPBIO221,Ceratopipra_rubrocapilla_PPBIO305 maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.maxmiss50.G0.vcf.gz -Oz -o maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.females.maxmiss50.G0.vcf.gz

#get heterozygosity per site (script from Kevin Blighe: https://www.biostars.org/p/291147/)
time paste <(/home/0_PROGRAMS/bcftools-1.17/bcftools view maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.females.maxmiss50.G0.vcf.gz | awk -F"\t" 'BEGIN {print "CHR\tPOS\tID\tREF\tALT"} !/^#/ {print $1"\t"$2"\t"$3"\t"$4"\t"$5}') <(/home/0_PROGRAMS/bcftools-1.17/bcftools query -f '[\t%SAMPLE=%GT]\n' maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.females.maxmiss50.G0.vcf.gz | awk 'BEGIN {print "nHet"} {print gsub(/0\|1|1\|0|0\/1|1\/0/, "")}') > VCF_QC/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.females.maxmiss50.G0.heterozygosity
#mean depth per site
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.females.maxmiss50.G0.vcf.gz --site-mean-depth --out VCF_QC/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.females.maxmiss50.G0

#repeat for males to compare
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples Ceratopipra_rubrocapilla_AMA340,Ceratopipra_rubrocapilla_AMA601,Ceratopipra_rubrocapilla_AMA656,Ceratopipra_rubrocapilla_AMANA034,Ceratopipra_rubrocapilla_AMANA137,Ceratopipra_rubrocapilla_B4615,Ceratopipra_rubrocapilla_B5082,Ceratopipra_rubrocapilla_BR036,Ceratopipra_rubrocapilla_BR36440,Ceratopipra_rubrocapilla_CAM117,Ceratopipra_rubrocapilla_CPE012,Ceratopipra_rubrocapilla_CPE025,Ceratopipra_rubrocapilla_CPE069,Ceratopipra_rubrocapilla_CPEII008,Ceratopipra_rubrocapilla_FPR015,Ceratopipra_rubrocapilla_FPR021,Ceratopipra_rubrocapilla_GAPTO277,Ceratopipra_rubrocapilla_MAM010,Ceratopipra_rubrocapilla_MAR092,Ceratopipra_rubrocapilla_MF005,Ceratopipra_rubrocapilla_MT123,Ceratopipra_rubrocapilla_OM112,Ceratopipra_rubrocapilla_PIME024,Ceratopipra_rubrocapilla_PIME062,Ceratopipra_rubrocapilla_PIME134,Ceratopipra_rubrocapilla_PIME460,Ceratopipra_rubrocapilla_PIME521,Ceratopipra_rubrocapilla_PPS041,Ceratopipra_rubrocapilla_PPS221,Ceratopipra_rubrocapilla_PPS352,Ceratopipra_rubrocapilla_PUC007,Ceratopipra_rubrocapilla_PUC008,Ceratopipra_rubrocapilla_TLP123,Ceratopipra_rubrocapilla_UFAC1007,Ceratopipra_rubrocapilla_UHE442,Ceratopipra_rubrocapilla_UNA018,Ceratopipra_rubrocapilla_UNA088,Ceratopipra_rubrocapilla_UNA110,Ceratopipra_rubrocapilla_UNA124 maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.maxmiss50.G0.vcf.gz -Oz -o maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.males.maxmiss50.G0.vcf.gz

#get heterozygosity per site (script from Kevin Blighe: https://www.biostars.org/p/291147/)
time paste <(/home/0_PROGRAMS/bcftools-1.17/bcftools view maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.males.maxmiss50.G0.vcf.gz | awk -F"\t" 'BEGIN {print "CHR\tPOS\tID\tREF\tALT"} !/^#/ {print $1"\t"$2"\t"$3"\t"$4"\t"$5}') <(/home/0_PROGRAMS/bcftools-1.17/bcftools query -f '[\t%SAMPLE=%GT]\n' maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.males.maxmiss50.G0.vcf.gz | awk 'BEGIN {print "nHet"} {print gsub(/0\|1|1\|0|0\/1|1\/0/, "")}') > VCF_QC/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.males.maxmiss50.G0.heterozygosity
#mean depth per site
/home/0_PROGRAMS/vcftools/bin/vcftools --gzvcf maxmissing50/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.males.maxmiss50.G0.vcf.gz --site-mean-depth --out VCF_QC/Ceratopipra.GBSPipfilbwa.chr_Z.diploid.males.maxmiss50.G0


```

Now, we have a list of sites that we want to exclude from downstream analyses due to those sites having elevated female heterozygosity.

Then, we will implement filters that are similar to the above filters for autosomes, but allowing a lower min genotype depth (3x) to allow the lower depth of hemizygous females. 
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping 
cat > filters
```
```
#as above, but lower min depth filter for chrZ to accommodate low-depth females (which are hemizygous for the Z)
Ceratopipra_rubrocapilla 37 Ceratopipra_rubrocapilla_UNA014,Ceratopipra_rubrocapilla_CPE027,Ceratopipra_rubrocapilla_CPE066,Ceratopipra_rubrocapilla_CPE067,Ceratopipra_rubrocapilla_CPE071,Ceratopipra_rubrocapilla_GAPTO140,Ceratopipra_rubrocapilla_GAPTO289,Ceratopipra_rubrocapilla_JRT021,Ceratopipra_rubrocapilla_MELO34,Ceratopipra_rubrocapilla_MPDS776,Ceratopipra_rubrocapilla_MT099,Ceratopipra_rubrocapilla_OM215,Ceratopipra_rubrocapilla_PPBIO312,Ceratopipra_rubrocapilla_UFAC1687,Ceratopipra_chloromeros_Fm433681,Ceratopipra_rubrocapilla_PUC007 0.2 3 20.0

```

```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping 

#get list of sites to exclude from other folder so that everything is together in this folder
mkdir -p sites_to_exclude
cp ../1.3c_genotyping/sites_to_exclude/chrZ_2heterozygousP0Females.tsv ./sites_to_exclude

#remove the pseudoautosomal region, which I found is probably located within the first 3,000,000 bp of the chromosome (this is a very conservative exaggeration of its extent - it is probably much smaller but I would rather remove too much than too little) (I identified it by looking at sequencing depth and heterozygosity of females, and Fst between males and females)

#filter invariant sites
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m1 -M1 -t chr_Z:3000000- --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz' #2m32.604s
#make a dataset with females diploidized
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz | sed -r -e "s@\t([0-9.]):@\t\1/\1:@g" | /home/0_PROGRAMS/bcftools-1.17/bcftools view -Oz -o ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz; time tabix ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz'
#remove sites flagged as heterozygous in a female with GQ>10 (chrW mapping to chrZ, or other collapsed paralogs/repeats)
cat filters | parallel --colsep " " '/home/0_PROGRAMS/bcftools-1.17/bcftools view --targets-file ^sites_to_exclude/chrZ_2heterozygousP0Females.tsv ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz -Oz -o ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz'
cat filters | parallel --colsep " " 'time tabix ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz'

#repeat to filter biallelic SNPs. Do not yet apply GQ filters, as these cannot be applied to invariants the same way and would bias pi/dxy downwards depending on sequencing depth
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m2 -M2 -v snps -t chr_Z:3000000- --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz ; /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz' #2m29.965s
#make a dataset with females diploidized
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz | sed -r -e "s@\t([0-9.]):@\t\1/\1:@g" | /home/0_PROGRAMS/bcftools-1.17/bcftools view -Oz -o ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz; time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz'
#remove sites flagged as heterozygous in a female with GQ>10 (chrW mapping to chrZ, or other collapsed paralogs/repeats)
cat filters | parallel --colsep " " '/home/0_PROGRAMS/bcftools-1.17/bcftools view --targets-file ^sites_to_exclude/chrZ_2heterozygousP0Females.tsv ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz -Oz -o ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz'
cat filters | parallel --colsep " " 'time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz'


#combine SNPs and invariants
conda activate basic
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools concat --threads 30 --allow-overlaps ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz -O z -o ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz ; time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz' #204m (360m)
zgrep -c -v "^#" fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.37_0.2_3_20.0.vcf.gz #73474 sites
```

Now we will filter chrZ for the Fst estimation just as we did for the autosomes.
```bash
#next repeat for the Z chromosome
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || AC==1 || AC>=AN-1 || QUAL<25'\'' ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz' #
#*****This dataset is used for chr Z fst*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.vcf.gz -- -t AC,AN | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''MAF<0.05'\'' -Oz -o filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.maf5.vcf.gz - '
cat filters | parallel --colsep " " time tabix ./filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmedP0.{2}_{4}_{5}_{6}.maf5.vcf.gz 
zgrep -c -v "^#" filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.37_0.2_3_20.0.maf5.vcf.gz #310
zgrep -c -v "^#" filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmedP0.37_0.2_3_20.0.maf5.vcf.gz #325
```

# Datasets with outgroups

Next, I will filter the phylogeneti genetics dataset that has *C. rubrocapilla* and the outgroup samples. To make my pipeline for flexible, I set it up to read an input text file that lists various filters to implement. This way I can reuse the pipeline for different datasets that need different filters. This file has 6 space-delimited columns:
1) Tagname to identify which input VCF to use (Ceratopipra_rubrocapilla or Ceratopipra).  
2) max average depth per site  
3) comma-separated list of samples to *keep* (the filter file above instead specified samples to exclude, but now I find it nicer for my records to instead state which sample to include)  
4) max missingness per site  
5) min sequencing depth per individual genotype  
6) min genotype quality per individual genotype  

```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping
cat > filters
```
```
Ceratopipra_rubrocapilla_imeriout_m07 49 Ceratopipra_erythrocephala_SGC288,Ceratopipra_erythrocephala_B7534,Ceratopipra_erythrocephala_B7539,Ceratopipra_erythrocephala_B7536,Ceratopipra_erythrocephala_B7537,Ceratopipra_rubrocapilla_CPE012,Ceratopipra_rubrocapilla_CPE025,Ceratopipra_rubrocapilla_CPE049,Ceratopipra_rubrocapilla_CPE069,Ceratopipra_rubrocapilla_CPEII008,Ceratopipra_rubrocapilla_UNA002,Ceratopipra_rubrocapilla_UNA018,Ceratopipra_rubrocapilla_UNA088,Ceratopipra_rubrocapilla_UNA110,Ceratopipra_rubrocapilla_UNA124,Ceratopipra_rubrocapilla_AMA340,Ceratopipra_rubrocapilla_AMA601,Ceratopipra_rubrocapilla_AMA656,Ceratopipra_rubrocapilla_B4615,Ceratopipra_rubrocapilla_B5082,Ceratopipra_rubrocapilla_CAM117,Ceratopipra_rubrocapilla_CUJ027,Ceratopipra_rubrocapilla_CUJ072,Ceratopipra_rubrocapilla_OM022,Ceratopipra_rubrocapilla_OM112,Ceratopipra_rubrocapilla_OM172,Ceratopipra_rubrocapilla_PUC008,Ceratopipra_rubrocapilla_PUC007,Ceratopipra_rubrocapilla_UFAC1007,Ceratopipra_rubrocapilla_UFAC305,Ceratopipra_rubrocapilla_UFAC476,Ceratopipra_rubrocapilla_AMZ610,Ceratopipra_rubrocapilla_FPR006,Ceratopipra_rubrocapilla_FPR015,Ceratopipra_rubrocapilla_FPR021,Ceratopipra_rubrocapilla_MAM010,Ceratopipra_rubrocapilla_MAR092,Ceratopipra_rubrocapilla_MPDS628,Ceratopipra_rubrocapilla_OP100,Ceratopipra_rubrocapilla_PIME024,Ceratopipra_rubrocapilla_PIME079,Ceratopipra_rubrocapilla_PIME460,Ceratopipra_rubrocapilla_BR033,Ceratopipra_rubrocapilla_BR036,Ceratopipra_rubrocapilla_MF005,Ceratopipra_rubrocapilla_PIME062,Ceratopipra_rubrocapilla_PIME134,Ceratopipra_rubrocapilla_PIME135,Ceratopipra_rubrocapilla_PIME169,Ceratopipra_rubrocapilla_PIME482,Ceratopipra_rubrocapilla_PIME522,Ceratopipra_rubrocapilla_PPS041,Ceratopipra_rubrocapilla_PPS174,Ceratopipra_rubrocapilla_PPS221,Ceratopipra_rubrocapilla_TLP025,Ceratopipra_rubrocapilla_TLP123,Ceratopipra_rubrocapilla_WM293,Ceratopipra_rubrocapilla_AMZ551,Ceratopipra_rubrocapilla_GAPTO259,Ceratopipra_rubrocapilla_GAPTO268,Ceratopipra_rubrocapilla_GAPTO277,Ceratopipra_rubrocapilla_GAPTO327,Ceratopipra_rubrocapilla_MT087,Ceratopipra_rubrocapilla_MT123,Ceratopipra_rubrocapilla_PPBIO221,Ceratopipra_rubrocapilla_PPBIO305,Ceratopipra_rubrocapilla_RDO007,Ceratopipra_rubrocapilla_UHE442 0.2 5 20
```

```bash
#after deciding on filters, filter invariant sites. These will be included with IQTree2, but not analyses that use only SNPs (eg., splitstree, ABBA BABA test)
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m1 -M1 -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples {3} -e '\''MQ<25'\'' maxmissing50/Ceratopipra.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #3m for rubricapilla #9m20.095s (20m18.336s) for all combined

#repeat to filter biallelic SNPs. Do not yet apply GQ filters, as these cannot be applied to invariants the same way and would bias pi/dxy downwards depending on sequencing depth. This is to avoid biasing branchlengths in IQtree2.
#filter SNPs with fair filters
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m2 -M2 -v snps -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples {3} -e '\''MQ<25'\'' maxmissing50/Ceratopipra.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #6m for rubricapilla #9m20.095s (20m18.336s) for all combined

#merge invariants with SNPs
#***This dataset is for IQtree2***
conda activate basic
cat filters | parallel --colsep " " time /home/0_PROGRAMS/bcftools-1.17/bcftools concat --threads 30 --allow-overlaps fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz -O z -o ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz #1m26.367s
cat filters | parallel --colsep " " time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz

#then, filter SNPs more strictly for analyses that do not need invariant sites, where we are not worried about biasing the relative proportion of variant vs invariant sites
#not thinned
#***This dataset is for Dsuite (ABBA BABA test) and Splitstree***
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || AC==1 || AC==AN-1 || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.gz' #0m19.856s 

#thin 1 SNP per 10 kb to reduce the effects of linkage. 
#**This dataset is used for SVDQuartets**
cat filters | parallel --colsep " " 'seed=$RANDOM ; echo "Random seed: "$seed ; time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet,AC,AN | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01 || AC==0 || AC==AN || AC==1" | /home/0_PROGRAMS/bcftools-1.17/bcftools +prune --nsites-per-win 1 -w 10000bp --nsites-per-win-mode "rand" --random-seed "$seed" -O z - 2> filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.{2}_{4}_{5}_{6}.vcf.gz' #0m22.314s


```

# Fasta alignment  

This section converts the VCF file to an alignment in fasta format.  
For a couple analyses, I want alignments in fasta format with invariants and variant sites. This will be used for the concatenated Maximum Likelihood tree with IQtree2.  

First, make a list of sample names to loop through.  
```bash
cat > samples 
```
```
Ceratopipra_erythrocephala_SGC288
Ceratopipra_erythrocephala_B7534
Ceratopipra_erythrocephala_B7539
Ceratopipra_erythrocephala_B7536
Ceratopipra_erythrocephala_B7537
Ceratopipra_rubrocapilla_CPE012
Ceratopipra_rubrocapilla_CPE025
Ceratopipra_rubrocapilla_CPE049
Ceratopipra_rubrocapilla_CPE069
Ceratopipra_rubrocapilla_CPEII008
Ceratopipra_rubrocapilla_UNA002
Ceratopipra_rubrocapilla_UNA018
Ceratopipra_rubrocapilla_UNA088
Ceratopipra_rubrocapilla_UNA110
Ceratopipra_rubrocapilla_UNA124
Ceratopipra_rubrocapilla_AMA340
Ceratopipra_rubrocapilla_AMA601
Ceratopipra_rubrocapilla_AMA656
Ceratopipra_rubrocapilla_B4615
Ceratopipra_rubrocapilla_B5082
Ceratopipra_rubrocapilla_BR36440
Ceratopipra_rubrocapilla_CAM117
Ceratopipra_rubrocapilla_CUJ027
Ceratopipra_rubrocapilla_CUJ072
Ceratopipra_rubrocapilla_OM022
Ceratopipra_rubrocapilla_OM112
Ceratopipra_rubrocapilla_OM172
Ceratopipra_rubrocapilla_PUC008
Ceratopipra_rubrocapilla_PUC007
Ceratopipra_rubrocapilla_UFAC1007
Ceratopipra_rubrocapilla_UFAC305
Ceratopipra_rubrocapilla_UFAC476
Ceratopipra_rubrocapilla_AMANA034
Ceratopipra_rubrocapilla_AMANA137
Ceratopipra_rubrocapilla_AMZ610
Ceratopipra_rubrocapilla_FPR006
Ceratopipra_rubrocapilla_FPR015
Ceratopipra_rubrocapilla_FPR021
Ceratopipra_rubrocapilla_MAM010
Ceratopipra_rubrocapilla_MAR092
Ceratopipra_rubrocapilla_MPDS628
Ceratopipra_rubrocapilla_OP100
Ceratopipra_rubrocapilla_PIME024
Ceratopipra_rubrocapilla_PIME079
Ceratopipra_rubrocapilla_PIME460
Ceratopipra_rubrocapilla_BR033
Ceratopipra_rubrocapilla_BR036
Ceratopipra_rubrocapilla_MF005
Ceratopipra_rubrocapilla_MSF107
Ceratopipra_rubrocapilla_PIME062
Ceratopipra_rubrocapilla_PIME134
Ceratopipra_rubrocapilla_PIME135
Ceratopipra_rubrocapilla_PIME169
Ceratopipra_rubrocapilla_PIME482
Ceratopipra_rubrocapilla_PIME521
Ceratopipra_rubrocapilla_PIME522
Ceratopipra_rubrocapilla_PPS041
Ceratopipra_rubrocapilla_PPS174
Ceratopipra_rubrocapilla_PPS221
Ceratopipra_rubrocapilla_PPS352
Ceratopipra_rubrocapilla_TLP025
Ceratopipra_rubrocapilla_TLP123
Ceratopipra_rubrocapilla_WM281
Ceratopipra_rubrocapilla_WM293
Ceratopipra_rubrocapilla_AMZ551
Ceratopipra_rubrocapilla_GAPTO259
Ceratopipra_rubrocapilla_GAPTO268
Ceratopipra_rubrocapilla_GAPTO277
Ceratopipra_rubrocapilla_GAPTO327
Ceratopipra_rubrocapilla_MT087
Ceratopipra_rubrocapilla_MT123
Ceratopipra_rubrocapilla_PPBIO221
Ceratopipra_rubrocapilla_PPBIO305
Ceratopipra_rubrocapilla_RDO007
Ceratopipra_rubrocapilla_UHE442
```
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

mkdir -p fairfilters/alignments
#get fasta from VCF
#this is a workaround to remove uncalled bases - I output them as "z" and then remove all the z's
#get rid of any indels, including sites that were just "considered" as indels. If any sites are overlapping, they will be skipped for some samples in bcftools consensus and ruin the alignment
#for example, this site is NOT excluded by bcftools view --include 'TYPE="snp" || TYPE="ref"'
#chr_24  805548  .       A       .       14168.2 .       DP=1074;MQ0F=0;AN=102;DP4=550,524,0,0;MQ=59;F_MISSING=0.271429  GT:DP:SP:AD     0/0:7:0:7       0/0:10:0:10     ./.:1:0:1       ./.:1:0:1       ./.:1:0:1       ./.:2:0:2       ./.:1:0:1       0/0:8:0:8       0/0:11:0:11     0/0:29:0:29     0/0:8:0:8       ./.:0:0:0       0/0:57:0:57     0/0:14:0:14     0/0:7:0:7       0/0:6:0:6       0/0:38:0:38     0/0:21:0:21     0/0:9:0:9       0/0:8:0:8       0/0:17:0:17     ./.:2:0:2       0/0:35:0:35     0/0:23:0:23     0/0:13:0:13     ./.:4:0:4       0/0:10:0:10     ./.:3:0:3       0/0:5:0:5       0/0:53:0:53     ./.:3:0:3       ./.:2:0:2       0/0:33:0:33     0/0:48:0:48     0/0:37:0:37     0/0:42:0:42     0/0:35:0:35     0/0:26:0:26     0/0:28:0:28     0/0:40:0:40     ./.:3:0:3       0/0:7:0:7       0/0:27:0:27     0/0:9:0:9       0/0:11:0:11     0/0:10:0:10     ./.:1:0:1       0/0:13:0:13     0/0:6:0:6       0/0:10:0:10     0/0:9:0:9       ./.:2:0:2       ./.:1:0:1       ./.:0:0:0       0/0:32:0:32     0/0:12:0:12     0/0:9:0:9       0/0:8:0:8       ./.:2:0:2       0/0:6:0:6       ./.:4:0:4       0/0:7:0:7       0/0:15:0:15     0/0:35:0:35     0/0:29:0:29     0/0:7:0:7       0/0:24:0:24     0/0:10:0:10     ./.:4:0:4       0/0:11:0:11
#chr_24  805548  .       ATCT    .       35.6358 .       INDEL;IDV=21;IMF=1;DP=1074;VDB=2.39482e-42;SGB=133.525;RPBZ=2.03077;MQBZ=-3.57201;MQSBZ=-1.77777;BQBZ=-1.89605;SCBZ=-1.51717;MQ0F=0;AN=100;DP4=512,464,36,51;MQ=59;F_MISSING=0.285714   GT:DP:SP:AD     0/0:7:0:7       0/0:10:0:3      ./.:1:0:1       ./.:1:0:1       ./.:1:0:0       ./.:2:0:2       ./.:1:0:1       0/0:8:0:6       0/0:11:0:11     0/0:29:0:29     0/0:8:0:8       ./.:0:0:0       0/0:57:0:57     0/0:14:16:9     0/0:7:0:7       0/0:6:0:6       0/0:38:0:38     0/0:21:0:0      0/0:9:0:9       0/0:8:3:4       0/0:17:0:17     ./.:2:0:2       0/0:35:0:35     0/0:21:0:8      0/0:13:0:13     ./.:4:0:4       ./.:4:0:4       ./.:3:0:3       0/0:5:0:5       0/0:53:0:53     ./.:3:0:3       ./.:2:0:2       0/0:33:0:33     0/0:48:0:48     0/0:37:0:37     0/0:42:0:42     0/0:35:0:35     0/0:26:0:26     0/0:28:0:28     0/0:40:0:40     ./.:3:0:3       0/0:7:0:7       0/0:26:6:10     0/0:9:0:9       0/0:11:0:11     0/0:10:0:10     ./.:1:0:1       0/0:13:0:13     0/0:6:0:6       0/0:10:0:10     0/0:9:0:9       ./.:2:0:2       ./.:1:0:1       ./.:0:0:0       0/0:32:0:32     0/0:12:5:7      0/0:9:0:9       0/0:8:0:8       ./.:2:0:2       0/0:6:0:6       ./.:4:0:4       0/0:7:0:7       0/0:15:0:15     0/0:35:0:35     0/0:29:0:29     0/0:7:0:7       0/0:24:0:13     0/0:9:0:7       ./.:4:0:4       0/0:11:0:11

#dataset 1 uses Ceratopipra erythrocephala as the outgroup. This is the dataset for IQtree2. I am generating several test datasets along with my chosen filters (Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20) so that I can ensure that results are robust to some of my bioinformatic choices.
#these additional datasets including relaxing the minimum depth filter from 5x to 3x and maximum average depth from 49X to 40X, or relaxing the missing data filter to include some individuals with more missing data
for VCF in Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeri.GBSPipfilbwa.allsites.filtered.for_pixy.auto.45_0.2_3_20 Ceratopipra_rubrocapilla_imeri.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20  Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 ; do zgrep -v "INDEL;" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/"$VCF".vcf.gz | bgzip > /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/"$VCF".noindels.vcf.gz ; /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/"$VCF".noindels.vcf.gz ; done  #remove indels

#this command can run multiple datasets in parallel (assuming they use the same samples)
#extract a concatenation of all the sites in the VCF file for each sample. Sites that are absent from the VCF are set to "z" so that they will be easy for me to remove. Missing data will be set to "-". I immediately use sed to remove all z's from the file - this removes the uncalled sites. If any sample names contain z, those would lose their z's.
parallel --jobs 88 --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools consensus --fasta-ref /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa --absent z --missing - --include '\''TYPE="snp" || TYPE="ref"'\'' fairfilters/{2}.noindels.vcf.gz --haplotype I --sample {1} --output fairfilters/alignments/{2}.{1}.fasta ; sed -i "s/z//g" fairfilters/alignments/{2}.{1}.fasta' :::: samples ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
#make sure it does not give any warnings for skipped variants! That will cause the alignment to shift out of frame!

#fix the formatting of the fasta and fold it to a line length of 60 bp. Ensure that the fasta headers are not more than 60 characters in length or they will get folded!
parallel 'cat samples | while read sample ; do if [ -f fairfilters/alignments/{1}."$sample".fasta ] ; then awk '\''/^>/ {printf("%s%s\t",(N>0?"\n":""),$0);N++;next;} {printf("%s",$0);} END {printf("\n");}'\'' fairfilters/alignments/{1}."$sample".fasta | tr "\t" "\n" | fold -w 60 > fairfilters/alignments/{1}."$sample".fa ; fi ; done' ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20

#make a fasta alignment with all samples, one per chromosome
parallel 'cat samples | while read sample ; do time if [ -f fairfilters/alignments/{2}."$sample".fa ] ; then /home/0_PROGRAMS/seqkit grep -p {1} fairfilters/alignments/{2}."$sample".fa | sed "s/>.*$/>$sample/g" >> fairfilters/alignments/{2}.{1}.fa ; fi ; done' ::: chr_2 chr_1 chr_3 chr_4 chr_1A chr_5 chr_7 chr_6 chr_8 chr_9 chr_12 chr_10 chr_11 chr_4A chr_13 chr_14 chr_20 chr_15 chr_18 chr_19 chr_17 chr_21 chr_24 chr_23 chr_26 chr_27 chr_28 chr_22 chr_25 chr_29 ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
#make a giant concatenated "genome" alignment of all samples
parallel 'cat samples | while read sample ; do time if [ -f fairfilters/alignments/{1}."$sample".fa ] ; then printf ">$sample\n" >> fairfilters/alignments/{1}.fasta ; grep -v ">" fairfilters/alignments/{1}."$sample".fa  >> fairfilters/alignments/{1}.fasta ; fi ; done' ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
```

While this will not be used in the published paper, I will also check that the use of a different outgroup would produce the same result, to ensure that our results are robust to choice of outgroup. I made a dataset with the outgroup as Ceratopipra chloromeros instead of Ceratopipra erythrocephala to check that results were the same, and they were. This dataset will not be made public because the samples are reserved for a project in-prep. Results were qualitatively identical.  

```bash
#repeat, with chloromeros outgroup
zgrep -v "INDEL;" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.vcf.gz | bgzip > /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.noindels.vcf.gz #remove indels
/home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.noindels.vcf.gz 
parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools consensus --fasta-ref /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa --absent z --missing - --include '\''TYPE="snp" || TYPE="ref"'\'' fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.noindels.vcf.gz --haplotype I --sample {1} --output fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.{1}.fasta ; sed -i "s/z//g" fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.{1}.fasta' :::: samples #62 mins
#fix the formatting of the fasta and fold it to a line length of 60 bp. Ensure that the fasta headers are not more than 60 characters in lengtho=!
cat samples | while read sample ; do awk '/^>/ {printf("%s%s\t",(N>0?"\n":""),$0);N++;next;} {printf("%s",$0);} END {printf("\n");}' fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fasta | tr "\t" "\n" | fold -w 60 > fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fa ; done

parallel 'cat samples | while read sample ; do /home/0_PROGRAMS/seqkit grep -p {1} fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fa | sed "s/>.*$/>$sample/g" >> fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.{1}.fa ; done' ::: chr_2 chr_1 chr_3 chr_4 chr_1A chr_5 chr_7 chr_6 chr_8 chr_9 chr_12 chr_10 chr_11 chr_4A chr_13 chr_14 chr_20 chr_15 chr_18 chr_19 chr_17 chr_21 chr_24 chr_23 chr_26 chr_27 chr_28 chr_22 chr_25 chr_29
cat samples | while read sample ; do printf ">$sample\n" >> fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.fasta ; grep -v ">" fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fa  >> fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.fasta ; done

```
