
# Datasets without outgroups

```bash
cat > filters
```
```
Ceratopipra_rubrocapilla 37 Ceratopipra_rubrocapilla_UNA014,Ceratopipra_rubrocapilla_CPE027,Ceratopipra_rubrocapilla_CPE066,Ceratopipra_rubrocapilla_CPE067,Ceratopipra_rubrocapilla_CPE071,Ceratopipra_rubrocapilla_GAPTO140,Ceratopipra_rubrocapilla_GAPTO289,Ceratopipra_rubrocapilla_JRT021,Ceratopipra_rubrocapilla_MELO34,Ceratopipra_rubrocapilla_MPDS776,Ceratopipra_rubrocapilla_MT099,Ceratopipra_rubrocapilla_OM215,Ceratopipra_rubrocapilla_PPBIO312,Ceratopipra_rubrocapilla_UFAC1687,Ceratopipra_chloromeros_Fm433681,Ceratopipra_rubrocapilla_PUC007 0.2 5 20.0

#as above, but lower min depth filter for chrZ to accommodate low-depth females (which are hemizygous for the Z)
Ceratopipra_rubrocapilla 37 Ceratopipra_rubrocapilla_UNA014,Ceratopipra_rubrocapilla_CPE027,Ceratopipra_rubrocapilla_CPE066,Ceratopipra_rubrocapilla_CPE067,Ceratopipra_rubrocapilla_CPE071,Ceratopipra_rubrocapilla_GAPTO140,Ceratopipra_rubrocapilla_GAPTO289,Ceratopipra_rubrocapilla_JRT021,Ceratopipra_rubrocapilla_MELO34,Ceratopipra_rubrocapilla_MPDS776,Ceratopipra_rubrocapilla_MT099,Ceratopipra_rubrocapilla_OM215,Ceratopipra_rubrocapilla_PPBIO312,Ceratopipra_rubrocapilla_UFAC1687,Ceratopipra_chloromeros_Fm433681,Ceratopipra_rubrocapilla_PUC007 0.2 3 20.0

```
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

mkdir -p fairfilters
#filter invariant sites
#first, filter the autosomal dataset
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m1 -M1 -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #3m for rubricapilla #9m20.095s (20m18.336s) for all combined

#repeat to filter biallelic SNPs. Do not yet apply GQ filters, as these cannot be applied to invariants the same way and would bias pi/dxy downwards depending on sequencing depth
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m2 -M2 -v snps -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #6m for rubricapilla #9m20.095s (20m18.336s) for all combined

#merge invariants with SNPs
conda activate basic
#*****This dataset is used for pi and dxy*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools concat --threads 30 --allow-overlaps fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz -O z -o ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #204m (360m)

```

We need to use slightly different strategies for chromosome Z to accommodate hemizygous female genotypes.
```bash
#remove the pseudoautosomal region, which I found is probably located within the first 3,000,000 bp of the chromosome (this is a very conservative exaggeration of its extent - it is probably much smaller but I would rather remove too much than too little) (I identified it by looking at sequencing depth and heterozygosity of females, and Fst between males and females)

#filter invariant sites
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m1 -M1 -t chr_Z:3000000- --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz' #2m32.604s
#make a dataset with females diploidized
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz | sed -r -e "s@\t([0-9.]):@\t\1/\1:@g" | /home/0_PROGRAMS/bcftools-1.17/bcftools view -Oz -o ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz; time tabix ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz'
#remove sites flagged as heterozygous in a female with GQ>10 (chrW mapping to chrZ, or other collapsed paralogs/repeats)
cat filters | parallel --colsep " " '/home/0_PROGRAMS/bcftools-1.17/bcftools view --targets-file ^sites_to_exclude/chrZ_2heterozygousFemales.tsv ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz -Oz -o ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz'
cat filters | parallel --colsep " " 'time tabix ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz'

#repeat to filter biallelic SNPs. Do not yet apply GQ filters, as these cannot be applied to invariants the same way and would bias pi/dxy downwards depending on sequencing depth
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m2 -M2 -v snps -t chr_Z:3000000- --samples ^{3} -e '\''MQ<25'\'' maxmissing50/{1}.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz ; /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz' #2m29.965s
#make a dataset with females diploidized
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.{2}_{4}_{5}_{6}.vcf.gz | sed -r -e "s@\t([0-9.]):@\t\1/\1:@g" | /home/0_PROGRAMS/bcftools-1.17/bcftools view -Oz -o ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz; time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz'
#remove sites flagged as heterozygous in a female with GQ>10 (chrW mapping to chrZ, or other collapsed paralogs/repeats)
cat filters | parallel --colsep " " '/home/0_PROGRAMS/bcftools-1.17/bcftools view --targets-file ^sites_to_exclude/chrZ_2heterozygousFemales.tsv ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.{2}_{4}_{5}_{6}.vcf.gz -Oz -o ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz'
cat filters | parallel --colsep " " 'time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz'


#combine SNPs and invariants
conda activate basic
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools concat --threads 30 --allow-overlaps ./fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz -O z -o ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz ; time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz' #204m (360m)
```

## Dataset for PCoA, STRUCTURE
My next dataset is filtered more heavily, thinned for linkage, and contains only biallelic SNPs.  

For the PCA we want the following filters:  
These are already applied for the pixy dataset:  
* biallelic SNPs only  
* autosomal and Z loci separated (in case they show different patterns)  
* no unplaced scaffolds  
* min mapping quality 25 (to reduce false SNPs due to mismapping)  
* max depth less than `mean_depth + 3*sqrt(mean_depth)` (to reduce collapsed duplicates being mistaken for SNPs)  
* min individual depth 5X (to reduce missed heterozygotes or false heterozygotes)  
* max missing data 18%  

These also need to be applied:  
* min genotype quality 25 (to reduce genotype errors)   
* no singletons (they are not informative)  
* none within genes or within 10 kb of genes (reduce (not eliminate) impact of selection)  
* each SNP spaced at least 5kb or 10 kb apart (reduce linkage. Making 2 datasets to ensure the results are not sensitive to that choice)    

Next, we will filter the SNPs more heavily
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping
mkdir -p filteredSNPs

#This dataset is for PCoA, and STRUCTURE: biallelic autosomal SNPs, min qual 25, min depth 5 (per sample), 
mkdir -p filteredSNPs/thinned_datasets

#Raising AC limit to 4
#*****This dataset is used for structure and pcoa*****
parallel --colsep " " 'seed=$RANDOM ; echo "Random seed: "$seed ; time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC<=4 || AC>=AN-4 || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" | /home/0_PROGRAMS/bcftools-1.17/bcftools +prune --nsites-per-win 1 -w 10000bp --nsites-per-win-mode "rand" --random-seed "$seed" -O z - 2> filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.{2}_{4}_{5}_{6}.vcf.gz' :::: filters
parallel --colsep " " 'echo {1}_{2}_{4}_{5}_{6} ; zgrep -v "^#" filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.{2}_{4}_{5}_{6}.vcf.gz | wc -l'  :::: filters
#Ceratopipra_rubrocapilla_37_0.2_5_20.0: 7865

```

## Dataset for Fst, Tajima's D
These datasets are similar to the above, but are not thinned for linkage, and need careful attention to the MAF filter (no MAF filter should go int oTajima's D or it will severely bias the results).  
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

#the dataset for Fst will not be thinned for linkage, but we will filter out sites with low minor allele frequency (make sure not to compare values to Fst estimates from datasets where this MAF filter was not implemented)
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || AC==1 || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.gz' #0m19.856s
#add MAF filter
#*****This dataset is used for fst*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''MAF<0.05'\'' -Oz -o filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.maf5.vcf.gz filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.gz'
cat filters | parallel --colsep " " time tabix filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.maf5.vcf.gz
zgrep -v "^#" filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.4_5_20.0.maf5.vcf.gz | wc -l #40514

#next repeat for the Z chromosome
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || AC==1 || AC>=AN-1 || QUAL<25'\'' ./fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz' #
#*****This dataset is used for chr Z fst*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.vcf.gz -- -t AC,AN | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''MAF<0.05'\'' -Oz -o filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.maf5.vcf.gz - '
cat filters | parallel --colsep " " time tabix ./filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.{2}_{4}_{5}_{6}.maf5.vcf.gz 
zgrep -c -v "^#" filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.37_0.2_3_20.0.maf5.vcf.gz #310

# now repeat but with no MAF filter (for Tajima's D)
#*****This dataset is used for tajima's D*****
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.{2}_{4}_{5}_{6}.vcf.gz' #0m19.856s
cat filters | parallel --colsep " " time tabix filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.{2}_{4}_{5}_{6}.vcf.gz
```




# Datasets with outgroups
```bash
#after deciding on filters, filter invariant sites. These will be included with IQTree2, but not analyses that use only SNPs (eg., splitstree, ABBA BABA test)
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m1 -M1 -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples {3} -e '\''MQ<25'\'' maxmissing50/Ceratopipra.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #3m for rubricapilla #9m20.095s (20m18.336s) for all combined

#repeat to filter biallelic SNPs. Do not yet apply GQ filters, as these cannot be applied to invariants the same way and would bias pi/dxy downwards depending on sequencing depth. This is to avoid biasing branchlengths in IQtree2.
#filter SNPs with fair filters
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -m2 -M2 -v snps -t chr_2,chr_1,chr_3,chr_4,chr_1A,chr_5,chr_7,chr_6,chr_8,chr_9,chr_12,chr_10,chr_11,chr_4A,chr_13,chr_14,chr_20,chr_15,chr_18,chr_19,chr_17,chr_21,chr_24,chr_23,chr_26,chr_27,chr_28,chr_22,chr_25,chr_29 --samples {3} -e '\''MQ<25'\'' maxmissing50/Ceratopipra.GBSPipfilbwa.allsites.maxmiss50.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e '\''FMT/DP<{5}'\'' | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t AC,AN,F_MISSING | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''F_MISSING > {4} || AVG(FMT/DP)>{2}'\'' -Oz -o fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz ; time /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz' #6m for rubricapilla #9m20.095s (20m18.336s) for all combined

#merge invariants with SNPs
#***This dataset is for IQtree2***
conda activate basic
cat filters | parallel --colsep " " time /home/0_PROGRAMS/bcftools-1.17/bcftools concat --threads 30 --allow-overlaps fairfilters/{1}.GBSPipfilbwa.invariants.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz -O z -o ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz #1m26.367s
cat filters | parallel --colsep " " time tabix ./fairfilters/{1}.GBSPipfilbwa.allsites.filtered.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz

#then, filter SNPs more strictly for analyses that do not need invariant sites, where we are not worried about biasing the relative proportion of variant vs invariant sites
#not thinned
#***This dataset is for Dsuite (ABBA BABA test) and Splitstree***
cat filters | parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''AC==0 || AC==AN || AC==1 || AC==AN-1 || QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01" -O z - 2> filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/{1}.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.{2}_{4}_{5}_{6}.vcf.gz' #0m19.856s 

#thin 1 SNP per 10 kb to reduce the effects of linkage. 
#**This dataset is used for SVDQuartets**
cat filters | parallel --colsep " " 'seed=$RANDOM ; echo "Random seed: "$seed ; time /home/0_PROGRAMS/bcftools-1.17/bcftools view -e '\''QUAL<25'\'' fairfilters/{1}.GBSPipfilbwa.allsites.for_pixy.auto.{2}_{4}_{5}_{6}.vcf.gz | /home/0_PROGRAMS/bcftools-1.17/bcftools filter -S . -e "FMT/GQ<{6}" | /home/0_PROGRAMS/bcftools-1.17/bcftools +fill-tags - --threads 4 -- -t F_MISSING,ExcHet,AC,AN | /home/0_PROGRAMS/bcftools-1.17/bcftools view -e "F_MISSING > {4} || ExcHet<0.01 || AC==0 || AC==AN || AC==1" | /home/0_PROGRAMS/bcftools-1.17/bcftools +prune --nsites-per-win 1 -w 10000bp --nsites-per-win-mode "rand" --random-seed "$seed" -O z - 2> filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.{2}_{4}_{5}_{6}.vcf.log > filteredSNPs/thinned_datasets/{1}.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.{2}_{4}_{5}_{6}.vcf.gz' #0m22.314s


```

# Fasta alignment
This section converts the VCF file to an alignment in fasta format.  
For a couple analyses, I want alignments in fasta format with invariants and variant sites. This will be used for the concatenated Maximum Likelihood tree with IQtree2.  

First, make a list of sample names to loop through.  
```bash
cat > samples 
```
```
Ceratopipra_erythrocephala_SGC288
Ceratopipra_erythrocephala_B7534
Ceratopipra_erythrocephala_B7539
Ceratopipra_erythrocephala_B7536
Ceratopipra_erythrocephala_B7537
Ceratopipra_rubrocapilla_CPE012
Ceratopipra_rubrocapilla_CPE025
Ceratopipra_rubrocapilla_CPE049
Ceratopipra_rubrocapilla_CPE069
Ceratopipra_rubrocapilla_CPEII008
Ceratopipra_rubrocapilla_UNA002
Ceratopipra_rubrocapilla_UNA018
Ceratopipra_rubrocapilla_UNA088
Ceratopipra_rubrocapilla_UNA110
Ceratopipra_rubrocapilla_UNA124
Ceratopipra_rubrocapilla_AMA340
Ceratopipra_rubrocapilla_AMA601
Ceratopipra_rubrocapilla_AMA656
Ceratopipra_rubrocapilla_B4615
Ceratopipra_rubrocapilla_B5082
Ceratopipra_rubrocapilla_BR36440
Ceratopipra_rubrocapilla_CAM117
Ceratopipra_rubrocapilla_CUJ027
Ceratopipra_rubrocapilla_CUJ072
Ceratopipra_rubrocapilla_OM022
Ceratopipra_rubrocapilla_OM112
Ceratopipra_rubrocapilla_OM172
Ceratopipra_rubrocapilla_PUC008
Ceratopipra_rubrocapilla_PUC007
Ceratopipra_rubrocapilla_UFAC1007
Ceratopipra_rubrocapilla_UFAC305
Ceratopipra_rubrocapilla_UFAC476
Ceratopipra_rubrocapilla_AMANA034
Ceratopipra_rubrocapilla_AMANA137
Ceratopipra_rubrocapilla_AMZ610
Ceratopipra_rubrocapilla_FPR006
Ceratopipra_rubrocapilla_FPR015
Ceratopipra_rubrocapilla_FPR021
Ceratopipra_rubrocapilla_MAM010
Ceratopipra_rubrocapilla_MAR092
Ceratopipra_rubrocapilla_MPDS628
Ceratopipra_rubrocapilla_OP100
Ceratopipra_rubrocapilla_PIME024
Ceratopipra_rubrocapilla_PIME079
Ceratopipra_rubrocapilla_PIME460
Ceratopipra_rubrocapilla_BR033
Ceratopipra_rubrocapilla_BR036
Ceratopipra_rubrocapilla_MF005
Ceratopipra_rubrocapilla_MSF107
Ceratopipra_rubrocapilla_PIME062
Ceratopipra_rubrocapilla_PIME134
Ceratopipra_rubrocapilla_PIME135
Ceratopipra_rubrocapilla_PIME169
Ceratopipra_rubrocapilla_PIME482
Ceratopipra_rubrocapilla_PIME521
Ceratopipra_rubrocapilla_PIME522
Ceratopipra_rubrocapilla_PPS041
Ceratopipra_rubrocapilla_PPS174
Ceratopipra_rubrocapilla_PPS221
Ceratopipra_rubrocapilla_PPS352
Ceratopipra_rubrocapilla_TLP025
Ceratopipra_rubrocapilla_TLP123
Ceratopipra_rubrocapilla_WM281
Ceratopipra_rubrocapilla_WM293
Ceratopipra_rubrocapilla_AMZ551
Ceratopipra_rubrocapilla_GAPTO259
Ceratopipra_rubrocapilla_GAPTO268
Ceratopipra_rubrocapilla_GAPTO277
Ceratopipra_rubrocapilla_GAPTO327
Ceratopipra_rubrocapilla_MT087
Ceratopipra_rubrocapilla_MT123
Ceratopipra_rubrocapilla_PPBIO221
Ceratopipra_rubrocapilla_PPBIO305
Ceratopipra_rubrocapilla_RDO007
Ceratopipra_rubrocapilla_UHE442
```
```bash
cd /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping

mkdir -p fairfilters/alignments
#get fasta from VCF
#this is a workaround to remove uncalled bases - I output them as "z" and then remove all the z's
#get rid of any indels, including sites that were just "considered" as indels. If any sites are overlapping, they will be skipped for some samples in bcftools consensus and ruin the alignment
#for example, this site is NOT excluded by bcftools view --include 'TYPE="snp" || TYPE="ref"'
#chr_24  805548  .       A       .       14168.2 .       DP=1074;MQ0F=0;AN=102;DP4=550,524,0,0;MQ=59;F_MISSING=0.271429  GT:DP:SP:AD     0/0:7:0:7       0/0:10:0:10     ./.:1:0:1       ./.:1:0:1       ./.:1:0:1       ./.:2:0:2       ./.:1:0:1       0/0:8:0:8       0/0:11:0:11     0/0:29:0:29     0/0:8:0:8       ./.:0:0:0       0/0:57:0:57     0/0:14:0:14     0/0:7:0:7       0/0:6:0:6       0/0:38:0:38     0/0:21:0:21     0/0:9:0:9       0/0:8:0:8       0/0:17:0:17     ./.:2:0:2       0/0:35:0:35     0/0:23:0:23     0/0:13:0:13     ./.:4:0:4       0/0:10:0:10     ./.:3:0:3       0/0:5:0:5       0/0:53:0:53     ./.:3:0:3       ./.:2:0:2       0/0:33:0:33     0/0:48:0:48     0/0:37:0:37     0/0:42:0:42     0/0:35:0:35     0/0:26:0:26     0/0:28:0:28     0/0:40:0:40     ./.:3:0:3       0/0:7:0:7       0/0:27:0:27     0/0:9:0:9       0/0:11:0:11     0/0:10:0:10     ./.:1:0:1       0/0:13:0:13     0/0:6:0:6       0/0:10:0:10     0/0:9:0:9       ./.:2:0:2       ./.:1:0:1       ./.:0:0:0       0/0:32:0:32     0/0:12:0:12     0/0:9:0:9       0/0:8:0:8       ./.:2:0:2       0/0:6:0:6       ./.:4:0:4       0/0:7:0:7       0/0:15:0:15     0/0:35:0:35     0/0:29:0:29     0/0:7:0:7       0/0:24:0:24     0/0:10:0:10     ./.:4:0:4       0/0:11:0:11
#chr_24  805548  .       ATCT    .       35.6358 .       INDEL;IDV=21;IMF=1;DP=1074;VDB=2.39482e-42;SGB=133.525;RPBZ=2.03077;MQBZ=-3.57201;MQSBZ=-1.77777;BQBZ=-1.89605;SCBZ=-1.51717;MQ0F=0;AN=100;DP4=512,464,36,51;MQ=59;F_MISSING=0.285714   GT:DP:SP:AD     0/0:7:0:7       0/0:10:0:3      ./.:1:0:1       ./.:1:0:1       ./.:1:0:0       ./.:2:0:2       ./.:1:0:1       0/0:8:0:6       0/0:11:0:11     0/0:29:0:29     0/0:8:0:8       ./.:0:0:0       0/0:57:0:57     0/0:14:16:9     0/0:7:0:7       0/0:6:0:6       0/0:38:0:38     0/0:21:0:0      0/0:9:0:9       0/0:8:3:4       0/0:17:0:17     ./.:2:0:2       0/0:35:0:35     0/0:21:0:8      0/0:13:0:13     ./.:4:0:4       ./.:4:0:4       ./.:3:0:3       0/0:5:0:5       0/0:53:0:53     ./.:3:0:3       ./.:2:0:2       0/0:33:0:33     0/0:48:0:48     0/0:37:0:37     0/0:42:0:42     0/0:35:0:35     0/0:26:0:26     0/0:28:0:28     0/0:40:0:40     ./.:3:0:3       0/0:7:0:7       0/0:26:6:10     0/0:9:0:9       0/0:11:0:11     0/0:10:0:10     ./.:1:0:1       0/0:13:0:13     0/0:6:0:6       0/0:10:0:10     0/0:9:0:9       ./.:2:0:2       ./.:1:0:1       ./.:0:0:0       0/0:32:0:32     0/0:12:5:7      0/0:9:0:9       0/0:8:0:8       ./.:2:0:2       0/0:6:0:6       ./.:4:0:4       0/0:7:0:7       0/0:15:0:15     0/0:35:0:35     0/0:29:0:29     0/0:7:0:7       0/0:24:0:13     0/0:9:0:7       ./.:4:0:4       0/0:11:0:11

#dataset 1 uses Ceratopipra erythrocephala as the outgroup. This is the dataset for IQtree2. I am generating several test datasets along with my chosen filters (Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20) so that I can ensure that results are robust to some of my bioinformatic choices.
#these additional datasets including relaxing the minimum depth filter from 5x to 3x and maximum average depth from 49X to 40X, or relaxing the missing data filter to include some individuals with more missing data
for VCF in Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeri.GBSPipfilbwa.allsites.filtered.for_pixy.auto.45_0.2_3_20 Ceratopipra_rubrocapilla_imeri.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20  Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 ; do zgrep -v "INDEL;" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/"$VCF".vcf.gz | bgzip > /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/"$VCF".noindels.vcf.gz ; /home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/"$VCF".noindels.vcf.gz ; done  #remove indels

#this command can run multiple datasets in parallel (assuming they use the same samples)
#extract a concatenation of all the sites in the VCF file for each sample. Sites that are absent from the VCF are set to "z" so that they will be easy for me to remove. Missing data will be set to "-". I immediately use sed to remove all z's from the file - this removes the uncalled sites. If any sample names contain z, those would lose their z's.
parallel --jobs 88 --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools consensus --fasta-ref /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa --absent z --missing - --include '\''TYPE="snp" || TYPE="ref"'\'' fairfilters/{2}.noindels.vcf.gz --haplotype I --sample {1} --output fairfilters/alignments/{2}.{1}.fasta ; sed -i "s/z//g" fairfilters/alignments/{2}.{1}.fasta' :::: samples ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
#make sure it does not give any warnings for skipped variants! That will cause the alignment to shift out of frame!

#fix the formatting of the fasta and fold it to a line length of 60 bp. Ensure that the fasta headers are not more than 60 characters in length or they will get folded!
parallel 'cat samples | while read sample ; do if [ -f fairfilters/alignments/{1}."$sample".fasta ] ; then awk '\''/^>/ {printf("%s%s\t",(N>0?"\n":""),$0);N++;next;} {printf("%s",$0);} END {printf("\n");}'\'' fairfilters/alignments/{1}."$sample".fasta | tr "\t" "\n" | fold -w 60 > fairfilters/alignments/{1}."$sample".fa ; fi ; done' ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20

#make a fasta alignment with all samples, one per chromosome
parallel 'cat samples | while read sample ; do time if [ -f fairfilters/alignments/{2}."$sample".fa ] ; then /home/0_PROGRAMS/seqkit grep -p {1} fairfilters/alignments/{2}."$sample".fa | sed "s/>.*$/>$sample/g" >> fairfilters/alignments/{2}.{1}.fa ; fi ; done' ::: chr_2 chr_1 chr_3 chr_4 chr_1A chr_5 chr_7 chr_6 chr_8 chr_9 chr_12 chr_10 chr_11 chr_4A chr_13 chr_14 chr_20 chr_15 chr_18 chr_19 chr_17 chr_21 chr_24 chr_23 chr_26 chr_27 chr_28 chr_22 chr_25 chr_29 ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
#make a giant concatenated "genome" alignment of all samples
parallel 'cat samples | while read sample ; do time if [ -f fairfilters/alignments/{1}."$sample".fa ] ; then printf ">$sample\n" >> fairfilters/alignments/{1}.fasta ; grep -v ">" fairfilters/alignments/{1}."$sample".fa  >> fairfilters/alignments/{1}.fasta ; fi ; done' ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
```

While this will not be used in the published paper, I will also check that the use of a different outgroup would produce the same result, to ensure that our results are robust to choice of outgroup. I made a dataset with the outgroup as Ceratopipra chloromeros instead of Ceratopipra erythrocephala to check that results were the same, and they were. This dataset will not be made public because the samples are reserved for a project in-prep. Results were qualitatively identical.  

```bash
#repeat, with chloromeros outgroup
zgrep -v "INDEL;" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.vcf.gz | bgzip > /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.noindels.vcf.gz #remove indels
/home/0_PROGRAMS/bcftools-1.17/bcftools index fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.noindels.vcf.gz 
parallel --colsep " " 'time /home/0_PROGRAMS/bcftools-1.17/bcftools consensus --fasta-ref /home/1_Else/Ceratopipra/1_Datasets/1.0_Reference_genome/0_reference/Pipra_filicauda.ref.fa --absent z --missing - --include '\''TYPE="snp" || TYPE="ref"'\'' fairfilters/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.4_5_20.noindels.vcf.gz --haplotype I --sample {1} --output fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.{1}.fasta ; sed -i "s/z//g" fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.{1}.fasta' :::: samples #62 mins
#fix the formatting of the fasta and fold it to a line length of 60 bp. Ensure that the fasta headers are not more than 60 characters in lengtho=!
cat samples | while read sample ; do awk '/^>/ {printf("%s%s\t",(N>0?"\n":""),$0);N++;next;} {printf("%s",$0);} END {printf("\n");}' fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fasta | tr "\t" "\n" | fold -w 60 > fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fa ; done

parallel 'cat samples | while read sample ; do /home/0_PROGRAMS/seqkit grep -p {1} fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fa | sed "s/>.*$/>$sample/g" >> fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.{1}.fa ; done' ::: chr_2 chr_1 chr_3 chr_4 chr_1A chr_5 chr_7 chr_6 chr_8 chr_9 chr_12 chr_10 chr_11 chr_4A chr_13 chr_14 chr_20 chr_15 chr_18 chr_19 chr_17 chr_21 chr_24 chr_23 chr_26 chr_27 chr_28 chr_22 chr_25 chr_29
cat samples | while read sample ; do printf ">$sample\n" >> fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.fasta ; grep -v ">" fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20."$sample".fa  >> fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.fasta ; done

```
