# STRUCTURE Analysis

This page describes running the STRUCTURE analysis to look for population structure in *Ceratopipra rubrocapilla*.

Here are some notes:
* make sure not to include close familial relatives (parent-offspring, siblings, etc) as this can confound detecting population structure
* filter data to reduce linkage, as each SNP is assumed to be independant
* try multiple runs starting from different seeds, since the program could get stuck on a single optimum
* try multiple values of K
* probably should remove singletons -> noninformative plus more likely to be errors. In fact, it is probably best to enforce a higher minor allele count filter to improve phylogenetic signal.
* use biallelic SNPs
* each pop should be represented by similar sample sizes

I will run it several times, and each time I will use a different random starting seed. 
There are a few different tools to downsample. `vcftools --thin` can do it, but it does not appear to be random since I get the same subset multiple times - it seems to just take the first site in the chunk. `bcftools prune` seems to do the trick! I have already generated the thinned datasets.  

# format data
First, the input data must be formatted for STRUCTURE. The input can have a few different formats, one STRUCTURE format has:  
1) A header with the names of the markers  
2) A row for each sample. In these rows, each locus is represented by two columns listing the alleles of the two diploid haplotypes. Heterozygotes are `1 2`, and homozygotes are `1 1` or `2 2`. Note that plink changed the "1" and "2" designation based on allele frequencies - it does not correspond to reference vs alternate in the VCF file.  
For example, here is a tiny file with 6 diploid samples and 3 SNPs.  
```
-1 34753 24805 
AMA340 1 2 2 2 2 2 2
AMA601 2 2 2 0 0 2 2
AMA656 3 2 2 0 0 0 0
AMANA034 4 0 0 0 0 2 2
AMANA137 5 2 2 0 0 0 0
AMZ551 6 2 2 0 0 2 2
```

I am generating the input file from a VCF file using plink.  
```bash
mkdir -p /home/1_Else/Ceratopipra/3_Structure/3.2aa_STRUCTURE
cd /home/1_Else/Ceratopipra/3_Structure/3.2aa_STRUCTURE

#first, gather a subsample of data. We shouldn't use the full dataset because STRUCTURE assumes that the input data is not linked
#I made a dataset that is pruned to 1 SNP per 10kb and only autosomal.

#count how many sites are retained if you have not already done so - we need to give this info to STRUCTURE
zgrep -c -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/thinned_datasets/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.37_0.2_5_20.0.vcf.gz #7865 sites

#format into the correct STRUCTURE format

mkdir -p data_Ceratopipra_rubrocapilla

#choose the dataset to use

#this is the dataset for the publication
VCF=Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.37_0.2_5_20.0.vcf.gz
CODE=Cerrub.AC4.37_0.2_5_20.0

#this dataset was used to compare the effect of lowering the min AC filter to 2 (removing only true singletons)
#zgrep -c -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/thinned_datasets/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned5k.forSVD.37_0.15_5_20.0.vcf.gz #10502 sites
#VCF=Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned5k.forSVD.37_0.15_5_20.0.vcf.gz
#CODE=Cerrub.37_0.15_5_20.0

#this dataset has removed the Atlantic Forest, to see whether the patterns change (59 samples)
#zgrep -c -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/thinned_datasets/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.37_0.1996_5_20.0.vcf.gz #5881 sites
#VCF=Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.37_0.1996_5_20.0.vcf.gz
#CODE=Cerrub.AC4.37_0.1996_5_20.0

#this dataset has removed the Atlantic Forest and Xingu, to see whether the patterns change
#VCF=Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.AC4.37_0.1997_5_20.0.vcf.gz
#CODE=Cerrub.AC4.37_0.1997_5_20.0

#this dataset has not been thinned - for comparison. 15040 variants and 69 samples 
#VCF=Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC4.37_0.2_5_20.0.vcf.gz
#CODE=Cerrub.AC4.unthinned.37_0.2_5_20.0
#zcat /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/"$VCF" | sed 's/_//g' > ./data_Ceratopipra_rubrocapilla/"$VCF".plinktemp.vcf

#firstly, it seems that plink cannot tolerate sample names with underscores in them, so I will take them out
zcat /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/thinned_datasets/"$VCF" | sed 's/_//g' > ./data_Ceratopipra_rubrocapilla/"$VCF".plinktemp.vcf
#Then run plink's convenient script to reformat the vcf into STRUCTURE's input format.
cd data_Ceratopipra_rubrocapilla
time /home/0_PROGRAMS/plink --vcf ./"$VCF".plinktemp.vcf --recode structure --out "$CODE" --allow-extra-chr --chr-set 30 #0m0.115s #I needed to specify that I had 30 chromosomes in the data, so it would not produce an error since it expects human data. #genotyping rate: 0.76

#remove the first row of blank labels
#convert missing data from "0" to "-9"
#also, I am trimming the prefix from my sample names to make them shorter since STRUCTURE truncates the names
tail -n +2 "$CODE".recode.strct_in | sed 's/ 0 / -9 /g; s/ 0 / -9 /g; s/ 0$/ -9/g; s/Ceratopiprarubrocapilla//g' > "$CODE".strct_in

#Ceratopipra_rubrocapilla_37_0.2_5_20.0: 7865 sites in the final input dataset
```

## Prepare param files  
STRUCTURE runs with many parameters, which are listed in parameter files. The main parameters are in the file mainparams, and a bunch of extra parameters are in extraparams.  
We need to tell structure how many samples and how many loci there are. In the format that Plink gave us, the two haplotypes of each sample are provided in the same row, rather than separate rows, so we need to set `ONEROWPERIND` to true. Plink also provided two rows of header. The first was completely meaningless so we removed it. We need to tell STRUCTURE that there is (one remaining) header row with the setting `MARKERNAMES`. Finally, we also need to tell it that Plink gave an extra column after the sample names, using the parameter `EXTRACOLS`. Key parameters to run structure are then the number of populations (`MAXPOPS`, burn-in (`BURNIN`), and # of MCMC reps (`NUMREPS`. For the template I set `BURNIN` and `NUMREPS` to 1000, which is too for a proper analysis (generally 100,000 or greater is used). I will modify those parameters later.  

```bash
cat > data_Ceratopipra_rubrocapilla/mainparams

#modify number of individuals and loci as needed for different datasets:  
sed 's/Cerrub.37_0.4_5_20.0/Cerrub.AC4.37_0.2_5_20.0/g; s/NUMLOCI  27385/NUMLOCI  7865/g; s/NUMINDS 70/NUMINDS 69/g' data_Ceratopipra_rubrocapilla/mainparams > data_Ceratopipra_rubrocapilla/mainparams.Cerrub.AC4.37_0.2_5_20.0 #modify template parameters to be customized for the input dataset

#parameters for the test datasets:
#sed 's/Cerrub.37_0.4_5_20.0/Cerrub.AC4.37_0.1996_5_20.0/g; s/NUMLOCI  27385/NUMLOCI  5881/g; s/NUMINDS 70/NUMINDS 59/g' data_Ceratopipra_rubrocapilla/mainparams > data_Ceratopipra_rubrocapilla/mainparams.Cerrub.AC4.37_0.1996_5_20.0 #modify template parameters to be customized for the input dataset
#sed 's/Cerrub.37_0.4_5_20.0/Cerrub.AC4.37_0.1997_5_20.0/g; s/NUMLOCI  27385/NUMLOCI  5329/g; s/NUMINDS 70/NUMINDS 51/g' data_Ceratopipra_rubrocapilla/mainparams > data_Ceratopipra_rubrocapilla/mainparams.Cerrub.AC4.37_0.1997_5_20.0 #modify template parameters to be customized for the input dataset
#sed 's/Cerrub.37_0.4_5_20.0/Cerrub.AC4.unthinned.37_0.2_5_20.0/g; s/NUMLOCI  27385/NUMLOCI  15040/g; s/NUMINDS 70/NUMINDS 69/g' data_Ceratopipra_rubrocapilla/mainparams > data_Ceratopipra_rubrocapilla/mainparams.Cerrub.AC4.unthinned.37_0.2_5_20.0 #modify template parameters to be customized for the input dataset

```
Here is my template for `mainparams`:  
```
KEY PARAMETERS FOR THE PROGRAM structure.  YOU WILL NEED TO SET THESE
IN ORDER TO RUN THE PROGRAM.  VARIOUS OPTIONS CAN BE ADJUSTED IN THE
FILE extraparams.


"(int)" means that this takes an integer value.
"(B)"   means that this variable is Boolean 
        (ie insert 1 for True, and 0 for False).
"(str)" means that this is a string (but not enclosed in quotes!) 

Data File

#define INFILE  Cerrub.37_0.4_5_20.0.strct_in // (str) name of input data file. max length 30.
#define OUTFILE Cerrub.37_0.4_5_20.0.res   // (str) name of output data file. max length 30.

#define NUMINDS 70   // (int) number of diploid individuals in data file
#define NUMLOCI  27385    // (int) number of loci in data file
#define ONEROWPERIND  1    // (B) Input file has one row per individual
#define MARKERNAMES  1    // (B) The top row of the data file contains a list of L names corresponding to the markers used

#define LABEL    1    // (B) Input file contains individual labels
#define POPDATA  0    // (B) Input file contains a population 
                             identifier
#define POPFLAG  0    // (B) Input file contains a flag which says 
                             whether to use popinfo when USEPOPINFO==1
#define PHENOTYPE 0   // (B) Input file contains phenotype information
#define EXTRACOLS 1   // (int) Number of additional columns of data 
                             before the genotype data start.
#define MISSING  -9 // (int) value given to missing genotype data
                  
Program Parameters

#define MAXPOPS 2     // (int) Model the data using this many populations (K)
#define BURNIN  1000   // (int) length of burnin period
#define NUMREPS 1000  // (int) number of MCMC reps after burnin

```

Next, set extra parameters in the file `extraparams`. Key parameters that I have modified are `POPALPHAS=1`, to allow each population to have a different value of alpha since my sample sizes are uneven. Also critical is setting `RANDOMIZE` to `0` so that STRUCTURE does not try to pick its own random seed. I will be running several runs at each value of K in parallel, starting at the exact same time. Since STRUCTURE picks a random number based on start time, all the runs would pick the same random number and be identical (I tested this). Instead, tell STRUCTURE NOT to randomize; I will provide my own random starting seed for each run.  
```bash
#cat > data_Ceratopipra_rubrocapilla/extraparams #this is identical to the file below, but without the line "#define POPALPHAS   1  // (B) Individual alpha for each population"
cat > data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS
cp data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS data_Ceratopipra_erythrocephala/extraparams_POPALPHAS #reuse the same parameter file template
```

```
EXTRA PARAMS FOR THE PROGRAM structure.  THE MOST IMPORTANT PARAMS ARE
IN THE FILE mainparams.

"(int)" means that this takes an integer value.
"(d)"   means that this is a double (ie, a Real number such as 3.14).
"(B)"   means that this variable is Boolean 
        (ie insert 1 for True, and 0 for False).

PROGRAM OPTIONS

#define INFERALPHA  1 // (B) Infer ALPHA (the admixture parameter)
#define POPALPHAS   1  // (B) Individual alpha for each population
#define FREQSCORR   1 // (B) allele frequencies are correlated among pops
#define COMPUTEPROB 1 // (B) Estimate the probability of the Data under 
                             the model.  This is used when choosing the 
                             best number of subpopulations.
#define NOADMIX     0 // (B) Use no admixture model
#define NOADMBURN   0 //(int) initial period of burnin with admixture,
                          before switching to no admixture model.  This 
                          sometimes leads to better convergence.
                          Set this to be less than the value of BURNIN.


USING PRIOR POPULATION INFO

#define USEPOPINFO  0  // (B) Use prior population information to assist 
                             clustering.  Need POPDATA=1.
#define GENSBACK    0  //(int) For use when inferring whether an indiv-
                         idual is an immigrant, or has an immigrant an-
                         cestor in the past GENSBACK generations.  eg, if 
                         GENSBACK==2, it tests for immigrant ancestry 
                         back to grandparents. 
#define MIGRPRIOR 0.05 //(d) prior prob that an individual is a migrant 
                             (used only when USEPOPINFO==1).  This should 
                             be small, eg 0.01 or 0.1.


OUTPUT OPTIONS

#define PRINTQHAT    1  // (B) Q-hat printed to a separate file.  Turn this 
                           on before using STRAT.
#define UPDATEFREQ   0  // (int) frequency of printing update on the screen.
                                 Set automatically if this is 0.
#define PRINTLIKES   0  // (B) print current likelihood to screen every rep
#define INTERMEDSAVE 0  // (int) number of saves to file during run
#define PRINTKLD     1  // (B) Print estimated Kullback-Leibler diver-
                              gence to screen during the run
#define ECHODATA     1  // (B) Print some of data file to screen to check
                              that the data entry is correct.
  (NEXT 3 ARE FOR COLLECTING DISTRIBUTION OF Q:)
#define ANCESTDIST   0  // (B) collect data about the distribution of an-
                              cestry coefficients (Q) for each individual
#define NUMBOXES   1000 // (int) the distribution of Q values is stored as 
                              a histogram with this number of boxes. 
#define ANCESTPINT 0.90 //(d) the size of the displayed probability  
                              interval on Q (values between 0.0--1.0)

PRIORS

#define ALPHA        1  // (d) Dirichlet parameter for degree of admixture 
                             (this is the initial value if INFERALPHA==1).
#define CORRA      2.0  // (d) CORRA, CORRB:  Prior parameters for parameter 
#define CORRB      2.0  // (d) controlling size of allele freq correlations.
                             The prior is a Gamma distribution with mean
			     CORRA*CORRB, and variance CORRA*CORRB^2. 
#define EPSILON    1.0  // (d) Dirichlet parameter for allele frequencies 
#define UNIFPRIORALPHA 1 // (B) use a uniform prior for alpha;
                                otherwise gamma prior
#define ALPHAMAX    20.0 // (d) max value of alpha if uniform prior
#define ALPHAPRIORA  1.0 // (only if UNIFPRIORALPHA==0): alpha has a gamma 
                            prior with mean A*B, and 
#define ALPHAPRIORB  2.0 // variance A*B^2.  Suggest A=1, B=2 


MISCELLANEOUS

#define ALPHAPROPSD 0.05 //(d) SD of proposal for updating alpha
#define FPROPSD     3.0  //(d) SD of proposal for updating freq correls
#define STARTATPOPINFO 0 //Use given populations as the initial condition 
                           for population origins.  (Need POPDATA==1).  It 
                           is assumed that the PopData in the input file 
                           are between 1 and k where k<=MAXPOPS.
#define RANDOMIZE    0  //(B) use new random seed for each run 
#define METROFREQ    10  //(int) Frequency of using Metropolis step to update
                           Q under admixture model (ie use the metr. move every
                           i steps).  If this is set to 0, it is never used.
                           (Proposal for each q^(i) sampled from prior.  The 
                           goal is to improve mixing for small alpha.)
#define REPORTHITRATE 0 //(B) report hit rate if using METROFREQ

```


# run STRUCTURE

Once we have our input data and parameter files, we can run structure! It will detect the param files.
I am running a few iterations at the same time, because each one uses only a single thread. Note that to ensure that the runs use a different starting seed, we must give it the seed ourself. If we use the built-in feature to choose a random starting seed, STRUCTURE will choose the seed based on the clock time... since we are starting our parallel runs at the exact same time, they would end up choosing the same starting seed (I tested this).  

I am running for 200,000 iterations of burnin followed by 1,000,000 iterations. At this length, all ten of my runs converged across K=2-4. My runs did not all converge when I ran for only 500,000 iterations, so the long length seems to be necessary for this dataset.  


First, I run a very quick run with only 10,000 burnin followed by 10,000 iterations, just to make sure that everything works as expected before I invest computing resources in a longer run.  
```bash
parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k2_10k10k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k2_10k10k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 2|g; s|NUMREPS 1000|NUMREPS 10000|g; s|BURNIN  1000|BURNIN  10000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k2_10k10k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k3_10k10k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k3_10k10k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 10000|g; s|BURNIN  1000|BURNIN  10000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k3_10k10k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_10k10k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_10k10k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 10000|g; s|BURNIN  1000|BURNIN  10000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_10k10k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k5_10k10k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k5_10k10k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 10000|g; s|BURNIN  1000|BURNIN  10000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k5_10k10k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k6_10k10k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k6_10k10k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 10000|g; s|BURNIN  1000|BURNIN  10000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k6_10k10k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
```

It looks like everything is working, it accepted the input files and produced preliminary results that look reasonable. Let's run for the full length of MCMC.  

Some runs appear not to have converged. Run for longer. I am running 10 iterations each for K=1-8 populations. K=1 is uninformative but is needed to estimate delta K in the Evanno method for choosing the best K. 8 is the maximum number of populations that I believe to be biologically realistic for this dataset based on a priori information about their biogeography (East Inambari, West Inambari, Rondonia, Tapajos, Headwaters, Xingu, and the two Atlantic Forest localities).  
```bash

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k1_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k1_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 1|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k1_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1372m52.254s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k2_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k2_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 2|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k2_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1574m57.926s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k3_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k3_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k3_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1771m29.318s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #2055m55.123s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k5_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k5_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k5_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #2187m15.556s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k6_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k6_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k6_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #2353m10.453s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k7_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k7_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 7|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k7_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #2504m1.080s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k8_200k1000k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k8_200k1000k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 8|g; s|NUMREPS 1000|NUMREPS 1000000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k8_200k1000k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #2698m23.513s


```
1,000,000 iterations seems to be sufficient. Both B=2, K=3, and K=4 converged across all 10 runs. K=5+ did not converge; maybe running for longer would help, but there might just not be a good fit to the data with K=5+.  


# Evaluate best K

We will evaluate the results with two methods: the Evanno Method and the MedMedK method. MedMedK may outperform Evanno when sampling schemes are uneven (for example, we have more samples from the western Amazon than Xingu)

First, run STRUCTURE Harvester to run the Evanno Method and to produce input files for clumpp.  
```bash
#make a folder to hold results
parallel mkdir -p Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/structureHarvester_in ::: Cerrub.AC4.37_0.2_5_20.0
#put STRUCTURE output files from each run together in one folder
parallel cp Ceratopipra_rubrocapilla/{1}.k{3}_200k1000k_POPALPHAS/run_{2}/*_f Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/structureHarvester_in/{1}.k{3}.{2}.res_f ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 ::: 1 2 3 4 5 6 7 8
#run structure harvester
time parallel /home/0_PROGRAMS/structureHarvester/structureHarvester.py --evanno --clumpp --dir Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/structureHarvester_in --out Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/structureHarvester_out ::: Cerrub.AC4.37_0.2_5_20.0 #0m9.830s #0m12.203s

```

## Evanno Method
Here are the results of STRUCTURE Harvester:
```
##########
# K     Reps    Mean LnP(K)     Stdev LnP(K)    Ln'(K)  |Ln''(K)|       Delta K
1       10      -275915.7900    35.1555 NA      NA      NA
2       10      -266991.5600    7.8907  8924.230000     7883.570000     999.100254
3       10      -265950.9000    20.0001 1040.660000     1597.040000     79.851556
4       10      -266507.2800    332.7296        -556.380000     38999.680000    117.211320
5       10      -306063.3400    72640.2208      -39556.060000   78319.680000    1.078186
6       10      -267299.7200    666.3043        38763.620000    41175.930000    61.797488
7       10      -269712.0300    503.4091        -2412.310000    961.490000      1.909957
8       10      -271162.8500    1116.0237       -1450.820000    NA      NA
```
Using the Evanno method, the best K will have the highest delta K. The highest value of Delta K is at K=2, so this is the number of populations favoured by the Evanno method. Note that K=3 has a higher lnP(K).  

Some relevant points to keep in mind:  
* the Evanno method can never support K=1, even if it would be the best biologicl description for the data.  
* "In many instances, discrete populations may not be the most valid biological description of individuals over the landscape, for example for continuously distributed populations with isolation by distance (Frantz et al. 2009). In cases where a single K value does not provide biological meaning, it is worth noting that structure can be better employed as a tool for visualizing population structure across different values of K (Jonathan Pritchard, personal communication)" (Gilbert 2016 "Identifying the number of population clusters with structure: problems and solutions")   

## MedMedK method  
This methid relies on a priori information about populations. We can use knowledge of the areas of endemism and sampling locations to assess the STRUCTURE runs. This method looks at how many of the STRUCTURE populations correspond to a priori groupings (eg based on geography) - that is, the group has at least a median of 50% assignment to that STRUCTURE population.  

For example, at K=2, The Atlantic Forest samples are assigned to one cluster, and the Inambari samples are assigned to a different cluster. This counts as 2 clusters. At K=3, the GAPTO Xingu samples are further assigned to a different cluster with more than 50% assignment, so that is another valid cluster. At K=4, one additional cluster *barely* has a median of greater than 50% assignment probability to Tapajos (median=0.5017-0.5074), so that is another valid cluster. At K=5 and higher, most runs have a ghost/spurious population that takes up most of the population assignment for all individuals. This is not a valid cluster. It counts as 1 cluster because no other populations have a median above 50% assignment in any a priori grouping. Under the MedMedK method, K=4 is favoured because a median of 4 clusters are supported across runs (all runs are actually qualitatively identical). If using MaxmedK, K=5 would be favoured because one run had 5 clusters (it separated the two Atlantic Forest localities); that method takes the max across all replicates rather than the more conservative median.  

We can report both K=2 and K=4 in the paper. They are likely showing different levels of structure in the data.  


Here are the number of clusters for the MedMedK method:  
K=2: 2 clusters  
K=3: 3 clusters  
K=4: 4 clusters  
K=5, 6, 7, or 8: 1 cluster (!)

Here is the breakdown for each of the 10 replicates for each value of K:  

| K | iteration | clusters (median assignment > 50% in any a priori grouping) |
| - | --------- | ----------------- |
| 2 | 1         | 2                 |
| 2 | 2         | 2                 |
| 2 | 3         | 2                 |
| 2 | 4         | 2                 |
| 2 | 5         | 2                 |
| 2 | 6         | 2                 |
| 2 | 7         | 2                 |
| 2 | 8         | 2                 |
| 2 | 9         | 2                 |
| 2 | 10        | 2                 |
| 3 | 1         | 3                 |
| 3 | 2         | 3                 |
| 3 | 3         | 3                 |
| 3 | 4         | 3                 |
| 3 | 5         | 3                 |
| 3 | 6         | 3                 |
| 3 | 7         | 3                 |
| 3 | 8         | 3                 |
| 3 | 9         | 3                 |
| 3 | 10        | 3                 |
| 4 | 1         | 4                 |
| 4 | 2         | 4                 |
| 4 | 3         | 4                 |
| 4 | 4         | 4                 |
| 4 | 5         | 4                 |
| 4 | 6         | 4                 |
| 4 | 7         | 4                 |
| 4 | 8         | 4                 |
| 4 | 9         | 4                 |
| 4 | 10        | 4                 |
| 5 | 1         | 1                 |
| 5 | 2         | 1                 |
| 5 | 3         | 1                 |
| 5 | 4         | 1                 |
| 5 | 5         | 1                 |
| 5 | 6         | 1                 |
| 5 | 7         | 1                 |
| 5 | 8         | 1                 |
| 5 | 9         | 5                 |
| 5 | 10        | 2                 |
| 6 | 1         | 1                 |
| 6 | 2         | 1                 |
| 6 | 3         | 1                 |
| 6 | 4         | 1                 |
| 6 | 5         | 1                 |
| 6 | 6         | 4                 |
| 6 | 7         | 1                 |
| 6 | 8         | 1                 |
| 6 | 9         | 1                 |
| 6 | 10        | 1                 |
| 7 | 1         | 1                 |
| 7 | 2         | 1                 |
| 7 | 3         | 1                 |
| 7 | 4         | 1                 |
| 7 | 5         | 1                 |
| 7 | 6         | 1                 |
| 7 | 7         | 1                 |
| 7 | 8         | 1                 |
| 7 | 9         | 1                 |
| 7 | 10        | 1                 |
| 8 | 1         | 1                 |
| 8 | 2         | 1                 |
| 8 | 3         | 1                 |
| 8 | 4         | 1                 |
| 8 | 5         | 1                 |
| 8 | 6         | 1                 |
| 8 | 7         | 1                 |
| 8 | 8         | 1                 |
| 8 | 9         | 1                 |
| 8 | 10        | 1                 |


# Combine runs

We can combine runs with clumpp.
Here is an example of the format for clumpp for K=3. There are five columns of sample information followed by columns of individual assignment for each value of k. Clumpp uses column 2 to get sample numbers, and ignores columns 1,3,4, and 5. Column 5 is usually just ":". Structure Harvester has already formatted these files for us, so we will just point clumpp towards the Structure Harvester output.  
```
  1   1 (2) 1 : 0.001 0.998 0.001
  2   2 (2) 1 : 0.001 0.996 0.004
  3   3 (26) 1 : 0.001 0.995 0.004
  4   4 (75) 1 : 0.005 0.925 0.070
  5   5 (53) 1 : 0.002 0.836 0.162
  6   6 (33) 1 : 0.093 0.732 0.175
  7   7 (43) 1 : 0.076 0.850 0.075
  8   8 (6) 1 : 0.001 0.997 0.002
  9   9 (11) 1 : 0.001 0.996 0.003
 10  10 (1) 1 : 0.030 0.636 0.334
```

clumpp requires a template file with several parameters set. Some of these can be modified while running clumpp on the command line.  
```bash
cat > clumpp.template
sed "s/^M 1/M 2/g" ./clumpp.template > clumpp_greedy.template #must make a paramfile with greedy set to 2 if wanting to use that option - changing it via command line doesn't work because the repeats option is ignored in the temlate file if M is set to 1 regardless of what it is set to in the command line.
```
```
DATATYPE 0 #0 means that the input samples are individuals (vs populations)                         
INDFILE template.indfile
POPFILE template.popfile
OUTFILE template.outfile
MISCFILE template.miscfile                
K 100 #value of K
C 100 #number of samples
R 100 #number of runs
M 1 #1 for fullsearch algorithm
W 1
S 2
GREEDY_OPTION 2
REPEATS 1000
PERMUTATIONFILE template.permutationfile
PRINT_PERMUTED_DATA 0                  
PERMUTED_DATAFILE template.perm_datafile  
PRINT_EVERY_PERM 0
EVERY_PERMFILE template.every_permfile
PRINT_RANDOM_INPUTORDER 0               
RANDOM_INPUTORDERFILE template.random_inputorderfile
OVERRIDE_WARNINGS 0
ORDER_BY_RUN 1
```

Then, run clumpp.  
```bash

#First, I run the fullsearch algorithm. This is the gold standard but too slow for K>3 with my dataset.  
parallel mkdir -p Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/clumpp_out ::: Cerrub.AC4.37_0.2_5_20.0
time parallel 'cd Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/structureHarvester_out ; /home/0_PROGRAMS/CLUMPP_Linux64.1.1.2/CLUMPP ../../../clumpp.template -k {2} -c 69  -r 10 -m 1 -i K{2}.indfile -o ../clumpp_out/k{2}.outfile -j ../clumpp_out/k{2}.miscfile' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 2 3 4 5 #K=2: almost instant; K=3: 5 mins. K=4 and 5 still had not finished after 8273m28.981s (6m20.534s) so I cancelled it.
#it gives a segfault for K=1; not a problem. /usr/bin/bash: line 1: 2660395 Segmentation fault      (core dumped) /home/0_PROGRAMS/CLUMPP_Linux64.1.1.2/CLUMPP ../../../clumpp.template -k 1 -c 69 -r 10 -m 1 -i K1.indfile -o ../../clumpp_out/k1.outfile -j ../../clumpp_out/k1.miscfile

#the output does not have sample names any more; get a list of the sample order from one of the structure runs
parallel 'sed "s/^ *//g" Ceratopipra_rubrocapilla/{1}.k2_200k1000k_POPALPHAS/run_1/Cerrub.AC4.37_0.2_5_20.0.res_q | cut -d " " -f 1 > Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/clumpp_out/sample_names' ::: Cerrub.AC4.37_0.2_5_20.0 #remove leading spaces, then take the first column - those should be the sample names.



#run with greedy search, as the fullsearch is prohibitavely slow for K higher than 3.
parallel mkdir -p Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/clumpp_out_greedy ::: Cerrub.AC4.37_0.2_5_20.0
time parallel 'sed "s/REPEATS 1000/REPEATS 10000/g" ./clumpp_greedy.template > Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/clumpp_out_greedy/clumpp_greedy.template' ::: Cerrub.AC4.37_0.2_5_20.0
time parallel 'cd Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/structureHarvester_out ; /home/0_PROGRAMS/CLUMPP_Linux64.1.1.2/CLUMPP ../clumpp_out_greedy/clumpp_greedy.template -k {2} -c 69  -r 10 -m 2 -i K{2}.indfile -o ../clumpp_out_greedy/k{2}.outfile -j ../clumpp_out_greedy/k{2}.miscfile' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 6 7 8 #2 3 4 5 #with 100 repeats: K=2 and K=3: almost instant; K=4 4 seconds; K=5 0m20.875s; with 1000 repeats: K=5 takes 3m22.562s. K=6 takes 23 min. K=7 took 3 hours 9 mins. K=8 took 1 day and 4 hours.
#K=3 gave the error "Error: some parameters have not been read correctly from paramfile ''../clumpp_out_greedy/clumpp_greedy.template'' DATATYPE was not set to a valid number."; I reran the command without any changes and it worked.

#the output does not have sample names any more; get a list of the sample order from one of the structure runs
parallel 'sed "s/^ *//g" Ceratopipra_rubrocapilla/{1}.k2_200k1000k_POPALPHAS/run_1/Cerrub.AC4.37_0.2_5_20.0.res_q | cut -d " " -f 1 > Ceratopipra_rubrocapilla/{1}_200k1000k_POPALPHAS/clumpp_out_greedy/sample_names' ::: Cerrub.AC4.37_0.2_5_20.0 #remove leading spaces, then take the first column - those should be the sample names.

```
* `-r`: Number of runs.
* `-c`: Number of individuals or populations.
* `-m`: "Method to be used (1 = FullSearch, 2 = Greedy, 3 = LargeKGreedy)"
* `-c`: Number of individuals or populations.
* `-j`: name of miscfile output file


# Plot results

I will plot the results in R; the code is in a separate .Rmd file with the same name prefix as this file.











# Archive: no Atlantic Forest
These test datasets were used to explore the dataset and ensure the results were robust.  

```bash
#min AC>4
parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k1_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k1_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 1|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k1_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 2|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #
parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_10k10k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_10k10k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 10000|g; s|BURNIN  1000|BURNIN  10000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_10k10k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #


parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k5_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k5_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k5_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k6_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k6_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k6_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k7_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k7_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 7|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k7_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k8_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k8_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 8|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k8_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1996_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

#NO XINGU

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 2|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1997_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1997_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1997_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1997_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.1997_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.1997_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #


```
# Archive: unthinned (no linkage thinning)

```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.2aa_STRUCTURE

#min AC>4

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 2|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.unthinned.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k2_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.unthinned.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.unthinned.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k3_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.unthinned.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 1000|g; s|BURNIN  1000|BURNIN  1000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.unthinned.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_1k1k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.unthinned.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

```

# Archive: shorter runs
These runs were too short; not all iterations reached convergence of the MCMC chain.
```bash
#min AC>4
parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k1_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k1_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 1|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k1_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #248m24.476s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k2_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k2_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 2|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k2_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #284m33.760s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k3_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k3_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k3_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #315m57.898s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #348m8.625s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k5_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k5_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k5_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #374m30.686s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k6_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k6_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k6_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #405m50.033s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k7_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k7_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 7|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k7_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #430m31.623s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k8_100k100k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k8_100k100k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 8|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k8_100k100k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #456m48.695s

```

Some runs appear not to have converged. Run for longer.
```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.2aa_STRUCTURE

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k1_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k1_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 1|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k1_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #785m40.580s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k2_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k2_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 2|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k2_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k3_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k3_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k3_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k4_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k4_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k4_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1217m35.297s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k5_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k5_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k5_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1355m20.872s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k6_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k6_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k6_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1458m1.062s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k7_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k7_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 7|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k7_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1564m17.431s

parallel 'mkdir -p Ceratopipra_rubrocapilla/{1}.k8_200k500k_POPALPHAS/run_{2} && cd Ceratopipra_rubrocapilla/{1}.k8_200k500k_POPALPHAS/run_{2} ; ln -s ../../../data_Ceratopipra_rubrocapilla/{1}.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 8|g; s|NUMREPS 1000|NUMREPS 500000|g; s|BURNIN  1000|BURNIN  200000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams.{1} > ./mainparams.{1}' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10
parallel 'cd Ceratopipra_rubrocapilla/{1}.k8_200k500k_POPALPHAS/run_{2} ; time structure -m mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' ::: Cerrub.AC4.37_0.2_5_20.0 ::: 1 2 3 4 5 6 7 8 9 10 #1656m35.241s

```


# Archive: inappropriate allele count filter
These datasets failed to produce biologically meaningful results. They are the same dataset as above but removing only singletons with AC=1; there is likely not ebough signal in these datasets.  
```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.2aa_STRUCTURE

mkdir -p Ceratopipra_rubrocapilla_k2/run1
cd Ceratopipra_rubrocapilla_k2/run1
ln -s ../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in .
sed "s|ITERATION|1|g" ../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams
#setup environment 
conda activate structure #conda install -y -c bioconda structure
time structure -m  mainparams -e ../../data_Ceratopipra_rubrocapilla/extraparams

#Ensure that it is interpreting the data correctly:
#Number of alleles per locus: min= 1; ave=2.0; max= 2


#run it quickly, just to ensure that the input files work and that a different seed is used in each run, before investing in a very long run.
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k2_10k10k/run_{1} && cd Ceratopipra_rubrocapilla/k2_10k10k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|NUMREPS 1000|NUMREPS 10000|g; s|BURNIN  1000|BURNIN  10000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k2_10k10k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #29m5

```
**Timing** with 10k burnin+10k MCMC, took maybe 10 mins. With 100k+100k, took 212m per run.

Then, run for longer. I am using 100,000 iterations as burnin and the length of the MCMC chain. Make sure that "RANDOMIZE" is set to "0" in the extraparams file, so that we can use a different random seed for each run (otherwise, they will all run at the exact same time and so receive the same seed, since STRUCTURE sets the seed based on time)
```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.2aa_STRUCTURE
conda activate structure #conda install -y -c bioconda structure

#Ceratopipra rubrocapilla
#run for longer with k=2
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k2_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/k2_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k2_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k3_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/k3_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k3_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k4_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/k4_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k4_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k5_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/k5_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k5_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k6_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/k6_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k6_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k7_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/k7_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 7|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k7_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #


#maxmissing 40%
#run for longer with k=2
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss40/k2_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss40/k2_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss40/k2_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #645m

#run for longer with k=3
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss40/k3_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss40/k3_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss40/k3_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #764m

#run for longer with k=4
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss40/k4_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss40/k4_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss40/k4_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #853m

#run for longer with k=5
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss40/k5_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss40/k5_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss40/k5_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #

#run for longer with k=6
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss40/k6_100k100k/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss40/k6_100k100k/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss40/k6_100k100k/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams -D $RANDOM' #

```

Next, I will run again with the setting `POPALPHAS` set to 1, which may be more appropriate than the default when sampling sizes are uneven.  
```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.2aa_STRUCTURE
conda activate structure #conda install -y -c bioconda structure

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k2_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k2_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.15_5_20.0.strct_in . ; sed "s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.15_5_20.0|g; s|NUMINDS 70|NUMINDS 66|g; s|NUMLOCI  27385|NUMLOCI  10502|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k2_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #358m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k3_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k3_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.15_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.15_5_20.0|g; s|NUMINDS 70|NUMINDS 66|g; s|NUMLOCI  27385|NUMLOCI  10502|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k3_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #360m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k4_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k4_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.15_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.15_5_20.0|g; s|NUMINDS 70|NUMINDS 66|g; s|NUMLOCI  27385|NUMLOCI  10502|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k4_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #401m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k5_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k5_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.15_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.15_5_20.0|g; s|NUMINDS 70|NUMINDS 66|g; s|NUMLOCI  27385|NUMLOCI  10502|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k5_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #449m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k6_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k6_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.15_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.15_5_20.0|g; s|NUMINDS 70|NUMINDS 66|g; s|NUMLOCI  27385|NUMLOCI  10502|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k6_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #498m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k7_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k7_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.15_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 7|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.15_5_20.0|g; s|NUMINDS 70|NUMINDS 66|g; s|NUMLOCI  27385|NUMLOCI  10502|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k7_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #552m

#20% missing data per locus, and three samples with >80% missing data
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss20/k1_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss20/k1_100k100k_POPALPHAS/run_{1} ; ln -s ../../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 1|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss20/k1_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #238m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss20/k2_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss20/k2_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss20/k2_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss20/k3_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss20/k3_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss20/k3_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #360m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss20/k4_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss20/k4_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss20/k4_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #401m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss20/k5_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss20/k5_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss20/k5_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #449m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss20/k6_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss20/k6_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss20/k6_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #498m

printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/maxmiss20/k7_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/maxmiss20/k7_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.2_5_20.0.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 7|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g; s|Cerrub.37_0.4_5_20.0|Cerrub.37_0.2_5_20.0|g; s|NUMINDS 70|NUMINDS 69|g; s|NUMLOCI  27385|NUMLOCI  12909|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/maxmiss20/k7_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #552m

#Ceratopipra rubrocapilla
#run for longer with k=2
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k2_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k2_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g; s|RANDOMIZE    1|RANDOMIZE    0|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k2_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #792m

#run for longer with k=3
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k3_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k3_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 3|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k3_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #922m

#run for longer with k=4
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k4_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k4_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 4|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k4_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #

#run for longer with k=5
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k5_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k5_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 5|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k5_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #

#run for longer with k=6
printf '%s\n' {1..10} |  parallel 'mkdir -p Ceratopipra_rubrocapilla/k6_100k100k_POPALPHAS/run_{1} && cd Ceratopipra_rubrocapilla/k6_100k100k_POPALPHAS/run_{1} ; ln -s ../../../data_Ceratopipra_rubrocapilla/Cerrub.37_0.4_5_20.strct_in . ; sed "s|MAXPOPS 2|MAXPOPS 6|g; s|NUMREPS 1000|NUMREPS 100000|g; s|BURNIN  1000|BURNIN  100000|g" ../../../data_Ceratopipra_rubrocapilla/mainparams > ./mainparams.{1}'
printf '%s\n' {1..10} |  parallel 'cd Ceratopipra_rubrocapilla/k6_100k100k_POPALPHAS/run_{1} ; time structure -m  mainparams.{1} -e ../../../data_Ceratopipra_rubrocapilla/extraparams_POPALPHAS -D $RANDOM' #


#min AC>4

```
