# Fst, dxy, and fst estimated with pixy

Here we are going to perform a genome scan of pi, dxy, and fst for *Ceratopipra rubrocapilla*. An easy way to do this is with the program pixy. There is excellent documentation for pixy [here](https://pixy.readthedocs.io/en/latest/about.html)

To run, we need a tab-delimited file with two columns. The first column lists sample names, and the second lists the population that they belong to.

First, we will separate our samples broadly into interfluve populations, with headwater populations separate
```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy
cat > Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt
sed -i 's/ /\t/g' Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt
```
Here are the contents pasted into Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt:
```
Ceratopipra_rubrocapilla_CPE012 AtlanticForest
Ceratopipra_rubrocapilla_CPE025 AtlanticForest
Ceratopipra_rubrocapilla_CPE049 AtlanticForest
Ceratopipra_rubrocapilla_CPE069 AtlanticForest
Ceratopipra_rubrocapilla_CPEII008 AtlanticForest
Ceratopipra_rubrocapilla_UNA002 AtlanticForest
Ceratopipra_rubrocapilla_UNA018 AtlanticForest
Ceratopipra_rubrocapilla_UNA088 AtlanticForest
Ceratopipra_rubrocapilla_UNA110 AtlanticForest
Ceratopipra_rubrocapilla_UNA124 AtlanticForest
Ceratopipra_rubrocapilla_AMA340 Inambari
Ceratopipra_rubrocapilla_AMA601 Inambari
Ceratopipra_rubrocapilla_AMA656 Inambari
Ceratopipra_rubrocapilla_B4615 Inambari
Ceratopipra_rubrocapilla_B5082 Inambari
Ceratopipra_rubrocapilla_BR36440 Inambari
Ceratopipra_rubrocapilla_CAM117 Inambari
Ceratopipra_rubrocapilla_CUJ027 Inambari
Ceratopipra_rubrocapilla_CUJ072 Inambari
Ceratopipra_rubrocapilla_OM022 Inambari
Ceratopipra_rubrocapilla_OM112 Inambari
Ceratopipra_rubrocapilla_OM172 Inambari
Ceratopipra_rubrocapilla_PUC008 Inambari
Ceratopipra_rubrocapilla_UFAC1007 Inambari
Ceratopipra_rubrocapilla_UFAC305 Inambari
Ceratopipra_rubrocapilla_UFAC476 Inambari
Ceratopipra_rubrocapilla_AMANA034 Rondonia
Ceratopipra_rubrocapilla_AMANA137 Rondonia
Ceratopipra_rubrocapilla_AMZ551 RondoniaHeadwaters
Ceratopipra_rubrocapilla_AMZ610 RondoniaHeadwaters
Ceratopipra_rubrocapilla_FPR006 Rondonia
Ceratopipra_rubrocapilla_FPR015 Rondonia
Ceratopipra_rubrocapilla_FPR021 Rondonia
Ceratopipra_rubrocapilla_MAM010 Rondonia
Ceratopipra_rubrocapilla_MAR092 Rondonia
Ceratopipra_rubrocapilla_MPDS628 Rondonia
Ceratopipra_rubrocapilla_OP100 RondoniaHeadwaters
Ceratopipra_rubrocapilla_PIME024 Rondonia
Ceratopipra_rubrocapilla_PIME079 Rondonia
Ceratopipra_rubrocapilla_PIME460 Rondonia
Ceratopipra_rubrocapilla_BR033 Tapajos
Ceratopipra_rubrocapilla_BR036 Tapajos
Ceratopipra_rubrocapilla_MF005 Tapajos
Ceratopipra_rubrocapilla_MSF107 Tapajos
Ceratopipra_rubrocapilla_PIME062 Rondonia
Ceratopipra_rubrocapilla_PIME134 Tapajos
Ceratopipra_rubrocapilla_PIME135 Tapajos
Ceratopipra_rubrocapilla_PIME169 Rondonia
Ceratopipra_rubrocapilla_PIME482 Tapajos
Ceratopipra_rubrocapilla_PIME521 Tapajos
Ceratopipra_rubrocapilla_PIME522 Tapajos
Ceratopipra_rubrocapilla_PPS041 TapajosHeadwaters
Ceratopipra_rubrocapilla_PPS174 Tapajos
Ceratopipra_rubrocapilla_PPS221 TapajosHeadwaters
Ceratopipra_rubrocapilla_PPS352 TapajosHeadwaters
Ceratopipra_rubrocapilla_TLP025 Tapajos
Ceratopipra_rubrocapilla_TLP123 Tapajos
Ceratopipra_rubrocapilla_WM281 Tapajos
Ceratopipra_rubrocapilla_WM293 Tapajos
Ceratopipra_rubrocapilla_GAPTO259 XinguBelem
Ceratopipra_rubrocapilla_GAPTO268 XinguBelem
Ceratopipra_rubrocapilla_GAPTO277 XinguBelem
Ceratopipra_rubrocapilla_GAPTO327 XinguBelem
Ceratopipra_rubrocapilla_MT087 XinguHeadwaters
Ceratopipra_rubrocapilla_MT123 XinguHeadwaters
Ceratopipra_rubrocapilla_PPBIO221 XinguBelem
Ceratopipra_rubrocapilla_PPBIO305 XinguBelem
Ceratopipra_rubrocapilla_RDO007 XinguBelem
Ceratopipra_rubrocapilla_UHE442 XinguBelem

```

Alternatively, I am also including a more fine-scale analysis of subpopulations (as above, but Atlantic Forest is divided into two subpopulations; Inambari is divided into East and West; XinguBelem is split into XinguN and XinguC
```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy
cat > Ceratopipra_rubrocapilla/subpopulations.txt
sed -i 's/ /\t/g' Ceratopipra_rubrocapilla/subpopulations.txt
```

```
Ceratopipra_rubrocapilla_CPE012 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE025 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE049 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE069 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPEII008 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_UNA002 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA018 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA088 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA110 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA124 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_AMA340 InambariW
Ceratopipra_rubrocapilla_AMA601 InambariW
Ceratopipra_rubrocapilla_AMA656 InambariW
Ceratopipra_rubrocapilla_B4615 InambariW
Ceratopipra_rubrocapilla_B5082 InambariW
Ceratopipra_rubrocapilla_BR36440 InambariE
Ceratopipra_rubrocapilla_CAM117 InambariW
Ceratopipra_rubrocapilla_CUJ027 InambariW
Ceratopipra_rubrocapilla_CUJ072 InambariW
Ceratopipra_rubrocapilla_OM022 InambariE
Ceratopipra_rubrocapilla_OM112 InambariE
Ceratopipra_rubrocapilla_OM172 InambariE
Ceratopipra_rubrocapilla_PUC008 InambariW
Ceratopipra_rubrocapilla_UFAC1007 InambariW
Ceratopipra_rubrocapilla_UFAC305 InambariE
Ceratopipra_rubrocapilla_UFAC476 InambariW
Ceratopipra_rubrocapilla_AMANA034 Rondonia
Ceratopipra_rubrocapilla_AMANA137 Rondonia
Ceratopipra_rubrocapilla_AMZ551 RondoniaHeadwaters
Ceratopipra_rubrocapilla_AMZ610 RondoniaHeadwaters
Ceratopipra_rubrocapilla_FPR006 Rondonia
Ceratopipra_rubrocapilla_FPR015 Rondonia
Ceratopipra_rubrocapilla_FPR021 Rondonia
Ceratopipra_rubrocapilla_MAM010 Rondonia
Ceratopipra_rubrocapilla_MAR092 Rondonia
Ceratopipra_rubrocapilla_MPDS628 Rondonia
Ceratopipra_rubrocapilla_OP100 RondoniaHeadwaters
Ceratopipra_rubrocapilla_PIME024 Rondonia
Ceratopipra_rubrocapilla_PIME079 Rondonia
Ceratopipra_rubrocapilla_PIME460 Rondonia
Ceratopipra_rubrocapilla_BR033 Tapajos
Ceratopipra_rubrocapilla_BR036 Tapajos
Ceratopipra_rubrocapilla_MF005 Tapajos
Ceratopipra_rubrocapilla_MSF107 Tapajos
Ceratopipra_rubrocapilla_PIME062 Rondonia
Ceratopipra_rubrocapilla_PIME134 Tapajos
Ceratopipra_rubrocapilla_PIME135 Tapajos
Ceratopipra_rubrocapilla_PIME169 Rondonia
Ceratopipra_rubrocapilla_PIME482 Tapajos
Ceratopipra_rubrocapilla_PIME521 Tapajos
Ceratopipra_rubrocapilla_PIME522 Tapajos
Ceratopipra_rubrocapilla_PPS041 TapajosHeadwaters
Ceratopipra_rubrocapilla_PPS174 Tapajos
Ceratopipra_rubrocapilla_PPS221 TapajosHeadwaters
Ceratopipra_rubrocapilla_PPS352 TapajosHeadwaters
Ceratopipra_rubrocapilla_TLP025 Tapajos
Ceratopipra_rubrocapilla_TLP123 Tapajos
Ceratopipra_rubrocapilla_WM281 Tapajos
Ceratopipra_rubrocapilla_WM293 Tapajos
Ceratopipra_rubrocapilla_GAPTO259 XinguC
Ceratopipra_rubrocapilla_GAPTO268 XinguC
Ceratopipra_rubrocapilla_GAPTO277 XinguC
Ceratopipra_rubrocapilla_GAPTO327 XinguC
Ceratopipra_rubrocapilla_MT087 XinguHeadwaters
Ceratopipra_rubrocapilla_MT123 XinguHeadwaters
Ceratopipra_rubrocapilla_PPBIO221 XinguN
Ceratopipra_rubrocapilla_PPBIO305 XinguN
Ceratopipra_rubrocapilla_RDO007 XinguC
Ceratopipra_rubrocapilla_UHE442 XinguN
```
Note the error "pandas.errors.ParserError: Too many columns specified: expected 2 and found 1" probably means your columns were not tab seperated.
## Populations with headwaters separated
This dataset separates Tapajos headwaters and Rondonia headwaters from their respective populations, since they may have a different ancestry than samples away from the headwaters region.  

```bash
mkdir -p /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy
cd /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy

conda activate pixy #conda install --yes -c conda-forge pixy #conda install --yes -c bioconda htslib
pixy --version #version 1.2.5.beta1

zgrep -c -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.2_5_20.0.vcf.gz #1125975 sites

#get list of autosomes to loop through
echo "chr_2 chr_1 chr_3 chr_4 chr_1A chr_5 chr_7 chr_6 chr_8 chr_9 chr_12 chr_10 chr_11 chr_4A chr_13 chr_14 chr_20 chr_15 chr_18 chr_19 chr_17 chr_21 chr_24 chr_23 chr_26 chr_27 chr_28 chr_22 chr_25 chr_29" | sed 's/ /\n/g' > scaff_names

#calculate pi, dxy, and fst using the dataset that has no minor allele frequency filter and contains both variant and invariant sites. Note that we will not use these fst estimates - we can filter SNPs more heavily for fst than would be appropriate for pi/dxy.
#Window size of 500,000
mkdir -p Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_headwaterslabelled
cat scaff_names | parallel time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.2_5_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_headwaterslabelled --output_prefix {1} --fst_type wc #16m12.386s

## Then, run the dataset that has been filtered for minor allele frequency (and otheradditional SNP filters) for Fst estimates.
zgrep -c -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.2_5_20.0.maf5.vcf.gz
#Window size of 500,000
mkdir -p Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_qualmaf5_headwaterslabelled
cat scaff_names | parallel time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.2_5_20.0.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_qualmaf5_headwaterslabelled --output_prefix {1} --fst_type wc #14m31.273s


#repeat for chrZ
#mitigate bias caused by chrW sequences in females by removing pseudoautosomal region and removing sites where any two females are heterozygous
time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZnopseudo.diploidized.2het_trimmed.37_0.2_3_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 2 --chromosomes chr_Z --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_headwaterslabelled --output_prefix chr_Znopseudo_2het_trimmed --fst_type wc #

time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.37_0.2_3_20.0.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 2 --chromosomes chr_Z --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_qualmaf5_headwaterslabelled --output_prefix chr_Znopseudo_2het_trimmed --fst_type wc #14m31.273s


#remove temp files to clear the space
rm Ceratopipra_rubrocapilla/*/pixy_tmpfile_*.tmp
```

# subpopulations
Next, I repeat while subdividing the populations into finer subpopulations

```bash
mkdir -p /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy
cd /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy

conda activate pixy #conda install --yes -c conda-forge pixy #conda install --yes -c bioconda htslib
pixy --version #version 1.2.5.beta1

#for dxy and pi
mkdir -p Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_subpops
cat scaff_names | parallel time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.2_5_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_subpops --output_prefix {1} --fst_type wc #15m

#for Fst
mkdir -p Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_qualmaf5_subpops
cat scaff_names | parallel time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.2_5_20.0.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_qualmaf5_subpops --output_prefix {1} --fst_type wc #

#repeat for chrZ
#mitigate bias caused by chrW sequences in females by removing pseudoautosomal region and removing sites where any two females are heterozygous
time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZnopseudo.diploidized.2het_trimmed.37_0.2_3_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 2 --chromosomes chr_Z --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_subpops --output_prefix chr_Znopseudo_2het_trimmed --fst_type wc #

time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZnopseudo.diploidized.2het_trimmed.37_0.2_3_20.0.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 2 --chromosomes chr_Z --output_folder Ceratopipra_rubrocapilla/window_500k_37_0.2_5_20.0_qualmaf5_subpops --output_prefix chr_Znopseudo_2het_trimmed --fst_type wc #14m31.273s

#remove temp files to clear the space
rm Ceratopipra_rubrocapilla/*/pixy_tmpfile_*.tmp
```


# Appendix: Testing other filters

To ensure that our results are robust, I also tried another dataset with the missing data filter relaxed from 0.2 to 0.4.  
I am also trying three different window sizes to better ensure that our results are robust to window size choice. I found that the patterns look qualitatively the same with all three window sizes, but that using a size of 100,000 was too noisy as there were many windows with very few sites in the GBS dataset. A window size of 500,000 seemed to be a good balance of having enough resolution to see patterns such as the dip of diversity and peak of Fst around centromeres, while still having enough sites per window such that it is not too noisy. I am also testing a dataset of chrZ which has not had the pseudoautosomal region removed, to see what effect that filter had.  

Results were qualitatively similar for all datasets in terms of the relative differences between populations, but the absolute values differ quite a bit, as expected - relaxing filters to allow more errors will inflate dxy/pi, and increasing stringency will lower it, possibly overshooting to the point of underestimation. We should therefore put little weight on the values themselves, only the patterns between population comparisons, as those patterns are very robust (eg., Atlantic Forest always has lower genetic diversity than all other populations and low dxy to Xingu, despite variation in the exact values across datasets).  

```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy

conda activate pixy #conda install --yes -c conda-forge pixy #conda install --yes -c bioconda htslib
pixy --version #version 1.2.5.beta1

#count how many sites are included
zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.4_5_20.0.vcf.gz | wc -l #2,936,625
zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZ.diploidized.37_0.4_3_20.0.vcf.gz | wc -l #171,172

## First, run the dataset that has NOT been filtered for minor allele frequency. this will be used for estimates of pi and dxy.
#Window size of 500,000
mkdir -p Ceratopipra_rubrocapilla/window_500k_headwaterslabelled
cat scaff_names | parallel time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.4_5_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_headwaterslabelled --output_prefix {1} --fst_type wc #16m12.386s
#run for chrZ that has not had the pseudoautosomal region removed
parallel time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZ.diploidized.37_0.4_3_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 5 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_headwaterslabelled --output_prefix {1} --fst_type wc ::: chr_Z #1m50.319s

## Then, run the dataset that has been filtered for minor allele frequency (and otheradditional SNP filters) for Fst estimates.
zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZ.diploidized.37_0.4_3_20.0.maf5.vcf.gz | wc -l #1552
zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.4_5_20.0.maf5.vcf.gz | wc -l #40514

#Window size of 500,000
mkdir -p Ceratopipra_rubrocapilla/window_500k_qualmaf5_headwaterslabelled
cat scaff_names | parallel time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.4_5_20.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_qualmaf5_headwaterslabelled --output_prefix {1} --fst_type wc #14m31.273s
#run for chrZ that has not had the pseudoautosomal region removed
parallel time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZ.diploidized.37_0.4_3_20.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 500000 --n_cores 5 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_qualmaf5_headwaterslabelled --output_prefix {1} --fst_type wc ::: chr_Z #1m36.303s


#Window size of 1 (individual SNPs), no MAF filter
mkdir -p Ceratopipra_rubrocapilla/window_1_headwaterslabelled
cat scaff_names | parallel time pixy --stats fst --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.4_5_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 1 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_1_headwaterslabelled --output_prefix {1} --fst_type wc #155m6.615s
#run for chrZ
parallel time pixy --stats fst --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZ.diploidized.37_0.4_3_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/populations_headwaterslabelled.txt --window_size 1 --n_cores 5 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_1_headwaterslabelled --output_prefix {1} --fst_type wc ::: chr_Z #155m6.615s

zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.4_5_20.0.vcf.gz | wc -l #2,936,625
zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZ.diploidized.37_0.4_3_20.0.vcf.gz | wc -l #171,172


#repeat for finer-scale subpopulations

#Window size of 500,000
mkdir -p Ceratopipra_rubrocapilla/window_500k_subpops
cat scaff_names | parallel time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.4_5_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_subpops --output_prefix {1} --fst_type wc #15m
#run for chrZ that has not had the pseudoautosomal region removed
parallel time pixy --stats pi fst dxy --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZ.diploidized.37_0.4_3_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 5 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_subpops --output_prefix {1} --fst_type wc ::: chr_Z 


 
#this dataset is more stringently filtered for genotype quality and excess heterozygosity
zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZ.diploidized.37_0.4_3_20.0.maf5.vcf.gz | wc -l #1552
zgrep -v "^#" /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.4_5_20.0.maf5.vcf.gz | wc -l #40514

#Window size of 500,000
mkdir -p Ceratopipra_rubrocapilla/window_500k_qualmaf5_subpops
cat scaff_names | parallel time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.37_0.4_5_20.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 2 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_qualmaf5_subpops --output_prefix {1} --fst_type wc #
#run for chrZ that has not had the pseudoautosomal region removed
parallel time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.chrZ.diploidized.37_0.4_3_20.maf5.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.txt --window_size 500000 --n_cores 5 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_500k_qualmaf5_subpops --output_prefix {1} --fst_type wc ::: chr_Z 


#remove temp files to clear the space
rm Ceratopipra_rubrocapilla/*/pixy_tmpfile_*.tmp
```










# Inambari vs Atlantic Forest
Now I want to calculate western Inambari vs Atlantic Forest, so I am using subpopulations but recombining the two AF localities. I am also excluding PUC008.
```bash
cd /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy
sed 's/AtlanticForest_Ibateguara/AtlanticForest/g; s/AtlanticForest_Ilheus/AtlanticForest/g' Ceratopipra_rubrocapilla/subpopulations.txt | grep -v "Ceratopipra_rubrocapilla_PUC008" > Ceratopipra_rubrocapilla/subpopulations.noPUC.AFcombined.txt

#Window size of 1 (individual SNPs), no MAF filter
mkdir -p Ceratopipra_rubrocapilla/window_1_subpopulationsAFcombined
cat scaff_names | parallel time pixy --stats fst --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.auto.37_0.4_5_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.noPUC.AFcombined.txt --window_size 1 --n_cores 4 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_1_subpopulationsAFcombined --output_prefix {1} --fst_type wc #530m6.763s (554m1.327s)
#run for chrZ
parallel time pixy --stats fst --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZ.diploidized.37_0.4_3_20.0.vcf.gz --populations Ceratopipra_rubrocapilla/subpopulations.noPUC.AFcombined.txt --window_size 1 --n_cores 10 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_1_subpopulationsAFcombined --output_prefix {1} --fst_type wc ::: chr_Z #128m21.090s (163m28.470s)


```


# Appendix: Identifying the pseudoautosomal region

Birds have Z and W sex chromosomes, with males generally having two Z chromosomes and females hemizygous for a Z and W. The W is quite divergent from the Z, but may be dynamic in some systems with strata at different stages of divergence from the Z (or other chromosomes that have translocated sequences to the W). If these divergent W sequences map to the Z chromosome of our reference genome (which is male and lacks a W), they can create false signals of diversity, differentiation, and divergence between populations. If populations differ in their proportions of males vs females, this could potentially severely bias our estimates. We should therefore identify and exclude these regions to reduce the impact of this bias. As one strategy, I will estimate Fst between males and females. Highly differentiated regions are likely to be flanking the pseudoautosomal region where W reads are able to map.


First, make a tab-delimited file with sample name in column 1 and sex in column 2.
```bash
cat > chrZ_investigation/populations.pruned.txt
sed -i 's/ /\t/g' chrZ_investigation/populations.pruned.txt
```
```
Ceratopipra_rubrocapilla_AMA340 M
Ceratopipra_rubrocapilla_AMA601 M
Ceratopipra_rubrocapilla_AMA656 M
Ceratopipra_rubrocapilla_AMANA034 M
Ceratopipra_rubrocapilla_AMANA137 M
Ceratopipra_rubrocapilla_AMZ551 F
Ceratopipra_rubrocapilla_AMZ610 F
Ceratopipra_rubrocapilla_B4615 M
Ceratopipra_rubrocapilla_B5082 M
Ceratopipra_rubrocapilla_BR033 F
Ceratopipra_rubrocapilla_BR036 M
Ceratopipra_rubrocapilla_BR36440 M
Ceratopipra_rubrocapilla_CAM117 M
Ceratopipra_rubrocapilla_CPE012 M
Ceratopipra_rubrocapilla_CPE025 M
Ceratopipra_rubrocapilla_CPE049 F
Ceratopipra_rubrocapilla_CPE069 M
Ceratopipra_rubrocapilla_CPEII008 M
Ceratopipra_rubrocapilla_CUJ027 F
Ceratopipra_rubrocapilla_CUJ072 F
Ceratopipra_rubrocapilla_FPR006 F
Ceratopipra_rubrocapilla_FPR015 M
Ceratopipra_rubrocapilla_FPR021 M
Ceratopipra_rubrocapilla_GAPTO259 F
Ceratopipra_rubrocapilla_GAPTO268 F
Ceratopipra_rubrocapilla_GAPTO277 M
Ceratopipra_rubrocapilla_GAPTO327 F
Ceratopipra_rubrocapilla_MAM010 M
Ceratopipra_rubrocapilla_MAR092 M
Ceratopipra_rubrocapilla_MF005 M
Ceratopipra_rubrocapilla_MPDS628 F
Ceratopipra_rubrocapilla_MSF107 F
Ceratopipra_rubrocapilla_MT087 F
Ceratopipra_rubrocapilla_MT123 M
Ceratopipra_rubrocapilla_OM022 F
Ceratopipra_rubrocapilla_OM112 M
Ceratopipra_rubrocapilla_OM172 F
Ceratopipra_rubrocapilla_OP100 F
Ceratopipra_rubrocapilla_PIME024 M
Ceratopipra_rubrocapilla_PIME062 M
Ceratopipra_rubrocapilla_PIME079 F
Ceratopipra_rubrocapilla_PIME134 M
Ceratopipra_rubrocapilla_PIME135 F
Ceratopipra_rubrocapilla_PIME169 F
Ceratopipra_rubrocapilla_PIME460 M
Ceratopipra_rubrocapilla_PIME482 F
Ceratopipra_rubrocapilla_PIME521 M
Ceratopipra_rubrocapilla_PIME522 F
Ceratopipra_rubrocapilla_PPBIO221 F
Ceratopipra_rubrocapilla_PPBIO305 F
Ceratopipra_rubrocapilla_PPS041 M
Ceratopipra_rubrocapilla_PPS174 F
Ceratopipra_rubrocapilla_PPS221 M
Ceratopipra_rubrocapilla_PPS352 M
Ceratopipra_rubrocapilla_PUC008 M
Ceratopipra_rubrocapilla_RDO007 F
Ceratopipra_rubrocapilla_TLP025 F
Ceratopipra_rubrocapilla_TLP123 M
Ceratopipra_rubrocapilla_UFAC1007 M
Ceratopipra_rubrocapilla_UFAC305 F
Ceratopipra_rubrocapilla_UFAC476 F
Ceratopipra_rubrocapilla_UHE442 M
Ceratopipra_rubrocapilla_UNA002 F
Ceratopipra_rubrocapilla_UNA018 M
Ceratopipra_rubrocapilla_UNA088 M
Ceratopipra_rubrocapilla_UNA110 M
Ceratopipra_rubrocapilla_UNA124 M
Ceratopipra_rubrocapilla_WM281 F
Ceratopipra_rubrocapilla_WM293 F
```

Then, run pixy. I am using a more relaxed dataset (allowing up to 40% missing data) since I am not trying to get accurate estimates, I just want to look at the distribution across the chromosome.
```bash
mkdir -p /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy/chrZ_investigation
cd /home/1_Else/Ceratopipra/3_Structure/3.3aa_pixy/chrZ_investigation

conda activate pixy #conda install --yes -c conda-forge pixy #conda install --yes -c bioconda htslib
pixy --version #version 1.2.5.beta1

parallel time pixy --stats fst --bypass_invariant_check yes --vcf /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.for_pixy.chrZ.diploidized.37_0.4_3_20.0.vcf.gz --populations populations.pruned.txt --window_size 1 --n_cores 50 --chromosomes \'{1}\' --output_folder Ceratopipra_rubrocapilla/window_1bp_chrZ --output_prefix {1} --fst_type wc ::: chr_Z 
#136m41.813s (168m19.549s)
```

There is a large peak near the start of chrZ, which is where the pseudoautosomal seems to be in other manakins as well. To be conservative, I will exclude the entire first 3 Mb of the chromosome. I'd rather exclude too much than too little. However, there are scattered peaks of high Fst in other areas as well, so we should still implement another filter: removing sites where females are called as heterozygous.