# A gradient across the Amazon

This page describes the pipeline used to calculate an index that describes how similar a sample is to the Western Inambari vs Atlantic Forest populations.  

To do this, I will polarize alleles according to whether they are at higher frequency in the Atlantic Forest or Western Inambari. Then, I will calculate the proportion of each sample's alleles that are more frequent in the Atlantic Forest vs Western Inambari.

First, I make a list of samples from the two populations that I will use to polarize the alleles. I will do 100 replicates in which I randomly susbsample 8 per populations.   
```bash
mkdir -p /home/1_Else/Ceratopipra/3_Structure/3.5aa_gradients/shuffled
cd /home/1_Else/Ceratopipra/3_Structure/3.5aa_gradients/shuffled

cat > AF.txt
```
```
Ceratopipra_rubrocapilla_CPE012
Ceratopipra_rubrocapilla_CPE025
Ceratopipra_rubrocapilla_CPE049
Ceratopipra_rubrocapilla_CPE069
Ceratopipra_rubrocapilla_CPEII008
Ceratopipra_rubrocapilla_UNA002
Ceratopipra_rubrocapilla_UNA018
Ceratopipra_rubrocapilla_UNA088
Ceratopipra_rubrocapilla_UNA110
Ceratopipra_rubrocapilla_UNA124
```
```bash
cat > Inambari.txt
```
```
Ceratopipra_rubrocapilla_AMA340
Ceratopipra_rubrocapilla_AMA601
Ceratopipra_rubrocapilla_AMA656
Ceratopipra_rubrocapilla_B4615
Ceratopipra_rubrocapilla_B5082
Ceratopipra_rubrocapilla_BR36440
Ceratopipra_rubrocapilla_CAM117
Ceratopipra_rubrocapilla_CUJ027
Ceratopipra_rubrocapilla_CUJ072
Ceratopipra_rubrocapilla_OM022
Ceratopipra_rubrocapilla_OM112
Ceratopipra_rubrocapilla_OM172
Ceratopipra_rubrocapilla_PUC008
Ceratopipra_rubrocapilla_UFAC1007
Ceratopipra_rubrocapilla_UFAC305
Ceratopipra_rubrocapilla_UFAC476
```

The first step is to repolarize the VCF according to which allele is higher frequency in the Atlantic Forest vs Inambari.
```bash
mkdir -p subsamples

#shuffle the order of the samples in the file randomly, so that we can later subset them by picking the first N lines. Save the order to a file so that our pipeline is easily replicable.
seq 1 100 | parallel 'shuf AF.txt > subsamples/AF.{1}.txt'
seq 1 100 | parallel 'shuf Inambari.txt > subsamples/Inambari.{1}.txt'

#choose which VCF file we will analyze, and choose a tagname that will be used to name output files, in case we want to run this multiple times without overwriting
VCF=/home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0.vcf.gz
TAGNAME=Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0_subsetted
Dataset=window_1_subpopulationsAFcombined_AC0.37_0.2_5_20.0

#make a directory within which to save the output
mkdir -p Ceratopipra_rubrocapilla/"$Dataset"

#count genotypes in Atlantic Forest. Select a subset of 8 samples randomly 100 times - take these as the first 8 lines of our shuffled sample list. First I check whether the file exists so that I don't accidentally rewrite it. Then I subset the VCF to keep only the 8 AF samples that were chosen randomly. There is a command within the command that takes the first 8 lines of the sample list and then formats it into a comma-separated list that bcftools expects. After subsetting the VCF, I pass it to my script for counting genotypes (homozygous ref, alt, or heterozygous)
seq 1 100 | parallel 'if [ ! -f Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.AtlanticForest.{1}.allelecounts ] ; then /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples $(head -n 8 subsamples/AF.{1}.txt | sed "s/^/,/g" | tr -d "\n" | sed "s/^,//g") '$VCF' | /home/0_PROGRAMS/elsemikk_scripts/count_genotypes.awk > Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.AtlanticForest.{1}.allelecounts ; else echo "Error: file already exists" ; fi'

#repeat for Inambari
seq 1 100 | parallel 'if [ ! -f Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.Inambari.{1}.allelecounts ] ; then /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples $(head -n 8 subsamples/Inambari.{1}.txt | sed "s/^/,/g" | tr -d "\n" | sed "s/^,//g") '$VCF' | /home/0_PROGRAMS/elsemikk_scripts/count_genotypes.awk > Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.Inambari.{1}.allelecounts ; else echo "Error: file already exists" ; fi'

#determine which population has a higher frequency of reference alleles. 
#Make a list of sites where AF is higher. Sites where the two are equal will get skipped over. These sites will have their polarity reversed
seq 1 100 | parallel 'paste Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.Inambari.{1}.allelecounts Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.AtlanticForest.{1}.allelecounts | awk -v OFS='\''\t'\'' '\''$2 == $7 && ((2*$5)+$4)/(2*($5+$4+$3)) < ((2*$10)+$9)/(2*($10+$9+$8)) {print $1, $2}'\'' > Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toReversePolarity.{1}.sites'

#and a list of sites where Inambari is higher. Sites where the two are equal will get skipped over. These sites will not have their polarity reversed
seq 1 100 | parallel 'paste Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.Inambari.{1}.allelecounts Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.sites.AtlanticForest.{1}.allelecounts | awk -v OFS='\''\t'\'' '\''$2 == $7 && ((2*$5)+$4)/(2*($5+$4+$3)) > ((2*$10)+$9)/(2*($10+$9+$8)) {print $1, $2}'\'' > Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toNOTReversePolarity.{1}.sites'

#reverse polarity of sites where Atlantic Forest has higher frequency of reference allele
seq 1 100 | parallel '/home/0_PROGRAMS/bcftools-1.17/bcftools view --regions-file Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toReversePolarity.{1}.sites '$VCF'  | /home/0_PROGRAMS/elsemikk_scripts/reverse_VCF_polarity.awk  | bgzip > Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toReversePolarity.repolarized.{1}.vcf.gz'
seq 1 100 | parallel '/home/0_PROGRAMS/bcftools-1.17/bcftools index Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toReversePolarity.repolarized.{1}.vcf.gz'

#extract the sites that do not need to have their polarity reversed
seq 1 100 | parallel '/home/0_PROGRAMS/bcftools-1.17/bcftools view --regions-file Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toNOTReversePolarity.{1}.sites '$VCF' -Oz -o Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toNOTReversePolarity.temp.{1}.vcf.gz'
seq 1 100 | parallel '/home/0_PROGRAMS/bcftools-1.17/bcftools index Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toNOTReversePolarity.temp.{1}.vcf.gz'

#put the reversed-polarity sites back together with the rest of the sites that did not need to be reversed
seq 1 100 | parallel '/home/0_PROGRAMS/bcftools-1.17/bcftools concat --allow-overlaps Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toNOTReversePolarity.temp.{1}.vcf.gz Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.toReversePolarity.repolarized.{1}.vcf.gz -Oz -o Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.repolarized.{1}.vcf.gz'

#finally, count the number of homozygous ref, alt, and heterozygous sites for each sample in the repolarized VCF file
seq 1 100 | parallel 'zcat Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.repolarized.{1}.vcf.gz | /home/0_PROGRAMS/elsemikk_scripts/count_genotypes_persample.awk > Ceratopipra_rubrocapilla/'$Dataset'/'$TAGNAME'.gradients.{1}.txt'

for sample in Ceratopipra_rubrocapilla_AMANA034 Ceratopipra_rubrocapilla_AMANA137 Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_FPR006 Ceratopipra_rubrocapilla_FPR015 Ceratopipra_rubrocapilla_FPR021 Ceratopipra_rubrocapilla_MAM010 Ceratopipra_rubrocapilla_MAR092 Ceratopipra_rubrocapilla_MPDS628 Ceratopipra_rubrocapilla_OP100 Ceratopipra_rubrocapilla_PIME024 Ceratopipra_rubrocapilla_PIME079 Ceratopipra_rubrocapilla_PIME460 Ceratopipra_rubrocapilla_BR033 Ceratopipra_rubrocapilla_BR036 Ceratopipra_rubrocapilla_MF005 Ceratopipra_rubrocapilla_MSF107 Ceratopipra_rubrocapilla_PIME062 Ceratopipra_rubrocapilla_PIME134 Ceratopipra_rubrocapilla_PIME135 Ceratopipra_rubrocapilla_PIME169 Ceratopipra_rubrocapilla_PIME482 Ceratopipra_rubrocapilla_PIME521 Ceratopipra_rubrocapilla_PIME522 Ceratopipra_rubrocapilla_PPS041 Ceratopipra_rubrocapilla_PPS174 Ceratopipra_rubrocapilla_PPS221 Ceratopipra_rubrocapilla_PPS352 Ceratopipra_rubrocapilla_TLP025 Ceratopipra_rubrocapilla_TLP123 Ceratopipra_rubrocapilla_WM281 Ceratopipra_rubrocapilla_WM293 Ceratopipra_rubrocapilla_GAPTO259 Ceratopipra_rubrocapilla_GAPTO268 Ceratopipra_rubrocapilla_GAPTO277 Ceratopipra_rubrocapilla_GAPTO327 Ceratopipra_rubrocapilla_MT087 Ceratopipra_rubrocapilla_MT123 Ceratopipra_rubrocapilla_PPBIO221 Ceratopipra_rubrocapilla_PPBIO305 Ceratopipra_rubrocapilla_RDO007 Ceratopipra_rubrocapilla_UHE442 ; do grep "$sample" Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.*.txt ; done

printf "SampleID\tHomAlt\tHet\tHomRef\n" > Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.averaged.txt
for sample in Ceratopipra_rubrocapilla_AMANA034 Ceratopipra_rubrocapilla_AMANA137 Ceratopipra_rubrocapilla_AMZ551 Ceratopipra_rubrocapilla_AMZ610 Ceratopipra_rubrocapilla_FPR006 Ceratopipra_rubrocapilla_FPR015 Ceratopipra_rubrocapilla_FPR021 Ceratopipra_rubrocapilla_MAM010 Ceratopipra_rubrocapilla_MAR092 Ceratopipra_rubrocapilla_MPDS628 Ceratopipra_rubrocapilla_OP100 Ceratopipra_rubrocapilla_PIME024 Ceratopipra_rubrocapilla_PIME079 Ceratopipra_rubrocapilla_PIME460 Ceratopipra_rubrocapilla_BR033 Ceratopipra_rubrocapilla_BR036 Ceratopipra_rubrocapilla_MF005 Ceratopipra_rubrocapilla_MSF107 Ceratopipra_rubrocapilla_PIME062 Ceratopipra_rubrocapilla_PIME134 Ceratopipra_rubrocapilla_PIME135 Ceratopipra_rubrocapilla_PIME169 Ceratopipra_rubrocapilla_PIME482 Ceratopipra_rubrocapilla_PIME521 Ceratopipra_rubrocapilla_PIME522 Ceratopipra_rubrocapilla_PPS041 Ceratopipra_rubrocapilla_PPS174 Ceratopipra_rubrocapilla_PPS221 Ceratopipra_rubrocapilla_PPS352 Ceratopipra_rubrocapilla_TLP025 Ceratopipra_rubrocapilla_TLP123 Ceratopipra_rubrocapilla_WM281 Ceratopipra_rubrocapilla_WM293 Ceratopipra_rubrocapilla_GAPTO259 Ceratopipra_rubrocapilla_GAPTO268 Ceratopipra_rubrocapilla_GAPTO277 Ceratopipra_rubrocapilla_GAPTO327 Ceratopipra_rubrocapilla_MT087 Ceratopipra_rubrocapilla_MT123 Ceratopipra_rubrocapilla_PPBIO221 Ceratopipra_rubrocapilla_PPBIO305 Ceratopipra_rubrocapilla_RDO007 Ceratopipra_rubrocapilla_UHE442 ; do grep "$sample" Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.*.txt | awk '{ homref += $2; het += $3; homalt += $4; n++ } END { if (n > 0) print "'$sample'\t" homref / n "\t" het / n "\t" homalt / n; }' >> Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.averaged.txt ; done

#for Atlantic Forest, only count iterations where a given sample was not included in the polarization
for sample in Ceratopipra_rubrocapilla_CPE012 Ceratopipra_rubrocapilla_CPE025 Ceratopipra_rubrocapilla_CPE049 Ceratopipra_rubrocapilla_CPE069 Ceratopipra_rubrocapilla_CPEII008 Ceratopipra_rubrocapilla_UNA002 Ceratopipra_rubrocapilla_UNA018 Ceratopipra_rubrocapilla_UNA088 Ceratopipra_rubrocapilla_UNA110 Ceratopipra_rubrocapilla_UNA124 ; do seq 1 100 | while read number ; do if tail -n 2 subsamples/AF."$number".txt | grep -q "$sample" ; then grep "$sample" Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients."$number".txt >>  Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.sample_"$sample".txt ; fi ; done ; done
wc -l Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.sample_*.txt
for sample in Ceratopipra_rubrocapilla_CPE012 Ceratopipra_rubrocapilla_CPE025 Ceratopipra_rubrocapilla_CPE049 Ceratopipra_rubrocapilla_CPE069 Ceratopipra_rubrocapilla_CPEII008 Ceratopipra_rubrocapilla_UNA002 Ceratopipra_rubrocapilla_UNA018 Ceratopipra_rubrocapilla_UNA088 Ceratopipra_rubrocapilla_UNA110 Ceratopipra_rubrocapilla_UNA124 ; do awk '{ homref += $2; het += $3; homalt += $4; n++ } END { if (n > 0) print "'$sample'\t" homref / n "\t" het / n "\t" homalt / n; }' Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.sample_"$sample".txt >> Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.averaged.txt ; done

#repeat for Inambari
for sample in Ceratopipra_rubrocapilla_AMA340 Ceratopipra_rubrocapilla_AMA601 Ceratopipra_rubrocapilla_AMA656 Ceratopipra_rubrocapilla_B4615 Ceratopipra_rubrocapilla_B5082 Ceratopipra_rubrocapilla_BR36440 Ceratopipra_rubrocapilla_CAM117 Ceratopipra_rubrocapilla_CUJ027 Ceratopipra_rubrocapilla_CUJ072 Ceratopipra_rubrocapilla_OM022 Ceratopipra_rubrocapilla_OM112 Ceratopipra_rubrocapilla_OM172 Ceratopipra_rubrocapilla_PUC008 Ceratopipra_rubrocapilla_UFAC1007 Ceratopipra_rubrocapilla_UFAC305 Ceratopipra_rubrocapilla_UFAC476 ; do seq 1 100 | while read number ; do if tail -n 8 subsamples/Inambari."$number".txt | grep -q "$sample" ; then grep "$sample" Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients."$number".txt >>  Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.sample_"$sample".txt ; fi ; done ; done
for sample in Ceratopipra_rubrocapilla_AMA340 Ceratopipra_rubrocapilla_AMA601 Ceratopipra_rubrocapilla_AMA656 Ceratopipra_rubrocapilla_B4615 Ceratopipra_rubrocapilla_B5082 Ceratopipra_rubrocapilla_BR36440 Ceratopipra_rubrocapilla_CAM117 Ceratopipra_rubrocapilla_CUJ027 Ceratopipra_rubrocapilla_CUJ072 Ceratopipra_rubrocapilla_OM022 Ceratopipra_rubrocapilla_OM112 Ceratopipra_rubrocapilla_OM172 Ceratopipra_rubrocapilla_PUC008 Ceratopipra_rubrocapilla_UFAC1007 Ceratopipra_rubrocapilla_UFAC305 Ceratopipra_rubrocapilla_UFAC476 ; do awk '{ homref += $2; het += $3; homalt += $4; n++ } END { if (n > 0) print "'$sample'\t" homref / n "\t" het / n "\t" homalt / n; }' Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.sample_"$sample".txt >> Ceratopipra_rubrocapilla/$Dataset/$TAGNAME.gradients.averaged.txt ; done

```


# Appendix
Here are my awk scripts used above:  

## count_genotypes

```bash
cat > /home/0_PROGRAMS/elsemikk_scripts/count_genotypes.awk
chmod +x /home/0_PROGRAMS/elsemikk_scripts/count_genotypes.awk

#run it like this:
zcat /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/thinned_datasets/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.37_0.2_8_20.00.vcf.gz | /home/0_PROGRAMS/elsemikk_scripts/count_genotypes.awk  | less
```
```awk
#!/usr/bin/awk -f 

BEGIN {
    FS = "\t"
    OFS = FS
}

#skip the header
/^#/ {
    next
}

{
	#make two variables to hold the counts of the genotypes. 
	homref=0
	het=0
	homalt=0

	#run through all the samples, which start at column 10 in a standard VCF file
	for (i = 10; i <= NF; i++) {
		split($(i), sample, ":") #split the column using ":" as a field separator. The genotype will be the first field. Name this new array "sample"
		if (sample[1] ~ /1\/1|1\|1/) { #test whether the genotype is 1/1 or 1|1
			homalt++ #add to the count
		}
		else if (sample[1] ~ /0\/1|1\/0|0\|1|1\|0/) { #test whether the genotype is 0/1 or 1/0 or 0|1 or 1|0
			het++ #add to the count
		}
		else if (sample[1] ~ /0\/0|0\|0/) { #test whether the genotype is 0/0 or 0|0
			homref++ #if it is any of the above genotypes, the site passes the test and we can stop looking through the rest of the samples
		}
	}

	#print results
	print $1,$2, homalt, het, homref
}
```

## count_genotypes_persample

```bash
cat > /home/0_PROGRAMS/elsemikk_scripts/count_genotypes_persample.awk
chmod +x /home/0_PROGRAMS/elsemikk_scripts/count_genotypes_persample.awk

#run it like this:
zcat /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/thinned_datasets/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.37_0.2_8_20.00.vcf.gz | /home/0_PROGRAMS/elsemikk_scripts/count_genotypes_persample.awk  | less
```
```awk
#!/usr/bin/awk -f 

BEGIN {
    FS = "\t"
    OFS = FS
}

#skip the header
/^##/ {
    next
}

#extract sample names and initialize genotype counts at zero
/^#CHROM/ {
	for (i = 10; i <= NF; i++) {
		samplename[i]=$(i)
		homalt[i]=0
		het[i]=0
		homref[i]=0
	}
}

{
	#run through genotypes of all the samples, which start at column 10 in a standard VCF file
	for (i = 10; i <= NF; i++) {
		

		split($(i), sample, ":") #split the column using ":" as a field separator. The genotype will be the first field. Name this new array "sample"
		if (sample[1] ~ /1\/1|1\|1/) { #test whether the genotype is 1/1 or 1|1
			homalt[i]++ #add to the count
		}
		else if (sample[1] ~ /0\/1|1\/0|0\|1|1\|0/) { #test whether the genotype is 0/1 or 1/0 or 0|1 or 1|0
			het[i]++ #add to the count
		}
		else if (sample[1] ~ /0\/0|0\|0/) { #test whether the genotype is 0/0 or 0|0
			homref[i]++ #if it is any of the above genotypes, the site passes the test and we can stop looking through the rest of the samples
		}
	}

}

#print results
END {
	print "SampleID", "HomAlt", "Het", "HomRef"
	for (i = 10; i <= NF; i++) {
		print samplename[i], homalt[i], het[i], homref[i]
	}
}

```

## reverse_VCF_polarity
```bash
cat > /home/0_PROGRAMS/elsemikk_scripts/reverse_VCF_polarity.awk
chmod +x /home/0_PROGRAMS/elsemikk_scripts/reverse_VCF_polarity.awk

#run it like this:
zcat /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/thinned_datasets/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.thinned10k.forSVD.37_0.2_8_20.00.vcf.gz | /home/0_PROGRAMS/elsemikk_scripts/reverse_VCF_polarity.awk  | less
```
```awk
#!/usr/bin/awk -f 

BEGIN {
    FS = "\t"
    OFS = FS
}

#print the header
/^#/ {
    print
    next
}


{
	#switch column 4 and column 5 to switch which allele is ref vs alt
	temp = $4
	$4 = $5
	$5 = temp
	#run through all the samples, which start at column 10 in a standard VCF file. Reverse their genotypes to match the repolarized alleles
	for (i = 10; i <= NF; i++) {
       split($(i), geno, ":")

       if (geno[1] ~ /0\/0|0\|0/) #test whether the genotype is 0/0 or 0|0
           geno[1] = "1/1"
       else if (geno[1] ~ /1\/1|1\|1/) #test whether the genotype is 1/1 or 1|1
           geno[1] = "0/0"
       $(i) = geno[1]
       #put the repolarized genotypes back together with their VCF fields
       for (j = 2; j <= length(geno); j++) {
           $(i) = $(i) ":" geno[j]
       }
    }
	#print results
	print
}
```

Here is the R code that I used to plot the data on a map:  
```R
setwd("/Users/elsemikkelsen/Documents/GitHub/Ceratopipra/Writing/Manuscript_drafts/Figures/Figure1.1_map")

# Load required libraries
library(sf)
library(ggplot2)
library(tidyverse)
library(ggrepel)
library(rgdal)
library(scatterpie)

#load species range, obtained with permission from BirdLife International
load("Cerrub_range.RData")

#read the shapefile of rivers
## Rivers
# Download the shapefile including rivers of Brazil from Natural Earth
#download.file("https://www.naturalearthdata.com/http//www.naturalearthdata.com/download/10m/physical/ne_10m_rivers_lake_centerlines.zip", "brazil_rivers.zip")
#unzip("brazil_rivers.zip", exdir = "brazil_rivers_shapefile")
brazil_rivers <- st_read("brazil_rivers_shapefile/ne_10m_rivers_lake_centerlines.shp")
# Filter to keep only major rivers of interest (otherwise the map would be too messy and hard to read with all the small rivers plotted)
brazil_rivers <- brazil_rivers[brazil_rivers$name %in% c("Amazonas", "Madeira", "Tapajós", "Xingu", "Tocantins", "Rio Negro", "Teles Pires", "Juruena", "Jamanxim", "Arapiuns", "Crepori", "Cupari", "Purus", "Baía de Marajó", "São Francisco", "Araguaia", "Madre de Dios", "Marañón", "Branco", "Orinoco", "Purús", "Ucayali", "Japurá", "Caquetá"), ] #"Braco Menor", "Mamoré", "Juruá", "Jari", "Putumayo", 

## Coastline
# Download the shapefile including the coastline in Brazil from Natural Earth
#download.file("https://www.naturalearthdata.com/http//www.naturalearthdata.com/download/10m/physical/ne_10m_coastline.zip", "brazil_coastline.zip")
#unzip("brazil_coastline.zip", exdir = "brazil_coastline_shapefile")
#read the shapefile of coastline
brazil_coastline <- st_read("brazil_coastline_shapefile/ne_10m_coastline.shp")

#load the Ceratopipra sample metadata
Ceratopipra_metadata <- read.delim(file="~/Documents/GitHub/Ceratopipra/3_Structure/3.1aa_PCA/Ceratopipra_metadata.txt", sep="\t", header=FALSE)
colnames(Ceratopipra_metadata)<-c("SampleID", "Species", "Population", "Locality", "Latitude", "Longitude", "Code", "Haplogroup") 
#ensure that latitude and longitude arter treated as numerics
Ceratopipra_metadata$Latitude <- as.numeric(Ceratopipra_metadata$Latitude)
Ceratopipra_metadata$Longitude <- as.numeric(Ceratopipra_metadata$Longitude)

#load the genetic gradient data
AF_I_highFst.gradients <- read.delim("~/Documents/GitHub/Ceratopipra/3_Structure/3.5aa_gradients/shuffled/Ceratopipra_rubrocapilla/window_1_subpopulationsAFcombined_AC0.37_0.2_5_20.0/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0_subsetted.gradients.averaged.txt", header=TRUE)

#exclude any outgroups, if they are included
samples_to_exclude <- c("none")

#add metadata to the dataset
AF_I_highFst.gradients <- merge(AF_I_highFst.gradients, Ceratopipra_metadata, by="SampleID")
AF_I_highFst.gradients$gradient <- (2*AF_I_highFst.gradients$HomAlt+AF_I_highFst.gradients$Het)/(2*(AF_I_highFst.gradients$HomRef+AF_I_highFst.gradients$Het+AF_I_highFst.gradients$HomAlt))
AF_I_highFst.gradients$sites <- AF_I_highFst.gradients$HomRef+AF_I_highFst.gradients$Het+AF_I_highFst.gradients$HomAlt
AF_I_highFst.gradients$index <- (2*AF_I_highFst.gradients$HomAlt+AF_I_highFst.gradients$Het)/(2*(AF_I_highFst.gradients$HomRef+AF_I_highFst.gradients$Het+AF_I_highFst.gradients$HomAlt))
AF_I_highFst.gradients$index_rescaled <- (AF_I_highFst.gradients$index-min(AF_I_highFst.gradients$index))
AF_I_highFst.gradients$index_rescaled <- (AF_I_highFst.gradients$index_rescaled/max(AF_I_highFst.gradients$index_rescaled))

#plot the map
ggplot() +
  geom_sf(data = Cerrub_range, color = "#ffcccb", fill = "#ffcccb") +
  geom_sf(data = brazil_rivers, color = "#38afcd", linewidth=1) +
  geom_sf(data = brazil_coastline, color = "black") +
  coord_sf(xlim = c(-80, -33), ylim = c(-24, 0), expand = FALSE) +
  geom_point(data=AF_I_highFst.gradients %>% filter(!SampleID %in% samples_to_exclude) %>% filter(sites > 10000), aes(x=Longitude, y=Latitude, color=index_rescaled), size=4, stroke=NA)+
  scale_color_viridis_c(option = "turbo")+
   theme_void()
```
After generating the map, I mannually jittered points that were overlapping so that all points were visible (I did this mannually so as to only jitter points that were overlapping, as little as possible, and to ensure that the jitter did not make samples appear to cross a river).

