# Tajima's D

This page describes the pipeline used to estimate Tajima's D for populations of *Ceratopipra rubrocapilla*. 

Tajima's D is essentially the difference between the observed and expected sequence diversity of a population. It uses a count of the number of segregating sites to estimate the amount of sequence diversity expected at equilibrium (theta), and compares this to the average number of pairwise differences between samples, which is expected to be equal to theta at equilibrium. After a selective sweep or recent growth in population size, there will be an excess of rare variants, and Tajima's D will be negative. Balancing selection or a recent decrease in population size can lead to positive Tajima's D (excess of common variants or deficit of rare variants).

Tajima's D can be calculated with the program VCF-kit. There is documentation about this function [here](https://github.com/AndersenLab/VCF-kit/blob/master/docs/tajima.md). vk will use biallelic SNPs. It will not consider multiallelic or invariant sites in the calculation.

The formula for Tajima's D is slightly different for haploid and diploid cases. VCF-kit calculates D for diploid sites, so I am not including the Z chromosome, which is hemizygous in females.  

I will use a dataset of filtered SNPs. 

## Estimate Tajima's D  
We generated this dataset: `/home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0.vcf.gz`  
Which has been filtered for:  
* biallelic SNPs  
* min qual 25  
* min GQ 20  
* max 20% missing data  
* remove sites with ExcHet<0.01  
* no minor allele count filter (important!)  
* max average depth 37 x per site  
* min 5 x depth per genotype

```bash
mkdir -p /home/1_Else/Ceratopipra/4_Demography/4.2_tajimasD
cd /home/1_Else/Ceratopipra/4_Demography/4.2_tajimasD

conda activate vcf-kit #conda create -n vcf-kit danielecook::vcf-kit=0.2.6 "bwa>=0.7.17" "samtools>=1.10" "bcftools>=1.10" "blast>=2.2.31" "muscle>=3.8.31" "primer3>=2.5.0"

mkdir -p Ceratopipra_rubrocapilla

# using 500 kb windows
#Inambari
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples Ceratopipra_rubrocapilla_AMA340,Ceratopipra_rubrocapilla_AMA601,Ceratopipra_rubrocapilla_AMA656,Ceratopipra_rubrocapilla_B4615,Ceratopipra_rubrocapilla_B5082,Ceratopipra_rubrocapilla_BR36440,Ceratopipra_rubrocapilla_CAM117,Ceratopipra_rubrocapilla_CUJ027,Ceratopipra_rubrocapilla_CUJ072,Ceratopipra_rubrocapilla_OM022,Ceratopipra_rubrocapilla_OM112,Ceratopipra_rubrocapilla_OM172,Ceratopipra_rubrocapilla_PUC008,Ceratopipra_rubrocapilla_UFAC1007,Ceratopipra_rubrocapilla_UFAC305,Ceratopipra_rubrocapilla_UFAC476 /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0.vcf.gz | vk tajima 500000 500000 - > Ceratopipra_rubrocapilla/Inambari.500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima #0m6.854s
#Rondonia (excluding headwaters) 
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples Ceratopipra_rubrocapilla_AMANA034,Ceratopipra_rubrocapilla_AMANA137,Ceratopipra_rubrocapilla_FPR006,Ceratopipra_rubrocapilla_FPR015,Ceratopipra_rubrocapilla_FPR021,Ceratopipra_rubrocapilla_MAM010,Ceratopipra_rubrocapilla_MAR092,Ceratopipra_rubrocapilla_MPDS628,Ceratopipra_rubrocapilla_PIME024,Ceratopipra_rubrocapilla_PIME079,Ceratopipra_rubrocapilla_PIME460,Ceratopipra_rubrocapilla_PIME062,Ceratopipra_rubrocapilla_PIME169 /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0.vcf.gz | vk tajima 500000 500000 - > Ceratopipra_rubrocapilla/Rondonia.500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima 
#Tapajos (excluding headwaters)
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples Ceratopipra_rubrocapilla_BR033,Ceratopipra_rubrocapilla_BR036,Ceratopipra_rubrocapilla_MF005,Ceratopipra_rubrocapilla_MSF107,Ceratopipra_rubrocapilla_PIME134,Ceratopipra_rubrocapilla_PIME135,Ceratopipra_rubrocapilla_PIME482,Ceratopipra_rubrocapilla_PIME521,Ceratopipra_rubrocapilla_PIME522,Ceratopipra_rubrocapilla_PPS174,Ceratopipra_rubrocapilla_TLP025,Ceratopipra_rubrocapilla_TLP123,Ceratopipra_rubrocapilla_WM281,Ceratopipra_rubrocapilla_WM293 /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0.vcf.gz | vk tajima 500000 500000 - > Ceratopipra_rubrocapilla/Tapajos.500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima 
#Xingu (excluding headwaters)
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples Ceratopipra_rubrocapilla_GAPTO259,Ceratopipra_rubrocapilla_GAPTO268,Ceratopipra_rubrocapilla_GAPTO277,Ceratopipra_rubrocapilla_GAPTO327,Ceratopipra_rubrocapilla_PPBIO221,Ceratopipra_rubrocapilla_PPBIO305,Ceratopipra_rubrocapilla_RDO007,Ceratopipra_rubrocapilla_UHE442 /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0.vcf.gz | vk tajima 500000 500000 - > Ceratopipra_rubrocapilla/Xingu.500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima
#Atlantic Forest
time /home/0_PROGRAMS/bcftools-1.17/bcftools view --samples Ceratopipra_rubrocapilla_CPE012,Ceratopipra_rubrocapilla_CPE025,Ceratopipra_rubrocapilla_CPE049,Ceratopipra_rubrocapilla_CPE069,Ceratopipra_rubrocapilla_CPEII008,Ceratopipra_rubrocapilla_UNA002,Ceratopipra_rubrocapilla_UNA018,Ceratopipra_rubrocapilla_UNA088,Ceratopipra_rubrocapilla_UNA110,Ceratopipra_rubrocapilla_UNA124 /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/Ceratopipra_rubrocapilla.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.AC0.37_0.2_5_20.0.vcf.gz | vk tajima 500000 500000 - > Ceratopipra_rubrocapilla/AtlanticForest.500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima #7s

#calculate mean values for each pop across the windows
for pop in AtlanticForest Xingu Tapajos Rondonia Inambari; do printf "$pop " >> Ceratopipra_rubrocapilla/Ceratopipra_rubrocapilla.500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima.txt ; awk 'BEGIN {sum=0; count=0;} NR>1 && $6 != "NA" {sum+=$6; count++;} END {print " " sum/count;}' Ceratopipra_rubrocapilla/"$pop".500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima >> Ceratopipra_rubrocapilla/Ceratopipra_rubrocapilla.500kb.genicintergenic.forSVD.AC0.37_0.2_5_20.0.tajima.txt ; done

```
* `vk tajima`: the first argument is the window size, the second is the step size; if they are the same size, then the windows will not overlap.  

Results:  
```
AtlanticForest  -0.653899
Xingu  -0.934163
Tapajos  -1.34484
Rondonia  -1.38684
Inambari  -1.50548
```
