# ABBA BABA tests

For this analysis, we need a dataset of high quality SNPs, and we need an outgroup.  
We will use *Ceratopipra erythrocephala* as our outgroup in the P4 position. An alternative would be to use the reference genome as the outgroup, but it is a bit distant; in general it is best to choose a P4 that is as close as possible to your ingroup without having too much incomplete lineage sorting (which would cause noise in interpretting which alleles are derived vs ancestral, i.e., BAAB would be counted as ABBA, and ABAB as BABA.  

We will use the same dataset as for Splitstree: `Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20`

First, we need to define what population each of our samples come from. We will make a file with two columns: the first is our sample names as they appear in the VCF file, and the second column assigns them to a population.  

```bash
mkdir -p /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA
cd /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA

cat > samples_m07.txt
sed -i 's/ /\t/g' Ceratopipra_rubrocapilla/samples_m07.txt #make tab separated
```
```
Ceratopipra_erythrocephala_SGC288 Imeri
Ceratopipra_erythrocephala_B7534 Imeri
Ceratopipra_erythrocephala_B7539 Imeri
Ceratopipra_erythrocephala_B7536 Imeri
Ceratopipra_erythrocephala_B7537 Imeri
Ceratopipra_rubrocapilla_CPE012 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE025 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE049 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE069 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPEII008 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_UNA002 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA018 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA088 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA110 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA124 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_AMA340 InambariW
Ceratopipra_rubrocapilla_AMA601 InambariW
Ceratopipra_rubrocapilla_AMA656 InambariW
Ceratopipra_rubrocapilla_B4615 InambariW
Ceratopipra_rubrocapilla_B5082 InambariW
Ceratopipra_rubrocapilla_CAM117 InambariW
Ceratopipra_rubrocapilla_CUJ027 InambariW
Ceratopipra_rubrocapilla_CUJ072 InambariW
Ceratopipra_rubrocapilla_OM022 InambariE
Ceratopipra_rubrocapilla_OM112 InambariE
Ceratopipra_rubrocapilla_OM172 InambariE
Ceratopipra_rubrocapilla_PUC008 InambariW
Ceratopipra_rubrocapilla_PUC007 InambariW
Ceratopipra_rubrocapilla_UFAC1007 InambariW
Ceratopipra_rubrocapilla_UFAC305 InambariE
Ceratopipra_rubrocapilla_UFAC476 InambariW
Ceratopipra_rubrocapilla_AMZ610 RondoniaHeadwaters
Ceratopipra_rubrocapilla_FPR006 Rondonia
Ceratopipra_rubrocapilla_FPR015 Rondonia
Ceratopipra_rubrocapilla_FPR021 Rondonia
Ceratopipra_rubrocapilla_MAM010 Rondonia
Ceratopipra_rubrocapilla_MAR092 Rondonia
Ceratopipra_rubrocapilla_MPDS628 Rondonia
Ceratopipra_rubrocapilla_OP100 RondoniaHeadwaters
Ceratopipra_rubrocapilla_PIME024 Rondonia
Ceratopipra_rubrocapilla_PIME079 Rondonia
Ceratopipra_rubrocapilla_PIME460 Rondonia
Ceratopipra_rubrocapilla_BR033 Tapajos
Ceratopipra_rubrocapilla_BR036 Tapajos
Ceratopipra_rubrocapilla_MF005 Tapajos
Ceratopipra_rubrocapilla_PIME062 Rondonia
Ceratopipra_rubrocapilla_PIME134 Tapajos
Ceratopipra_rubrocapilla_PIME135 Tapajos
Ceratopipra_rubrocapilla_PIME169 Rondonia
Ceratopipra_rubrocapilla_PIME482 Tapajos
Ceratopipra_rubrocapilla_PIME522 Tapajos
Ceratopipra_rubrocapilla_PPS041 TapajosHeadwaters
Ceratopipra_rubrocapilla_PPS174 Tapajos
Ceratopipra_rubrocapilla_PPS221 TapajosHeadwaters
Ceratopipra_rubrocapilla_TLP025 Tapajos
Ceratopipra_rubrocapilla_TLP123 Tapajos
Ceratopipra_rubrocapilla_WM293 Tapajos
Ceratopipra_rubrocapilla_AMZ551 RondoniaHeadwaters
Ceratopipra_rubrocapilla_GAPTO259 XinguBelemC
Ceratopipra_rubrocapilla_GAPTO268 XinguBelemC
Ceratopipra_rubrocapilla_GAPTO277 XinguBelemC
Ceratopipra_rubrocapilla_GAPTO327 XinguBelemC
Ceratopipra_rubrocapilla_MT087 Headwaters
Ceratopipra_rubrocapilla_MT123 Headwaters
Ceratopipra_rubrocapilla_PPBIO221 XinguBelemN
Ceratopipra_rubrocapilla_PPBIO305 XinguBelemN
Ceratopipra_rubrocapilla_RDO007 XinguBelemC
Ceratopipra_rubrocapilla_UHE442 XinguBelemN
```
Then, I want a text file listing the order of populations to plot when I view the results.  
```bash
cat > Ceratopipra_rubrocapilla/populations.txt #without outgroup
```
```
AtlanticForest_Ibateguara
AtlanticForest_Ilheus
XinguBelemC
XinguBelemN
Tapajos
Headwaters
TapajosHeadwaters
RondoniaHeadwaters
Rondonia
InambariE
InambariW
```
Then, run Dsuite.  
```bash
cd /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA
#Imeri as outgroup for rubrocapilla
DATASET=Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20 #82218 variants

#try a few different filters to ensure that results are robust to bioinformatic choices
#DATASET=Cryptopipo_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.intergenic.forSVD.49_0.2_5_20
#DATASET=Cryptopipo_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20 #58548 variants

mkdir -p  /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA/Ceratopipra_rubrocapilla/"$DATASET"
cd /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA/Ceratopipra_rubrocapilla/"$DATASET"

#get VCF file here
ln -s /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/"$DATASET".vcf.gz .

#specify the outgroup in the sample population file
sed 's/Imeri/Outgroup/g' ../samples_m07.txt > samples_ID.txt

#specify the backbone phylogeny upon which to do the ABBA BABA tests (from SVDQuartets tree)
echo "((((((((AtlanticForest_Ibateguara,AtlanticForest_Ilheus),XinguBelemC),XinguBelemN),Tapajos),((TapajosHeadwaters,Headwaters),RondoniaHeadwaters)),Rondonia),(InambariW,InambariE)),Outgroup);" > Ceratopipra.nwk

#run Dsuite
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios "$DATASET".vcf.gz samples_ID.txt -t Ceratopipra.nwk -n "$DATASET" #1m54.835s

#make a plot to visualize the results
#We have to remove column 5 of the data because the script was made before there was a column 5
cut -f 1,2,3,4,6,7 samples_ID_"$DATASET"_tree.txt > plotting_data.txt
ruby /home/0_PROGRAMS/mmatschiner/plot_d.rb plotting_data.txt ../populations.txt 0.1 Dsuite_autosomes.svg

#Run Fbranch
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch Ceratopipra.nwk samples_ID_"$DATASET"_tree.txt > fbranch.txt

#plot the results!
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch.txt Ceratopipra.nwk

#repeat with alternate tree
echo "((((((((AtlanticForest_Ibateguara,AtlanticForest_Ilheus),XinguBelemC),XinguBelemN),Tapajos),(Rondonia,((TapajosHeadwaters,Headwaters),RondoniaHeadwaters))),InambariE),InambariW),Outgroup);" > Ceratopipra.alternate.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios "$DATASET".vcf.gz samples_ID.txt -t Ceratopipra.alternate.nwk -n "$DATASET".alternate #1m54.835s

```

Next, I will combine the two Atlantic Forest localities together - we have no reason to have those two subpopulations separate (I wanted to check whether the two showed different patterns to make sure we weren't overlooking anything, eg., introgression into one of the two Atlantic Forest localities. Since they do not show different patterns, we should combine them to simplify the paper - no need to show an entire second set of Atlantic Forest results that do not contribute anything)  

```bash
#Imeri as outgroup for rubrocapilla
DATASET=Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20 #82218 variants

#try a few different filters to ensure that results are robust to bioinformatic choices
#DATASET=Cryptopipo_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.intergenic.forSVD.49_0.2_5_20
#DATASET=Cryptopipo_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20 #58548 variants

mkdir -p  /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA/Ceratopipra_rubrocapilla/"$DATASET"
cd /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA/Ceratopipra_rubrocapilla/"$DATASET"

ln -s /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/"$DATASET".vcf.gz .

sed 's/Imeri/Outgroup/g' ../samples_m07.txt > samples_ID.txt
sed 's/AtlanticForest_Ibateguara/AtlanticForest/g; s/AtlanticForest_Ilheus/AtlanticForest/g' samples_ID.txt > samples_ID_10subpops.txt
grep -v "AtlanticForest_Ibateguara" ../populations.txt | sed 's/AtlanticForest_Ilheus/AtlanticForest/g' > populations_10subpops.txt
#specify the backbone phylogeny (from SVDQuartets tree)
echo "(((((((AtlanticForest,XinguBelemC),XinguBelemN),Tapajos),((TapajosHeadwaters,Headwaters),RondoniaHeadwaters)),Rondonia),(InambariW,InambariE)),Outgroup);" > Ceratopipra_10subpops.nwk

#run Dsuite
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios "$DATASET".vcf.gz samples_ID_10subpops.txt -t Ceratopipra_10subpops.nwk -n "$DATASET"_10subpops #1m54.835s

#make a plot to visualize the results
#We have to remove column 5 of the data because the script was made before there was a column 5
cut -f 1,2,3,4,6,7 samples_ID_10subpops_"$DATASET"_10subpops_tree.txt > plotting_data_10subpops.txt
ruby /home/0_PROGRAMS/mmatschiner/plot_d.rb plotting_data_10subpops.txt populations_10subpops.txt 0.1 Dsuite_autosomes_10subpops.svg

#Run Fbranch
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch Ceratopipra_10subpops.nwk samples_ID_10subpops_"$DATASET"_10subpops_tree.txt > fbranch_10subpops.txt

#plot the results!
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch_10subpops.txt Ceratopipra_10subpops.nwk


#repeat with alternate tree
echo "(((((((AtlanticForest,XinguBelemC),XinguBelemN),Tapajos),(Rondonia,((TapajosHeadwaters,Headwaters),RondoniaHeadwaters))),InambariE),InambariW),Outgroup);" > Ceratopipra.alternate_10subpops.nwk
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios "$DATASET".vcf.gz samples_ID_10subpops.txt -t Ceratopipra.alternate_10subpops.nwk -n "$DATASET".alternate_10subpops #1m54.835s

```

Next, I will split the Tapajos into north and south to see if those subpopulations differ in any levels of gene flow.

```bash
cd /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA/Ceratopipra_rubrocapilla/Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20

cat > samples_m07_splitTapajos.txt
sed -i 's/ /\t/g' samples_m07_splitTapajos.txt #make tab separated
#specify the outgroup
sed -i 's/Imeri/Outgroup/g' ./samples_m07_splitTapajos.txt 

```
```
Ceratopipra_rubrocapilla_BR033 TapajosC
Ceratopipra_rubrocapilla_BR036 TapajosC
Ceratopipra_rubrocapilla_MF005 TapajosN
Ceratopipra_rubrocapilla_PIME134 TapajosN
Ceratopipra_rubrocapilla_PIME135 TapajosN
Ceratopipra_rubrocapilla_PIME482 TapajosC
Ceratopipra_rubrocapilla_PIME522 TapajosC
Ceratopipra_rubrocapilla_PPS041 TapajosHeadwaters
Ceratopipra_rubrocapilla_PPS174 TapajosS
Ceratopipra_rubrocapilla_PPS221 TapajosHeadwaters
Ceratopipra_rubrocapilla_TLP025 TapajosS
Ceratopipra_rubrocapilla_TLP123 TapajosS
Ceratopipra_rubrocapilla_WM293 TapajosN
Ceratopipra_erythrocephala_SGC288 Imeri
Ceratopipra_erythrocephala_B7534 Imeri
Ceratopipra_erythrocephala_B7539 Imeri
Ceratopipra_erythrocephala_B7536 Imeri
Ceratopipra_erythrocephala_B7537 Imeri
Ceratopipra_rubrocapilla_CPE012 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE025 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE049 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPE069 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_CPEII008 AtlanticForest_Ibateguara
Ceratopipra_rubrocapilla_UNA002 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA018 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA088 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA110 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_UNA124 AtlanticForest_Ilheus
Ceratopipra_rubrocapilla_AMA340 InambariW
Ceratopipra_rubrocapilla_AMA601 InambariW
Ceratopipra_rubrocapilla_AMA656 InambariW
Ceratopipra_rubrocapilla_B4615 InambariW
Ceratopipra_rubrocapilla_B5082 InambariW
Ceratopipra_rubrocapilla_CAM117 InambariW
Ceratopipra_rubrocapilla_CUJ027 InambariW
Ceratopipra_rubrocapilla_CUJ072 InambariW
Ceratopipra_rubrocapilla_OM022 InambariE
Ceratopipra_rubrocapilla_OM112 InambariE
Ceratopipra_rubrocapilla_OM172 InambariE
Ceratopipra_rubrocapilla_PUC008 InambariW
Ceratopipra_rubrocapilla_PUC007 InambariW
Ceratopipra_rubrocapilla_UFAC1007 InambariW
Ceratopipra_rubrocapilla_UFAC305 InambariE
Ceratopipra_rubrocapilla_UFAC476 InambariW
Ceratopipra_rubrocapilla_AMZ610 RondoniaHeadwaters
Ceratopipra_rubrocapilla_FPR006 Rondonia
Ceratopipra_rubrocapilla_FPR015 Rondonia
Ceratopipra_rubrocapilla_FPR021 Rondonia
Ceratopipra_rubrocapilla_MAM010 Rondonia
Ceratopipra_rubrocapilla_MAR092 Rondonia
Ceratopipra_rubrocapilla_MPDS628 Rondonia
Ceratopipra_rubrocapilla_OP100 RondoniaHeadwaters
Ceratopipra_rubrocapilla_PIME024 Rondonia
Ceratopipra_rubrocapilla_PIME079 Rondonia
Ceratopipra_rubrocapilla_PIME460 Rondonia
Ceratopipra_rubrocapilla_PIME062 Rondonia
Ceratopipra_rubrocapilla_PIME169 Rondonia
Ceratopipra_rubrocapilla_AMZ551 RondoniaHeadwaters
Ceratopipra_rubrocapilla_GAPTO259 XinguBelemC
Ceratopipra_rubrocapilla_GAPTO268 XinguBelemC
Ceratopipra_rubrocapilla_GAPTO277 XinguBelemC
Ceratopipra_rubrocapilla_GAPTO327 XinguBelemC
Ceratopipra_rubrocapilla_MT087 Headwaters
Ceratopipra_rubrocapilla_MT123 Headwaters
Ceratopipra_rubrocapilla_PPBIO221 XinguBelemN
Ceratopipra_rubrocapilla_PPBIO305 XinguBelemN
Ceratopipra_rubrocapilla_RDO007 XinguBelemC
Ceratopipra_rubrocapilla_UHE442 XinguBelemN
```
```bash
#Imeri as outgroup for rubrocapilla
#try a few different filters to ensure that results are robust to bioinformatic choices
#DATASET=Cryptopipo_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.intergenic.forSVD.49_0.2_5_20
#DATASET=Cryptopipo_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20 #58548 variants
DATASET=Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.genicintergenic.forSVD.49_0.2_5_20 #82218 variants

mkdir -p  /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA/Ceratopipra_rubrocapilla/"$DATASET"
cd /home/1_Else/Ceratopipra/5_Introgression/5.1aa_Dsuite_ABBABABA/Ceratopipra_rubrocapilla/"$DATASET"

ln -s /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/filteredSNPs/"$DATASET".vcf.gz .


#specify the backbone phylogeny (from SVDQuartets tree)
echo "((((((((AtlanticForest_Ibateguara,AtlanticForest_Ilheus),XinguBelemC),XinguBelemN),((TapajosN,TapajosC),TapajosS)),((TapajosHeadwaters,Headwaters),RondoniaHeadwaters)),Rondonia),(InambariW,InambariE)),Outgroup);" > Ceratopipra_splitTapajos.nwk

time /home/0_PROGRAMS/Dsuite/Build/Dsuite Dtrios "$DATASET".vcf.gz samples_m07_splitTapajos.txt -t Ceratopipra_splitTapajos.nwk -n "$DATASET"_splitTapajos #1m54.835s

#Run Fbranch
time /home/0_PROGRAMS/Dsuite/Build/Dsuite Fbranch Ceratopipra_splitTapajos.nwk samples_m07_splitTapajos_"$DATASET"_splitTapajos_tree.txt > fbranch_splitTapajos.txt

#plot the results!
/home/0_PROGRAMS/Dsuite/utils/dtools.py fbranch_splitTapajos.txt Ceratopipra_splitTapajos.nwk

```
