# Maximum likelihood tree
This page is the code for generated a maximum likelihood phylogeny from the concatenated dataset. Note that concatenated trees can combine genes with many different histories, especially amongst closely related organisms, which can be problematic. I am including it in addition to a coalescent tree which is better able to handle gene tree discordance. Nevertheless, maximum likelihood concatenated trees are still standard to include.  

To properly estimate branch lengths, both variant and invariant sites should be included. When incorporating long alignments with many invariants sites, it is much faster to use the '-t PARS -ninit 2' option (see discussion here: https://bitsandbugs.org/2019/11/06/two-easy-ways-to-run-iq-tree-with-the-correct-number-of-constant-sites/ "`From Leo – For whole genome phylogenetic trees, you can feed IQ-TREE your whole genome alignment (including constant sites) and it will run in a reasonable time if you pass it the following params `-t PARS -ninit 2`. If you use the IQ-TREE default parsimony options then it doesn’t remove repeated patterns (including the many invariant positions), and will take a very, very long time for larger alignments.`").

I have generated an alignment with the following filters:  
* autosomes only  
* SNPs and invariants only (no indels)
* min MQ 25
* min individual depth of 5X per site
* max missingness of 20% per site
* max average depth 49X per site

```bash
#set up directory
mkdir -p /home/1_Else/Ceratopipra/2_Phylogeny/2.2aa_IQtree2
cd /home/1_Else/Ceratopipra/2_Phylogeny/2.2aa_IQtree2

#link the fasta file here
mkdir -p datasets

#this is the dataset to use
parallel ln -s /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/alignments/{1} ./datasets ::: Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.fasta 




##### Test datasets #####

#I am also testing a few more datasets to ensure that the patterns are robust to bioinformatic choices. All of these datasets gave qualitatively similar results, although with some conflict at poorly-supported nodes.

#these datasets are designed to test the impact of missing data and of min depth filters. I included a min depth of 3 (40_0.2_3_20) or 5 (49_0.2_5_20) (with the max average depth adjusted to 40 or 9 respectively), and varied relaxing my filters for missing data. M07 excludes 40% missing and above; M08 excludes 45% and up; M09 excludes 55% and up, and no-m excludes only 69% and up. Datasets with more missing data run much slower, so lenient cutoffs run much slower than stricter cutoffs for missing data. Note that with a min depth of 3X, there are likely to be many genotyping errors, but the phylogeny remained very similar.
parallel ln -s /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/alignments/{1} ./datasets :::  Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20.fasta Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.fasta Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20.fasta Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20.fasta Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.fasta Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20.fasta Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.fasta

####Explanation of these test datasets:  
#m08 allows Ceratopipra_rubrocapilla_AMANA137 and Ceratopipra_rubrocapilla_PIME521
#m09 then adds Ceratopipra_rubrocapilla_WM281 and Ceratopipra_rubrocapilla_MSF107
#no-m (Ceratopipra_rubrocapilla_imeriout) then adds Ceratopipra_rubrocapilla_AMANA034 and Ceratopipra_rubrocapilla_PPS352 and Ceratopipra_rubrocapilla_BR36440 which have a lot of missing data

#these datasets have similar amounts of data, with less data as you allow samples with more missingness (because more sites fail the missingness filter)  
#Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.log:17948 parsimony-informative, 16274 singleton sites, 2096573 constant sites
#Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.log:17116 parsimony-informative, 15523 singleton sites, 2004433 constant sites
#Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.log:14937 parsimony-informative, 13523 singleton sites, 1773171 constant sites
#Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.log:12765 parsimony-informative, 11507 singleton sites, 1543694 constant sites

#test a dataset with chloromeros as outgroup to ensure results are robust to choice of outgroup. (Warning: Inambari is in geographic contact with chloromeros, so using it as an outgroup could be problematic if Inambari introgressed with chloromeros. This will be tested in a future project). Results are qualitatively the same.
ln -s /home/1_Else/Ceratopipra/1_Datasets/1.3a_genotyping/fairfilters/alignments/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.fasta ./datasets


```

# Run IQTree2
Then, run IQtree2. I am running a few different datasets to ensure that the topology is robust to inclusion of samples with a lot of missing data, and to choice of min depth filter (described above).  
```bash
cd /home/1_Else/Ceratopipra/2_Phylogeny/2.2aa_IQtree2
#first, do a single run to get a quick look of the data and ensure that it works, before investing hours into doing a full 1000 bootstraps.
parallel --jobs 6 time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/{1}.fasta -t PARS -ninit 2 -nt AUTO ::: Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
#It ran with no errors, we can proceed with the full run

#now with bootstrapping. I will do 1000 bootstrap replicates to get a good estimate of support for each node.
#I will not waste resources to bootstrap the datasets with min 3X depth, those were just to ensure the results were consistent, so a single estimate will do.
#I will bootstrap my 4 additional datasets that allow more missing data, because I am curious where those high-missingness samples would get placed and how much support their placement will receive
#the dataset for publication is Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 ; the others are not suitable for publication as they have a lot of missing data.
mkdir -p bootstrapped_trees
parallel --jobs 5 time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/{1}.fasta -t PARS -ninit 2 -nt AUTO -B 1000 --prefix {1}.boot1000 ::: Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 
mv ./Ceratopipra_rubrocapilla_imeriout*.GBSPipfilbwa.allsites.filtered.for_pixy.auto.*.boot1000* bootstrapped_trees #organize them into a folder
#Timing:
#543m55.858s (6289m47.766s) for Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20; 
#747m29.020s (5721m7.739s) for Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 
#1484m23.891s (5777m32.500s) for Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 
#allowing just a few samples with high missingness dramatically increased computational time in terms of clock time, although the "user time" did not increase as much.



###Test additional datasets to ensure results robust###

#then, test some additional datasets with different filters to ensure the tree looks robust in overall patterns (minor differences as this point are not meaningful since we have not done a bootstrap to evaluate support)

#test the inclusion of samples with more missing data and different min and max depth filters. Results qualitatively the same. 
parallel --jobs 6 time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/{1}.fasta -t PARS -ninit 2 -nt AUTO ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20
#331m37.361s (331m37.361s) for Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.fasta
#6516m52.119s (24837m0.841s) for Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.40_0.2_3_20.fasta
#allowing genotypes with 3X vs 5X dramatically increased the time it took to run.

#Run full bootstrapped analysis with the datasets that have different missingness filters. Results qualitatively the same.
parallel --jobs 5 time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/{1}.fasta -t PARS -ninit 2 -nt AUTO -B 1000 --prefix {1}.boot1000 ::: Ceratopipra_rubrocapilla_imeriout.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m08.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 Ceratopipra_rubrocapilla_imeriout_m09.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20 

#test chloromeros as outgroup. Results qualitatively the same.
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.fasta -t PARS -ninit 2 -nt 2 #1580m8.207s (3115m28.382s)
#run one more time with -NT auto, I want to compare the time cost
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/Ceratopipra_rubrocapilla_chloromeros.GBSPipfilbwa.allsites.filtered.for_pixy.40_0.4_5_20.fasta -t PARS -ninit 2 -nt AUTO #1198m45.863s (15914m29.050s) #BEST NUMBER OF THREADS: 14
#including -nt AUTO picked 14 threads. It ran in 20 hours, vs 26 hours when I specified -nt 2, so the cost of picking the number of threads was worth it.

```

To make it easier to read, label samples with their population
```bash
#just population names
sed 's|Ceratopipra_rubrocapilla_CPE012|Ceratopipra_rubrocapilla_CPE012.AtlanticForest|g; s|Ceratopipra_rubrocapilla_CPE025|Ceratopipra_rubrocapilla_CPE025.AtlanticForest|g; s|Ceratopipra_rubrocapilla_CPE049|Ceratopipra_rubrocapilla_CPE049.AtlanticForest|g; s|Ceratopipra_rubrocapilla_CPE069|Ceratopipra_rubrocapilla_CPE069.AtlanticForest|g; s|Ceratopipra_rubrocapilla_CPEII008|Ceratopipra_rubrocapilla_CPEII008.AtlanticForest|g; s|Ceratopipra_rubrocapilla_UNA002|Ceratopipra_rubrocapilla_UNA002.AtlanticForest|g; s|Ceratopipra_rubrocapilla_UNA018|Ceratopipra_rubrocapilla_UNA018.AtlanticForest|g; s|Ceratopipra_rubrocapilla_UNA088|Ceratopipra_rubrocapilla_UNA088.AtlanticForest|g; s|Ceratopipra_rubrocapilla_UNA110|Ceratopipra_rubrocapilla_UNA110.AtlanticForest|g; s|Ceratopipra_rubrocapilla_UNA124|Ceratopipra_rubrocapilla_UNA124.AtlanticForest|g; s|Ceratopipra_rubrocapilla_AMA340|Ceratopipra_rubrocapilla_AMA340.Inambari|g; s|Ceratopipra_rubrocapilla_AMA601|Ceratopipra_rubrocapilla_AMA601.Inambari|g; s|Ceratopipra_rubrocapilla_AMA656|Ceratopipra_rubrocapilla_AMA656.Inambari|g; s|Ceratopipra_rubrocapilla_B4615|Ceratopipra_rubrocapilla_B4615.Inambari|g; s|Ceratopipra_rubrocapilla_B5082|Ceratopipra_rubrocapilla_B5082.Inambari|g; s|Ceratopipra_rubrocapilla_BR36440|Ceratopipra_rubrocapilla_BR36440.Inambari|g; s|Ceratopipra_rubrocapilla_CAM117|Ceratopipra_rubrocapilla_CAM117.Inambari|g; s|Ceratopipra_rubrocapilla_CUJ027|Ceratopipra_rubrocapilla_CUJ027.Inambari|g; s|Ceratopipra_rubrocapilla_CUJ072|Ceratopipra_rubrocapilla_CUJ072.Inambari|g; s|Ceratopipra_rubrocapilla_OM022|Ceratopipra_rubrocapilla_OM022.Inambari|g; s|Ceratopipra_rubrocapilla_OM112|Ceratopipra_rubrocapilla_OM112.Inambari|g; s|Ceratopipra_rubrocapilla_OM172|Ceratopipra_rubrocapilla_OM172.Inambari|g; s|Ceratopipra_rubrocapilla_PUC008|Ceratopipra_rubrocapilla_PUC008.Inambari|g; s|Ceratopipra_rubrocapilla_UFAC1007|Ceratopipra_rubrocapilla_UFAC1007.Inambari|g; s|Ceratopipra_rubrocapilla_UFAC305|Ceratopipra_rubrocapilla_UFAC305.Inambari|g; s|Ceratopipra_rubrocapilla_UFAC476|Ceratopipra_rubrocapilla_UFAC476.Inambari|g; s|Ceratopipra_rubrocapilla_AMANA034|Ceratopipra_rubrocapilla_AMANA034.Rondonia|g; s|Ceratopipra_rubrocapilla_AMANA137|Ceratopipra_rubrocapilla_AMANA137.Rondonia|g; s|Ceratopipra_rubrocapilla_AMZ610|Ceratopipra_rubrocapilla_AMZ610.Rondonia|g; s|Ceratopipra_rubrocapilla_FPR006|Ceratopipra_rubrocapilla_FPR006.Rondonia|g; s|Ceratopipra_rubrocapilla_FPR015|Ceratopipra_rubrocapilla_FPR015.Rondonia|g; s|Ceratopipra_rubrocapilla_FPR021|Ceratopipra_rubrocapilla_FPR021.Rondonia|g; s|Ceratopipra_rubrocapilla_MAM010|Ceratopipra_rubrocapilla_MAM010.Rondonia|g; s|Ceratopipra_rubrocapilla_MAR092|Ceratopipra_rubrocapilla_MAR092.Rondonia|g; s|Ceratopipra_rubrocapilla_MPDS628|Ceratopipra_rubrocapilla_MPDS628.Rondonia|g; s|Ceratopipra_rubrocapilla_OP100|Ceratopipra_rubrocapilla_OP100.Rondonia|g; s|Ceratopipra_rubrocapilla_PIME024|Ceratopipra_rubrocapilla_PIME024.Rondonia|g; s|Ceratopipra_rubrocapilla_PIME079|Ceratopipra_rubrocapilla_PIME079.Rondonia|g; s|Ceratopipra_rubrocapilla_PIME460|Ceratopipra_rubrocapilla_PIME460.Rondonia|g; s|Ceratopipra_rubrocapilla_BR033|Ceratopipra_rubrocapilla_BR033.Tapajos|g; s|Ceratopipra_rubrocapilla_BR036|Ceratopipra_rubrocapilla_BR036.Tapajos|g; s|Ceratopipra_rubrocapilla_MF005|Ceratopipra_rubrocapilla_MF005.Tapajos|g; s|Ceratopipra_rubrocapilla_MSF107|Ceratopipra_rubrocapilla_MSF107.Tapajos|g; s|Ceratopipra_rubrocapilla_PIME062|Ceratopipra_rubrocapilla_PIME062.Tapajos|g; s|Ceratopipra_rubrocapilla_PIME134|Ceratopipra_rubrocapilla_PIME134.Tapajos|g; s|Ceratopipra_rubrocapilla_PIME135|Ceratopipra_rubrocapilla_PIME135.Tapajos|g; s|Ceratopipra_rubrocapilla_PIME169|Ceratopipra_rubrocapilla_PIME169.Rondonia|g; s|Ceratopipra_rubrocapilla_PIME482|Ceratopipra_rubrocapilla_PIME482.Tapajos|g; s|Ceratopipra_rubrocapilla_PIME521|Ceratopipra_rubrocapilla_PIME521.Tapajos|g; s|Ceratopipra_rubrocapilla_PIME522|Ceratopipra_rubrocapilla_PIME522.Tapajos|g; s|Ceratopipra_rubrocapilla_PPS041|Ceratopipra_rubrocapilla_PPS041.Tapajos|g; s|Ceratopipra_rubrocapilla_PPS174|Ceratopipra_rubrocapilla_PPS174.Tapajos|g; s|Ceratopipra_rubrocapilla_PPS221|Ceratopipra_rubrocapilla_PPS221.Tapajos|g; s|Ceratopipra_rubrocapilla_PPS352|Ceratopipra_rubrocapilla_PPS352.Tapajos|g; s|Ceratopipra_rubrocapilla_TLP025|Ceratopipra_rubrocapilla_TLP025.Tapajos|g; s|Ceratopipra_rubrocapilla_TLP123|Ceratopipra_rubrocapilla_TLP123.Tapajos|g; s|Ceratopipra_rubrocapilla_WM281|Ceratopipra_rubrocapilla_WM281.Tapajos|g; s|Ceratopipra_rubrocapilla_WM293|Ceratopipra_rubrocapilla_WM293.Tapajos|g; s|Ceratopipra_rubrocapilla_AMZ551|Ceratopipra_rubrocapilla_AMZ551.Rondonia|g; s|Ceratopipra_rubrocapilla_UNA014|Ceratopipra_rubrocapilla_UNA014.Unknown|g; s|Ceratopipra_rubrocapilla_GAPTO259|Ceratopipra_rubrocapilla_GAPTO259.XinguBelem|g; s|Ceratopipra_rubrocapilla_GAPTO268|Ceratopipra_rubrocapilla_GAPTO268.XinguBelem|g; s|Ceratopipra_rubrocapilla_GAPTO277|Ceratopipra_rubrocapilla_GAPTO277.XinguBelem|g; s|Ceratopipra_rubrocapilla_GAPTO327|Ceratopipra_rubrocapilla_GAPTO327.XinguBelem|g; s|Ceratopipra_rubrocapilla_MT087|Ceratopipra_rubrocapilla_MT087.XinguBelem|g; s|Ceratopipra_rubrocapilla_MT123|Ceratopipra_rubrocapilla_MT123.XinguBelem|g; s|Ceratopipra_rubrocapilla_PPBIO221|Ceratopipra_rubrocapilla_PPBIO221.XinguBelem|g; s|Ceratopipra_rubrocapilla_PPBIO305|Ceratopipra_rubrocapilla_PPBIO305.XinguBelem|g; s|Ceratopipra_rubrocapilla_RDO007|Ceratopipra_rubrocapilla_RDO007.XinguBelem|g; s|Ceratopipra_rubrocapilla_UHE442|Ceratopipra_rubrocapilla_UHE442.XinguBelem|g' */*Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20*boot*contree

#population names with more specific localities
ls *treefile | while read f ; do echo "$f" ; sed 's|Ceratopipra_rubrocapilla_UNA002|Ceratopipra_rubrocapilla_UNA002.AtlanticForest_Ilheus|g;s|Ceratopipra_rubrocapilla_UNA018|Ceratopipra_rubrocapilla_UNA018.AtlanticForest_Ilheus|g;s|Ceratopipra_rubrocapilla_UNA088|Ceratopipra_rubrocapilla_UNA088.AtlanticForest_Ilheus|g;s|Ceratopipra_rubrocapilla_UNA110|Ceratopipra_rubrocapilla_UNA110.AtlanticForest_Ilheus|g;s|Ceratopipra_rubrocapilla_UNA124|Ceratopipra_rubrocapilla_UNA124.AtlanticForest_Ilheus|g;s|Ceratopipra_rubrocapilla_MT087|Ceratopipra_rubrocapilla_MT087.Xingu/Belém_Querencia|g;s|Ceratopipra_rubrocapilla_MT123|Ceratopipra_rubrocapilla_MT123.Xingu/Belém_Querencia|g;s|Ceratopipra_rubrocapilla_PPS352|Ceratopipra_rubrocapilla_PPS352.Tapajos_UniãoDoSul|g;s|Ceratopipra_rubrocapilla_PPS041|Ceratopipra_rubrocapilla_PPS041.Tapajos_Marcelândia|g;s|Ceratopipra_rubrocapilla_OP100|Ceratopipra_rubrocapilla_OP100.Rondonia_Guajará-Mirim|g;s|Ceratopipra_rubrocapilla_AMZ551|Ceratopipra_rubrocapilla_AMZ551.Rondonia_NovaBandeirantes|g;s|Ceratopipra_rubrocapilla_AMZ610|Ceratopipra_rubrocapilla_AMZ610.Rondonia_NovBandeirantes|g;s|Ceratopipra_rubrocapilla_PPS221|Ceratopipra_rubrocapilla_PPS221.Tapajos_GuarantãdoNorte|g;s|Ceratopipra_rubrocapilla_UFAC1687|Ceratopipra_rubrocapilla_UFAC1687.Inambari_SENADORGUIOMARD|g;s|Ceratopipra_rubrocapilla_UFAC305|Ceratopipra_rubrocapilla_UFAC305.Inambari_SENADORGUIOMARD|g;s|Ceratopipra_rubrocapilla_BR36440|Ceratopipra_rubrocapilla_BR36440.Inambari_FortdoABUNÃ|g;s|Ceratopipra_rubrocapilla_TLP123|Ceratopipra_rubrocapilla_TLP123.Tapajos_Paranaíta|g;s|Ceratopipra_rubrocapilla_TLP025|Ceratopipra_rubrocapilla_TLP025.Tapajos_Paranaíta|g;s|Ceratopipra_rubrocapilla_MSF107|Ceratopipra_rubrocapilla_MSF107.Tapajos_Altamira|g;s|Ceratopipra_rubrocapilla_PPS174|Ceratopipra_rubrocapilla_PPS174.Tapajos_Altamira|g;s|Ceratopipra_rubrocapilla_CPE012|Ceratopipra_rubrocapilla_CPE012.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_CPE025|Ceratopipra_rubrocapilla_CPE025.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_CPE049|Ceratopipra_rubrocapilla_CPE049.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_CPE066|Ceratopipra_rubrocapilla_CPE066.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_CPE067|Ceratopipra_rubrocapilla_CPE067.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_CPE069|Ceratopipra_rubrocapilla_CPE069.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_CPE071|Ceratopipra_rubrocapilla_CPE071.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_CPEII008|Ceratopipra_rubrocapilla_CPEII008.AtlanticForest_Ibateguara|g;s|Ceratopipra_rubrocapilla_OM215|Ceratopipra_rubrocapilla_OM215.Rondonia_MachaDOESTE|g;s|Ceratopipra_rubrocapilla_MAR092|Ceratopipra_rubrocapilla_MAR092.Rondonia_Manicoré|g;s|Ceratopipra_rubrocapilla_BR036|Ceratopipra_rubrocapilla_BR036.Tapajos_Altamira|g;s|Ceratopipra_rubrocapilla_UFAC1007|Ceratopipra_rubrocapilla_UFAC1007.Inambari_PORTOWALTER|g;s|Ceratopipra_rubrocapilla_BR033|Ceratopipra_rubrocapilla_BR033.Tapajos_Altamira|g;s|Ceratopipra_rubrocapilla_UFAC476|Ceratopipra_rubrocapilla_UFAC476.Inambari_TARAUACÁ|g;s|Ceratopipra_rubrocapilla_GAPTO140|Ceratopipra_rubrocapilla_GAPTO140.Xingu/Belém_Araguaina|g;s|Ceratopipra_rubrocapilla_GAPTO259|Ceratopipra_rubrocapilla_GAPTO259.Xingu/Belém_Araguaina|g;s|Ceratopipra_rubrocapilla_GAPTO268|Ceratopipra_rubrocapilla_GAPTO268.Xingu/Belém_Araguaina|g;s|Ceratopipra_rubrocapilla_GAPTO277|Ceratopipra_rubrocapilla_GAPTO277.Xingu/Belém_Araguaina|g;s|Ceratopipra_rubrocapilla_GAPTO289|Ceratopipra_rubrocapilla_GAPTO289.Xingu/Belém_Araguaina|g;s|Ceratopipra_rubrocapilla_GAPTO327|Ceratopipra_rubrocapilla_GAPTO327.Xingu/Belém_Araguaina|g;s|Ceratopipra_rubrocapilla_CAM117|Ceratopipra_rubrocapilla_CAM117.Inambari_Guajará-AM|g;s|Ceratopipra_rubrocapilla_OM022|Ceratopipra_rubrocapilla_OM022.Inambari_Humaitá|g;s|Ceratopipra_rubrocapilla_OM112|Ceratopipra_rubrocapilla_OM112.Inambari_Humaitá|g;s|Ceratopipra_rubrocapilla_OM172|Ceratopipra_rubrocapilla_OM172.Inambari_Humaitá|g;s|Ceratopipra_rubrocapilla_MPDS628|Ceratopipra_rubrocapilla_MPDS628.Rondonia_HUMAITÁ|g;s|Ceratopipra_rubrocapilla_PIME521|Ceratopipra_rubrocapilla_PIME521.Tapajos_NovoProgresso|g;s|Ceratopipra_rubrocapilla_PIME522|Ceratopipra_rubrocapilla_PIME522.Tapajos_NovoProgresso|g;s|Ceratopipra_rubrocapilla_MPDS776|Ceratopipra_rubrocapilla_MPDS776.Rondonia_HUMAITÁ|g;s|Ceratopipra_rubrocapilla_PIME169|Ceratopipra_rubrocapilla_PIME169.Rondonia_Jacareacanga|g;s|Ceratopipra_rubrocapilla_PIME482|Ceratopipra_rubrocapilla_PIME482.Tapajos_Jacareacanga|g;s|Ceratopipra_rubrocapilla_AMANA137|Ceratopipra_rubrocapilla_AMANA137.Rondonia_JACAREACANGA|g;s|Ceratopipra_rubrocapilla_CUJ072|Ceratopipra_rubrocapilla_CUJ072.Inambari_RDSCujubim|g;s|Ceratopipra_rubrocapilla_PIME460|Ceratopipra_rubrocapilla_PIME460.Rondonia_JACAREACANGA|g;s|Ceratopipra_rubrocapilla_AMANA034|Ceratopipra_rubrocapilla_AMANA034.Rondonia_ITAITUBA|g;s|Ceratopipra_rubrocapilla_CUJ027|Ceratopipra_rubrocapilla_CUJ027.Inambari_RDSCujubim|g;s|Ceratopipra_rubrocapilla_PUC007|Ceratopipra_rubrocapilla_PUC007.Inambari_Tefe|g;s|Ceratopipra_rubrocapilla_PUC008|Ceratopipra_rubrocapilla_PUC008.Inambari_Tefe|g;s|Ceratopipra_rubrocapilla_RDO007|Ceratopipra_rubrocapilla_RDO007.Xingu/Belém_RondondoPará|g;s|Ceratopipra_rubrocapilla_AMA340|Ceratopipra_rubrocapilla_AMA340.Inambari_AtalaiadoNorte|g;s|Ceratopipra_rubrocapilla_AMA601|Ceratopipra_rubrocapilla_AMA601.Inambari_AtalaiadoNorte|g;s|Ceratopipra_rubrocapilla_AMA656|Ceratopipra_rubrocapilla_AMA656.Inambari_AtalaiadoNorte|g;s|Ceratopipra_rubrocapilla_MF005|Ceratopipra_rubrocapilla_MF005.Tapajos_Trairão|g;s|Ceratopipra_rubrocapilla_PIME062|Ceratopipra_rubrocapilla_PIME062.Tapajos_Itaituba|g;s|Ceratopipra_rubrocapilla_PIME079|Ceratopipra_rubrocapilla_PIME079.Rondonia_ITAITUBA|g;s|Ceratopipra_rubrocapilla_B4615|Ceratopipra_rubrocapilla_B4615.Inambari_Loreto|g;s|Ceratopipra_rubrocapilla_B5082|Ceratopipra_rubrocapilla_B5082.Inambari_Loreto|g;s|Ceratopipra_rubrocapilla_FPR006|Ceratopipra_rubrocapilla_FPR006.Rondonia_Maués|g;s|Ceratopipra_rubrocapilla_FPR015|Ceratopipra_rubrocapilla_FPR015.Rondonia_Maués|g;s|Ceratopipra_rubrocapilla_FPR021|Ceratopipra_rubrocapilla_FPR021.Rondonia_Maués|g;s|Ceratopipra_rubrocapilla_PIME134|Ceratopipra_rubrocapilla_PIME134.Tapajos_Placas|g;s|Ceratopipra_rubrocapilla_PIME135|Ceratopipra_rubrocapilla_PIME135.Tapajos_Placas|g;s|Ceratopipra_rubrocapilla_UHE442|Ceratopipra_rubrocapilla_UHE442.Xingu/Belém_Sen.JoséPorf.|g;s|Ceratopipra_rubrocapilla_WM281|Ceratopipra_rubrocapilla_WM281.Tapajos_Belterra|g;s|Ceratopipra_rubrocapilla_WM293|Ceratopipra_rubrocapilla_WM293.Tapajos_Belterra|g;s|Ceratopipra_rubrocapilla_MAM010|Ceratopipra_rubrocapilla_MAM010.Rondonia_AVEIRO|g;s|Ceratopipra_rubrocapilla_JRT021|Ceratopipra_rubrocapilla_JRT021.Rondonia_JURUTI|g;s|Ceratopipra_rubrocapilla_PIME024|Ceratopipra_rubrocapilla_PIME024.Rondonia_Santarém|g;s|Ceratopipra_rubrocapilla_PPBIO221|Ceratopipra_rubrocapilla_PPBIO221.Xingu/Belém_Portel|g;s|Ceratopipra_rubrocapilla_PPBIO305|Ceratopipra_rubrocapilla_PPBIO305.Xingu/Belém_Portel|g;s|Ceratopipra_rubrocapilla_PPBIO312|Ceratopipra_rubrocapilla_PPBIO312.Xingu/Belém_Portel|g;s|Ceratopipra_rubrocapilla_CPE027|Ceratopipra_rubrocapilla_CPE027.Unknown_Unknown|g;s|Ceratopipra_rubrocapilla_MELO34|Ceratopipra_rubrocapilla_MELO34.Unknown_Unknown|g;s|Ceratopipra_rubrocapilla_MT099|Ceratopipra_rubrocapilla_MT099.Unknown_Unknown|g;s|Ceratopipra_rubrocapilla_UNA014|Ceratopipra_rubrocapilla_UNA014.Unknown_Unknown|g' "$f" ; done

#add sex of sample to ensure that samples don't group by sex or have different branch lengths
sed 's/Ceratopipra_chloromeros_Fm433681/F_Ceratopipra_chloromeros_Fm433681/g' */*chrZnopseudo*boot*treefile | sed 's/Ceratopipra_rubrocapilla_AMA340/M_Ceratopipra_rubrocapilla_AMA340/g' | sed 's/Ceratopipra_rubrocapilla_AMA601/M_Ceratopipra_rubrocapilla_AMA601/g' | sed 's/Ceratopipra_rubrocapilla_AMA656/M_Ceratopipra_rubrocapilla_AMA656/g' | sed 's/Ceratopipra_rubrocapilla_AMANA034/M_Ceratopipra_rubrocapilla_AMANA034/g' | sed 's/Ceratopipra_rubrocapilla_AMANA137/M_Ceratopipra_rubrocapilla_AMANA137/g' | sed 's/Ceratopipra_rubrocapilla_AMZ551/F_Ceratopipra_rubrocapilla_AMZ551/g' | sed 's/Ceratopipra_rubrocapilla_AMZ610/F_Ceratopipra_rubrocapilla_AMZ610/g' | sed 's/Ceratopipra_rubrocapilla_B4615/M_Ceratopipra_rubrocapilla_B4615/g' | sed 's/Ceratopipra_rubrocapilla_B5082/M_Ceratopipra_rubrocapilla_B5082/g' | sed 's/Ceratopipra_rubrocapilla_BR033/F_Ceratopipra_rubrocapilla_BR033/g' | sed 's/Ceratopipra_rubrocapilla_BR036/M_Ceratopipra_rubrocapilla_BR036/g' | sed 's/Ceratopipra_rubrocapilla_BR36440/M_Ceratopipra_rubrocapilla_BR36440/g' | sed 's/Ceratopipra_rubrocapilla_CAM117/M_Ceratopipra_rubrocapilla_CAM117/g' | sed 's/Ceratopipra_rubrocapilla_CPE012/M_Ceratopipra_rubrocapilla_CPE012/g' | sed 's/Ceratopipra_rubrocapilla_CPE025/M_Ceratopipra_rubrocapilla_CPE025/g' | sed 's/Ceratopipra_rubrocapilla_CPE027/M_Ceratopipra_rubrocapilla_CPE027/g' | sed 's/Ceratopipra_rubrocapilla_CPE049/F_Ceratopipra_rubrocapilla_CPE049/g' | sed 's/Ceratopipra_rubrocapilla_CPE066/F_Ceratopipra_rubrocapilla_CPE066/g' | sed 's/Ceratopipra_rubrocapilla_CPE067/F_Ceratopipra_rubrocapilla_CPE067/g' | sed 's/Ceratopipra_rubrocapilla_CPE069/M_Ceratopipra_rubrocapilla_CPE069/g' | sed 's/Ceratopipra_rubrocapilla_CPE071/F_Ceratopipra_rubrocapilla_CPE071/g' | sed 's/Ceratopipra_rubrocapilla_CPEII008/M_Ceratopipra_rubrocapilla_CPEII008/g' | sed 's/Ceratopipra_rubrocapilla_CUJ027/F_Ceratopipra_rubrocapilla_CUJ027/g' | sed 's/Ceratopipra_rubrocapilla_CUJ072/F_Ceratopipra_rubrocapilla_CUJ072/g' | sed 's/Ceratopipra_rubrocapilla_FPR006/F_Ceratopipra_rubrocapilla_FPR006/g' | sed 's/Ceratopipra_rubrocapilla_FPR015/M_Ceratopipra_rubrocapilla_FPR015/g' | sed 's/Ceratopipra_rubrocapilla_FPR021/M_Ceratopipra_rubrocapilla_FPR021/g' | sed 's/Ceratopipra_rubrocapilla_GAPTO140/F_Ceratopipra_rubrocapilla_GAPTO140/g' | sed 's/Ceratopipra_rubrocapilla_GAPTO259/F_Ceratopipra_rubrocapilla_GAPTO259/g' | sed 's/Ceratopipra_rubrocapilla_GAPTO268/F_Ceratopipra_rubrocapilla_GAPTO268/g' | sed 's/Ceratopipra_rubrocapilla_GAPTO277/M_Ceratopipra_rubrocapilla_GAPTO277/g' | sed 's/Ceratopipra_rubrocapilla_GAPTO289/F_Ceratopipra_rubrocapilla_GAPTO289/g' | sed 's/Ceratopipra_rubrocapilla_GAPTO327/F_Ceratopipra_rubrocapilla_GAPTO327/g' | sed 's/Ceratopipra_rubrocapilla_JRT021/M_Ceratopipra_rubrocapilla_JRT021/g' | sed 's/Ceratopipra_rubrocapilla_MAM010/M_Ceratopipra_rubrocapilla_MAM010/g' | sed 's/Ceratopipra_rubrocapilla_MAR092/M_Ceratopipra_rubrocapilla_MAR092/g' | sed 's/Ceratopipra_rubrocapilla_MELO34/F_Ceratopipra_rubrocapilla_MELO34/g' | sed 's/Ceratopipra_rubrocapilla_MF005/M_Ceratopipra_rubrocapilla_MF005/g' | sed 's/Ceratopipra_rubrocapilla_MPDS628/F_Ceratopipra_rubrocapilla_MPDS628/g' | sed 's/Ceratopipra_rubrocapilla_MPDS776/F_Ceratopipra_rubrocapilla_MPDS776/g' | sed 's/Ceratopipra_rubrocapilla_MSF107/F_Ceratopipra_rubrocapilla_MSF107/g' | sed 's/Ceratopipra_rubrocapilla_MT087/F_Ceratopipra_rubrocapilla_MT087/g' | sed 's/Ceratopipra_rubrocapilla_MT099/F_Ceratopipra_rubrocapilla_MT099/g' | sed 's/Ceratopipra_rubrocapilla_MT123/M_Ceratopipra_rubrocapilla_MT123/g' | sed 's/Ceratopipra_rubrocapilla_OM022/F_Ceratopipra_rubrocapilla_OM022/g' | sed 's/Ceratopipra_rubrocapilla_OM112/M_Ceratopipra_rubrocapilla_OM112/g' | sed 's/Ceratopipra_rubrocapilla_OM172/F_Ceratopipra_rubrocapilla_OM172/g' | sed 's/Ceratopipra_rubrocapilla_OM215/F_Ceratopipra_rubrocapilla_OM215/g' | sed 's/Ceratopipra_rubrocapilla_OP100/F_Ceratopipra_rubrocapilla_OP100/g' | sed 's/Ceratopipra_rubrocapilla_PIME024/M_Ceratopipra_rubrocapilla_PIME024/g' | sed 's/Ceratopipra_rubrocapilla_PIME062/M_Ceratopipra_rubrocapilla_PIME062/g' | sed 's/Ceratopipra_rubrocapilla_PIME079/F_Ceratopipra_rubrocapilla_PIME079/g' | sed 's/Ceratopipra_rubrocapilla_PIME134/M_Ceratopipra_rubrocapilla_PIME134/g' | sed 's/Ceratopipra_rubrocapilla_PIME135/F_Ceratopipra_rubrocapilla_PIME135/g' | sed 's/Ceratopipra_rubrocapilla_PIME169/F_Ceratopipra_rubrocapilla_PIME169/g' | sed 's/Ceratopipra_rubrocapilla_PIME460/M_Ceratopipra_rubrocapilla_PIME460/g' | sed 's/Ceratopipra_rubrocapilla_PIME482/F_Ceratopipra_rubrocapilla_PIME482/g' | sed 's/Ceratopipra_rubrocapilla_PIME521/M_Ceratopipra_rubrocapilla_PIME521/g' | sed 's/Ceratopipra_rubrocapilla_PIME522/F_Ceratopipra_rubrocapilla_PIME522/g' | sed 's/Ceratopipra_rubrocapilla_PPBIO221/F_Ceratopipra_rubrocapilla_PPBIO221/g' | sed 's/Ceratopipra_rubrocapilla_PPBIO305/F_Ceratopipra_rubrocapilla_PPBIO305/g' | sed 's/Ceratopipra_rubrocapilla_PPBIO312/F_Ceratopipra_rubrocapilla_PPBIO312/g' | sed 's/Ceratopipra_rubrocapilla_PPS041/M_Ceratopipra_rubrocapilla_PPS041/g' | sed 's/Ceratopipra_rubrocapilla_PPS174/F_Ceratopipra_rubrocapilla_PPS174/g' | sed 's/Ceratopipra_rubrocapilla_PPS221/M_Ceratopipra_rubrocapilla_PPS221/g' | sed 's/Ceratopipra_rubrocapilla_PPS352/M_Ceratopipra_rubrocapilla_PPS352/g' | sed 's/Ceratopipra_rubrocapilla_PUC007/M_Ceratopipra_rubrocapilla_PUC007/g' | sed 's/Ceratopipra_rubrocapilla_PUC008/M_Ceratopipra_rubrocapilla_PUC008/g' | sed 's/Ceratopipra_rubrocapilla_RDO007/F_Ceratopipra_rubrocapilla_RDO007/g' | sed 's/Ceratopipra_rubrocapilla_TLP025/F_Ceratopipra_rubrocapilla_TLP025/g' | sed 's/Ceratopipra_rubrocapilla_TLP123/M_Ceratopipra_rubrocapilla_TLP123/g' | sed 's/Ceratopipra_rubrocapilla_UFAC1007/M_Ceratopipra_rubrocapilla_UFAC1007/g' | sed 's/Ceratopipra_rubrocapilla_UFAC1687/M_Ceratopipra_rubrocapilla_UFAC1687/g' | sed 's/Ceratopipra_rubrocapilla_UFAC305/F_Ceratopipra_rubrocapilla_UFAC305/g' | sed 's/Ceratopipra_rubrocapilla_UFAC476/F_Ceratopipra_rubrocapilla_UFAC476/g' | sed 's/Ceratopipra_rubrocapilla_UHE442/M_Ceratopipra_rubrocapilla_UHE442/g' | sed 's/Ceratopipra_rubrocapilla_UNA002/F_Ceratopipra_rubrocapilla_UNA002/g' | sed 's/Ceratopipra_rubrocapilla_UNA014/F_Ceratopipra_rubrocapilla_UNA014/g' | sed 's/Ceratopipra_rubrocapilla_UNA018/M_Ceratopipra_rubrocapilla_UNA018/g' | sed 's/Ceratopipra_rubrocapilla_UNA088/M_Ceratopipra_rubrocapilla_UNA088/g' | sed 's/Ceratopipra_rubrocapilla_UNA110/M_Ceratopipra_rubrocapilla_UNA110/g' | sed 's/Ceratopipra_rubrocapilla_UNA124/M_Ceratopipra_rubrocapilla_UNA124/g' | sed 's/Ceratopipra_rubrocapilla_WM281/F_Ceratopipra_rubrocapilla_WM281/g' | sed 's/Ceratopipra_rubrocapilla_WM293/F_Ceratopipra_rubrocapilla_WM293/g'  
```

# results

Here is the tree with 1000 bootstrap replicates:  

```
Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.treefile
(Ceratopipra_erythrocephala_SGC288:0.0012662246,(((Ceratopipra_erythrocephala_B7534:0.0008580458,Ceratopipra_erythrocephala_B7539:0.0009988556)84:0.0002345157,Ceratopipra_erythrocephala_B7537:0.0012218539)91:0.0001871447,Ceratopipra_erythrocephala_B7536:0.0012098688)74:0.0001798127,((((((((((((((Ceratopipra_rubrocapilla_CPE012.AtlanticForest_Ibateguara:0.0012185407,Ceratopipra_rubrocapilla_CPE025.AtlanticForest_Ibateguara:0.0012124607)95:0.0002464256,Ceratopipra_rubrocapilla_CPE049.AtlanticForest_Ibateguara:0.0011713475)79:0.0002177244,(Ceratopipra_rubrocapilla_CPE069.AtlanticForest_Ibateguara:0.0013048013,Ceratopipra_rubrocapilla_CPEII008.AtlanticForest_Ibateguara:0.0013116434)80:0.0002599020)100:0.0003738673,(((Ceratopipra_rubrocapilla_UNA002.AtlanticForest_Ilheus:0.0013105834,Ceratopipra_rubrocapilla_UNA018.AtlanticForest_Ilheus:0.0012965870)80:0.0002682017,(Ceratopipra_rubrocapilla_UNA110.AtlanticForest_Ilheus:0.0012993742,Ceratopipra_rubrocapilla_UNA124.AtlanticForest_Ilheus:0.0009630296)96:0.0002838801)75:0.0002194098,Ceratopipra_rubrocapilla_UNA088.AtlanticForest_Ilheus:0.0008305389)100:0.0004597035)100:0.0003491755,(((Ceratopipra_rubrocapilla_GAPTO259.Xingu/Belém_Araguaina:0.0012693477,(Ceratopipra_rubrocapilla_GAPTO277.Xingu/Belém_Araguaina:0.0011463953,Ceratopipra_rubrocapilla_GAPTO327.Xingu/Belém_Araguaina:0.0012488911)88:0.0001653159)86:0.0002148385,Ceratopipra_rubrocapilla_GAPTO268.Xingu/Belém_Araguaina:0.0014423979)100:0.0004251144,Ceratopipra_rubrocapilla_RDO007.Xingu/Belém_RondondoPará:0.0010634091)96:0.0001710784)100:0.0002154832,((Ceratopipra_rubrocapilla_PPBIO221.Xingu/Belém_Portel:0.0010755509,Ceratopipra_rubrocapilla_PPBIO305.Xingu/Belém_Portel:0.0011149767)100:0.0002416387,Ceratopipra_rubrocapilla_UHE442.Xingu/Belém_Sen.JoséPorf.:0.0005771532)100:0.0001790036)99:0.0001796752,((Ceratopipra_rubrocapilla_MF005.Tapajos_Trairão:0.0012028013,Ceratopipra_rubrocapilla_PIME135.Tapajos_Placas:0.0012690092)89:0.0001766094,((Ceratopipra_rubrocapilla_PIME134.Tapajos_Placas:0.0011954254,Ceratopipra_rubrocapilla_PIME482.Tapajos_Jacareacanga:0.0011235458)93:0.0001067622,Ceratopipra_rubrocapilla_WM293.Tapajos_Belterra:0.0006484232)91:0.0001134128)89:0.0001545129)87:0.0001696094,((Ceratopipra_rubrocapilla_BR033.Tapajos_Altamira:0.0013299697,Ceratopipra_rubrocapilla_PIME522.Tapajos_NovoProgresso:0.0010966984)70:0.0001665545,((Ceratopipra_rubrocapilla_BR036.Tapajos_Altamira:0.0012012842,Ceratopipra_rubrocapilla_PPS174.Tapajos_Altamira:0.0020228706)35:0.0001267291,(Ceratopipra_rubrocapilla_TLP025.Tapajos_Paranaíta:0.0011252261,Ceratopipra_rubrocapilla_TLP123.Tapajos_Paranaíta:0.0012572091)43:0.0001384186)79:0.0001751674)72:0.0001214210)97:0.0001591538,((Ceratopipra_rubrocapilla_MAM010.Rondonia_AVEIRO:0.0011760071,Ceratopipra_rubrocapilla_PIME460.Rondonia_JACAREACANGA:0.0012477778)59:0.0001831437,Ceratopipra_rubrocapilla_PIME024.Rondonia_Santarém:0.0012590338)41:0.0002007700)31:0.0000925171,((Ceratopipra_rubrocapilla_AMZ610.Rondonia_NovBandeirantes:0.0007981536,Ceratopipra_rubrocapilla_AMZ551.Rondonia_NovaBandeirantes:0.0006677981)81:0.0001550294,(Ceratopipra_rubrocapilla_OP100.Rondonia_Guajará-Mirim:0.0013524448,((Ceratopipra_rubrocapilla_PPS041.Tapajos_Marcelândia:0.0010032442,Ceratopipra_rubrocapilla_PPS221.Tapajos_GuarantãdoNorte:0.0020079597)95:0.0001998414,(Ceratopipra_rubrocapilla_MT087.Xingu/Belém_Querencia:0.0012651158,Ceratopipra_rubrocapilla_MT123.Xingu/Belém_Querencia:0.0013550080)99:0.0002217456)96:0.0002153520)75:0.0001163411)81:0.0001546434)33:0.0001145078,((Ceratopipra_rubrocapilla_FPR006.Rondonia_Maués:0.0020802131,(Ceratopipra_rubrocapilla_MAR092.Rondonia_Manicoré:0.0010606107,Ceratopipra_rubrocapilla_PIME079.Rondonia_ITAITUBA:0.0009771403)88:0.0001466573)67:0.0001615230,(((Ceratopipra_rubrocapilla_FPR015.Rondonia_Maués:0.0010992178,(Ceratopipra_rubrocapilla_MPDS628.Rondonia_HUMAITÁ:0.0012867729,Ceratopipra_rubrocapilla_PIME169.Rondonia_Jacareacanga:0.0010575974)77:0.0001277320)62:0.0001094415,Ceratopipra_rubrocapilla_PIME062.Tapajos_Itaituba:0.0012931003)33:0.0001149544,Ceratopipra_rubrocapilla_FPR021.Rondonia_Maués:0.0013070409)34:0.0001058247)34:0.0001449662)73:0.0001443346,((Ceratopipra_rubrocapilla_OM022.Inambari_Humaitá:0.0012444093,Ceratopipra_rubrocapilla_OM112.Inambari_Humaitá:0.0016660924)94:0.0002080557,(Ceratopipra_rubrocapilla_OM172.Inambari_Humaitá:0.0012137998,Ceratopipra_rubrocapilla_UFAC305.Inambari_SENADORGUIOMARD:0.0013226060)89:0.0001713309)65:0.0001602818)65:0.0001429586,(((Ceratopipra_rubrocapilla_AMA340.Inambari_AtalaiadoNorte:0.0009521167,(Ceratopipra_rubrocapilla_CUJ027.Inambari_RDSCujubim:0.0010963597,Ceratopipra_rubrocapilla_UFAC476.Inambari_TARAUACÁ:0.0011410506)82:0.0001267398)65:0.0001530770,((Ceratopipra_rubrocapilla_AMA656.Inambari_AtalaiadoNorte:0.0010580789,Ceratopipra_rubrocapilla_B4615.Inambari_Loreto:0.0011579445)53:0.0001502522,Ceratopipra_rubrocapilla_CAM117.Inambari_Guajará-AM:0.0013117552)61:0.0001873550)60:0.0001494843,((Ceratopipra_rubrocapilla_AMA601.Inambari_AtalaiadoNorte:0.0011004711,Ceratopipra_rubrocapilla_B5082.Inambari_Loreto:0.0012017416)68:0.0001302483,(Ceratopipra_rubrocapilla_CUJ072.Inambari_RDSCujubim:0.0010474520,Ceratopipra_rubrocapilla_UFAC1007.Inambari_PORTOWALTER:0.0010185649)63:0.0001280901)57:0.0001574029)68:0.0001808251)19:0.0001164004,(Ceratopipra_rubrocapilla_PUC008.Inambari_Tefe:0.0008821188,Ceratopipra_rubrocapilla_PUC007.Inambari_Tefe:0.0008969995)100:0.0006659482)100:0.0015189295);
```
And here is the consensus tree (essentially identical):
```
bootstrapped_trees/Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.contree

(Ceratopipra_erythrocephala_SGC288:0.0012688568,(((Ceratopipra_erythrocephala_B7534:0.0008606733,Ceratopipra_erythrocephala_B7539:0.0009990725)84:0.0002337955,Ceratopipra_erythrocephala_B7537:0.0012241885)91:0.0001898646,Ceratopipra_erythrocephala_B7536:0.0012090094)74:0.0001776486,(((((((((((((Ceratopipra_rubrocapilla_CPE012:0.0012208265,Ceratopipra_rubrocapilla_CPE025:0.0012114925)95:0.0002469069,Ceratopipra_rubrocapilla_CPE049:0.0011719941)79:0.0002171466,(Ceratopipra_rubrocapilla_CPE069:0.0013058170,Ceratopipra_rubrocapilla_CPEII008:0.0013115967)80:0.0002628283)100:0.0003743040,(((Ceratopipra_rubrocapilla_UNA002:0.0013113280,Ceratopipra_rubrocapilla_UNA018:0.0012977721)80:0.0002682179,(Ceratopipra_rubrocapilla_UNA110:0.0012997234,Ceratopipra_rubrocapilla_UNA124:0.0009633959)96:0.0002836162)75:0.0002202719,Ceratopipra_rubrocapilla_UNA088:0.0008301283)100:0.0004597785)100:0.0003458660,(((Ceratopipra_rubrocapilla_GAPTO259:0.0012727478,(Ceratopipra_rubrocapilla_GAPTO277:0.0011468124,Ceratopipra_rubrocapilla_GAPTO327:0.0012496771)88:0.0001643893)86:0.0002157161,Ceratopipra_rubrocapilla_GAPTO268:0.0014409353)100:0.0004269908,Ceratopipra_rubrocapilla_RDO007:0.0010628787)96:0.0001746061)100:0.0002152961,((Ceratopipra_rubrocapilla_PPBIO221:0.0010695065,Ceratopipra_rubrocapilla_PPBIO305:0.0011166432)100:0.0002464451,Ceratopipra_rubrocapilla_UHE442:0.0005827128)100:0.0001745678)99:0.0001797200,((Ceratopipra_rubrocapilla_MF005:0.0012010832,Ceratopipra_rubrocapilla_PIME135:0.0012710813)89:0.0001793334,((Ceratopipra_rubrocapilla_PIME134:0.0011959465,Ceratopipra_rubrocapilla_PIME482:0.0011205071)93:0.0001102290,Ceratopipra_rubrocapilla_WM293:0.0006483817)91:0.0001133144)89:0.0001486660)87:0.0001757885,((Ceratopipra_rubrocapilla_BR033:0.0013328576,Ceratopipra_rubrocapilla_PIME522:0.0010984080)70:0.0001696060,((Ceratopipra_rubrocapilla_BR036:0.0012221185,Ceratopipra_rubrocapilla_TLP123:0.0012685640)52:0.0001231872,(Ceratopipra_rubrocapilla_PPS174:0.0020321636,Ceratopipra_rubrocapilla_TLP025:0.0011226771)42:0.0001193445)79:0.0001719616)72:0.0001191993)97:0.0001718164,((Ceratopipra_rubrocapilla_AMZ610:0.0007967698,Ceratopipra_rubrocapilla_AMZ551:0.0006686508)81:0.0001569208,(Ceratopipra_rubrocapilla_OP100:0.0013602842,((Ceratopipra_rubrocapilla_PPS041:0.0010089084,Ceratopipra_rubrocapilla_PPS221:0.0020072793)95:0.0001889039,(Ceratopipra_rubrocapilla_MT087:0.0012670244,Ceratopipra_rubrocapilla_MT123:0.0013542688)99:0.0002290086)96:0.0002106656)75:0.0001170867)81:0.0001472816)44:0.0000928284,((Ceratopipra_rubrocapilla_MAM010:0.0011858369,Ceratopipra_rubrocapilla_PIME460:0.0012441643)59:0.0001775923,Ceratopipra_rubrocapilla_PIME024:0.0012592239)41:0.0002158578)33:0.0001006533,((Ceratopipra_rubrocapilla_FPR006:0.0020842677,(Ceratopipra_rubrocapilla_MAR092:0.0010621646,Ceratopipra_rubrocapilla_PIME079:0.0009843212)88:0.0001417448)67:0.0001628110,((Ceratopipra_rubrocapilla_FPR015:0.0011071583,(Ceratopipra_rubrocapilla_MPDS628:0.0012904992,Ceratopipra_rubrocapilla_PIME169:0.0010571548)77:0.0001330179)62:0.0001568037,(Ceratopipra_rubrocapilla_FPR021:0.0012280551,Ceratopipra_rubrocapilla_PIME062:0.0012501091)46:0.0001349288)34:0.0001284570)34:0.0001420588)73:0.0001385975,((Ceratopipra_rubrocapilla_OM022:0.0012479072,Ceratopipra_rubrocapilla_OM112:0.0016683366)94:0.0002084770,(Ceratopipra_rubrocapilla_OM172:0.0012131873,Ceratopipra_rubrocapilla_UFAC305:0.0013233637)89:0.0001696571)65:0.0001566905)65:0.0001202474,((((Ceratopipra_rubrocapilla_AMA340:0.0009535274,(Ceratopipra_rubrocapilla_CUJ027:0.0010950371,Ceratopipra_rubrocapilla_UFAC476:0.0011395847)82:0.0001290971)65:0.0001511287,((Ceratopipra_rubrocapilla_AMA656:0.0010565867,Ceratopipra_rubrocapilla_B4615:0.0011539415)53:0.0001529623,Ceratopipra_rubrocapilla_CAM117:0.0013099970)61:0.0001910226)60:0.0001472106,((Ceratopipra_rubrocapilla_AMA601:0.0011023470,Ceratopipra_rubrocapilla_B5082:0.0012025937)68:0.0001257855,(Ceratopipra_rubrocapilla_CUJ072:0.0010494837,Ceratopipra_rubrocapilla_UFAC1007:0.0010156022)63:0.0001336552)57:0.0001541775)68:0.0001523075,(Ceratopipra_rubrocapilla_PUC008:0.0008845655,Ceratopipra_rubrocapilla_PUC007:0.0009052248)100:0.0007171343)40:0.0000957107)100:0.0015821806);

```

To make the figure:  
1) Open in Figtree v1.4.4  
2) check "Align tip labels"  
3) Select outgroup (*Ceratopipra erythrocephala*) -> click "reroot"  
4) check "node labels". Display: Label (this is the bootstrap value)  

Then, I mannually collapsed any nodes with less than 50% bootstrap support. We cannot have any confidence in those nodes, so I consider it better to present a polytomy at those nodes to reflect the fact that we really do not know the branch order.    

## Appendix: log

Here is the log. Note the highest amount of missing data comes from Ceratopipra_rubrocapilla_AMZ610 at 52.31%. It would be next on the chopping block if raising the missingness threshold. Note that my initial filter allowed only 40% missingness - AMZ610 was initially at 39% missing but increased due to my min depth filter that converted low-depth genotype calls to missing data. It is a bit tricky because missingness per sample changes with different missingness per site filters, and I want to implement per sample filters first so that I can get the best sites for the remaining cohort. 

The next up that just made the cut for exclusion in the m07 dataset was Ceratopipra_rubrocapilla_AMANA137, which went to 55%, 59%, or 62% missing data depending what other samples were included (i.e., the no-m, m09, or m08  dataset) (despite having the same filters on genotypes and sites. Having more samples with high missingness means that the sites that are retained with the 20% missing data filter are more likely to have that 20% taken up by higher-missingness samples, so there is less opportunity for sites where Ceratopipra_rubrocapilla_AMANA137 has missing data to be retained. When other samples have low missingness, sites where Ceratopipra_rubrocapilla_AMANA137 has missing data can be retained more often, so the sample ends up with higher missingness due to inclusion of those sites)
```
IQ-TREE multicore version 2.1.3 COVID-edition for Linux 64-bit built Apr 21 2021
Developed by Bui Quang Minh, James Barbetti, Nguyen Lam Tung,
Olga Chernomor, Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    weirlab60 (AVX, 503 GB RAM)
Command: /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.fasta -redo -t PARS -ninit 2 -nt AUTO -B 1000 --prefix Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000
Seed:    175964 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Tue Jul 25 00:18:12 2023
Kernel:  AVX - auto-detect threads (120 CPU cores detected)

Reading alignment file datasets/Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.fasta ... Fasta format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 68 sequences with 2130795 columns, 305776 distinct patterns
17948 parsimony-informative, 16274 singleton sites, 2096573 constant sites
                                   Gap/Ambiguity  Composition  p-value
   1  Ceratopipra_erythrocephala_SGC288    1.09%    passed     33.95%
   2  Ceratopipra_erythrocephala_B7534     1.34%    passed     41.36%
   3  Ceratopipra_erythrocephala_B7539     1.21%    passed     42.85%
   4  Ceratopipra_erythrocephala_B7536     1.37%    passed     39.88%
   5  Ceratopipra_erythrocephala_B7537     1.26%    passed     43.19%
   6  Ceratopipra_rubrocapilla_CPE012      1.04%    passed     38.29%
   7  Ceratopipra_rubrocapilla_CPE025      1.12%    passed     53.56%
   8  Ceratopipra_rubrocapilla_CPE049      8.14%    passed     43.04%
   9  Ceratopipra_rubrocapilla_CPE069      0.90%    passed     40.04%
  10  Ceratopipra_rubrocapilla_CPEII008    0.96%    passed     49.26%
  11  Ceratopipra_rubrocapilla_UNA002      1.06%    passed     45.22%
  12  Ceratopipra_rubrocapilla_UNA018      0.96%    passed     26.07%
  13  Ceratopipra_rubrocapilla_UNA088      7.10%    failed      0.09%
  14  Ceratopipra_rubrocapilla_UNA110      0.97%    passed     35.31%
  15  Ceratopipra_rubrocapilla_UNA124      3.80%    passed     88.03%
  16  Ceratopipra_rubrocapilla_AMA340      1.92%    passed     65.46%
  17  Ceratopipra_rubrocapilla_AMA601      4.67%    passed     82.31%
  18  Ceratopipra_rubrocapilla_AMA656     37.19%    failed      0.00%
  19  Ceratopipra_rubrocapilla_B4615       9.08%    failed      0.00%
  20  Ceratopipra_rubrocapilla_B5082      13.86%    failed      0.00%
  21  Ceratopipra_rubrocapilla_CAM117      0.84%    passed     34.73%
  22  Ceratopipra_rubrocapilla_CUJ027      5.97%    passed     70.85%
  23  Ceratopipra_rubrocapilla_CUJ072     12.68%    failed      0.09%
  24  Ceratopipra_rubrocapilla_OM022       0.84%    passed     48.63%
  25  Ceratopipra_rubrocapilla_OM112       1.07%    passed     48.71%
  26  Ceratopipra_rubrocapilla_OM172       0.86%    passed     45.94%
  27  Ceratopipra_rubrocapilla_PUC008      0.78%    passed     44.32%
  28  Ceratopipra_rubrocapilla_PUC007      0.82%    passed     42.86%
  29  Ceratopipra_rubrocapilla_UFAC1007   46.74%    failed      0.00%
  30  Ceratopipra_rubrocapilla_UFAC305     0.95%    passed     41.47%
  31  Ceratopipra_rubrocapilla_UFAC476    32.00%    failed      3.52%
  32  Ceratopipra_rubrocapilla_AMZ610     52.31%    failed      0.00%
  33  Ceratopipra_rubrocapilla_FPR006      1.13%    passed     47.08%
  34  Ceratopipra_rubrocapilla_FPR015     20.77%    failed      0.00%
  35  Ceratopipra_rubrocapilla_FPR021      0.77%    passed     51.60%
  36  Ceratopipra_rubrocapilla_MAM010      7.82%    passed     95.89%
  37  Ceratopipra_rubrocapilla_MAR092     22.00%    failed      0.00%
  38  Ceratopipra_rubrocapilla_MPDS628    43.14%    failed      0.00%
  39  Ceratopipra_rubrocapilla_OP100       0.94%    passed     46.79%
  40  Ceratopipra_rubrocapilla_PIME024     0.83%    passed     44.68%
  41  Ceratopipra_rubrocapilla_PIME079    33.38%    passed     52.05%
  42  Ceratopipra_rubrocapilla_PIME460     2.11%    passed     98.21%
  43  Ceratopipra_rubrocapilla_BR033       2.49%    passed      5.89%
  44  Ceratopipra_rubrocapilla_BR036       6.95%    failed      0.00%
  45  Ceratopipra_rubrocapilla_MF005       0.76%    passed     47.93%
  46  Ceratopipra_rubrocapilla_PIME062     0.79%    passed     38.73%
  47  Ceratopipra_rubrocapilla_PIME134     6.89%    failed      0.00%
  48  Ceratopipra_rubrocapilla_PIME135     0.86%    passed     41.51%
  49  Ceratopipra_rubrocapilla_PIME169     7.88%    passed     99.73%
  50  Ceratopipra_rubrocapilla_PIME482    14.03%    failed      0.00%
  51  Ceratopipra_rubrocapilla_PIME522     4.52%    passed     98.66%
  52  Ceratopipra_rubrocapilla_PPS041     31.55%    failed      0.00%
  53  Ceratopipra_rubrocapilla_PPS174     17.36%    failed      0.00%
  54  Ceratopipra_rubrocapilla_PPS221     20.11%    failed      0.00%
  55  Ceratopipra_rubrocapilla_TLP025     13.56%    failed      0.00%
  56  Ceratopipra_rubrocapilla_TLP123      1.65%    passed     99.98%
  57  Ceratopipra_rubrocapilla_WM293       2.85%    passed     83.04%
  58  Ceratopipra_rubrocapilla_AMZ551     43.17%    failed      0.00%
  59  Ceratopipra_rubrocapilla_GAPTO259    5.46%    failed      0.00%
  60  Ceratopipra_rubrocapilla_GAPTO268    3.51%    failed      0.18%
  61  Ceratopipra_rubrocapilla_GAPTO277   40.33%    failed      0.00%
  62  Ceratopipra_rubrocapilla_GAPTO327    2.76%    failed      2.74%
  63  Ceratopipra_rubrocapilla_MT087       0.90%    passed     56.76%
  64  Ceratopipra_rubrocapilla_MT123       0.86%    passed     43.16%
  65  Ceratopipra_rubrocapilla_PPBIO221   42.30%    failed      0.00%
  66  Ceratopipra_rubrocapilla_PPBIO305   23.08%    failed      0.00%
  67  Ceratopipra_rubrocapilla_RDO007      4.71%    passed     80.46%
  68  Ceratopipra_rubrocapilla_UHE442      3.41%    passed     61.53%
WARNING: 1 sequences contain more than 50% gaps/ambiguity
****  TOTAL                               10.16%  25 sequences failed composition chi2 test (p-value<5%; df=3)
NOTE: minimal branch length is reduced to 0.000000046931 for long alignment

Creating fast initial parsimony tree by random order stepwise addition...
14.145 seconds, parsimony score: 99114 (based on 34222 sites)
Measuring multi-threading efficiency up to 120 CPU cores
8 trees examined
Threads: 1 / Time: 120.489 sec / Speedup: 1.000 / Efficiency: 100% / LogL: -3872056
Threads: 2 / Time: 69.996 sec / Speedup: 1.721 / Efficiency: 86% / LogL: -3872056
Threads: 3 / Time: 51.215 sec / Speedup: 2.353 / Efficiency: 78% / LogL: -3872056
Threads: 4 / Time: 45.218 sec / Speedup: 2.665 / Efficiency: 67% / LogL: -3872056
Threads: 5 / Time: 39.769 sec / Speedup: 3.030 / Efficiency: 61% / LogL: -3872056
Threads: 6 / Time: 33.928 sec / Speedup: 3.551 / Efficiency: 59% / LogL: -3872056
Threads: 7 / Time: 31.493 sec / Speedup: 3.826 / Efficiency: 55% / LogL: -3872056
Threads: 8 / Time: 24.555 sec / Speedup: 4.907 / Efficiency: 61% / LogL: -3872056
Threads: 9 / Time: 23.635 sec / Speedup: 5.098 / Efficiency: 57% / LogL: -3872056
Threads: 10 / Time: 23.213 sec / Speedup: 5.191 / Efficiency: 52% / LogL: -3872056
Threads: 11 / Time: 21.422 sec / Speedup: 5.624 / Efficiency: 51% / LogL: -3872056
Threads: 12 / Time: 18.032 sec / Speedup: 6.682 / Efficiency: 56% / LogL: -3872056
Threads: 13 / Time: 17.292 sec / Speedup: 6.968 / Efficiency: 54% / LogL: -3872056
Threads: 14 / Time: 18.428 sec / Speedup: 6.538 / Efficiency: 47% / LogL: -3872056
BEST NUMBER OF THREADS: 12

Perform fast likelihood tree search using GTR+I+G model...
Estimate model parameters (epsilon = 5.000)
Perform nearest neighbor interchange...
Estimate model parameters (epsilon = 1.000)
1. Initial log-likelihood: -3577407.424
2. Current log-likelihood: -3577404.503
3. Current log-likelihood: -3577403.166
Optimal log-likelihood: -3577402.231
Rate parameters:  A-C: 1.06491  A-G: 9.43054  A-T: 1.00564  C-G: 1.08138  C-T: 9.39275  G-T: 1.00000
Base frequencies:  A: 0.233  C: 0.267  G: 0.267  T: 0.233
Proportion of invariable sites: 0.972
Gamma shape alpha: 0.376
Parameters optimization took 3 rounds (37.090 sec)
Time for fast ML tree search: 907.556 seconds

NOTE: ModelFinder requires 7908 MB RAM!
ModelFinder will test up to 286 DNA models (sample size: 2130795) ...
 No. Model         -LnL         df  AIC          AICc         BIC
  1  GTR+F         3852938.293  141 7706158.586  7706158.605  7707931.239
  2  GTR+F+I       3602439.332  142 7205162.663  7205162.682  7206947.888
  3  GTR+F+G4      3760078.232  142 7520440.464  7520440.483  7522225.689
  4  GTR+F+I+G4    3577400.278  143 7155086.555  7155086.575  7156884.352
  5  GTR+F+R2      3581470.826  143 7163227.653  7163227.672  7165025.450
  6  GTR+F+R3      3581469.683  145 7163229.367  7163229.386  7165052.307
 17  SYM+I+G4      3582295.235  140 7164870.469  7164870.488  7166630.550
 30  TVM+F+I+G4    3577401.342  142 7155086.685  7155086.704  7156871.910
 43  TVMe+I+G4     3582295.980  139 7164869.959  7164869.977  7166617.468
 56  TIM3+F+I+G4   3577400.087  141 7155082.175  7155082.193  7156854.827
 69  TIM3e+I+G4    3582328.692  138 7164933.383  7164933.401  7166668.320
 82  TIM2+F+I+G4   3577406.521  141 7155095.042  7155095.060  7156867.694
 95  TIM2e+I+G4    3582354.199  138 7164984.397  7164984.415  7166719.334
108  TIM+F+I+G4    3577406.698  141 7155095.397  7155095.416  7156868.050
121  TIMe+I+G4     3582382.592  138 7165041.184  7165041.202  7166776.121
134  TPM3u+F+I+G4  3577401.115  140 7155082.230  7155082.248  7156842.311
147  TPM3+F+I+G4   3577401.075  140 7155082.149  7155082.168  7156842.230
160  TPM2u+F+I+G4  3577407.744  140 7155095.489  7155095.507  7156855.569
173  TPM2+F+I+G4   3577407.744  140 7155095.489  7155095.507  7156855.569
186  K3Pu+F+I+G4   3577407.881  140 7155095.763  7155095.781  7156855.843
199  K3P+I+G4      3582383.909  137 7165041.817  7165041.835  7166764.182
212  TN+F+I+G4     3577408.008  140 7155096.016  7155096.035  7156856.097
225  TNe+I+G4      3582385.341  137 7165044.681  7165044.699  7166767.046
238  HKY+F+I+G4    3577407.966  139 7155093.931  7155093.950  7156841.440
251  K2P+I+G4      3582386.683  136 7165045.366  7165045.383  7166755.159
264  F81+F+I+G4    3627107.087  138 7254490.175  7254490.193  7256225.112
277  JC+I+G4       3631433.971  135 7263137.943  7263137.960  7264835.164
Akaike Information Criterion:           TPM3+F+I+G4
Corrected Akaike Information Criterion: TPM3+F+I+G4
Bayesian Information Criterion:         HKY+F+I+G4
Best-fit model: HKY+F+I+G4 chosen according to BIC

All model information printed to Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.model.gz
CPU time for ModelFinder: 34027.049 seconds (9h:27m:7s)
Wall-clock time for ModelFinder: 3240.193 seconds (0h:54m:0s)
Generating 1000 samples for ultrafast bootstrap (seed: 175964)...

NOTE: 3863 MB RAM (3 GB) is required!
Estimate model parameters (epsilon = 0.100)
Thoroughly optimizing +I+G parameters from 10 start values...
Init pinv, alpha: 0.000, 0.382 / Estimate: 0.000, 0.020 / LogL: -3760176.665
Init pinv, alpha: 0.109, 0.382 / Estimate: 0.973, 0.380 / LogL: -3577408.074
Init pinv, alpha: 0.219, 0.382 / Estimate: 0.973, 0.380 / LogL: -3577408.052
Init pinv, alpha: 0.328, 0.382 / Estimate: 0.973, 0.381 / LogL: -3577408.041
Init pinv, alpha: 0.437, 0.382 / Estimate: 0.973, 0.380 / LogL: -3577408.059
Init pinv, alpha: 0.547, 0.382 / Estimate: 0.973, 0.380 / LogL: -3577408.089
Init pinv, alpha: 0.656, 0.382 / Estimate: 0.973, 0.381 / LogL: -3577408.045
Init pinv, alpha: 0.765, 0.382 / Estimate: 0.973, 0.380 / LogL: -3577408.060
Init pinv, alpha: 0.875, 0.382 / Estimate: 0.973, 0.380 / LogL: -3577408.054
Init pinv, alpha: 0.984, 0.382 / Estimate: 0.973, 0.384 / LogL: -3577408.052
Optimal pinv,alpha: 0.973, 0.381 / LogL: -3577408.041

Parameters optimization took 6065.685 sec
Computing ML distances based on estimated model parameters...
Computing ML distances took 0.337306 sec (of wall-clock time) 4.021044 sec(of CPU time)
Computing RapidNJ tree took 0.008080 sec (of wall-clock time) 0.087715 sec (of CPU time)
Log-likelihood of RapidNJ tree: -3578261.241
--------------------------------------------------------------------
|             INITIALIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------

Do NNI search on 2 best initial trees
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 1: -3577375.012
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 2: -3577188.254
Finish initializing candidate tree set (4)
Current best tree score: -3577188.254 / CPU time: 109.523
Number of iterations: 2
--------------------------------------------------------------------
|               OPTIMIZING CANDIDATE TREE SET                      |
--------------------------------------------------------------------
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 8: -3577139.830
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 10: -3577030.396
Iteration 10 / LogL: -3577030.396 / Time: 1h:50m:29s
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 12: -3577003.083
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 13: -3576953.990
Iteration 20 / LogL: -3577133.249 / Time: 1h:57m:34s
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 23: -3576922.077
Iteration 30 / LogL: -3576928.784 / Time: 2h:3m:36s (6h:36m:24s left)
Iteration 40 / LogL: -3576948.488 / Time: 2h:9m:52s (4h:36m:23s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 43: -3576913.571
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 45: -3576908.734
Iteration 50 / LogL: -3576961.405 / Time: 2h:15m:58s (4h:23m:38s left)
Log-likelihood cutoff on original alignment: -3577208.056
Iteration 60 / LogL: -3576933.073 / Time: 2h:22m:26s (3h:25m:12s left)
Iteration 70 / LogL: -3576941.795 / Time: 2h:28m:43s (2h:41m:39s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 79: -3576906.568
Iteration 80 / LogL: -3576938.857 / Time: 2h:34m:43s (3h:13m:53s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 85: -3576891.281
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 88: -3576882.391
Iteration 90 / LogL: -3576903.428 / Time: 2h:42m:9s (2h:58m:32s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 100: -3576866.589
Iteration 100 / LogL: -3576866.589 / Time: 2h:48m:28s (2h:50m:10s left)
Log-likelihood cutoff on original alignment: -3577208.056
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.937
NOTE: UFBoot does not converge, continue at least 100 more iterations
Iteration 110 / LogL: -3576986.217 / Time: 2h:54m:54s (2h:24m:24s left)
Iteration 120 / LogL: -3576902.279 / Time: 3h:1m:5s (2h:1m:44s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 123: -3576862.300
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 128: -3576854.504
Iteration 130 / LogL: -3576917.786 / Time: 3h:7m:47s (2h:22m:40s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 137: -3576840.186
Iteration 140 / LogL: -3576907.044 / Time: 3h:14m:52s (2h:15m:59s left)
Iteration 150 / LogL: -3576872.844 / Time: 3h:20m:53s (1h:57m:17s left)
Log-likelihood cutoff on original alignment: -3577139.873
Iteration 160 / LogL: -3576942.370 / Time: 3h:27m:23s (1h:40m:26s left)
Iteration 170 / LogL: -3576847.257 / Time: 3h:34m:18s (1h:24m:57s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 173: -3576828.003
Iteration 180 / LogL: -3576876.271 / Time: 3h:40m:52s (1h:54m:45s left)
Iteration 190 / LogL: -3576881.196 / Time: 3h:47m:32s (1h:39m:55s left)
Iteration 200 / LogL: -3576861.645 / Time: 3h:54m:29s (1h:26m:1s left)
Log-likelihood cutoff on original alignment: -3577139.873
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.936
NOTE: UFBoot does not converge, continue at least 100 more iterations
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 201: -3576827.622
Iteration 210 / LogL: -3576968.307 / Time: 4h:1m:14s (1h:45m:2s left)
Iteration 220 / LogL: -3576988.445 / Time: 4h:8m:14s (1h:31m:48s left)
Estimate model parameters (epsilon = 0.100)
BETTER TREE FOUND at iteration 221: -3576825.166
Iteration 230 / LogL: -3576833.342 / Time: 4h:15m:5s (1h:41m:22s left)
Iteration 240 / LogL: -3576831.470 / Time: 4h:22m:39s (1h:29m:0s left)
Iteration 250 / LogL: -3576828.856 / Time: 4h:30m:6s (1h:17m:0s left)
Log-likelihood cutoff on original alignment: -3577121.218
Iteration 260 / LogL: -3576911.698 / Time: 4h:37m:2s (1h:5m:14s left)
Iteration 270 / LogL: -3576979.600 / Time: 4h:43m:16s (0h:53m:42s left)
Iteration 280 / LogL: -3576860.044 / Time: 4h:49m:44s (0h:42m:34s left)
Iteration 290 / LogL: -3576906.951 / Time: 4h:55m:38s (0h:31m:42s left)
Iteration 300 / LogL: -3576900.185 / Time: 5h:1m:57s (0h:21m:12s left)
Log-likelihood cutoff on original alignment: -3577121.218
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.976
NOTE: UFBoot does not converge, continue at least 100 more iterations
Iteration 310 / LogL: -3576934.674 / Time: 5h:8m:1s (1h:29m:42s left)
Iteration 320 / LogL: -3576886.856 / Time: 5h:14m:37s (1h:18m:54s left)
Iteration 330 / LogL: -3576846.467 / Time: 5h:21m:1s (1h:8m:18s left)
Iteration 340 / LogL: -3576827.685 / Time: 5h:27m:16s (0h:57m:55s left)
Estimate model parameters (epsilon = 0.100)
UPDATE BEST LOG-LIKELIHOOD: -3576823.978
Iteration 350 / LogL: -3576901.947 / Time: 5h:33m:19s (0h:47m:45s left)
Log-likelihood cutoff on original alignment: -3577121.218
Iteration 360 / LogL: -3576827.719 / Time: 5h:39m:35s (0h:37m:50s left)
Iteration 370 / LogL: -3576825.219 / Time: 5h:45m:49s (0h:28m:6s left)
Iteration 380 / LogL: -3576998.217 / Time: 5h:52m:1s (0h:18m:34s left)
Iteration 390 / LogL: -3576901.550 / Time: 5h:58m:3s (0h:9m:12s left)
Iteration 400 / LogL: -3576889.598 / Time: 6h:4m:16s (0h:0m:0s left)
Log-likelihood cutoff on original alignment: -3577121.218
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.976
NOTE: UFBoot does not converge, continue at least 100 more iterations
Iteration 410 / LogL: -3576928.267 / Time: 6h:10m:41s (1h:21m:34s left)
Iteration 420 / LogL: -3576841.667 / Time: 6h:17m:25s (1h:12m:3s left)
Iteration 430 / LogL: -3576845.744 / Time: 6h:23m:27s (1h:2m:34s left)
Iteration 440 / LogL: -3576883.780 / Time: 6h:29m:35s (0h:53m:14s left)
Iteration 450 / LogL: -3576827.769 / Time: 6h:35m:35s (0h:44m:3s left)
Log-likelihood cutoff on original alignment: -3577121.218
Iteration 460 / LogL: -3576881.325 / Time: 6h:42m:14s (0h:35m:3s left)
Iteration 470 / LogL: -3576824.101 / Time: 6h:48m:27s (0h:26m:7s left)
Iteration 480 / LogL: -3576865.764 / Time: 6h:55m:7s (0h:17m:19s left)
Iteration 490 / LogL: -3576922.280 / Time: 7h:1m:4s (0h:8m:36s left)
Iteration 500 / LogL: -3576892.390 / Time: 7h:7m:57s (0h:0m:0s left)
Log-likelihood cutoff on original alignment: -3577121.218
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.987
NOTE: UFBoot does not converge, continue at least 100 more iterations
Iteration 510 / LogL: -3576862.030 / Time: 7h:13m:40s (1h:16m:40s left)
Iteration 520 / LogL: -3576827.666 / Time: 7h:19m:53s (1h:7m:48s left)
Iteration 530 / LogL: -3576851.662 / Time: 7h:25m:31s (0h:58m:57s left)
Iteration 540 / LogL: -3576831.458 / Time: 7h:31m:39s (0h:50m:16s left)
Iteration 550 / LogL: -3576827.777 / Time: 7h:37m:44s (0h:41m:41s left)
Log-likelihood cutoff on original alignment: -3577121.218
Iteration 560 / LogL: -3576929.412 / Time: 7h:43m:43s (0h:33m:10s left)
Iteration 570 / LogL: -3576893.095 / Time: 7h:50m:4s (0h:24m:47s left)
Iteration 580 / LogL: -3576828.106 / Time: 7h:56m:21s (0h:16m:27s left)
Iteration 590 / LogL: -3576824.011 / Time: 8h:2m:40s (0h:8m:11s left)
Iteration 600 / LogL: -3577004.531 / Time: 8h:9m:23s (0h:0m:0s left)
Log-likelihood cutoff on original alignment: -3577121.218
NOTE: Bootstrap correlation coefficient of split occurrence frequencies: 0.995
TREE SEARCH COMPLETED AFTER 600 ITERATIONS / Time: 8h:9m:23s

--------------------------------------------------------------------
|                    FINALIZING TREE SEARCH                        |
--------------------------------------------------------------------
Performs final model parameters optimization
Estimate model parameters (epsilon = 0.010)
1. Initial log-likelihood: -3576823.978
Optimal log-likelihood: -3576823.978
Rate parameters:  A-C: 1.00000  A-G: 9.02405  A-T: 1.00000  C-G: 1.00000  C-T: 9.02405  G-T: 1.00000
Base frequencies:  A: 0.233  C: 0.267  G: 0.267  T: 0.233
Proportion of invariable sites: 0.973
Gamma shape alpha: 0.382
Parameters optimization took 1 rounds (10.903 sec)
BEST SCORE FOUND : -3576823.978
Creating bootstrap support values...
Split supports printed to NEXUS file Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.splits.nex
Total tree length: 0.094

Total number of iterations: 600
CPU time used for tree search: 272145.428 sec (75h:35m:45s)
Wall-clock time used for tree search: 23218.639 sec (6h:26m:58s)
Total CPU time used: 343181.645 sec (95h:19m:41s)
Total wall-clock time used: 29375.119 sec (8h:9m:35s)

Computing bootstrap consensus tree...
Reading input file Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.splits.nex...
68 taxa and 723 splits.
Consensus tree written to Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.contree
Reading input trees file Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.contree
Log-likelihood of consensus tree: -3576841.575

Analysis results written to: 
  IQ-TREE report:                Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.iqtree
  Maximum-likelihood tree:       Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.treefile
  Likelihood distances:          Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.mldist

Ultrafast bootstrap approximation results written to:
  Split support values:          Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.splits.nex
  Consensus tree:                Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.contree
  Screen log file:               Ceratopipra_rubrocapilla_imeriout_m07.GBSPipfilbwa.allsites.filtered.for_pixy.auto.49_0.2_5_20.boot1000.log

Date and Time: Tue Jul 25 09:22:07 2023
```

