# ML mtDNA tree

This page describes how the maximum likelihood tree of mtDNA sequences was generated.  

# Dataset  
I received the data in the file `Crub_CYTB_2015_out.fas`. There are 141 *Ceratopipra rubrocapilla* sequences in this alignment. Most samples have no missing data.  

Before proceeding, I corrected minor spelling differences in sample names so that they are an exact match between GBS and mtDNA data:  
0P022 -> OP022  
BR364040 -> BR36440  
pime62 -> PIME062  
pime79 -> PIME079  
TLPB025 -> TLP025  
TLPB123 -> TLP123  

Next, I selected an outgroup. *Ceratopipra chloromeros* is the closest relative of *C. rubrocapilla* and so would make the best outgroup. We have these samples available:  
* B27638 (already in this alignment)  
* Ceratopipra_chloromeros_B106768 (in Ceratopipra2016.fas)  
* Ceratopipra_chloromeros_B9014 (in Ceratopipra2016.fas)  

I removed a sample of *C. erythrocephala* to simplify the analysis (this paper is only focused on *Ceratopipra rubrocapilla*), and added the other two chloromeros so that there are 3 chloromeros as the outgroup. Note that our GBS data used *C. erythrocephala* as an outgroup instead of *C. chloromeros* - that was because I was concerned that western Inambari samples could have introgressed with *C. chloromeros* given their close geographic contact and that this could bias the nuclear phylogeny or ABBA BABA test. I am not concerned about this with mtDNA since the mtDNA is non-recombining and so introgressed samples should be easy to spot.   

I then checked for stop codons, which can be indicative of numts. I translated the sequence using the vertebrate mitochondrial code in reading frame 3 and I found two samples with a stop codon in their sequence: OM172 and PPS081. These could just be sequencing errors, but I will exclude the whole sequence to be safe. It is extremely unlikely that a genuine bird mitochondrial cytb sequence would contain such an early stop codon, as this would undoubtably be fatal. I did run the phylogenies with these samples as well; they do not change any interpretations of the results, as the sequences are just placed embedded within the most common clade (haplotype 1).  

I saved this alignment as 'Crubrocapilla_Cytb954bp_3chloromerosout.fasta' with the two possible numts (for record keeping purposes, not to be used), and 'Crubrocapilla_Cytb954bp_filtered_3chloromerosout.fasta' without the 2 samples that are possible numts.  
Then, I made the names more informative by adding areas of endemism:  
```bash
sed 's/B27638.*$/B27638_Cchl/g; s/B9014.*$/B9014_Cchl/g; s/B106768.*$/B106768_Cchl/g; s/AMA340.*$/AMA340_In/g; s/AMA601.*$/AMA601_In/g; s/AMA656.*$/AMA656_In/g; s/AMANA034.*$/AMANA034_Ro/g; s/AMANA137.*$/AMANA137_Ro/g; s/AMZ610.*$/AMZ610_Ro/g; s/ARA010.*$/ARA010_Ro/g; s/ARA109.*$/ARA109_Ro/g; s/ARAII010.*$/ARAII010_Ro/g; s/ARAII026.*$/ARAII026_Ro/g; s/ARAII031.*$/ARAII031_Ro/g; s/ARAII033.*$/ARAII033_Ro/g; s/ARAII062.*$/ARAII062_Ro/g; s/ARAII069.*$/ARAII069_Ro/g; s/BR163177.*$/BR163177_Ta/g; s/BR36440.*$/BR36440_In/g; s/CAM009.*$/CAM009_In/g; s/CAM039.*$/CAM039_In/g; s/CAM087.*$/CAM087_In/g; s/CAM097.*$/CAM097_In/g; s/CAM107.*$/CAM107_In/g; s/CAM117.*$/CAM117_In/g; s/CAR005.*$/CAR005_Xi/g; s/CH288.*$/CH288_Ta/g; s/CH289.*$/CH289_Ta/g; s/CH290.*$/CH290_Ta/g; s/CPE012.*$/CPE012_At/g; s/CPE025.*$/CPE025_At/g; s/CPE049.*$/CPE049_At/g; s/CPE066.*$/CPE066_At/g; s/CPE067.*$/CPE067_At/g; s/CPE069.*$/CPE069_At/g; s/CPE071.*$/CPE071_At/g; s/CPEII008.*$/CPEII008_At/g; s/CUJ027.*$/CUJ027_In/g; s/CUJ072.*$/CUJ072_In/g; s/CUJ079.*$/CUJ079_In/g; s/FLJA001.*$/FLJA001_Ta/g; s/FLJA085.*$/FLJA085_Ta/g; s/FPR006.*$/FPR006_Ro/g; s/FPR015.*$/FPR015_Ro/g; s/FPR021.*$/FPR021_Ro/g; s/GAPTO140.*$/GAPTO140_Xi/g; s/GAPTO259.*$/GAPTO259_Xi/g; s/GAPTO268.*$/GAPTO268_Xi/g; s/GAPTO277.*$/GAPTO277_Xi/g; s/GAPTO289.*$/GAPTO289_Xi/g; s/GAPTO327.*$/GAPTO327_Xi/g; s/JATA145.*$/JATA145_Ro/g; s/JATA201.*$/JATA201_Ro/g; s/JATB060.*$/JATB060_Ta/g; s/JATC147.*$/JATC147_Ro/g; s/JATC148.*$/JATC148_Ro/g; s/JATC149.*$/JATC149_Ro/g; s/JRT021.*$/JRT021_Ro/g; s/JRT022.*$/JRT022_Ro/g; s/JTW1135.*$/JTW1135_Ta/g; s/JTW1220.*$/JTW1220_Ta/g; s/JTW1230.*$/JTW1230_Ta/g; s/JUT094.*$/JUT094_In/g; s/JUT237.*$/JUT237_In/g; s/MAM010.*$/MAM010_Ro/g; s/MAM011.*$/MAM011_Ro/g; s/MAR021.*$/MAR021_Ro/g; s/MAR092.*$/MAR092_Ro/g; s/MF005.*$/MF005_Ta/g; s/MFNC067.*$/MFNC067_Xi/g; s/MPDS1220.*$/MPDS1220_Ta/g; s/MPDS496.*$/MPDS496_Ro/g; s/MPDS608.*$/MPDS608_Ro/g; s/MPDS628.*$/MPDS628_Ro/g; s/MPDS672.*$/MPDS672_Ro/g; s/MPDS776.*$/MPDS776_Ro/g; s/MSF052.*$/MSF052_Ta/g; s/MSF107.*$/MSF107_Ta/g; s/MSF162.*$/MSF162_Ta/g; s/MSF198.*$/MSF198_Ta/g; s/MT087.*$/MT087_Xi/g; s/MT123.*$/MT123_Xi/g; s/MTMA033.*$/MTMA033_Xi/g; s/MTMA035.*$/MTMA035_Xi/g; s/OM022.*$/OM022_In/g; s/OM112.*$/OM112_In/g; s/OM172.*$/OM172_In/g; s/OM215.*$/OM215_Ro/g; s/OP077.*$/OP077_Ro/g; s/OP100.*$/OP100_Ro/g; s/PIME024.*$/PIME024_Ro/g; s/PIME062.*$/PIME062_Ro/g; s/PIME079.*$/PIME079_Ro/g; s/PIME134.*$/PIME134_Ta/g; s/PIME135.*$/PIME135_Ta/g; s/PIME169.*$/PIME169_Ro/g; s/PIME450.*$/PIME450_Ro/g; s/PIME460.*$/PIME460_Ro/g; s/PIME482.*$/PIME482_Ta/g; s/PIME521.*$/PIME521_Ta/g; s/PIME522.*$/PIME522_Ta/g; s/PPBIO221.*$/PPBIO221_Xi/g; s/PPBIO305.*$/PPBIO305_Xi/g; s/PPBIO312.*$/PPBIO312_Xi/g; s/PPS072.*$/PPS072_Ta/g; s/PPS081.*$/PPS081_Ta/g; s/PPS145.*$/PPS145_Ta/g; s/PPS174.*$/PPS174_Ta/g; s/PPS352.*$/PPS352_Ta/g; s/PUC007.*$/PUC007_In/g; s/PUC008.*$/PUC008_In/g; s/PUC023.*$/PUC023_In/g; s/PUC174.*$/PUC174_In/g; s/RUR038.*$/RUR038_In/g; s/TLP025.*$/TLP025_Ta/g; s/TLP123.*$/TLP123_Ta/g; s/TLPA267.*$/TLPA267_Ta/g; s/TLPA360.*$/TLPA360_Ta/g; s/TLPB026.*$/TLPB026_Ta/g; s/TLPB027.*$/TLPB027_Ta/g; s/TLPB221.*$/TLPB221_Ta/g; s/TLPC022.*$/TLPC022_Ta/g; s/TLPC036.*$/TLPC036_Ta/g; s/TLPC040.*$/TLPC040_Ta/g; s/TLPC041.*$/TLPC041_Ta/g; s/TLPC068.*$/TLPC068_Ta/g; s/TLPC077.*$/TLPC077_Ta/g; s/TLPC184.*$/TLPC184_Ta/g; s/TLPC185.*$/TLPC185_Ta/g; s/UFAC095.*$/UFAC095_In/g; s/UFAC1007.*$/UFAC1007_In/g; s/UFAC1687.*$/UFAC1687_In/g; s/UFAC305.*$/UFAC305_In/g; s/UFAC476.*$/UFAC476_In/g; s/UHE442.*$/UHE442_Xi/g; s/UNA002.*$/UNA002_At/g; s/UNA018.*$/UNA018_At/g; s/UNA088.*$/UNA088_At/g; s/UNA110.*$/UNA110_At/g; s/UNA124.*$/UNA124_At/g; s/WM281.*$/WM281_Ta/g; s/WM293.*$/WM293_Ta/g; s/WM294.*$/WM294_Ta/g; s/WM345.*$/WM345_Ta/g' Crubrocapilla_Cytb954bp_filtered_3chloromerosout.fasta > Crubrocapilla_Cytb954bp_filtered_3chloromerosout.renamed.fasta
```

With OM172 and PPS081 (the two posible numts) removed, we now have 139 *Ceratopipra rubrocapilla* and three outgroup *Ceratopipra chloromeros*, for 141 samples in total.  

# Phylogeny  
I am first going to make a maximum likelihood phylogeny using IQtree2.  

First, setup a folder:  
```bash
#set up directory
mkdir -p /home/1_Else/Ceratopipra/2_Phylogeny/2.4aa_ML_mtDNAtree
cd /home/1_Else/Ceratopipra/2_Phylogeny/2.4aa_ML_mtDNAtree

#link the fasta file here
mkdir -p datasets
#I placed the alignment into this folder.
```

Then, run IQtree2. I am using 1000 bootstrap replicates to assess support for each node. IQTree2 will use BIC to select the best fit model of sequence evolution.      
```bash
#with the two possible numts removed
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/Crubrocapilla_Cytb954bp_filtered_3chloromerosout.renamed.fasta -nt AUTO -B 1000 --prefix datasets/Crubrocapilla_Cytb954bp_filtered_3chloromerosout.renamed.boot100 #5m36.612s
#Best-fit model: HKY+F+G4 chosen according to BIC





#These datasets are tests to ensure that our results are robust

#with the two possible numts, to see where they would be placed. They are embedded into the most common haplogroup.  
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/Crubrocapilla_Cytb954bp_3chloromerosout.renamed.fasta -nt AUTO -B 1000 --prefix datasets/Crubrocapilla_Cytb954bp_3chloromerosout.renamed.boot100 #5m0.692s #Best-fit model: TN+F+G4 chosen according to BIC
#testing different outgroups to ensure that our conclusions are robust to the choice of outgroup. Yes, results are qualitatively identical
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/ceratopipra.fas -nt AUTO #9m20.575s
#testing dataset with cytb and ND2/ND3 combined
#A12V3 and A15V3 stick out vey far - these two are missing the first part of the alignment (cytb), only have ND
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/Ceratopipra2016.fas -nt AUTO 
#testing ND2/ND3 alone
time /home/0_PROGRAMS/iqtree-2.1.3-Linux/bin/iqtree2 -s datasets/Crub_ND2_ND3_2014.fas -nt AUTO -B 1000
#ND2 and ND3 add no more insight, they recapitulate the patterns reconstructed by cytb. Since most samples are missing ND2/ND3, we will exclude it and only include cytb to avoid biasing branch lengths of those samples that only have cytb in a mixed alignment.
```


Here is `datasets/Crubrocapilla_Cytb954bp_filtered_3chloromerosout.renamed.boot100.treefile`
```
(B27638_Cchl:0.0045354010,(B9014_Cchl:0.0009781833,B106768_Cchl:0.0055872473)100:0.0079036456,((((((((((((((((((((CAM107_In:0.0045443950,FLJA085_Ta:0.0000010000)51:0.0021572772,(TLPB027_Ta:0.0000029950,OP077_Ro:0.0000010000)97:0.0010751049)5:0.0000027679,(((((MT123_Xi:0.0000025843,((TLPC068_Ta:0.0021618498,CUJ079_In:0.0010757955)19:0.0000010000,TLPC041_Ta:0.0000010000)43:0.0010747826)32:0.0010736896,MSF198_Ta:0.0021628874)7:0.0000028504,(((FLJA001_Ta:0.0000000000,MAR092_Ro:0.0000000000):0.0000000000,FPR021_Ro:0.0000000000):0.0000028786,ARAII031_Ro:0.0000010000)39:0.0000021734)11:0.0000023007,(TLPA360_Ta:0.0000010000,JUT094_In:0.0000010000)94:0.0010749775)18:0.0010749906,MSF107_Ta:0.0021619256)1:0.0000020449)5:0.0000010000,((((WM281_Ta:0.0000028152,JRT021_Ro:0.0021561320)92:0.0010763225,MAR021_Ro:0.0021644223)10:0.0000020453,((((ARAII069_Ro:0.0000010000,FPR015_Ro:0.0000010000)62:0.0000010000,CUJ027_In:0.0010727688)95:0.0010740319,JATA145_Ro:0.0010731046)52:0.0000010000,UFAC1687_In:0.0000010000)94:0.0010751916)2:0.0000020432,(AMANA034_Ro:0.0010752032,UFAC476_In:0.0010752112)7:0.0000010000)3:0.0000010000)4:0.0000010000,((((((((((((((((((MT087_Xi:0.0000000000,JTW1220_Ta:0.0000000000):0.0000000000,TLP123_Ta:0.0000000000):0.0000000000,JTW1230_Ta:0.0000000000):0.0000000000,TLPB221_Ta:0.0000000000):0.0000010000,PPS145_Ta:0.0000010000)51:0.0000010000,(TLPC077_Ta:0.0000010000,AMZ610_Ro:0.0013169436)57:0.0000010000)97:0.0021600316,FPR006_Ro:0.0010773628)27:0.0000020506,JUT237_In:0.0010773674)42:0.0000010000,MPDS1220_Ta:0.0000010000)4:0.0000010000,CH290_Ta:0.0000010000)7:0.0000010000,(((MF005_Ta:0.0000000000,PUC008_In:0.0000000000):0.0000000000,OP100_Ro:0.0000000000):0.0000000000,ARAII033_Ro:0.0000000000):0.0000010000)22:0.0000010000,PUC174_In:0.0021673006)93:0.0010762793,PIME450_Ro:0.0021604281)10:0.0000020371,(JATC147_Ro:0.0000010000,AMANA137_Ro:0.0000010000)97:0.0010753284)13:0.0000010000,PPS174_Ta:0.0000010000)7:0.0000010000,((BR163177_Ta:0.0010716947,CAM117_In:0.0023958274)36:0.0000053914,(ARA010_Ro:0.0010762669,AMA340_In:0.0010764843)10:0.0000010000)10:0.0000010000)4:0.0000010000,((((((((((((((PIME079_Ro:0.0000000000,MSF162_Ta:0.0000000000):0.0000000000,MAM011_Ro:0.0000000000):0.0000000000,OM112_In:0.0000000000):0.0000000000,CUJ072_In:0.0000000000):0.0000000000,MPDS628_Ro:0.0000000000):0.0000000000,PPS352_Ta:0.0000000000):0.0000000000,UFAC305_In:0.0000000000):0.0000000000,JATC148_Ro:0.0000000000):0.0000000000,CAM039_In:0.0000000000):0.0000000000,ARAII010_Ro:0.0000000000):0.0000000000,CAM087_In:0.0000000000):0.0000000000,TLPC036_Ta:0.0000000000):0.0000000000,UFAC1007_In:0.0000000000):0.0000010000,((PIME482_Ta:0.0043591016,MPDS496_Ro:0.0021600508)17:0.0000010000,PIME460_Ro:0.0010803344)8:0.0000010000)9:0.0000010000)5:0.0000010000,(PUC007_In:0.0021625883,(RUR038_In:0.0010798883,MPDS776_Ro:0.0010854402)92:0.0010793007)7:0.0000010000)3:0.0000010000)5:0.0000010000,(((WM293_Ta:0.0010823471,PIME134_Ta:0.0000010000)97:0.0021645233,((WM345_Ta:0.0000010000,PIME522_Ta:0.0010773183)61:0.0000010000,PIME135_Ta:0.0010762004)97:0.0010761926)10:0.0000020455,(OM022_In:0.0010746354,((OM215_Ro:0.0010747232,BR36440_In:0.0032400519)20:0.0000020406,UFAC095_In:0.0000010000)11:0.0000020362)67:0.0010779568)3:0.0000010000)13:0.0000020448,JATC149_Ro:0.0010770808)47:0.0010759249,MPDS608_Ro:0.0010759262)8:0.0000020430,(TLPB026_Ta:0.0000010000,MAM010_Ro:0.0000010000)38:0.0000010000)3:0.0000010000,(MTMA033_Xi:0.0010758787,PIME062_Ro:0.0010758870)11:0.0000010000)11:0.0000010000,MTMA035_Xi:0.0000010000)8:0.0000010000,((((((((MFNC067_Xi:0.0000000000,PIME521_Ta:0.0000000000):0.0000000000,TLPC184_Ta:0.0000000000):0.0000000000,ARAII062_Ro:0.0000000000):0.0000000000,TLPC185_Ta:0.0000000000):0.0000000000,CAM097_In:0.0000000000):0.0000000000,AMA601_In:0.0000000000):0.0000000000,PIME169_Ro:0.0000000000):0.0000000000,MPDS672_Ro:0.0000000000):0.0000010000)7:0.0000010000,ARAII026_Ro:0.0010747567)3:0.0000010000,((TLPC040_Ta:0.0010758602,(JTW1135_Ta:0.0000010000,TLP025_Ta:0.0010770044)61:0.0000010000)98:0.0010758849,JATB060_Ta:0.0010756864)14:0.0000020280)9:0.0000024593,((((((((GAPTO259_Xi:0.0000000000,TLPC022_Ta:0.0000000000):0.0000000000,GAPTO327_Xi:0.0000000000):0.0000000000,PIME024_Ro:0.0000000000):0.0000000000,CH288_Ta:0.0000000000):0.0000000000,MSF052_Ta:0.0000000000):0.0000000000,TLPA267_Ta:0.0000000000):0.0000010000,GAPTO277_Xi:0.0000010000)61:0.0000024677,CH289_Ta:0.0000026439)97:0.0010758736)11:0.0000020394,PUC023_In:0.0021585548)56:0.0010757189,(WM294_Ta:0.0010884588,((JRT022_Ro:0.0000010000,JATA201_Ro:0.0000010000)64:0.0000024035,ARA109_Ro:0.0010855879)94:0.0010764513)96:0.0021700734)32:0.0000020925,PPS072_Ta:0.0010756195)98:0.0033083992,((((UNA018_At:0.0010787172,UHE442_Xi:0.0032530439)23:0.0000010000,GAPTO140_Xi:0.0010798098)14:0.0000010000,(((((((((((UNA002_At:0.0000000000,UNA124_At:0.0000000000):0.0000000000,CPEII008_At:0.0000000000):0.0000010000,UNA110_At:0.0000010000)84:0.0000020356,UNA088_At:0.0010806300)98:0.0010806218,CPE066_At:0.0000025215)99:0.0021678480,(PPBIO221_Xi:0.0000010000,CAR005_Xi:0.0010780382)98:0.0010782916)66:0.0000020463,CPE012_At:0.0010811210)97:0.0010785564,(CPE025_At:0.0010858029,PPBIO305_Xi:0.0032562533)41:0.0000020279)16:0.0000010000,(CPE049_At:0.0000010000,CPE067_At:0.0000010000)33:0.0000010000)8:0.0000010000,((CPE069_At:0.0000000000,GAPTO268_Xi:0.0000000000):0.0000000000,GAPTO289_Xi:0.0000000000):0.0000010000)2:0.0000010000,CPE071_At:0.0000010000)7:0.0000010000)25:0.0000023549,PPBIO312_Xi:0.0010798158)84:0.0021792477)85:0.0009623432,(CAM009_In:0.0022798958,AMA656_In:0.0000010000)97:0.0022943303)100:0.0227644076);

#This is the phylogeny above, With population labels removed for a cleaner slate for presenting as a figure in the manuscript:
(B27638:0.0045354010,(B9014:0.0009781833,B106768:0.0055872473)100:0.0079036456,((((((((((((((((((((CAM107:0.0045443950,FLJA085:0.0000010000)51:0.0021572772,(TLPB027:0.0000029950,OP077:0.0000010000)97:0.0010751049)5:0.0000027679,(((((MT123:0.0000025843,((TLPC068:0.0021618498,CUJ079:0.0010757955)19:0.0000010000,TLPC041:0.0000010000)43:0.0010747826)32:0.0010736896,MSF198:0.0021628874)7:0.0000028504,(((FLJA001:0.0000000000,MAR092:0.0000000000):0.0000000000,FPR021:0.0000000000):0.0000028786,ARAII031:0.0000010000)39:0.0000021734)11:0.0000023007,(TLPA360:0.0000010000,JUT094:0.0000010000)94:0.0010749775)18:0.0010749906,MSF107:0.0021619256)1:0.0000020449)5:0.0000010000,((((WM281:0.0000028152,JRT021:0.0021561320)92:0.0010763225,MAR021:0.0021644223)10:0.0000020453,((((ARAII069:0.0000010000,FPR015:0.0000010000)62:0.0000010000,CUJ027:0.0010727688)95:0.0010740319,JATA145:0.0010731046)52:0.0000010000,UFAC1687:0.0000010000)94:0.0010751916)2:0.0000020432,(AMANA034:0.0010752032,UFAC476:0.0010752112)7:0.0000010000)3:0.0000010000)4:0.0000010000,((((((((((((((((((MT087:0.0000000000,JTW1220:0.0000000000):0.0000000000,TLP123:0.0000000000):0.0000000000,JTW1230:0.0000000000):0.0000000000,TLPB221:0.0000000000):0.0000010000,PPS145:0.0000010000)51:0.0000010000,(TLPC077:0.0000010000,AMZ610:0.0013169436)57:0.0000010000)97:0.0021600316,FPR006:0.0010773628)27:0.0000020506,JUT237:0.0010773674)42:0.0000010000,MPDS1220:0.0000010000)4:0.0000010000,CH290:0.0000010000)7:0.0000010000,(((MF005:0.0000000000,PUC008:0.0000000000):0.0000000000,OP100:0.0000000000):0.0000000000,ARAII033:0.0000000000):0.0000010000)22:0.0000010000,PUC174:0.0021673006)93:0.0010762793,PIME450:0.0021604281)10:0.0000020371,(JATC147:0.0000010000,AMANA137:0.0000010000)97:0.0010753284)13:0.0000010000,PPS174:0.0000010000)7:0.0000010000,((BR163177:0.0010716947,CAM117:0.0023958274)36:0.0000053914,(ARA010:0.0010762669,AMA340:0.0010764843)10:0.0000010000)10:0.0000010000)4:0.0000010000,((((((((((((((PIME079:0.0000000000,MSF162:0.0000000000):0.0000000000,MAM011:0.0000000000):0.0000000000,OM112:0.0000000000):0.0000000000,CUJ072:0.0000000000):0.0000000000,MPDS628:0.0000000000):0.0000000000,PPS352:0.0000000000):0.0000000000,UFAC305:0.0000000000):0.0000000000,JATC148:0.0000000000):0.0000000000,CAM039:0.0000000000):0.0000000000,ARAII010:0.0000000000):0.0000000000,CAM087:0.0000000000):0.0000000000,TLPC036:0.0000000000):0.0000000000,UFAC1007:0.0000000000):0.0000010000,((PIME482:0.0043591016,MPDS496:0.0021600508)17:0.0000010000,PIME460:0.0010803344)8:0.0000010000)9:0.0000010000)5:0.0000010000,(PUC007:0.0021625883,(RUR038:0.0010798883,MPDS776:0.0010854402)92:0.0010793007)7:0.0000010000)3:0.0000010000)5:0.0000010000,(((WM293:0.0010823471,PIME134:0.0000010000)97:0.0021645233,((WM345:0.0000010000,PIME522:0.0010773183)61:0.0000010000,PIME135:0.0010762004)97:0.0010761926)10:0.0000020455,(OM022:0.0010746354,((OM215:0.0010747232,BR36440:0.0032400519)20:0.0000020406,UFAC095:0.0000010000)11:0.0000020362)67:0.0010779568)3:0.0000010000)13:0.0000020448,JATC149:0.0010770808)47:0.0010759249,MPDS608:0.0010759262)8:0.0000020430,(TLPB026:0.0000010000,MAM010:0.0000010000)38:0.0000010000)3:0.0000010000,(MTMA033:0.0010758787,PIME062:0.0010758870)11:0.0000010000)11:0.0000010000,MTMA035:0.0000010000)8:0.0000010000,((((((((MFNC067:0.0000000000,PIME521:0.0000000000):0.0000000000,TLPC184:0.0000000000):0.0000000000,ARAII062:0.0000000000):0.0000000000,TLPC185:0.0000000000):0.0000000000,CAM097:0.0000000000):0.0000000000,AMA601:0.0000000000):0.0000000000,PIME169:0.0000000000):0.0000000000,MPDS672:0.0000000000):0.0000010000)7:0.0000010000,ARAII026:0.0010747567)3:0.0000010000,((TLPC040:0.0010758602,(JTW1135:0.0000010000,TLP025:0.0010770044)61:0.0000010000)98:0.0010758849,JATB060:0.0010756864)14:0.0000020280)9:0.0000024593,((((((((GAPTO259:0.0000000000,TLPC022:0.0000000000):0.0000000000,GAPTO327:0.0000000000):0.0000000000,PIME024:0.0000000000):0.0000000000,CH288:0.0000000000):0.0000000000,MSF052:0.0000000000):0.0000000000,TLPA267:0.0000000000):0.0000010000,GAPTO277:0.0000010000)61:0.0000024677,CH289:0.0000026439)97:0.0010758736)11:0.0000020394,PUC023:0.0021585548)56:0.0010757189,(WM294:0.0010884588,((JRT022:0.0000010000,JATA201:0.0000010000)64:0.0000024035,ARA109:0.0010855879)94:0.0010764513)96:0.0021700734)32:0.0000020925,PPS072:0.0010756195)98:0.0033083992,((((UNA018:0.0010787172,UHE442:0.0032530439)23:0.0000010000,GAPTO140:0.0010798098)14:0.0000010000,(((((((((((UNA002:0.0000000000,UNA124:0.0000000000):0.0000000000,CPEII008:0.0000000000):0.0000010000,UNA110:0.0000010000)84:0.0000020356,UNA088:0.0010806300)98:0.0010806218,CPE066:0.0000025215)99:0.0021678480,(PPBIO221:0.0000010000,CAR005:0.0010780382)98:0.0010782916)66:0.0000020463,CPE012:0.0010811210)97:0.0010785564,(CPE025:0.0010858029,PPBIO305:0.0032562533)41:0.0000020279)16:0.0000010000,(CPE049:0.0000010000,CPE067:0.0000010000)33:0.0000010000)8:0.0000010000,((CPE069:0.0000000000,GAPTO268:0.0000000000):0.0000000000,GAPTO289:0.0000000000):0.0000010000)2:0.0000010000,CPE071:0.0000010000)7:0.0000010000)25:0.0000023549,PPBIO312:0.0010798158)84:0.0021792477)85:0.0009623432,(CAM009:0.0022798958,AMA656:0.0000010000)97:0.0022943303)100:0.0227644076);

```


For comparison, here is `datasets/Crubrocapilla_Cytb954bp_3chloromerosout.renamed.boot100.treefile`:  
(the one with two possible numts)  
```
(B27638_Cchl:0.0043023722,(B9014_Cchl:0.0008828241,B106768_Cchl:0.0055910581)99:0.0080600932,(((((((((((((((((CAM107_In:0.0045343104,FLJA085_Ta:0.0000023169)52:0.0021397018,MPDS496_Ro:0.0021441492)13:0.0000020260,(((((WM345_Ta:0.0000010000,PIME522_Ta:0.0010703366)54:0.0000026760,PIME135_Ta:0.0010688324)96:0.0010688396,(PIME482_Ta:0.0043293343,PIME450_Ro:0.0021444472)10:0.0000010000)5:0.0000020023,(BR163177_Ta:0.0010688833,UFAC476_In:0.0010672488)5:0.0000010000)9:0.0000010000,PPS174_Ta:0.0000010000)8:0.0000010000)7:0.0000010000,((((((((((((((((MT087_Xi:0.0000000000,JTW1220_Ta:0.0000000000):0.0000000000,TLP123_Ta:0.0000000000):0.0000000000,JTW1230_Ta:0.0000000000):0.0000000000,TLPB221_Ta:0.0000000000):0.0000010000,PPS145_Ta:0.0000010000)43:0.0000010000,(TLPC077_Ta:0.0000010000,AMZ610_Ro:0.0013129603)58:0.0000010000)97:0.0021442288,FPR006_Ro:0.0010695867)26:0.0000010000,JUT237_In:0.0010695981)22:0.0000010000,((((MF005_Ta:0.0000000000,OP100_Ro:0.0000000000):0.0000000000,PUC008_In:0.0000000000):0.0000000000,ARAII033_Ro:0.0000000000):0.0000010000,MPDS1220_Ta:0.0000010000)32:0.0000010000)17:0.0000010000,CH290_Ta:0.0000010000)20:0.0000010000,PUC174_In:0.0021499080)95:0.0010678964,(WM281_Ta:0.0000010000,JRT021_Ro:0.0021378653)96:0.0010680557)10:0.0000010000,(MSF107_Ta:0.0021468340,((((ARAII069_Ro:0.0000010000,FPR015_Ro:0.0000010000)62:0.0000010000,CUJ027_In:0.0010630276)97:0.0010648715,JATA145_Ro:0.0010634199)49:0.0000010000,UFAC1687_In:0.0000010000)95:0.0010664631)10:0.0000010000)6:0.0000010000,(((((((((((((PIME079_Ro:0.0000000000,PPS352_Ta:0.0000000000):0.0000000000,UFAC1007_In:0.0000000000):0.0000000000,CAM039_In:0.0000000000):0.0000000000,MPDS628_Ro:0.0000000000):0.0000000000,TLPC036_Ta:0.0000000000):0.0000000000,MAM011_Ro:0.0000000000):0.0000000000,CUJ072_In:0.0000000000):0.0000000000,CAM087_In:0.0000000000):0.0000000000,OM112_In:0.0000000000):0.0000000000,ARAII010_Ro:0.0000000000):0.0000000000,UFAC305_In:0.0000000000):0.0000000000,MSF162_Ta:0.0000000000):0.0000000000,JATC148_Ro:0.0000000000):0.0000010000)6:0.0000010000,((TLPB027_Ta:0.0000010000,OP077_Ro:0.0000010000)98:0.0010664273,(JATC147_Ro:0.0000010000,AMANA137_Ro:0.0000010000)98:0.0010664254)13:0.0000020374)11:0.0000010000,((RUR038_In:0.0010756954,MPDS776_Ro:0.0010784466)92:0.0010696669,(OM022_In:0.0010688128,((OM215_Ro:0.0010690721,BR36440_In:0.0032199006)28:0.0000010000,UFAC095_In:0.0000010000)12:0.0000010000)71:0.0010707149)10:0.0000010000)3:0.0000010000)2:0.0000010000,(((((((WM293_Ta:0.0010773588,PIME134_Ta:0.0000023332)96:0.0021495003,CAM117_In:0.0023933684)15:0.0000020416,PUC007_In:0.0021478169)0:0.0000010000,(PPS081_Ta:0.0023946573,AMA340_In:0.0010680050)0:0.0000010000)3:0.0000010000,JATC149_Ro:0.0010693578)1:0.0000010000,MAR021_Ro:0.0021505112)6:0.0000010000,(AMANA034_Ro:0.0010663726,OM172_In:0.0032257324)12:0.0000010000)4:0.0000010000)4:0.0000010000,(((MT123_Xi:0.0000020370,(TLPC068_Ta:0.0021471951,(TLPC041_Ta:0.0000010000,CUJ079_In:0.0010684231)23:0.0000010000)36:0.0010667758)41:0.0010651372,((((FLJA001_Ta:0.0000000000,FPR021_Ro:0.0000000000):0.0000000000,MAR092_Ro:0.0000000000):0.0000010000,ARAII031_Ro:0.0000010000)54:0.0000010000,(MSF198_Ta:0.0021414160,(TLPA360_Ta:0.0000010000,JUT094_In:0.0000010000)91:0.0010669512)11:0.0000020755)16:0.0000020437)21:0.0010669761,PIME460_Ro:0.0010718523)5:0.0000010000)12:0.0000010000,ARA010_Ro:0.0010685858)49:0.0010685622,JATB060_Ta:0.0010683862)16:0.0000010000,(((((((((((GAPTO259_Xi:0.0000000000,MSF052_Ta:0.0000000000):0.0000000000,CH288_Ta:0.0000000000):0.0000000000,PIME024_Ro:0.0000000000):0.0000000000,TLPC022_Ta:0.0000000000):0.0000000000,GAPTO327_Xi:0.0000000000):0.0000000000,TLPA267_Ta:0.0000000000):0.0000010000,GAPTO277_Xi:0.0000010000)55:0.0000010000,CH289_Ta:0.0000010000)97:0.0010685036,ARAII026_Ro:0.0010668835)20:0.0000020390,(TLPB026_Ta:0.0000010000,MAM010_Ro:0.0000010000)41:0.0000010000)6:0.0000010000,(MTMA033_Xi:0.0010685392,(TLPC040_Ta:0.0010685077,(JTW1135_Ta:0.0000010000,TLP025_Ta:0.0010700335)57:0.0000010000)97:0.0010685287)10:0.0000010000)9:0.0000010000)7:0.0000010000,MTMA035_Xi:0.0000010000)4:0.0000010000,((((((((MFNC067_Xi:0.0000000000,AMA601_In:0.0000000000):0.0000000000,PIME169_Ro:0.0000000000):0.0000000000,TLPC185_Ta:0.0000000000):0.0000000000,TLPC184_Ta:0.0000000000):0.0000000000,ARAII062_Ro:0.0000000000):0.0000000000,PIME521_Ta:0.0000000000):0.0000000000,CAM097_In:0.0000000000):0.0000000000,MPDS672_Ro:0.0000000000):0.0000010000)7:0.0000010000,(PIME062_Ro:0.0010684911,MPDS608_Ro:0.0010684931)5:0.0000010000)2:0.0000010000,PUC023_In:0.0021434840)55:0.0010683525,(WM294_Ta:0.0010793561,((JRT022_Ro:0.0000010000,JATA201_Ro:0.0000010000)62:0.0000010000,ARA109_Ro:0.0010784585)92:0.0010750742)98:0.0021554153)27:0.0000010000,PPS072_Ta:0.0010682927)98:0.0032725400,(((((((UNA018_At:0.0010690996,GAPTO140_Xi:0.0010707617)11:0.0000010000,((((((((UNA002_At:0.0000000000,CPEII008_At:0.0000000000):0.0000000000,UNA124_At:0.0000000000):0.0000010000,UNA110_At:0.0000010000)83:0.0000010000,UNA088_At:0.0010722401)98:0.0010722768,CPE066_At:0.0000020554)99:0.0021492675,(PPBIO221_Xi:0.0000010000,CAR005_Xi:0.0010685423)99:0.0010687402)64:0.0000021294,CPE012_At:0.0010715894)98:0.0010689301,UHE442_Xi:0.0032196613)20:0.0000010000)3:0.0000010000,(CPE049_At:0.0000010000,CPE067_At:0.0000010000)30:0.0000010000)7:0.0000010000,((CPE069_At:0.0000000000,GAPTO268_Xi:0.0000000000):0.0000000000,GAPTO289_Xi:0.0000000000):0.0000010000)3:0.0000010000,CPE071_At:0.0000010000)6:0.0000010000,(PPBIO312_Xi:0.0010707735,PPBIO305_Xi:0.0032252955)7:0.0000010000)6:0.0000010000,CPE025_At:0.0010754339)93:0.0021573233)93:0.0010210147,(CAM009_In:0.0022553469,AMA656_In:0.0000010000)95:0.0022025244)100:0.0225687738);

#With population labels removed for a cleaner slate:
(B27638:0.0043023722,(B9014:0.0008828241,B106768:0.0055910581)99:0.0080600932,(((((((((((((((((CAM107:0.0045343104,FLJA085:0.0000023169)52:0.0021397018,MPDS496:0.0021441492)13:0.0000020260,(((((WM345:0.0000010000,PIME522:0.0010703366)54:0.0000026760,PIME135:0.0010688324)96:0.0010688396,(PIME482:0.0043293343,PIME450:0.0021444472)10:0.0000010000)5:0.0000020023,(BR163177:0.0010688833,UFAC476:0.0010672488)5:0.0000010000)9:0.0000010000,PPS174:0.0000010000)8:0.0000010000)7:0.0000010000,((((((((((((((((MT087:0.0000000000,JTW1220:0.0000000000):0.0000000000,TLP123:0.0000000000):0.0000000000,JTW1230:0.0000000000):0.0000000000,TLPB221:0.0000000000):0.0000010000,PPS145:0.0000010000)43:0.0000010000,(TLPC077:0.0000010000,AMZ610:0.0013129603)58:0.0000010000)97:0.0021442288,FPR006:0.0010695867)26:0.0000010000,JUT237:0.0010695981)22:0.0000010000,((((MF005:0.0000000000,OP100:0.0000000000):0.0000000000,PUC008:0.0000000000):0.0000000000,ARAII033:0.0000000000):0.0000010000,MPDS1220:0.0000010000)32:0.0000010000)17:0.0000010000,CH290:0.0000010000)20:0.0000010000,PUC174:0.0021499080)95:0.0010678964,(WM281:0.0000010000,JRT021:0.0021378653)96:0.0010680557)10:0.0000010000,(MSF107:0.0021468340,((((ARAII069:0.0000010000,FPR015:0.0000010000)62:0.0000010000,CUJ027:0.0010630276)97:0.0010648715,JATA145:0.0010634199)49:0.0000010000,UFAC1687:0.0000010000)95:0.0010664631)10:0.0000010000)6:0.0000010000,(((((((((((((PIME079:0.0000000000,PPS352:0.0000000000):0.0000000000,UFAC1007:0.0000000000):0.0000000000,CAM039:0.0000000000):0.0000000000,MPDS628:0.0000000000):0.0000000000,TLPC036:0.0000000000):0.0000000000,MAM011:0.0000000000):0.0000000000,CUJ072:0.0000000000):0.0000000000,CAM087:0.0000000000):0.0000000000,OM112:0.0000000000):0.0000000000,ARAII010:0.0000000000):0.0000000000,UFAC305:0.0000000000):0.0000000000,MSF162:0.0000000000):0.0000000000,JATC148:0.0000000000):0.0000010000)6:0.0000010000,((TLPB027:0.0000010000,OP077:0.0000010000)98:0.0010664273,(JATC147:0.0000010000,AMANA137:0.0000010000)98:0.0010664254)13:0.0000020374)11:0.0000010000,((RUR038:0.0010756954,MPDS776:0.0010784466)92:0.0010696669,(OM022:0.0010688128,((OM215:0.0010690721,BR36440:0.0032199006)28:0.0000010000,UFAC095:0.0000010000)12:0.0000010000)71:0.0010707149)10:0.0000010000)3:0.0000010000)2:0.0000010000,(((((((WM293:0.0010773588,PIME134:0.0000023332)96:0.0021495003,CAM117:0.0023933684)15:0.0000020416,PUC007:0.0021478169)0:0.0000010000,(PPS081:0.0023946573,AMA340:0.0010680050)0:0.0000010000)3:0.0000010000,JATC149:0.0010693578)1:0.0000010000,MAR021:0.0021505112)6:0.0000010000,(AMANA034:0.0010663726,OM172:0.0032257324)12:0.0000010000)4:0.0000010000)4:0.0000010000,(((MT123:0.0000020370,(TLPC068:0.0021471951,(TLPC041:0.0000010000,CUJ079:0.0010684231)23:0.0000010000)36:0.0010667758)41:0.0010651372,((((FLJA001:0.0000000000,FPR021:0.0000000000):0.0000000000,MAR092:0.0000000000):0.0000010000,ARAII031:0.0000010000)54:0.0000010000,(MSF198:0.0021414160,(TLPA360:0.0000010000,JUT094:0.0000010000)91:0.0010669512)11:0.0000020755)16:0.0000020437)21:0.0010669761,PIME460:0.0010718523)5:0.0000010000)12:0.0000010000,ARA010:0.0010685858)49:0.0010685622,JATB060:0.0010683862)16:0.0000010000,(((((((((((GAPTO259:0.0000000000,MSF052:0.0000000000):0.0000000000,CH288:0.0000000000):0.0000000000,PIME024:0.0000000000):0.0000000000,TLPC022:0.0000000000):0.0000000000,GAPTO327:0.0000000000):0.0000000000,TLPA267:0.0000000000):0.0000010000,GAPTO277:0.0000010000)55:0.0000010000,CH289:0.0000010000)97:0.0010685036,ARAII026:0.0010668835)20:0.0000020390,(TLPB026:0.0000010000,MAM010:0.0000010000)41:0.0000010000)6:0.0000010000,(MTMA033:0.0010685392,(TLPC040:0.0010685077,(JTW1135:0.0000010000,TLP025:0.0010700335)57:0.0000010000)97:0.0010685287)10:0.0000010000)9:0.0000010000)7:0.0000010000,MTMA035:0.0000010000)4:0.0000010000,((((((((MFNC067:0.0000000000,AMA601:0.0000000000):0.0000000000,PIME169:0.0000000000):0.0000000000,TLPC185:0.0000000000):0.0000000000,TLPC184:0.0000000000):0.0000000000,ARAII062:0.0000000000):0.0000000000,PIME521:0.0000000000):0.0000000000,CAM097:0.0000000000):0.0000000000,MPDS672:0.0000000000):0.0000010000)7:0.0000010000,(PIME062:0.0010684911,MPDS608:0.0010684931)5:0.0000010000)2:0.0000010000,PUC023:0.0021434840)55:0.0010683525,(WM294:0.0010793561,((JRT022:0.0000010000,JATA201:0.0000010000)62:0.0000010000,ARA109:0.0010784585)92:0.0010750742)98:0.0021554153)27:0.0000010000,PPS072:0.0010682927)98:0.0032725400,(((((((UNA018:0.0010690996,GAPTO140:0.0010707617)11:0.0000010000,((((((((UNA002:0.0000000000,CPEII008:0.0000000000):0.0000000000,UNA124:0.0000000000):0.0000010000,UNA110:0.0000010000)83:0.0000010000,UNA088:0.0010722401)98:0.0010722768,CPE066:0.0000020554)99:0.0021492675,(PPBIO221:0.0000010000,CAR005:0.0010685423)99:0.0010687402)64:0.0000021294,CPE012:0.0010715894)98:0.0010689301,UHE442:0.0032196613)20:0.0000010000)3:0.0000010000,(CPE049:0.0000010000,CPE067:0.0000010000)30:0.0000010000)7:0.0000010000,((CPE069:0.0000000000,GAPTO268:0.0000000000):0.0000000000,GAPTO289:0.0000000000):0.0000010000)3:0.0000010000,CPE071:0.0000010000)6:0.0000010000,(PPBIO312:0.0010707735,PPBIO305:0.0032252955)7:0.0000010000)6:0.0000010000,CPE025:0.0010754339)93:0.0021573233)93:0.0010210147,(CAM009:0.0022553469,AMA656:0.0000010000)95:0.0022025244)100:0.0225687738);

```

To make the figure, I openned it in FigTree, then:  
* selected the branch containing the three *C. chloromeros* and clicked "Reroot"  
* selected "align tip labels"  
* rotated a few clades to make the tree look more balanced to my eye - doesn't change the relationships in the tree, just makes it subjectively look nicer to look at in a figure, in my opinion.  
* Node labels -> Display: "label" (these are bootstrap values)  
* File -> Save as -> datasets/Crubrocapilla_Cytb954bp_3chloromerosout.renamed.boot100.rooted.treefile  
* File -> Export PDF -> Crubrocapilla_Cytb954bp_3chloromerosout.renamed.boot100.rooted.treefile.pdf  
Then, I removed extra bootstrap values for polytomy nodes. Each node is given a bootstrap value because the underlying tree must be bifurcating, even if some branches have zero branch length (a polytomy) because the sequences are identical or there were no mutations observed at that particular branch. Bootstrap values for zero branch length nodes are not biologically meaningful (in this dataset, those are "clades" supported by no sites) and only add clutter, so I will collapse those into a polytomy.  

There are many identical haplotypes, some within samples from different interfluves:   
```
NOTE: UNA110_At is identical to UNA002_At but kept for subsequent analysis
NOTE: CPE067_At is identical to CPE049_At but kept for subsequent analysis
NOTE: CPE071_At is identical to CPE069_At but kept for subsequent analysis
NOTE: MTMA035_Xi is identical to MFNC067_Xi but kept for subsequent analysis
NOTE: GAPTO277_Xi is identical to GAPTO259_Xi but kept for subsequent analysis
NOTE: PPS145_Ta is identical to MT087_Xi but kept for subsequent analysis
NOTE: PPS174_Ta is identical to PIME079_Ro but kept for subsequent analysis
NOTE: MPDS1220_Ta is identical to MF005_Ta but kept for subsequent analysis
NOTE: ARAII031_Ro is identical to FLJA001_Ta but kept for subsequent analysis
NOTE: OP077_Ro is identical to TLPB027_Ta but kept for subsequent analysis
NOTE: JUT094_In is identical to TLPA360_Ta but kept for subsequent analysis
NOTE: FPR015_Ro is identical to ARAII069_Ro but kept for subsequent analysis
NOTE: JATA201_Ro is identical to JRT022_Ro but kept for subsequent analysis
NOTE: AMANA137_Ro is identical to JATC147_Ro but kept for subsequent analysis
NOTE: 40 identical sequences (see below) will be ignored for subsequent analysis
NOTE: UNA124_At (identical to UNA002_At) is ignored but added at the end
NOTE: CPEII008_At (identical to UNA002_At) is ignored but added at the end
NOTE: GAPTO268_Xi (identical to CPE069_At) is ignored but added at the end
NOTE: GAPTO289_Xi (identical to CPE069_At) is ignored but added at the end
NOTE: PIME169_Ro (identical to MFNC067_Xi) is ignored but added at the end
NOTE: PIME521_Ta (identical to MFNC067_Xi) is ignored but added at the end
NOTE: TLPC184_Ta (identical to MFNC067_Xi) is ignored but added at the end
NOTE: TLPC185_Ta (identical to MFNC067_Xi) is ignored but added at the end
NOTE: ARAII062_Ro (identical to MFNC067_Xi) is ignored but added at the end
NOTE: MPDS672_Ro (identical to MFNC067_Xi) is ignored but added at the end
NOTE: AMA601_In (identical to MFNC067_Xi) is ignored but added at the end
NOTE: CAM097_In (identical to MFNC067_Xi) is ignored but added at the end
NOTE: GAPTO327_Xi (identical to GAPTO259_Xi) is ignored but added at the end
NOTE: CH288_Ta (identical to GAPTO259_Xi) is ignored but added at the end
NOTE: MSF052_Ta (identical to GAPTO259_Xi) is ignored but added at the end
NOTE: TLPC022_Ta (identical to GAPTO259_Xi) is ignored but added at the end
NOTE: TLPA267_Ta (identical to GAPTO259_Xi) is ignored but added at the end
NOTE: PIME024_Ro (identical to GAPTO259_Xi) is ignored but added at the end
NOTE: JTW1230_Ta (identical to MT087_Xi) is ignored but added at the end
NOTE: JTW1220_Ta (identical to MT087_Xi) is ignored but added at the end
NOTE: TLPB221_Ta (identical to MT087_Xi) is ignored but added at the end
NOTE: TLP123_Ta (identical to MT087_Xi) is ignored but added at the end
NOTE: TLPC036_Ta (identical to PIME079_Ro) is ignored but added at the end
NOTE: PPS352_Ta (identical to PIME079_Ro) is ignored but added at the end
NOTE: MSF162_Ta (identical to PIME079_Ro) is ignored but added at the end
NOTE: ARAII010_Ro (identical to PIME079_Ro) is ignored but added at the end
NOTE: MAM011_Ro (identical to PIME079_Ro) is ignored but added at the end
NOTE: JATC148_Ro (identical to PIME079_Ro) is ignored but added at the end
NOTE: MPDS628_Ro (identical to PIME079_Ro) is ignored but added at the end
NOTE: CUJ072_In (identical to PIME079_Ro) is ignored but added at the end
NOTE: OM112_In (identical to PIME079_Ro) is ignored but added at the end
NOTE: CAM039_In (identical to PIME079_Ro) is ignored but added at the end
NOTE: CAM087_In (identical to PIME079_Ro) is ignored but added at the end
NOTE: UFAC1007_In (identical to PIME079_Ro) is ignored but added at the end
NOTE: UFAC305_In (identical to PIME079_Ro) is ignored but added at the end
NOTE: ARAII033_Ro (identical to MF005_Ta) is ignored but added at the end
NOTE: OP100_Ro (identical to MF005_Ta) is ignored but added at the end
NOTE: PUC008_In (identical to MF005_Ta) is ignored but added at the end
NOTE: FPR021_Ro (identical to FLJA001_Ta) is ignored but added at the end
NOTE: MAR092_Ro (identical to FLJA001_Ta) is ignored but added at the end
```

# Notes
There are six mutations that (almost) fully separate haplogroups 1 and 2:  
* T134C; T is in east but also chl, men, and ery, and AMA656, CAM009, PPS072.TA5S, JRT022.RO5N, WM294.TA5N, JATA201.RO5S, ARA109.RO5N, all other wests have C.
* T416C; T in all east and 2/4 chl.
* A641T; A in all east and AMA656 and CAM009 and cornuta, chl, and ery; C in men; T in most west. 
* C647T; C in cornuta, T in chl, ery, men, and most west. C in east and MT087, and TLPB221.TA5S, JTW1230.TA5S, JTW1220.TA5S, TLPC077.TA5S, PPS145.TA5S, TLPB123.TA5S, AMZ610.RO5N
* T689C & T749C; both mutations have identical site patterns. C in chloromeros, T in other species. C in 3 ery... wait a minute, those are actually chloromeros haplotypes: B27528.N,B27638.N,B27671.N. Also C in 3 westerns: CAM107.IN5S, AMA656.IN5N, and CAM009.IN5S.  

There are 6 mutations separating east from west. But, 2 western samples share 3 of those mutations. But, those 3 seem to be the ancestral state in east, so could be shared ancestral state with those 2 being split earliest from the rest of the westerns. It looks like those two western samples are a third moderately diverged haplogroup. The phylogeny supports this.      

There are also 4 mutations in ND2/ND3 that distinguish the two haplotype groups. Neither sample with haplogroup3 was sequenced for ND2/ND3 unfortunately.  


# Calculating Mitochondrial Genetic Distance  

In this analysis, we will calculate genetic distances between each cytochrome b sequence. We can calculate distances using the alignment we generated for the ML phylogeny.    

```bash
#Clean alignment
#Emboss will count IUPAC ambiguity characters towards distances in the distance matrix. We do not want this because it will inflate distances for low quality sequences. In our dataset, ambiguity characters are more likely to be reference alleles with poor quality calls from the Sanger sequence, not variants that should contribute to differences between sequences. We will replace any ambiguous bases with gaps so that they do not contribute to distances. Note that this would introduce a bias if derived/rare alleles were more likely to be called as ambiguous (eg., if there were strong reference bias), but this is not very plausible for our Sanger data which is not reference-based.  
#Check for any non-ATCG characters in the alignment
cd /home/1_Else/Ceratopipra/2_Phylogeny/2.4aa_ML_mtDNAtree/datasets/
grep -v ">" Crubrocapilla_Cytb954bp_3chloromerosout.renamed.fasta | tr -d "TACGtacg" | grep -v "^$" | less
#there are no N's in this sequence, otherwise N's should be replaced by gaps. The only non-ATCG character is this alignment is "?", which represents uncertain nucleotides. I will replace it with "-"
sed 's/?/-/g' Crubrocapilla_Cytb954bp_3chloromerosout.renamed.fasta > Crubrocapilla_Cytb954bp_3chloromerosout.renamed.noN.fasta #keep in mind that it is case sensitive and there are more possible ambiguity codes - these are the only ones that I have in my dataset.
#check again to make sure they are really gone
grep -v ">" Crubrocapilla_Cytb954bp_3chloromerosout.renamed.noN.fasta | tr -d "TACGtacg" | grep -v "^$" | less

#kimura 2 parameter distances
/home/0_PROGRAMS/EMBOSS-6.6.0/bin/distmat -sequence Crubrocapilla_Cytb954bp_3chloromerosout.renamed.noN.fasta -nucmethod 2 -stdout -outfile Crubrocapilla_Cytb954bp_3chloromerosout.renamed.noN.fasta.distmat
```
* `-nucmethod`: 2 for kimura 2 parameter, 0 for uncorrected p.  

I placed the distance matrix into a folder `distmat` within `2_Phylogeny/2.4aa_ML_mtDNAtree/`.

Note that this distance matrix contains the two possible numts (OM172 and PPS081); I will not include these when I calculate the mean distance between the three haplogroups.  

## Calculating the averages  
Here I have taken the subset of the samples to use for the calculation of the average distances between samples:  

Here is a subset of the distance matrix for haplotypes 2 vs 3:  
```
	AMA656_In	CAM009_In
UNA018_At 6	0.63	0.89
UNA002_At 7	0.95	1.23
UNA110_At 8	0.95	1.23
UNA088_At 9	1.06	1.35
UNA124_At 10	0.95	1.23
CPEII008_At 11	0.95	1.23
CPE066_At 12	0.85	1.12
CPE012_At 13	0.74	0.89
CPE025_At 14	0.64	0.89
CPE049_At 15	0.53	0.78
CPE067_At 16	0.53	0.78
CPE069_At 17	0.53	0.78
CPE071_At 18	0.53	0.78
UHE442_Xi 19	0.63	0.89
PPBIO312_Xi 20	0.63	0.89
PPBIO221_Xi 21	0.74	1.01
PPBIO305_Xi 22	0.85	1.01
GAPTO140_Xi 23	0.63	0.89
GAPTO268_Xi 24	0.53	0.78
CAR005_Xi 25	0.85	1.12
GAPTO289_Xi 26	0.53	0.78
```
Average: 0.851904762 = 0.85% divergence  


Here is a subset of the distance matrix for haplotypes 1 vs 3:  
```
	MFNC067_Xi 27	GAPTO259_Xi 28	GAPTO277_Xi 29	GAPTO327_Xi 30	MT087_Xi 31	MT123_Xi 32	MTMA033_Xi 33	MTMA035_Xi 34	WM281_Ta 35	WM293_Ta 36	WM345_Ta 37	WM294_Ta 38	PIME134_Ta 39	PIME135_Ta 40	PIME062_Ro 41	PIME079_Ro 42	MF005_Ta 43	PIME482_Ta 44	PIME169_Ro 45	MPDS1220_Ta 46	FLJA001_Ta 47	FLJA085_Ta 48	PIME521_Ta 49	PIME522_Ta 50	BR163177_Ta 51	MSF107_Ta 52	PPS174_Ta 53	CH289_Ta 54	CH290_Ta 55	CH288_Ta 56	TLPC036_Ta 57	TLPC040_Ta 58	PPS145_Ta 59	PPS081_Ta 60	JTW1230_Ta 61	JTW1220_Ta 62	PPS072_Ta 63	MSF052_Ta 64	JTW1135_Ta 65	MSF198_Ta 66	PPS352_Ta 67	MSF162_Ta 68	TLPC022_Ta 69	TLPB221_Ta 70	TLP025_Ta 71	TLP123_Ta 72	TLPB026_Ta 73	TLPB027_Ta 74	TLPA267_Ta 75	TLPC068_Ta 76	TLPC077_Ta 77	TLPA360_Ta 78	TLPC041_Ta 79	TLPC184_Ta 80	TLPC185_Ta 81	ARAII062_Ro 82	ARAII069_Ro 83	PIME024_Ro 84	JRT021_Ro 85	JRT022_Ro 86	ARA109_Ro 87	ARAII010_Ro 88	ARAII026_Ro 89	ARAII033_Ro 90	ARAII031_Ro 91	ARA010_Ro 92	MAM010_Ro 93	MAM011_Ro 94	FPR006_Ro 95	FPR015_Ro 96	FPR021_Ro 97	AMZ610_Ro 98	JATC149_Ro 99	JATC148_Ro 100	JATC147_Ro 101	AMANA034_Ro 102	PIME460_Ro 103	PIME450_Ro 104	JATA145_Ro 105	JATA201_Ro 106	AMANA137_Ro 107	MPDS608_Ro 108	MPDS628_Ro 109	MPDS672_Ro 110	MAR021_Ro 111	MAR092_Ro 112	JATB060_Ta 113	MPDS496_Ro 114	OP077_Ro 115	OP100_Ro 116	AMA340_In 117	AMA601_In 118	PUC008_In 120	PUC007_In 121	PUC174_In 122	PUC023_In 123	RUR038_In 124	MPDS776_Ro 125	JUT237_In 126	JUT094_In 127	CUJ027_In 128	CUJ079_In 129	CUJ072_In 130	OM112_In 131	OM172_In 132	CAM039_In 133	CAM087_In 134	CAM097_In 135	CAM117_In 136	UFAC476_In 137	UFAC1007_In 138	UFAC305_In 139	UFAC1687_In 140	OM022_In 141	OM215_Ro 142	BR36440_In 143	UFAC095_In 144	CAM107_In 4
CAM009_In 5	1	1.12	1.12	1.12	1.46	1.34	1.12	1	1.23	1.46	1.23	1.12	1.34	1.34	1.12	1.12	1.23	1.57	1	1.23	1.23	1.23	1	1.34	1.23	1.34	1.12	1.12	1.23	1.12	1.12	1.23	1.46	1.47	1.46	1.46	1	1.12	1.12	1.46	1.12	1.12	1.12	1.46	1.12	1.46	1.03	1.23	1.12	1.46	1.44	1.34	1.23	1	1	1	1.34	1.12	1.46	1.12	1.23	1.12	1.12	1.23	1.23	1.12	0.86	1.12	1.34	1.34	1.23	1.78	1.23	1.12	1.23	1.23	1.23	1.34	1.34	1.12	1.23	1.12	1.12	1	1.23	1.23	1.12	1.34	1.23	1.23	1.23	1	1.23	1.34	1.46	1.23	1.34	1.34	1.34	1.34	1.46	1.12	1.12	1.12	1.46	1.12	1.12	1	1.23	1.23	1.12	1.12	1.23	1.34	1.34	1.34	1.23	1.16
AMA656_In 119	0.74	0.84	0.84	0.84	1.17	1.06	0.84	0.74	0.95	1.16	0.95	0.95	1.06	1.06	0.84	0.84	0.95	1.27	0.74	0.95	0.95	1.06	0.74	1.06	0.95	1.06	0.84	0.85	0.96	0.84	0.84	0.95	1.17	1.16	1.17	1.17	0.74	0.84	0.84	1.16	0.84	0.84	0.84	1.17	0.95	1.17	0.8	0.95	0.84	1.17	1.31	1.06	0.95	0.74	0.74	0.74	1.06	0.84	1.17	0.95	1.06	0.84	0.84	0.95	0.95	0.95	0.74	0.84	1.06	1.06	0.95	1.52	0.95	0.84	0.95	0.95	0.95	1.06	1.06	0.95	0.95	0.84	0.84	0.74	0.95	0.95	0.84	1.06	0.95	0.95	0.95	0.74	0.95	1.06	1.16	0.95	1.06	1.06	1.06	1.06	1.17	0.84	0.84	0.84	1.17	0.84	0.84	0.74	0.92	0.95	0.84	0.84	0.95	1.06	1.06	1.06	0.95	1.01
```
Average: 1.086465517 = 1.09% divergence (not counting the columns for OM172 and PPS081, possible numts)  


Here is a subset of the distance matrix for haplotypes 2 vs 1:  
```
	MFNC067_Xi 27	GAPTO259_Xi 28	GAPTO277_Xi 29	GAPTO327_Xi 30	MT087_Xi 31	MT123_Xi 32	MTMA033_Xi 33	MTMA035_Xi 34	WM281_Ta 35	WM293_Ta 36	WM345_Ta 37	WM294_Ta 38	PIME134_Ta 39	PIME135_Ta 40	PIME062_Ro 41	PIME079_Ro 42	MF005_Ta 43	PIME482_Ta 44	PIME169_Ro 45	MPDS1220_Ta 46	FLJA001_Ta 47	FLJA085_Ta 48	PIME521_Ta 49	PIME522_Ta 50	BR163177_Ta 51	MSF107_Ta 52	PPS174_Ta 53	CH289_Ta 54	CH290_Ta 55	CH288_Ta 56	TLPC036_Ta 57	TLPC040_Ta 58	PPS145_Ta 59	PPS081_Ta 60	JTW1230_Ta 61	JTW1220_Ta 62	PPS072_Ta 63	MSF052_Ta 64	JTW1135_Ta 65	MSF198_Ta 66	PPS352_Ta 67	MSF162_Ta 68	TLPC022_Ta 69	TLPB221_Ta 70	TLP025_Ta 71	TLP123_Ta 72	TLPB026_Ta 73	TLPB027_Ta 74	TLPA267_Ta 75	TLPC068_Ta 76	TLPC077_Ta 77	TLPA360_Ta 78	TLPC041_Ta 79	TLPC184_Ta 80	TLPC185_Ta 81	ARAII062_Ro 82	ARAII069_Ro 83	PIME024_Ro 84	JRT021_Ro 85	JRT022_Ro 86	ARA109_Ro 87	ARAII010_Ro 88	ARAII026_Ro 89	ARAII033_Ro 90	ARAII031_Ro 91	ARA010_Ro 92	MAM010_Ro 93	MAM011_Ro 94	FPR006_Ro 95	FPR015_Ro 96	FPR021_Ro 97	AMZ610_Ro 98	JATC149_Ro 99	JATC148_Ro 100	JATC147_Ro 101	AMANA034_Ro 102	PIME460_Ro 103	PIME450_Ro 104	JATA145_Ro 105	JATA201_Ro 106	AMANA137_Ro 107	MPDS608_Ro 108	MPDS628_Ro 109	MPDS672_Ro 110	MAR021_Ro 111	MAR092_Ro 112	JATB060_Ta 113	MPDS496_Ro 114	OP077_Ro 115	OP100_Ro 116	AMA340_In 117	AMA601_In 118	PUC008_In 120	PUC007_In 121	PUC174_In 122	PUC023_In 123	RUR038_In 124	MPDS776_Ro 125	JUT237_In 126	JUT094_In 127	CUJ027_In 128	CUJ079_In 129	CUJ072_In 130	OM112_In 131	OM172_In 132	CAM039_In 133	CAM087_In 134	CAM097_In 135	CAM117_In 136	UFAC476_In 137	UFAC1007_In 138	UFAC305_In 139	UFAC1687_In 140	OM022_In 141	OM215_Ro 142	BR36440_In 143	UFAC095_In 144	CAM107_In 4
UNA018_At 6	0.74	0.84	0.84	0.84	0.95	1.06	0.84	0.74	0.95	1.16	0.95	0.95	1.06	1.06	0.84	0.84	0.95	1.27	0.74	0.95	0.95	1.06	0.74	1.06	0.95	1.06	0.84	0.85	0.96	0.84	0.84	0.95	0.95	1.16	0.95	0.95	0.74	0.84	0.84	1.16	0.84	0.84	0.84	0.95	0.95	0.95	0.8	0.95	0.84	1.17	1.17	1.06	0.95	0.74	0.74	0.74	1.06	0.84	1.17	0.95	1.06	0.84	0.84	0.95	0.95	0.95	0.86	0.84	1.06	1.06	0.95	1.26	0.95	0.84	0.95	0.95	0.95	1.06	1.06	0.95	0.95	0.84	0.84	0.74	1.06	0.95	0.84	1.06	0.95	0.95	0.95	0.74	0.95	1.06	1.16	0.95	1.06	1.06	1.06	1.06	1.17	0.84	0.84	0.84	1.17	0.84	0.84	0.74	1.16	0.95	0.84	0.84	0.95	1.06	1.06	1.06	0.95	1.16
UNA002_At 7	1.06	1.17	1.17	1.17	1.27	1.38	1.17	1.06	1.27	1.49	1.27	1.27	1.38	1.38	1.17	1.17	1.27	1.6	1.06	1.27	1.27	1.38	1.06	1.38	1.27	1.38	1.17	1.18	1.28	1.17	1.17	1.27	1.27	1.51	1.27	1.27	1.06	1.17	1.17	1.49	1.17	1.17	1.17	1.27	1.27	1.27	1.15	1.27	1.17	1.49	1.57	1.38	1.27	1.06	1.06	1.06	1.38	1.17	1.49	1.27	1.39	1.17	1.17	1.27	1.27	1.27	1.23	1.17	1.17	1.38	1.27	1.65	1.27	1.17	1.27	1.27	1.27	1.38	1.38	1.27	1.27	1.17	1.17	1.06	1.38	1.27	1.17	1.38	1.27	1.27	1.06	1.06	1.27	1.38	1.49	1.27	1.38	1.38	1.38	1.38	1.49	1.17	1.17	1.17	1.49	1.17	1.17	1.06	1.51	1.27	1.17	1.17	1.27	1.38	1.38	1.38	1.27	1.6
UNA110_At 8	1.06	1.17	1.17	1.17	1.27	1.38	1.17	1.06	1.27	1.49	1.27	1.27	1.38	1.38	1.17	1.17	1.27	1.6	1.06	1.27	1.27	1.38	1.06	1.38	1.27	1.38	1.17	1.18	1.28	1.17	1.17	1.27	1.27	1.51	1.27	1.27	1.06	1.17	1.17	1.49	1.17	1.17	1.17	1.27	1.27	1.27	1.15	1.27	1.17	1.49	1.57	1.38	1.27	1.06	1.06	1.06	1.38	1.17	1.49	1.27	1.39	1.17	1.17	1.27	1.27	1.27	1.23	1.17	1.17	1.38	1.27	1.65	1.27	1.17	1.27	1.27	1.27	1.38	1.38	1.27	1.27	1.17	1.17	1.06	1.38	1.27	1.17	1.38	1.27	1.27	1.06	1.06	1.27	1.38	1.49	1.27	1.38	1.38	1.38	1.38	1.49	1.17	1.17	1.17	1.49	1.17	1.17	1.06	1.51	1.27	1.17	1.17	1.27	1.38	1.38	1.38	1.27	1.6
UNA088_At 9	1.17	1.27	1.27	1.27	1.38	1.49	1.27	1.17	1.38	1.59	1.38	1.38	1.49	1.49	1.27	1.27	1.38	1.7	1.17	1.38	1.38	1.49	1.17	1.49	1.38	1.49	1.27	1.28	1.39	1.27	1.27	1.38	1.38	1.63	1.38	1.38	1.17	1.27	1.27	1.59	1.27	1.27	1.27	1.38	1.38	1.38	1.26	1.38	1.27	1.6	1.57	1.49	1.38	1.17	1.17	1.17	1.49	1.27	1.6	1.38	1.49	1.27	1.27	1.38	1.38	1.38	1.36	1.27	1.27	1.49	1.38	1.78	1.38	1.27	1.38	1.38	1.38	1.49	1.49	1.38	1.38	1.27	1.27	1.17	1.49	1.38	1.27	1.49	1.38	1.38	1.17	1.17	1.38	1.49	1.59	1.38	1.49	1.49	1.49	1.49	1.6	1.27	1.27	1.27	1.6	1.27	1.27	1.17	1.63	1.38	1.27	1.27	1.38	1.49	1.49	1.49	1.38	1.6
UNA124_At 10	1.06	1.17	1.17	1.17	1.27	1.38	1.17	1.06	1.27	1.49	1.27	1.27	1.38	1.38	1.17	1.17	1.27	1.6	1.06	1.27	1.27	1.38	1.06	1.38	1.27	1.38	1.17	1.18	1.28	1.17	1.17	1.27	1.27	1.51	1.27	1.27	1.06	1.17	1.17	1.49	1.17	1.17	1.17	1.27	1.27	1.27	1.15	1.27	1.17	1.49	1.57	1.38	1.27	1.06	1.06	1.06	1.38	1.17	1.49	1.27	1.39	1.17	1.17	1.27	1.27	1.27	1.23	1.17	1.17	1.38	1.27	1.65	1.27	1.17	1.27	1.27	1.27	1.38	1.38	1.27	1.27	1.17	1.17	1.06	1.38	1.27	1.17	1.38	1.27	1.27	1.06	1.06	1.27	1.38	1.49	1.27	1.38	1.38	1.38	1.38	1.49	1.17	1.17	1.17	1.49	1.17	1.17	1.06	1.51	1.27	1.17	1.17	1.27	1.38	1.38	1.38	1.27	1.6
CPEII008_At 11	1.06	1.17	1.17	1.17	1.27	1.38	1.17	1.06	1.27	1.49	1.27	1.27	1.38	1.38	1.17	1.17	1.27	1.6	1.06	1.27	1.27	1.38	1.06	1.38	1.27	1.38	1.17	1.18	1.28	1.17	1.17	1.27	1.27	1.51	1.27	1.27	1.06	1.17	1.17	1.49	1.17	1.17	1.17	1.27	1.27	1.27	1.15	1.27	1.17	1.49	1.57	1.38	1.27	1.06	1.06	1.06	1.38	1.17	1.49	1.27	1.39	1.17	1.17	1.27	1.27	1.27	1.23	1.17	1.17	1.38	1.27	1.65	1.27	1.17	1.27	1.27	1.27	1.38	1.38	1.27	1.27	1.17	1.17	1.06	1.38	1.27	1.17	1.38	1.27	1.27	1.06	1.06	1.27	1.38	1.49	1.27	1.38	1.38	1.38	1.38	1.49	1.17	1.17	1.17	1.49	1.17	1.17	1.06	1.51	1.27	1.17	1.17	1.27	1.38	1.38	1.38	1.27	1.6
CPE066_At 12	0.96	1.06	1.06	1.06	1.17	1.28	1.06	0.96	1.17	1.38	1.17	1.17	1.28	1.28	1.06	1.06	1.17	1.49	0.96	1.17	1.17	1.28	0.96	1.28	1.17	1.28	1.06	1.07	1.17	1.06	1.06	1.17	1.17	1.4	1.17	1.17	0.96	1.06	1.06	1.38	1.06	1.06	1.06	1.17	1.17	1.17	1.03	1.17	1.06	1.39	1.44	1.28	1.17	0.96	0.96	0.96	1.28	1.06	1.39	1.17	1.28	1.06	1.06	1.17	1.17	1.17	1.11	1.06	1.06	1.28	1.17	1.52	1.17	1.06	1.17	1.17	1.17	1.28	1.28	1.17	1.17	1.06	1.06	0.96	1.28	1.17	1.06	1.28	1.17	1.17	1.17	0.96	1.17	1.28	1.38	1.17	1.28	1.28	1.28	1.28	1.39	1.06	1.06	1.06	1.39	1.06	1.06	0.96	1.4	1.17	1.06	1.06	1.17	1.28	1.28	1.28	1.17	1.45
CPE012_At 13	0.85	0.96	0.96	0.96	1.06	1.17	0.96	0.85	1.06	1.28	1.06	1.06	1.17	1.17	0.96	0.96	1.06	1.39	0.85	1.06	1.06	0.96	0.85	1.17	1.06	1.17	0.96	0.96	1.06	0.96	0.96	1.06	1.06	1.28	1.06	1.06	0.85	0.96	0.96	1.28	0.96	0.96	0.96	1.06	1.06	1.06	0.8	1.06	0.96	1.28	1.17	1.17	1.06	0.85	0.85	0.85	1.17	0.96	1.28	1.06	1.17	0.96	0.96	1.06	1.06	1.06	0.86	0.96	1.17	1.17	1.06	1.26	1.06	0.96	1.06	1.06	1.06	1.17	1.17	1.06	1.06	0.96	0.96	0.85	1.17	1.06	0.96	1.17	1.06	1.06	1.06	0.85	1.06	1.17	1.28	1.06	1.17	1.17	1.17	1.17	1.28	0.96	0.96	0.96	1.28	0.96	0.96	0.85	1.28	1.06	0.96	0.96	1.06	1.17	1.17	1.17	1.06	1.16
CPE025_At 14	0.74	0.85	0.85	0.85	0.96	1.06	0.85	0.74	0.96	1.17	0.96	0.96	1.06	1.06	0.85	0.85	0.96	1.28	0.74	0.96	0.96	1.06	0.74	1.06	0.96	1.06	0.85	0.85	0.96	0.85	0.85	0.96	0.96	1.16	0.96	0.96	0.74	0.85	0.85	1.17	0.85	0.85	0.85	0.96	0.96	0.96	0.8	0.96	0.85	1.17	1.17	1.06	0.96	0.74	0.74	0.74	1.06	0.85	1.17	0.96	1.07	0.85	0.85	0.96	0.96	0.96	0.86	0.85	1.06	1.06	0.96	1.26	0.96	0.85	0.96	0.96	0.96	1.06	1.06	0.96	0.96	0.85	0.85	0.74	1.06	0.96	0.85	1.06	0.96	0.96	0.96	0.74	0.96	1.06	1.17	0.96	1.06	1.06	1.06	1.06	1.17	0.85	0.85	0.85	1.17	0.85	0.85	0.74	1.16	0.96	0.85	0.85	0.96	1.06	1.06	1.06	0.96	1.16
CPE049_At 15	0.64	0.74	0.74	0.74	0.85	0.96	0.74	0.64	0.85	1.06	0.85	0.85	0.96	0.96	0.74	0.74	0.85	1.17	0.64	0.85	0.85	0.96	0.64	0.96	0.85	0.96	0.74	0.75	0.85	0.74	0.74	0.85	0.85	1.05	0.85	0.85	0.64	0.74	0.74	1.06	0.74	0.74	0.74	0.85	0.85	0.85	0.69	0.85	0.74	1.06	1.04	0.96	0.85	0.64	0.64	0.64	0.96	0.74	1.06	0.85	0.96	0.74	0.74	0.85	0.85	0.85	0.74	0.74	0.96	0.96	0.85	1.14	0.85	0.74	0.85	0.85	0.85	0.96	0.96	0.85	0.85	0.74	0.74	0.64	0.95	0.85	0.74	0.96	0.85	0.85	0.85	0.64	0.85	0.96	1.06	0.85	0.96	0.96	0.96	0.96	1.06	0.74	0.74	0.74	1.06	0.74	0.74	0.64	1.05	0.85	0.74	0.74	0.85	0.95	0.95	0.95	0.85	1.01
CPE067_At 16	0.64	0.74	0.74	0.74	0.85	0.96	0.74	0.64	0.85	1.06	0.85	0.85	0.96	0.96	0.74	0.74	0.85	1.17	0.64	0.85	0.85	0.96	0.64	0.96	0.85	0.96	0.74	0.75	0.85	0.74	0.74	0.85	0.85	1.05	0.85	0.85	0.64	0.74	0.74	1.06	0.74	0.74	0.74	0.85	0.85	0.85	0.69	0.85	0.74	1.06	1.04	0.96	0.85	0.64	0.64	0.64	0.96	0.74	1.06	0.85	0.96	0.74	0.74	0.85	0.85	0.85	0.74	0.74	0.96	0.96	0.85	1.14	0.85	0.74	0.85	0.85	0.85	0.96	0.96	0.85	0.85	0.74	0.74	0.64	0.95	0.85	0.74	0.96	0.85	0.85	0.85	0.64	0.85	0.96	1.06	0.85	0.96	0.96	0.96	0.96	1.06	0.74	0.74	0.74	1.06	0.74	0.74	0.64	1.05	0.85	0.74	0.74	0.85	0.95	0.95	0.95	0.85	1.01
CPE069_At 17	0.63	0.74	0.74	0.74	0.84	0.95	0.74	0.63	0.84	1.06	0.84	0.84	0.95	0.95	0.74	0.74	0.84	1.17	0.63	0.84	0.84	0.95	0.63	0.95	0.84	0.95	0.74	0.75	0.85	0.74	0.74	0.84	0.84	1.04	0.84	0.84	0.63	0.74	0.74	1.06	0.74	0.74	0.74	0.84	0.84	0.84	0.69	0.84	0.74	1.06	1.04	0.95	0.84	0.63	0.63	0.63	0.95	0.74	1.06	0.84	0.96	0.74	0.74	0.84	0.84	0.84	0.74	0.74	0.95	0.95	0.84	1.14	0.84	0.74	0.84	0.84	0.84	0.95	0.95	0.84	0.84	0.74	0.74	0.63	0.95	0.84	0.74	0.95	0.84	0.84	0.84	0.63	0.84	0.95	1.06	0.84	0.95	0.95	0.95	0.95	1.06	0.74	0.74	0.74	1.06	0.74	0.74	0.63	1.04	0.84	0.74	0.74	0.84	0.95	0.95	0.95	0.84	1.01
CPE071_At 18	0.63	0.74	0.74	0.74	0.84	0.95	0.74	0.63	0.84	1.06	0.84	0.84	0.95	0.95	0.74	0.74	0.84	1.17	0.63	0.84	0.84	0.95	0.63	0.95	0.84	0.95	0.74	0.75	0.85	0.74	0.74	0.84	0.84	1.04	0.84	0.84	0.63	0.74	0.74	1.06	0.74	0.74	0.74	0.84	0.84	0.84	0.69	0.84	0.74	1.06	1.04	0.95	0.84	0.63	0.63	0.63	0.95	0.74	1.06	0.84	0.96	0.74	0.74	0.84	0.84	0.84	0.74	0.74	0.95	0.95	0.84	1.14	0.84	0.74	0.84	0.84	0.84	0.95	0.95	0.84	0.84	0.74	0.74	0.63	0.95	0.84	0.74	0.95	0.84	0.84	0.84	0.63	0.84	0.95	1.06	0.84	0.95	0.95	0.95	0.95	1.06	0.74	0.74	0.74	1.06	0.74	0.74	0.63	1.04	0.84	0.74	0.74	0.84	0.95	0.95	0.95	0.84	1.01
UHE442_Xi 19	0.95	1.06	1.06	1.06	1.17	1.27	1.06	0.95	1.17	1.38	1.17	1.17	1.27	1.27	1.06	1.06	1.17	1.49	0.95	1.17	1.17	1.27	0.95	1.27	1.17	1.27	1.06	1.07	1.17	1.06	1.06	1.17	1.17	1.39	1.17	1.17	0.95	1.06	1.06	1.38	1.06	1.06	1.06	1.17	1.17	1.17	1.03	1.17	1.06	1.38	1.44	1.27	1.17	0.95	0.95	0.95	1.27	1.06	1.38	1.17	1.28	1.06	1.06	1.17	1.17	1.17	1.11	1.06	1.27	1.27	1.17	1.52	1.17	1.06	1.17	1.17	1.16	1.27	1.27	1.17	1.17	1.06	1.06	0.95	1.16	1.17	1.06	1.27	1.17	1.17	1.17	0.95	1.17	1.27	1.38	1.17	1.27	1.27	1.27	1.27	1.38	1.06	1.06	1.06	1.38	1.06	1.06	0.95	1.39	1.17	1.06	1.06	1.17	1.27	1.27	1.27	1.16	1.45
PPBIO312_Xi 20	0.74	0.84	0.84	0.84	0.95	1.06	0.84	0.74	0.95	1.16	0.95	0.95	1.06	1.06	0.84	0.84	0.95	1.27	0.74	0.95	0.95	1.06	0.74	1.06	0.95	1.06	0.84	0.85	0.96	0.84	0.84	0.95	0.95	1.16	0.95	0.95	0.74	0.84	0.84	1.16	0.84	0.84	0.84	0.95	0.95	0.95	0.8	0.95	0.84	0.95	1.17	1.06	0.95	0.74	0.74	0.74	1.06	0.84	1.17	0.95	1.06	0.84	0.84	0.95	0.95	0.95	0.86	0.84	1.06	1.06	0.95	1.26	0.95	0.84	0.95	0.95	0.95	1.06	1.06	0.95	0.95	0.84	0.84	0.74	1.06	0.95	0.84	1.06	0.95	0.95	0.95	0.74	0.95	1.06	1.16	0.95	1.06	1.06	0.84	1.06	1.17	0.84	0.84	0.84	1.17	0.84	0.84	0.74	1.16	0.95	0.84	0.84	0.95	1.06	1.06	1.06	0.95	1.16
PPBIO221_Xi 21	0.84	0.95	0.95	0.95	1.06	1.17	0.95	0.84	1.06	1.27	1.06	1.06	1.17	1.17	0.95	0.95	1.06	1.38	0.84	1.06	1.06	1.17	0.84	1.17	1.06	1.17	0.95	0.96	1.06	0.95	0.95	0.84	1.06	1.28	1.06	1.06	0.84	0.95	0.95	1.27	0.95	0.95	0.95	1.06	1.06	1.06	0.92	1.06	0.95	1.27	1.31	1.17	1.06	0.84	0.84	0.84	1.17	0.95	1.27	1.06	1.17	0.95	0.95	1.06	1.06	1.06	0.98	0.95	1.17	1.17	1.06	1.39	1.06	0.95	1.06	1.06	1.06	1.17	1.17	1.06	1.06	0.95	0.95	0.84	1.16	1.06	0.95	1.17	1.06	1.06	1.06	0.84	1.06	1.17	1.06	1.06	1.17	1.17	1.17	1.17	1.27	0.95	0.95	0.95	1.27	0.95	0.95	0.84	1.27	1.06	0.95	0.95	1.06	1.16	1.16	1.16	1.06	1.3
PPBIO305_Xi 22	0.95	1.06	1.06	1.06	1.17	1.27	1.06	0.95	1.17	1.38	1.17	1.17	1.27	1.27	1.06	1.06	1.17	1.49	0.95	1.17	1.17	1.27	0.95	1.27	1.17	1.27	1.06	0.96	1.17	1.06	1.06	1.17	1.17	1.39	1.17	1.17	0.95	1.06	1.06	1.38	1.06	1.06	1.06	1.17	1.17	1.17	0.92	1.17	1.06	1.38	1.17	1.27	1.17	0.95	0.95	0.95	1.27	1.06	1.38	1.17	1.28	1.06	1.06	1.17	1.17	1.17	0.98	1.06	1.27	1.27	1.17	1.39	1.17	1.06	1.17	1.17	1.16	1.27	1.27	1.17	1.17	1.06	1.06	0.95	1.27	1.17	1.06	1.27	1.17	1.17	1.17	0.95	1.17	1.27	1.38	1.17	1.27	1.27	1.27	1.27	1.38	1.06	1.06	1.06	1.38	1.06	1.06	0.95	1.39	1.17	1.06	1.06	1.17	1.27	1.27	1.27	1.16	1.16
GAPTO140_Xi 23	0.74	0.84	0.84	0.84	0.95	1.06	0.84	0.74	0.95	0.95	0.95	0.95	0.84	1.06	0.84	0.84	0.95	1.27	0.74	0.95	0.95	1.06	0.74	1.06	0.95	1.06	0.84	0.85	0.96	0.84	0.84	0.95	0.95	1.04	0.95	0.95	0.74	0.84	0.84	1.16	0.84	0.84	0.84	0.95	0.95	0.95	0.8	0.95	0.84	1.17	1.17	0.84	0.95	0.74	0.74	0.74	1.06	0.84	1.17	0.95	1.06	0.84	0.84	0.95	0.95	0.95	0.86	0.84	1.06	1.06	0.95	1.14	0.95	0.84	0.95	0.95	0.95	1.06	1.06	0.95	0.95	0.84	0.84	0.74	1.06	0.95	0.84	1.06	0.95	0.95	0.95	0.74	0.95	1.06	1.16	0.95	1.06	1.06	1.06	0.84	1.17	0.84	0.84	0.84	1.17	0.84	0.84	0.74	1.04	0.95	0.84	0.84	0.95	1.06	1.06	1.06	0.95	1.01
GAPTO268_Xi 24	0.63	0.74	0.74	0.74	0.84	0.95	0.74	0.63	0.84	1.06	0.84	0.84	0.95	0.95	0.74	0.74	0.84	1.17	0.63	0.84	0.84	0.95	0.63	0.95	0.84	0.95	0.74	0.75	0.85	0.74	0.74	0.84	0.84	1.04	0.84	0.84	0.63	0.74	0.74	1.06	0.74	0.74	0.74	0.84	0.84	0.84	0.69	0.84	0.74	1.06	1.04	0.95	0.84	0.63	0.63	0.63	0.95	0.74	1.06	0.84	0.96	0.74	0.74	0.84	0.84	0.84	0.74	0.74	0.95	0.95	0.84	1.14	0.84	0.74	0.84	0.84	0.84	0.95	0.95	0.84	0.84	0.74	0.74	0.63	0.95	0.84	0.74	0.95	0.84	0.84	0.84	0.63	0.84	0.95	1.06	0.84	0.95	0.95	0.95	0.95	1.06	0.74	0.74	0.74	1.06	0.74	0.74	0.63	1.04	0.84	0.74	0.74	0.84	0.95	0.95	0.95	0.84	1.01
CAR005_Xi 25	0.95	1.06	1.06	1.06	1.17	1.27	1.06	0.95	1.17	1.38	1.17	1.17	1.27	1.27	1.06	1.06	1.17	1.49	0.95	1.17	1.17	1.27	0.95	1.27	1.17	1.27	1.06	1.07	1.17	1.06	1.06	0.95	1.17	1.39	1.17	1.17	0.95	1.06	1.06	1.38	1.06	1.06	1.06	1.17	1.17	1.17	1.03	1.17	1.06	1.38	1.44	1.27	1.17	0.95	0.95	0.95	1.27	1.06	1.38	1.17	1.28	1.06	1.06	1.17	1.17	1.17	1.11	1.06	1.27	1.27	1.17	1.52	1.17	1.06	1.17	1.17	1.16	1.27	1.27	1.17	1.17	1.06	1.06	0.95	1.27	1.17	1.06	1.27	1.17	1.17	1.17	0.95	1.17	1.27	1.16	1.17	1.27	1.27	1.27	1.27	1.38	1.06	1.06	1.06	1.38	1.06	1.06	0.95	1.39	1.17	1.06	1.06	1.17	1.27	1.27	1.27	1.16	1.45
GAPTO289_Xi 26	0.63	0.74	0.74	0.74	0.84	0.95	0.74	0.63	0.84	1.06	0.84	0.84	0.95	0.95	0.74	0.74	0.84	1.17	0.63	0.84	0.84	0.95	0.63	0.95	0.84	0.95	0.74	0.75	0.85	0.74	0.74	0.84	0.84	1.04	0.84	0.84	0.63	0.74	0.74	1.06	0.74	0.74	0.74	0.84	0.84	0.84	0.69	0.84	0.74	1.06	1.04	0.95	0.84	0.63	0.63	0.63	0.95	0.74	1.06	0.84	0.96	0.74	0.74	0.84	0.84	0.84	0.74	0.74	0.95	0.95	0.84	1.14	0.84	0.74	0.84	0.84	0.84	0.95	0.95	0.84	0.84	0.74	0.74	0.63	0.95	0.84	0.74	0.95	0.84	0.84	0.84	0.63	0.84	0.95	1.06	0.84	0.95	0.95	0.95	0.95	1.06	0.74	0.74	0.74	1.06	0.74	0.74	0.63	1.04	0.84	0.74	0.74	0.84	0.95	0.95	0.95	0.84	1.01
```
Average: 1.045036946 = 1.05% divergence.  